diff --git a/toolkit/components/extensions/test/xpcshell/test_ext_runtime_getBrowserInfo.js b/toolkit/components/extensions/test/xpcshell/test_ext_runtime_getBrowserInfo.js
index 3f3b8f8e95fc69ce09b143a7865e5a52bfaf9200..166d72d94c017c6704d602e0ed45e7ab934ec10d 100644
--- a/toolkit/components/extensions/test/xpcshell/test_ext_runtime_getBrowserInfo.js
+++ b/toolkit/components/extensions/test/xpcshell/test_ext_runtime_getBrowserInfo.js
@@ -15,7 +15,10 @@ add_task(async function test_getBrowserInfo() {
     browser.test.assertEq(info.vendor, "Mozilla", "vendor is Mozilla");
     browser.test.assertEq(info.version, "48", "version is correct");
     browser.test.assertEq(info.buildID, "20160315", "buildID is correct");
-
+    // Assert that the zen object is present and has the expected properties
+    browser.test.assertTrue(info.zen, "zen object is present");
+    browser.test.assertTrue(info.zen.version, "zen version is present");
+    browser.test.assertTrue(info.zen.platform, "zen platform is present");
     browser.test.notifyPass("runtime.getBrowserInfo");
   }
 
