diff --git a/toolkit/content/aboutSupport.xhtml b/toolkit/content/aboutSupport.xhtml
index 7f57218f3573488445ea363e0c4532b292d53167..511f578b4ae38a496ca936adf2fef1f587249f47 100644
--- a/toolkit/content/aboutSupport.xhtml
+++ b/toolkit/content/aboutSupport.xhtml
@@ -10,6 +10,7 @@
 
 <html xmlns="http://www.w3.org/1999/xhtml">
   <head>
+    <script src="chrome://browser/content/zenThemeModifier.js"></script>
     <meta http-equiv="Content-Security-Policy" content="default-src chrome:; object-src 'none'" />
     <meta name="color-scheme" content="light dark" />
     <title data-l10n-id="page-title"/>
