diff --git a/toolkit/content/aboutSupport.xhtml b/toolkit/content/aboutSupport.xhtml
index 05e3d72a6e67900e9b3d9bfd35c89b056c7ca747..55408e77830492765c976dd967ca14a7b74cba97 100644
--- a/toolkit/content/aboutSupport.xhtml
+++ b/toolkit/content/aboutSupport.xhtml
@@ -31,6 +31,7 @@
 #ifndef ANDROID
     <link rel="localization" href="toolkit/featuregates/features.ftl"/>
 #endif
+    <script src="chrome://browser/content/zenThemeModifier.js"></script>
   </head>
 
   <body class="wide-container">
