diff --git a/toolkit/content/aboutSupport.xhtml b/toolkit/content/aboutSupport.xhtml
index d3de7d001973434c09deb5723e73ea72f134362b..736b4e7e647d2f80697a0721c765d2aa82507583 100644
--- a/toolkit/content/aboutSupport.xhtml
+++ b/toolkit/content/aboutSupport.xhtml
@@ -27,6 +27,7 @@
 #ifndef ANDROID
     <link rel="localization" href="toolkit/featuregates/features.ftl"/>
 #endif
+  <script src="chrome://browser/content/zenThemeModifier.js"></script>
   </head>
 
   <body class="wide-container">
