diff --git a/toolkit/content/aboutSupport.xhtml b/toolkit/content/aboutSupport.xhtml
index d28ca0dd0939c9a9c43bc7c6b3c7184172c09ed8..112d060540e254e1766c145913e21819f62439f8 100644
--- a/toolkit/content/aboutSupport.xhtml
+++ b/toolkit/content/aboutSupport.xhtml
@@ -30,6 +30,7 @@
     <link rel="localization" href="preview/genai.ftl" />
 #endif
 #endif
+  <script src="chrome://browser/content/zenThemeModifier.js"></script>
   </head>
 
   <body class="wide-container">
