diff --git a/toolkit/content/jar.mn b/toolkit/content/jar.mn
index 8b18c945253d00603891cc5f0b989476318e219c..c5368eec8d9a760c19a827fe7a4b9afb4b702976 100644
--- a/toolkit/content/jar.mn
+++ b/toolkit/content/jar.mn
@@ -143,3 +143,7 @@ toolkit.jar:
 # Third party files
    content/global/third_party/d3/d3.js                (/third_party/js/d3/d3.js)
    content/global/third_party/cfworker/json-schema.js (/third_party/js/cfworker/json-schema.js)
+
+# Zen modules
+   content/global/zen-xul.css
+   content/global/zen-fullscreen-override.css
