diff --git a/layout/base/nsPresContext.h b/layout/base/nsPresContext.h
index b7cd5cecf67462581e78a8346196da25c3b93742..0bef9f83491c1f6850be8ad97707ab1f08178a0a 100644
--- a/layout/base/nsPresContext.h
+++ b/layout/base/nsPresContext.h
@@ -561,6 +561,8 @@ class nsPresContext : public nsISupports,
   void UpdateForcedColors(bool aNotify = true);
 
  public:
+  nscolor mZenBoostsPresContextData{0};
+  bool mZenBoostsInvert{false};
   float GetFullZoom() { return mFullZoom; }
   /**
    * Device full zoom differs from full zoom because it gets the zoom from
