diff --git a/layout/base/nsPresContext.cpp b/layout/base/nsPresContext.cpp
index edae3bacba3a029cfb4038e1de4b36b366488385..40b7b7b66edeeb99a5208d989077e6a39162c7a4 100644
--- a/layout/base/nsPresContext.cpp
+++ b/layout/base/nsPresContext.cpp
@@ -20,6 +20,7 @@
 #include "base/basictypes.h"
 #include "gfxPlatform.h"
 #include "gfxTextRun.h"
+#include "mozilla/nsZenBoostsBackend.h"
 #include "mozilla/AnimationEventDispatcher.h"
 #include "mozilla/ContentBlockingAllowList.h"
 #include "mozilla/CycleCollectedJSContext.h"
@@ -917,6 +918,13 @@ void nsPresContext::RecomputeBrowsingContextDependentData() {
   }
 
   auto* top = browsingContext->Top();
+
+  nsCOMPtr<zen::nsZenBoostsBackend> zenBackend(
+    do_GetService(ZEN_BOOSTS_BACKEND_CONTRACTID));
+  if (zenBackend) {
+    zenBackend->RecomputeBrowsingContextDependentData(this, top);
+  }
+
   SetColorSchemeOverride([&] {
     auto overriden = top->PrefersColorSchemeOverride();
     if (overriden != PrefersColorSchemeOverride::None) {
