diff --git a/layout/base/nsPresContext.cpp b/layout/base/nsPresContext.cpp
index c1cc9a6505ffb3f0019ce007c8d17366364bdd86..f83e4ecb6379a3b12edcef152affa7030aa5f9aa 100644
--- a/layout/base/nsPresContext.cpp
+++ b/layout/base/nsPresContext.cpp
@@ -20,6 +20,7 @@
 #include "base/basictypes.h"
 #include "gfxPlatform.h"
 #include "gfxTextRun.h"
+#include "mozilla/nsZenBoostsBackend.h"
 #include "mozilla/AnimationEventDispatcher.h"
 #include "mozilla/ContentBlockingAllowList.h"
 #include "mozilla/CycleCollectedJSContext.h"
@@ -914,6 +915,12 @@ void nsPresContext::RecomputeBrowsingContextDependentData() {
     SetOverrideDPPX(browsingContext->OverrideDPPX());
   }
 
+  nsCOMPtr<zen::nsZenBoostsBackend> zenBackend(
+    do_GetService(ZEN_BOOSTS_BACKEND_CONTRACTID));
+  if (zenBackend) {
+    zenBackend->RecomputeBrowsingContextDependentData(this);
+  }
+
   auto* top = browsingContext->Top();
   SetColorSchemeOverride([&] {
     auto overriden = top->PrefersColorSchemeOverride();
