diff --git a/layout/base/nsPresContext.h b/layout/base/nsPresContext.h
index 659c2a237280794c19de2fbb54b3b3070590cc30..b47ef9d53ddcdd904c1383fd136cf13a2224f347 100644
--- a/layout/base/nsPresContext.h
+++ b/layout/base/nsPresContext.h
@@ -566,6 +566,7 @@ class nsPresContext : public nsISupports,
   void UpdateForcedColors(bool aNotify = true);
 
  public:
+  nscolor mZenBoostsPresContextData;
   float GetFullZoom() { return mFullZoom; }
   /**
    * Device full zoom differs from full zoom because it gets the zoom from
