diff --git a/layout/base/nsPresContext.h b/layout/base/nsPresContext.h
index cabe267af71a442fba3aa9dbb4d5f742242e8313..e919600e632422b44d6f7dabadf1f70745994254 100644
--- a/layout/base/nsPresContext.h
+++ b/layout/base/nsPresContext.h
@@ -561,6 +561,8 @@ class nsPresContext : public nsISupports,
   void UpdateForcedColors(bool aNotify = true);
 
  public:
+  nscolor mZenBoostsPresContextData{0};
+  bool mZenBoostsInvert;
   float GetFullZoom() { return mFullZoom; }
   /**
    * Device full zoom differs from full zoom because it gets the zoom from
