diff --git a/layout/base/nsPresContext.cpp b/layout/base/nsPresContext.cpp
index 222015b9617edcbcd349214c1610517a8617fdd9..aa576f37091b354a5c4cb8be307adef09b54a8f4 100644
--- a/layout/base/nsPresContext.cpp
+++ b/layout/base/nsPresContext.cpp
@@ -19,6 +19,7 @@
 #include "base/basictypes.h"
 #include "gfxPlatform.h"
 #include "gfxTextRun.h"
+#include "mozilla/nsZenBoostsBackend.h"
 #include "mozilla/AnimationEventDispatcher.h"
 #include "mozilla/ContentBlockingAllowList.h"
 #include "mozilla/CycleCollectedJSContext.h"
@@ -913,6 +914,11 @@ void nsPresContext::RecomputeBrowsingContextDependentData() {
   }
 
   auto* top = browsingContext->Top();
+
+  if (auto zenBackend = zen::nsZenBoostsBackend::GetInstance()) {
+    zenBackend->RecomputeBrowsingContextDependentData(this, top);
+  }
+
   SetColorSchemeOverride([&] {
     auto overriden = top->PrefersColorSchemeOverride();
     if (browsingContext == top &&
