diff --git a/layout/base/nsPresContext.cpp b/layout/base/nsPresContext.cpp
index e7e242d41f2584a5e9238b9a594e9fd9ca25882b..d5a0d017e41c1b7f87ec3675b05a99d3077277d5 100644
--- a/layout/base/nsPresContext.cpp
+++ b/layout/base/nsPresContext.cpp
@@ -19,6 +19,7 @@
 #include "base/basictypes.h"
 #include "gfxPlatform.h"
 #include "gfxTextRun.h"
+#include "mozilla/nsZenBoostsBackend.h"
 #include "mozilla/AnimationEventDispatcher.h"
 #include "mozilla/ContentBlockingAllowList.h"
 #include "mozilla/CycleCollectedJSContext.h"
@@ -917,6 +918,11 @@ void nsPresContext::RecomputeBrowsingContextDependentData() {
   }
 
   auto* top = browsingContext->Top();
+
+  if (auto zenBackend = zen::nsZenBoostsBackend::GetInstance()) {
+    zenBackend->RecomputeBrowsingContextDependentData(this, top);
+  }
+
   SetColorSchemeOverride([&] {
     auto overriden = top->PrefersColorSchemeOverride();
     if (overriden != PrefersColorSchemeOverride::None) {
