diff --git a/layout/base/PresShell.cpp b/layout/base/PresShell.cpp
index 9d651a17a91d2975017f5e21f1f533c13cc17659..ebc173c4fb6208eb771118f8edd40a4383364d55 100644
--- a/layout/base/PresShell.cpp
+++ b/layout/base/PresShell.cpp
@@ -137,6 +137,7 @@
 #include "mozilla/layers/WebRenderLayerManager.h"
 #include "mozilla/layers/WebRenderUserData.h"
 #include "mozilla/layout/ScrollAnchorContainer.h"
+#include "mozilla/nsZenBoostsBackend.h"
 #include "nsAnimationManager.h"
 #include "nsAutoLayoutPhase.h"
 #include "nsCOMArray.h"
@@ -5614,7 +5615,8 @@ nscolor PresShell::GetDefaultBackgroundColorToDraw() const {
   if (!mPresContext) {
     return NS_RGB(255, 255, 255);
   }
-  return mPresContext->DefaultBackgroundColor();
+  return zen::nsZenBoostsBackend::ResolveStyleColor(
+      mPresContext->DefaultBackgroundColor(), GetRootFrame());
 }
 
 void PresShell::UpdateCanvasBackground() {
