diff --git a/browser/base/jar.mn b/browser/base/jar.mn
index 7eaca95f7e876809552da5a3af3449b7e7b1a34d..00f1e3047c59565c8830a40da3a85f3b228f7c93 100644
--- a/browser/base/jar.mn
+++ b/browser/base/jar.mn
@@ -110,3 +110,5 @@ browser.jar:
 
 # L10n resources and overrides.
 % override chrome://global/locale/appstrings.properties chrome://browser/locale/appstrings.properties
+
+#include content/zen-assets.jar.inc.mn
