diff --git a/browser/base/jar.mn b/browser/base/jar.mn
index b1f7121a1f3926fe3eb89540edb7dcbb68734039..61c98964cf5efd0a603500cf36559713e1f231f7 100644
--- a/browser/base/jar.mn
+++ b/browser/base/jar.mn
@@ -109,3 +109,5 @@ browser.jar:
 
 # L10n resources and overrides.
 % override chrome://global/locale/appstrings.properties chrome://browser/locale/appstrings.properties
+
+#include content/zen-assets.jar.inc.mn
