diff --git a/browser/base/jar.mn b/browser/base/jar.mn
index f5a94b47c95304399bce8159dd9fb5d2911ea60a..f63cb861e8807294d597a341ddb408f236ee2df2 100644
--- a/browser/base/jar.mn
+++ b/browser/base/jar.mn
@@ -106,3 +106,5 @@ browser.jar:
 
 # L10n resources and overrides.
 % override chrome://global/locale/appstrings.properties chrome://browser/locale/appstrings.properties
+
+#include content/zen-assets.jar.inc.mn
