diff --git a/browser/base/jar.mn b/browser/base/jar.mn
index c724d243dc62a039d75aaee25cc956193589d9dc..04819ec5e17bd2023dbfb3e9dd812f832d9a7e92 100644
--- a/browser/base/jar.mn
+++ b/browser/base/jar.mn
@@ -103,3 +103,6 @@ browser.jar:
 
 # L10n resources and overrides.
 % override chrome://global/locale/appstrings.properties chrome://browser/locale/appstrings.properties
+
+#include content/zen-avatars/jar.inc.mn
+#include content/zen-assets.jar.inc.mn
\ No newline at end of file
