diff --git a/browser/base/moz.build b/browser/base/moz.build
index 803f8a5b62608af58fc43bd9fab298a63fa549a4..2e00cb9d02dade5cd9b8d3dea5d948ac6e829c26 100644
--- a/browser/base/moz.build
+++ b/browser/base/moz.build
@@ -82,3 +82,5 @@ DEFINES["MOZ_APP_VERSION_DISPLAY"] = CONFIG["MOZ_APP_VERSION_DISPLAY"]
 DEFINES["APP_LICENSE_BLOCK"] = "%s/content/overrides/app-license.html" % SRCDIR
 
 JAR_MANIFESTS += ["jar.mn"]
+
+DIRS += ["../../zen"]
