diff --git a/servo/components/style/values/generics/border.rs b/servo/components/style/values/generics/border.rs
--- a/servo/components/style/values/generics/border.rs
+++ b/servo/components/style/values/generics/border.rs
@@ -296,9 +296,9 @@
     pub fn all(s: S) -> Self {
         Self {
             top_left: s.clone(),
             top_right: s.clone(),
             bottom_right: s.clone(),
-            bottom_left: s.clone(),
+            bottom_left: s,
         }
     }
 }

