diff --git a/gfx/wr/swgl/src/glsl.h b/gfx/wr/swgl/src/glsl.h
--- a/gfx/wr/swgl/src/glsl.h
+++ b/gfx/wr/swgl/src/glsl.h
@@ -1599,10 +1599,17 @@
     x += a.x;
     y += a.y;
     z += a.z;
     return *this;
   }
+
+  vec3& operator*=(Float a) {
+    x *= a;
+    y *= a;
+    z *= a;
+    return *this;
+  }
 };
 
 vec3_scalar force_scalar(const vec3& v) {
   return vec3_scalar{force_scalar(v.x), force_scalar(v.y), force_scalar(v.z)};
 }

