diff --git a/gfx/wr/swgl/src/glsl.h b/gfx/wr/swgl/src/glsl.h
--- a/gfx/wr/swgl/src/glsl.h
+++ b/gfx/wr/swgl/src/glsl.h
@@ -469,10 +469,15 @@
   vec2_scalar_ref& operator=(const vec2_scalar& a) {
     x = a.x;
     y = a.y;
     return *this;
   }
+  vec2_scalar_ref& operator+=(vec2_scalar a) {
+    x += a.x;
+    y += a.y;
+    return *this;
+  }
   vec2_scalar_ref& operator*=(vec2_scalar a) {
     x *= a.x;
     y *= a.y;
     return *this;
   }

