diff --git a/gfx/wr/webrender/res/border_shared.glsl b/gfx/wr/webrender/res/border_shared.glsl
new file mode 100644
--- /dev/null
+++ b/gfx/wr/webrender/res/border_shared.glsl
@@ -0,0 +1,47 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include gpu_buffer
+
+#define SEGMENT_TOP_LEFT        0
+#define SEGMENT_TOP_RIGHT       1
+#define SEGMENT_BOTTOM_RIGHT    2
+#define SEGMENT_BOTTOM_LEFT     3
+#define SEGMENT_LEFT            4
+#define SEGMENT_TOP             5
+#define SEGMENT_RIGHT           6
+#define SEGMENT_BOTTOM          7
+
+#ifdef WR_VERTEX_SHADER
+
+PER_INSTANCE in vec2 aTaskOrigin;
+PER_INSTANCE in int aFlags;
+PER_INSTANCE in int aGpuDataAddress;
+PER_INSTANCE in vec4 aClipParams1;
+PER_INSTANCE in vec4 aClipParams2;
+
+struct BorderInstanceGpuData {
+    vec4 rect;
+    vec4 color0;
+    vec4 color1;
+    vec2 widths;
+    vec2 radii;
+    float shape;
+};
+
+BorderInstanceGpuData fetch_gpu_data(int index) {
+    BorderInstanceGpuData data;
+
+    vec4 texels[5] = fetch_from_gpu_buffer_5f(index);
+    data.rect = texels[0];
+    data.color0 = texels[1];
+    data.color1 = texels[2];
+    data.widths = texels[3].xy;
+    data.radii = texels[3].zw;
+    data.shape = texels[4].x;
+
+    return data;
+}
+
+#endif
diff --git a/gfx/wr/webrender/res/cs_border_segment.glsl b/gfx/wr/webrender/res/cs_border_segment.glsl
--- a/gfx/wr/webrender/res/cs_border_segment.glsl
+++ b/gfx/wr/webrender/res/cs_border_segment.glsl
@@ -1,10 +1,10 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#include shared,rect,ellipse
+#include shared,rect,border_shared,ellipse
 
 // For edges, the colors are the same. For corners, these
 // are the colors of each edge making up the corner.
 flat varying mediump vec4 vColor00;
 flat varying mediump vec4 vColor01;
@@ -44,19 +44,10 @@
 flat varying mediump vec4 vClipParams2;
 
 // Local space position
 varying highp vec2 vPos;
 
-#define SEGMENT_TOP_LEFT        0
-#define SEGMENT_TOP_RIGHT       1
-#define SEGMENT_BOTTOM_RIGHT    2
-#define SEGMENT_BOTTOM_LEFT     3
-#define SEGMENT_LEFT            4
-#define SEGMENT_TOP             5
-#define SEGMENT_RIGHT           6
-#define SEGMENT_BOTTOM          7
-
 // Border styles as defined in webrender_api/types.rs
 #define BORDER_STYLE_NONE         0
 #define BORDER_STYLE_SOLID        1
 #define BORDER_STYLE_DOUBLE       2
 #define BORDER_STYLE_DOTTED       3
@@ -72,20 +63,10 @@
 #define CLIP_DASH_EDGE   2
 #define CLIP_DOT         3
 
 #ifdef WR_VERTEX_SHADER
 
-PER_INSTANCE in vec2 aTaskOrigin;
-PER_INSTANCE in vec4 aRect;
-PER_INSTANCE in vec4 aColor0;
-PER_INSTANCE in vec4 aColor1;
-PER_INSTANCE in int aFlags;
-PER_INSTANCE in vec2 aWidths;
-PER_INSTANCE in vec2 aRadii;
-PER_INSTANCE in vec4 aClipParams1;
-PER_INSTANCE in vec4 aClipParams2;
-
 vec2 get_outer_corner_scale(int segment) {
     vec2 p;
 
     switch (segment) {
         case SEGMENT_TOP_LEFT:
@@ -153,16 +134,18 @@
 
     return result;
 }
 
 void main(void) {
+    BorderInstanceGpuData data = fetch_gpu_data(aGpuDataAddress);
+
     int segment = aFlags & 0xff;
     int style0 = (aFlags >> 8) & 0xff;
     int style1 = (aFlags >> 16) & 0xff;
     int clip_mode = (aFlags >> 24) & 0x0f;
 
-    vec2 size = aRect.zw - aRect.xy;
+    vec2 size = data.rect.zw - data.rect.xy;
     vec2 outer_scale = get_outer_corner_scale(segment);
     vec2 outer = outer_scale * size;
     vec2 clip_sign = 1.0 - 2.0 * outer_scale;
 
     // Set some flags used by the FS to determine the
@@ -176,19 +159,19 @@
             edge_axis = ivec2(0, 1);
             edge_reference = outer;
             break;
         case SEGMENT_TOP_RIGHT:
             edge_axis = ivec2(1, 0);
-            edge_reference = vec2(outer.x - aWidths.x, outer.y);
+            edge_reference = vec2(outer.x - data.widths.x, outer.y);
             break;
         case SEGMENT_BOTTOM_RIGHT:
             edge_axis = ivec2(0, 1);
-            edge_reference = outer - aWidths;
+            edge_reference = outer - data.widths;
             break;
         case SEGMENT_BOTTOM_LEFT:
             edge_axis = ivec2(1, 0);
-            edge_reference = vec2(outer.x, outer.y - aWidths.y);
+            edge_reference = vec2(outer.x, outer.y - data.widths.y);
             break;
         case SEGMENT_TOP:
         case SEGMENT_BOTTOM:
             edge_axis = ivec2(1, 1);
             break;
@@ -199,23 +182,23 @@
     }
 
     vSegmentClipMode = vec2(float(segment), float(clip_mode));
     vStyleEdgeAxis = vec4(float(style0), float(style1), float(edge_axis.x), float(edge_axis.y));
 
-    vPartialWidths = vec4(aWidths / 3.0, aWidths / 2.0);
+    vPartialWidths = vec4(data.widths / 3.0, data.widths / 2.0);
     vPos = size * aPosition.xy;
 
-    vec4[2] color0 = get_colors_for_side(aColor0, style0);
+    vec4[2] color0 = get_colors_for_side(data.color0, style0);
     vColor00 = color0[0];
     vColor01 = color0[1];
-    vec4[2] color1 = get_colors_for_side(aColor1, style1);
+    vec4[2] color1 = get_colors_for_side(data.color1, style1);
     vColor10 = color1[0];
     vColor11 = color1[1];
-    vClipCenter_Sign = vec4(outer + clip_sign * aRadii, clip_sign);
-    vClipRadii = vec4(aRadii, max(aRadii - aWidths, 0.0));
-    vColorLine = vec4(outer, aWidths.y * -clip_sign.y, aWidths.x * clip_sign.x);
-    vEdgeReference = vec4(edge_reference, edge_reference + aWidths);
+    vClipCenter_Sign = vec4(outer + clip_sign * data.radii, clip_sign);
+    vClipRadii = vec4(data.radii, max(data.radii - data.widths, 0.0));
+    vColorLine = vec4(outer, data.widths.y * -clip_sign.y, data.widths.x * clip_sign.x);
+    vEdgeReference = vec4(edge_reference, edge_reference + data.widths);
     vClipParams1 = aClipParams1;
     vClipParams2 = aClipParams2;
 
     // For the case of dot and dash clips, optimize the number of pixels that
     // are hit to just include the dot itself.
@@ -234,17 +217,17 @@
         // This is a gross approximation which works out because dashes don't have
         // a strong curvature and we will overshoot by inflating the geometry by
         // this amount on each side (sqrt(2) * length(dash) would be enough and we
         // compute 2 * approx_length(dash)).
         float dash_length = length(aClipParams1.xy - aClipParams2.xy);
-        float width = max(aWidths.x, aWidths.y);
+        float width = max(data.widths.x, data.widths.y);
         // expand by a small amout for AA just like we do for dots.
         vec2 r = vec2(max(dash_length, width)) + 2.0;
         vPos = clamp(vPos, center - r, center + r);
     }
 
-    gl_Position = uTransform * vec4(aTaskOrigin + aRect.xy + vPos, 0.0, 1.0);
+    gl_Position = uTransform * vec4(aTaskOrigin + data.rect.xy + vPos, 0.0, 1.0);
 }
 #endif
 
 #ifdef WR_FRAGMENT_SHADER
 vec4 evaluate_color_for_style_in_corner(
diff --git a/gfx/wr/webrender/res/cs_border_solid.glsl b/gfx/wr/webrender/res/cs_border_solid.glsl
--- a/gfx/wr/webrender/res/cs_border_solid.glsl
+++ b/gfx/wr/webrender/res/cs_border_solid.glsl
@@ -1,10 +1,10 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#include shared,rect,ellipse
+#include shared,rect,border_shared,ellipse
 
 #define DONT_MIX 0
 #define MIX_AA 1
 #define MIX_NO_AA 2
 
@@ -37,27 +37,12 @@
 flat varying highp vec2 vVerticalClipRadii;
 
 // Local space position
 varying highp vec2 vPos;
 
-#define SEGMENT_TOP_LEFT        0
-#define SEGMENT_TOP_RIGHT       1
-#define SEGMENT_BOTTOM_RIGHT    2
-#define SEGMENT_BOTTOM_LEFT     3
-
 #ifdef WR_VERTEX_SHADER
 
-PER_INSTANCE in vec2 aTaskOrigin;
-PER_INSTANCE in vec4 aRect;
-PER_INSTANCE in vec4 aColor0;
-PER_INSTANCE in vec4 aColor1;
-PER_INSTANCE in int aFlags;
-PER_INSTANCE in vec2 aWidths;
-PER_INSTANCE in vec2 aRadii;
-PER_INSTANCE in vec4 aClipParams1;
-PER_INSTANCE in vec4 aClipParams2;
-
 vec2 get_outer_corner_scale(int segment) {
     vec2 p;
 
     switch (segment) {
         case SEGMENT_TOP_LEFT:
@@ -80,15 +65,17 @@
 
     return p;
 }
 
 void main(void) {
+    BorderInstanceGpuData data = fetch_gpu_data(aGpuDataAddress);
+
     int segment = aFlags & 0xff;
     bool do_aa = ((aFlags >> 24) & 0xf0) != 0;
 
     vec2 outer_scale = get_outer_corner_scale(segment);
-    vec2 size = aRect.zw - aRect.xy;
+    vec2 size = data.rect.zw - data.rect.xy;
     vec2 outer = outer_scale * size;
     vec2 clip_sign = 1.0 - 2.0 * outer_scale;
 
     int mix_colors;
     switch (segment) {
@@ -105,15 +92,15 @@
     }
 
     vMixColors.x = mix_colors;
     vPos = size * aPosition.xy;
 
-    vColor0 = aColor0;
-    vColor1 = aColor1;
-    vClipCenter_Sign = vec4(outer + clip_sign * aRadii, clip_sign);
-    vClipRadii = vec4(aRadii, max(aRadii - aWidths, 0.0));
-    vColorLine = vec4(outer, aWidths.y * -clip_sign.y, aWidths.x * clip_sign.x);
+    vColor0 = data.color0;
+    vColor1 = data.color1;
+    vClipCenter_Sign = vec4(outer + clip_sign * data.radii, clip_sign);
+    vClipRadii = vec4(data.radii, max(data.radii - data.widths, 0.0));
+    vColorLine = vec4(outer, data.widths.y * -clip_sign.y, data.widths.x * clip_sign.x);
 
     vec2 horizontal_clip_sign = vec2(-clip_sign.x, clip_sign.y);
     vHorizontalClipCenter_Sign = vec4(aClipParams1.xy +
                                       horizontal_clip_sign * aClipParams1.zw,
                                       horizontal_clip_sign);
@@ -123,11 +110,11 @@
     vVerticalClipCenter_Sign = vec4(aClipParams2.xy +
                                     vertical_clip_sign * aClipParams2.zw,
                                     vertical_clip_sign);
     vVerticalClipRadii = aClipParams2.zw;
 
-    gl_Position = uTransform * vec4(aTaskOrigin + aRect.xy + vPos, 0.0, 1.0);
+    gl_Position = uTransform * vec4(aTaskOrigin + data.rect.xy + vPos, 0.0, 1.0);
 }
 #endif
 
 #ifdef WR_FRAGMENT_SHADER
 void main(void) {
diff --git a/gfx/wr/webrender/src/border.rs b/gfx/wr/webrender/src/border.rs
--- a/gfx/wr/webrender/src/border.rs
+++ b/gfx/wr/webrender/src/border.rs
@@ -6,13 +6,14 @@
 use api::{NormalBorder as ApiNormalBorder, RepeatMode};
 use api::units::*;
 use crate::clip::ClipNodeId;
 use crate::ellipse::Ellipse;
 use euclid::vec2;
+use crate::renderer::GpuBufferBuilderF;
 use crate::scene_building::SceneBuilder;
 use crate::spatial_tree::SpatialNodeIndex;
-use crate::gpu_types::{BorderInstance, BorderSegment, BrushFlags};
+use crate::gpu_types::{BorderInstance, BorderInstanceGpuData, BorderSegment, BrushFlags};
 use crate::prim_store::{BorderSegmentInfo, BrushSegment, NinePatchDescriptor};
 use crate::prim_store::borders::NormalBorderPrim;
 use crate::util::{lerp, RectHelpers};
 use crate::internal_types::LayoutPrimitiveInfo;
 use crate::segment::EdgeMask;
@@ -125,10 +126,11 @@
 #[cfg_attr(feature = "capture", derive(Serialize))]
 #[cfg_attr(feature = "replay", derive(Deserialize))]
 pub struct BorderSegmentCacheKey {
     pub size: LayoutSizeAu,
     pub radius: LayoutSizeAu,
+    pub shape: u32,
     pub side0: BorderSideAu,
     pub side1: BorderSideAu,
     pub segment: BorderSegment,
     pub do_aa: bool,
     pub h_adjacent_corner_outer: LayoutPointAu,
@@ -762,10 +764,11 @@
         ),
         border.left,
         border.top,
         LayoutSize::new(widths.left, widths.top),
         border.radius.top_left,
+        border.radius.shape_top_left,
         BorderSegment::TopLeft,
         EdgeMask::TOP | EdgeMask::LEFT,
         rect.top_right(),
         border.radius.top_right,
         rect.bottom_left(),
@@ -789,10 +792,11 @@
         ),
         border.top,
         border.right,
         LayoutSize::new(widths.right, widths.top),
         border.radius.top_right,
+        border.radius.shape_top_right,
         BorderSegment::TopRight,
         EdgeMask::TOP | EdgeMask::RIGHT,
         rect.min,
         border.radius.top_left,
         rect.max,
@@ -816,10 +820,11 @@
         ),
         border.right,
         border.bottom,
         LayoutSize::new(widths.right, widths.bottom),
         border.radius.bottom_right,
+        border.radius.shape_bottom_right,
         BorderSegment::BottomRight,
         EdgeMask::BOTTOM | EdgeMask::RIGHT,
         rect.bottom_left(),
         border.radius.bottom_left,
         rect.top_right(),
@@ -843,10 +848,11 @@
         ),
         border.bottom,
         border.left,
         LayoutSize::new(widths.left, widths.bottom),
         border.radius.bottom_left,
+        border.radius.shape_bottom_left,
         BorderSegment::BottomLeft,
         EdgeMask::BOTTOM | EdgeMask::LEFT,
         rect.max,
         border.radius.bottom_right,
         rect.min,
@@ -881,30 +887,37 @@
     color1: ColorF,
     segment: BorderSegment,
     instances: &mut Vec<BorderInstance>,
     widths: DeviceSize,
     radius: DeviceSize,
+    shape: f32,
     do_aa: bool,
     h_adjacent_corner_outer: DevicePoint,
     h_adjacent_corner_radius: DeviceSize,
     v_adjacent_corner_outer: DevicePoint,
     v_adjacent_corner_radius: DeviceSize,
+    gpu_buffer_builder: &mut GpuBufferBuilderF,
 ) {
     let base_flags = (segment as i32) |
                      ((style0 as i32) << 8) |
                      ((style1 as i32) << 16) |
                      ((do_aa as i32) << 28);
 
-    let base_instance = BorderInstance {
-        task_origin: DevicePoint::zero(),
+    let instance_gpu_data = BorderInstanceGpuData {
         local_rect: task_rect,
-        flags: base_flags,
         color0: color0.premultiplied(),
         color1: color1.premultiplied(),
         widths,
         radius,
+        shape
+    };
+
+    let base_instance = BorderInstance {
+        task_origin: DevicePoint::zero(),
+        flags: base_flags,
         clip_params: [0.0; 8],
+        gpu_data_address: instance_gpu_data.write(gpu_buffer_builder)
     };
 
     match segment {
         BorderSegment::TopLeft |
         BorderSegment::TopRight |
@@ -1018,10 +1031,11 @@
     non_overlapping_rect: LayoutRect,
     side0: BorderSide,
     side1: BorderSide,
     widths: LayoutSize,
     radius: LayoutSize,
+    shape: f32,
     segment: BorderSegment,
     edge_flags: EdgeMask,
     h_adjacent_corner_outer: LayoutPoint,
     h_adjacent_corner_radius: LayoutSize,
     v_adjacent_corner_outer: LayoutPoint,
@@ -1137,10 +1151,11 @@
             do_aa,
             side0: side0.into(),
             side1: side1.into(),
             segment,
             radius: radius.to_au(),
+            shape: shape.to_bits(),
             size: widths.to_au(),
             h_adjacent_corner_outer: (h_corner_outer - image_rect.min).to_point().to_au(),
             h_adjacent_corner_radius: h_corner_radius.to_au(),
             v_adjacent_corner_outer: (v_corner_outer - image_rect.min).to_point().to_au(),
             v_adjacent_corner_radius: v_corner_radius.to_au(),
@@ -1200,10 +1215,11 @@
         cache_key: BorderSegmentCacheKey {
             do_aa,
             side0: side.into(),
             side1: side.into(),
             radius: LayoutSizeAu::zero(),
+            shape: 0,
             size: size.to_au(),
             segment,
             h_adjacent_corner_outer: LayoutPointAu::zero(),
             h_adjacent_corner_radius: LayoutSizeAu::zero(),
             v_adjacent_corner_outer: LayoutPointAu::zero(),
@@ -1217,10 +1233,11 @@
 pub fn build_border_instances(
     cache_key: &BorderSegmentCacheKey,
     cache_size: DeviceIntSize,
     border: &ApiNormalBorder,
     scale: LayoutToDeviceScale,
+    gpu_buffer_builder: &mut GpuBufferBuilderF,
 ) -> Vec<BorderInstance> {
     let mut instances = Vec::new();
 
     let (side0, side1, flip0, flip1) = match cache_key.segment {
         BorderSegment::Left => (&border.left, &border.left, false, false),
@@ -1247,10 +1264,11 @@
     let color0 = side0.border_color(flip0);
     let color1 = side1.border_color(flip1);
 
     let widths = (LayoutSize::from_au(cache_key.size) * scale).ceil();
     let radius = (LayoutSize::from_au(cache_key.radius) * scale).ceil();
+    let shape = f32::from_bits(cache_key.shape);
 
     let h_corner_outer = (LayoutPoint::from_au(cache_key.h_adjacent_corner_outer) * scale).round();
     let h_corner_radius = (LayoutSize::from_au(cache_key.h_adjacent_corner_radius) * scale).ceil();
     let v_corner_outer = (LayoutPoint::from_au(cache_key.v_adjacent_corner_outer) * scale).round();
     let v_corner_radius = (LayoutSize::from_au(cache_key.v_adjacent_corner_radius) * scale).ceil();
@@ -1263,15 +1281,17 @@
         color1,
         cache_key.segment,
         &mut instances,
         widths,
         radius,
+        shape,
         border.do_aa,
         h_corner_outer,
         h_corner_radius,
         v_corner_outer,
         v_corner_radius,
+        gpu_buffer_builder,
     );
 
     instances
 }
 
diff --git a/gfx/wr/webrender/src/gpu_types.rs b/gfx/wr/webrender/src/gpu_types.rs
--- a/gfx/wr/webrender/src/gpu_types.rs
+++ b/gfx/wr/webrender/src/gpu_types.rs
@@ -192,22 +192,40 @@
     Top,
     Right,
     Bottom,
 }
 
+pub struct BorderInstanceGpuData {
+    pub local_rect: DeviceRect,
+    pub color0: PremultipliedColorF,
+    pub color1: PremultipliedColorF,
+    pub widths: DeviceSize,
+    pub radius: DeviceSize,
+    pub shape: f32,
+}
+
+impl BorderInstanceGpuData {
+    pub fn write(&self, gpu_buffer_builder: &mut GpuBufferBuilderF) -> GpuBufferAddress {
+        let mut writer = gpu_buffer_builder.write_blocks(5);
+        writer.push_one(self.local_rect);
+        writer.push_one(self.color0);
+        writer.push_one(self.color1);
+        writer.push_one([self.widths.width, self.widths.height, self.radius.width, self.radius.height]);
+        writer.push_one([self.shape, 0.0, 0.0, 0.0]);
+
+        writer.finish()
+    }
+}
+
 #[derive(Debug, Clone)]
 #[repr(C)]
 #[cfg_attr(feature = "capture", derive(Serialize))]
 #[cfg_attr(feature = "replay", derive(Deserialize))]
 pub struct BorderInstance {
     pub task_origin: DevicePoint,
-    pub local_rect: DeviceRect,
-    pub color0: PremultipliedColorF,
-    pub color1: PremultipliedColorF,
     pub flags: i32,
-    pub widths: DeviceSize,
-    pub radius: DeviceSize,
+    pub gpu_data_address: GpuBufferAddress,
     pub clip_params: [f32; 8],
 }
 
 #[derive(Copy, Clone, Debug)]
 #[cfg_attr(feature = "capture", derive(Serialize))]
diff --git a/gfx/wr/webrender/src/prim_store/borders.rs b/gfx/wr/webrender/src/prim_store/borders.rs
--- a/gfx/wr/webrender/src/prim_store/borders.rs
+++ b/gfx/wr/webrender/src/prim_store/borders.rs
@@ -230,19 +230,20 @@
                 false,          // TODO(gw): We don't calculate opacity for borders yet!
                 RenderTaskParent::Surface,
                 &mut frame_state.frame_gpu_data.f32,
                 frame_state.rg_builder,
                 &mut frame_state.surface_builder,
-                &mut |rg_builder, _| {
+                &mut |rg_builder, gpu_buffer_builder| {
                     rg_builder.add().init(RenderTask::new_dynamic(
                         cache_size,
                         RenderTaskKind::new_border_segment(
                             build_border_instances(
                                 &segment.cache_key,
                                 cache_size,
                                 &self.border,
                                 scale,
+                                gpu_buffer_builder,
                             )
                         ),
                     ))
                 }
             );
diff --git a/gfx/wr/webrender/src/renderer/vertex.rs b/gfx/wr/webrender/src/renderer/vertex.rs
--- a/gfx/wr/webrender/src/renderer/vertex.rs
+++ b/gfx/wr/webrender/src/renderer/vertex.rs
@@ -63,16 +63,12 @@
 
     pub const BORDER: VertexDescriptor = VertexDescriptor {
         vertex_attributes: &[VertexAttribute::quad_instance_vertex()],
         instance_attributes: &[
             VertexAttribute::f32x2("aTaskOrigin"),
-            VertexAttribute::f32x4("aRect"),
-            VertexAttribute::f32x4("aColor0"),
-            VertexAttribute::f32x4("aColor1"),
             VertexAttribute::i32("aFlags"),
-            VertexAttribute::f32x2("aWidths"),
-            VertexAttribute::f32x2("aRadii"),
+            VertexAttribute::gpu_buffer_address("aGpuDataAddress"),
             VertexAttribute::f32x4("aClipParams1"),
             VertexAttribute::f32x4("aClipParams2"),
         ],
     };
 

