diff --git a/gfx/wr/swgl/src/glsl.h b/gfx/wr/swgl/src/glsl.h
--- a/gfx/wr/swgl/src/glsl.h
+++ b/gfx/wr/swgl/src/glsl.h
@@ -1611,10 +1611,14 @@
 
 vec3_scalar make_vec3(const vec2_scalar& v, float z) {
   return vec3_scalar{v.x, v.y, z};
 }
 
+vec3_scalar make_vec3(float x, const vec2_scalar& v) {
+  return vec3_scalar{x, v.x, v.y};
+}
+
 vec3_scalar make_vec3(float x, float y, float z) {
   return vec3_scalar{x, y, z};
 }
 
 vec3_scalar make_vec3(int32_t x, int32_t y, float z) {

