diff --git a/browser/components/preferences/jar.mn b/browser/components/preferences/jar.mn
index aec7888f59eeb5dbffc62b861fe223159ac42586..a593f3015469c11c762ae01465964251a77ddb45 100644
--- a/browser/components/preferences/jar.mn
+++ b/browser/components/preferences/jar.mn
@@ -69,5 +69,7 @@ browser.jar:
    content/browser/preferences/widgets/sync-engines-list.css          (widgets/sync-engine-list/sync-engines-list.css)
    content/browser/preferences/widgets/update-information.mjs          (widgets/update-information/update-information.mjs)
    content/browser/preferences/widgets/update-information.css          (widgets/update-information/update-information.css)
+
+   content/browser/preferences/zen-settings.js
    content/browser/preferences/widgets/update-state.mjs          (widgets/update-state/update-state.mjs)
    content/browser/preferences/widgets/update-state.css          (widgets/update-state/update-state.css)
