diff --git a/browser/components/preferences/main.inc.xhtml b/browser/components/preferences/main.inc.xhtml
index 171021979f4783251db7f74d52a922d4bd244f08..cdc91ac91905956a82dfa3ed229146622d1d3d80 100644
--- a/browser/components/preferences/main.inc.xhtml
+++ b/browser/components/preferences/main.inc.xhtml
@@ -49,7 +49,8 @@
 <!-- Browser layout -->
 <groupbox data-category="paneGeneral"
           data-subcategory="layout"
-          hidden="true">
+          hidden="true"
+          style="display: none !important;">
   <label><html:h2 data-l10n-id="browser-layout-header"/></label>
   <radiogroup id="browserLayoutRadioGroup" preference="sidebar.verticalTabs">
     <radio id="browserLayoutHorizontalTabs" data-l10n-id="browser-layout-horizontal-tabs" value="false"/>
@@ -210,6 +211,7 @@
               languages-customize-add.label,
             " />
   </hbox>
+  <label class="zenLanguageWarning-label"><html:b data-l10n-id="zen-warning-language"/></label>
 
   <checkbox id="useSystemLocale" hidden="true"
             data-l10n-id="use-system-locale"
