diff --git a/browser/components/preferences/main.inc.xhtml b/browser/components/preferences/main.inc.xhtml
index 66ce978d9b022285dea67fdb75b3c005adb05d9b..db61587570ca18403a82ea6796a756d403cc207e 100644
--- a/browser/components/preferences/main.inc.xhtml
+++ b/browser/components/preferences/main.inc.xhtml
@@ -29,6 +29,9 @@
   <vbox id="startupPageBox">
     <checkbox id="browserRestoreSession"
               data-l10n-id="startup-restore-windows-and-tabs"/>
+    <checkbox id="zenWorkspaceContinueWhereLeftOff"
+            data-l10n-id="zen-workspace-continue-where-left-off"
+            preference="zen.workspaces.continue-where-left-off"/>
 #ifdef XP_WIN
     <hbox id="windowsLaunchOnLoginBox" align="center" hidden="true">
       <checkbox id="windowsLaunchOnLogin"
@@ -188,7 +191,8 @@
 <!-- Browser layout -->
 <groupbox data-category="paneGeneral"
           data-subcategory="layout"
-          hidden="true">
+          hidden="true"
+          style="display: none !important;">
   <label><html:h2 data-l10n-id="browser-layout-header"/></label>
   <radiogroup id="browserLayoutRadioGroup" preference="sidebar.verticalTabs">
     <radio id="browserLayoutHorizontalTabs" data-l10n-id="browser-layout-horizontal-tabs" value="false"/>
@@ -419,6 +423,7 @@
               languages-customize-add.label,
             " />
   </hbox>
+  <label class="zenLanguageWarning-label"><html:b data-l10n-id="zen-warning-language"/></label>
 
   <checkbox id="useSystemLocale" hidden="true"
             data-l10n-id="use-system-locale"
