diff --git a/browser/components/preferences/main.js b/browser/components/preferences/main.js
index 6f99cbf973d414ffd4036d17740abe963629db37..520dcce0adb06c5ebfee8c76f72f7d7b40e106da 100644
--- a/browser/components/preferences/main.js
+++ b/browser/components/preferences/main.js
@@ -1729,6 +1729,11 @@ SettingGroupManager.registerGroups({
         id: "browserRestoreSession",
         l10nId: "startup-restore-windows-and-tabs",
       },
+      {
+        id: "zenWorkspaceContinueWhereLeftOff",
+        l10nId: "zen-workspace-continue-where-left-off",
+        control: "moz-checkbox",
+      },
       {
         id: "windowsLaunchOnLogin",
         l10nId: "windows-launch-on-login",
@@ -3133,7 +3138,7 @@ function getBundleForLocales(newLocales) {
     ])
   );
   return new Localization(
-    ["browser/preferences/preferences.ftl", "branding/brand.ftl"],
+    ["browser/preferences/preferences.ftl", "branding/brand.ftl", "browser/preferences/zen-preferences.ftl"],
     false,
     undefined,
     locales
