diff --git a/browser/components/preferences/main.inc.xhtml b/browser/components/preferences/main.inc.xhtml
index 891cf7108cf4606c99a902afa420e744b9690b87..ceda60d9f2c2d26d95606d80303cc32ab8c5160d 100644
--- a/browser/components/preferences/main.inc.xhtml
+++ b/browser/components/preferences/main.inc.xhtml
@@ -29,6 +29,11 @@
   <vbox id="startupPageBox">
     <checkbox id="browserRestoreSession"
               data-l10n-id="startup-restore-windows-and-tabs"/>
+    <vbox class="indent">
+      <checkbox id="zenWorkspaceContinueWhereLeftOff"
+              data-l10n-id="zen-workspace-continue-where-left-off"
+              preference="zen.workspaces.continue-where-left-off"/>
+    </vbox>
 #ifdef XP_WIN
     <hbox id="windowsLaunchOnLoginBox" align="center" hidden="true">
       <checkbox id="windowsLaunchOnLogin"
@@ -125,6 +130,14 @@
     <checkbox id="ctrlTabRecentlyUsedOrder" data-l10n-id="ctrl-tab-recently-used-order"
               preference="browser.ctrlTab.sortByRecentlyUsed"/>
 
+    <checkbox id="ctrlTabRestrictToSection" data-l10n-id="ctrl-tab-restrict-to-section"
+              preference="browser.ctrlTab.restrictToSection"/>
+
+    <description id="ctrlTabRestrictInfo"
+                 class="indent description-deemphasized"
+                 data-l10n-id="ctrl-tab-restrict-warning"
+                 hidden="true"/>
+
     <checkbox id="linkTargeting" data-l10n-id="open-new-link-as-tabs"
               preference="browser.link.open_newwindow"/>
 
@@ -189,7 +202,8 @@
 <!-- Browser layout -->
 <groupbox data-category="paneGeneral"
           data-subcategory="layout"
-          hidden="true">
+          hidden="true"
+          style="display: none !important;">
   <label><html:h2 data-l10n-id="browser-layout-header"/></label>
   <radiogroup id="browserLayoutRadioGroup" preference="sidebar.verticalTabs">
     <radio id="browserLayoutHorizontalTabs" data-l10n-id="browser-layout-horizontal-tabs" value="false"/>
@@ -420,6 +434,7 @@
               languages-customize-add.label,
             " />
   </hbox>
+  <label class="zenLanguageWarning-label"><html:b data-l10n-id="zen-warning-language"/></label>
 
   <checkbox id="useSystemLocale" hidden="true"
             data-l10n-id="use-system-locale"
