diff --git a/browser/components/preferences/main.js b/browser/components/preferences/main.js
index 8f8da3c4cabb4a05b50431b1feeff8f1f1abcf6e..97c651cb3728353b9a0d09d1aaa9ba87f8c05b57 100644
--- a/browser/components/preferences/main.js
+++ b/browser/components/preferences/main.js
@@ -2658,6 +2658,11 @@ function createStartupConfig(hidden = false) {
         id: "browserRestoreSession",
         l10nId: "startup-restore-windows-and-tabs",
       },
+      {
+        id: "zenWorkspaceContinueWhereLeftOff",
+        l10nId: "zen-workspace-continue-where-left-off",
+        control: "moz-checkbox",
+      },
       {
         id: "windowsLaunchOnLogin",
         l10nId: "windows-launch-on-login",
@@ -5174,7 +5179,7 @@ function getBundleForLocales(newLocales) {
     ])
   );
   return new Localization(
-    ["browser/preferences/preferences.ftl", "branding/brand.ftl"],
+    ["browser/preferences/preferences.ftl", "branding/brand.ftl", "browser/preferences/zen-preferences.ftl"],
     false,
     undefined,
     locales
