diff --git a/browser/components/preferences/preferences.js b/browser/components/preferences/preferences.js
index c76a0647e14bac806d9428483bb565b3e10c31e7..2188f7808aca090f92574c7f5c55a28a9c5bb3f8 100644
--- a/browser/components/preferences/preferences.js
+++ b/browser/components/preferences/preferences.js
@@ -131,6 +131,7 @@ ChromeUtils.defineLazyGetter(this, "gSubDialog", function () {
       styleSheets: [
         "chrome://browser/skin/preferences/dialog.css",
         "chrome://browser/skin/preferences/preferences.css",
+        "chrome://browser/skin/preferences/zen-preferences.css",
       ],
       resizeCallback: async ({ title, frame }) => {
         // Search within main document and highlight matched keyword.
@@ -437,6 +438,8 @@ const CONFIG_PANES = Object.freeze({
   tabsBrowsing: {
     l10nId: "tabs-browsing-section",
     groupIds: [
+      "defaultBrowserHome",
+      "startupHome",
       "browserLayout",
       "tabs",
       "pageNavigation",
@@ -483,7 +486,7 @@ function register_module(categoryName, categoryObject) {
       }
       this._initted = true;
       let template = document.getElementById("template-" + categoryName);
-      if (template && !srdSectionPrefs.all) {
+      if (template && (!srdSectionPrefs.all || categoryName.startsWith("paneZen"))) {
         // Replace the template element with the nodes inside of it.
         template.replaceWith(template.content);
 
@@ -527,6 +530,10 @@ function init_all() {
   register_module("paneHome", gHomePane);
   register_module("paneSearch", gSearchPane);
   register_module("panePrivacy", gPrivacyPane);
+  register_module("paneZenLooks", gZenLooksAndFeel);
+  register_module("paneZenTabManagement", gZenWorkspacesSettings);
+  register_module("paneZenCKS", gZenCKSSettings);
+  register_module("paneZenMarketplace", gZenMarketplaceManager);
 
   // Restore the cached Firefox Labs nav button visibility so it shows
   // immediately when recipes are expected to be available, before
@@ -656,9 +663,9 @@ async function gotoPref(
   let redesignEnabled = srdSectionPrefs.all;
   let categories = document.getElementById("categories");
   const kDefaultCategoryInternalName = redesignEnabled
-    ? "paneSync"
+    ? "paneTabsBrowsing"
     : "paneGeneral";
-  const kDefaultCategory = redesignEnabled ? "sync" : "general";
+  const kDefaultCategory = redesignEnabled ? "tabsBrowsing" : "general";
   let hash = document.location.hash;
   let category = aCategory || hash.substring(1) || kDefaultCategoryInternalName;
 
