diff --git a/browser/components/preferences/preferences.xhtml b/browser/components/preferences/preferences.xhtml
index eee227822a772bec128d6adbe4f6aca90ed810f6..54d0f2df0c422a7d8ae0fbe10ef6c072bd7d6453 100644
--- a/browser/components/preferences/preferences.xhtml
+++ b/browser/components/preferences/preferences.xhtml
@@ -43,6 +43,8 @@
   />
   <link rel="stylesheet" href="chrome://browser/skin/preferences/privacy.css" />
 
+#include zen-preferences-links.xhtml
+
   <link rel="localization" href="branding/brand.ftl"/>
   <link rel="localization" href="browser/browser.ftl"/>
     <!-- Used by fontbuilder.js -->
@@ -112,6 +114,17 @@
           <label class="category-name" flex="1" data-l10n-id="pane-general-title"></label>
         </richlistitem>
 
+        <richlistitem id="category-zen-looks"
+                      class="category"
+                      value="paneZenLooks"
+                      helpTopic="prefs-main"
+                      data-l10n-id="category-zen-looks"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-zen-looks-title"></label>
+        </richlistitem>
+
         <richlistitem id="category-home"
                       class="category"
                       value="paneHome"
@@ -248,6 +261,7 @@
 #include sync.inc.xhtml
 #include experimental.inc.xhtml
 #include moreFromMozilla.inc.xhtml
+#include zenLooksAndFeel.inc.xhtml
         </vbox>
       </vbox>
     </vbox>
