diff --git a/browser/components/preferences/preferences.xhtml b/browser/components/preferences/preferences.xhtml
index 3c4b3225c672bcb172e7f946bc5f98d424e21e84..d10824470c564b6631333f197de897c21ccc19f5 100644
--- a/browser/components/preferences/preferences.xhtml
+++ b/browser/components/preferences/preferences.xhtml
@@ -42,6 +42,8 @@
   <link rel="stylesheet" href="chrome://browser/content/preferences/widgets/setting-pane.css" />
   <link rel="stylesheet" href="chrome://browser/content/preferences/widgets/setting-control.css" />
 
+#include zen-preferences-links.xhtml
+
   <link rel="localization" href="branding/brand.ftl"/>
   <link rel="localization" href="browser/browser.ftl"/>
     <!-- Used by fontbuilder.js -->
@@ -126,6 +128,26 @@
         iconsrc="chrome://browser/skin/preferences/category-general.svg"
         data-l10n-id="pane-general-title">
       </html:moz-page-nav-button>
+      <html:moz-page-nav-button id="category-zen-looks"
+        view="paneZenLooks"
+        iconsrc="chrome://browser/skin/customize.svg"
+        data-l10n-id="pane-zen-looks-title">
+      </html:moz-page-nav-button>
+      <html:moz-page-nav-button id="category-zen-tabs-management"
+        view="paneZenTabManagement"
+        iconsrc="chrome://browser/skin/window.svg"
+        data-l10n-id="pane-zen-tabs-title">
+      </html:moz-page-nav-button>
+      <html:moz-page-nav-button id="category-zen-CKS"
+        view="paneZenCKS"
+        iconsrc="chrome://browser/skin/quickactions.svg"
+        data-l10n-id="pane-zen-CKS-title">
+      </html:moz-page-nav-button>
+      <html:moz-page-nav-button id="category-zen-marketplace"
+        view="paneZenMarketplace"
+        iconsrc="chrome://mozapps/skin/extensions/category-themes.svg"
+        data-l10n-id="pane-zen-marketplace-title">
+      </html:moz-page-nav-button>
       <html:moz-page-nav-button id="category-home"
         view="paneHome"
         iconsrc="chrome://browser/skin/home.svg"
@@ -218,6 +240,10 @@
 #include sync.inc.xhtml
 #include experimental.inc.xhtml
 #include moreFromMozilla.inc.xhtml
+#include zenLooksAndFeel.inc.xhtml
+#include zenKeyboardShortcuts.inc.xhtml
+#include zenTabsManagement.inc.xhtml
+#include zenMarketplace.inc.xhtml
         </vbox>
       </vbox>
     </vbox>
