diff --git a/browser/components/preferences/preferences.js b/browser/components/preferences/preferences.js
index b5af9af69fd715012af5c0613e0627dd9cf9c3b7..7f86bac8cce278c5b7b7e1230230b3a27b7cf49a 100644
--- a/browser/components/preferences/preferences.js
+++ b/browser/components/preferences/preferences.js
@@ -117,6 +117,7 @@ ChromeUtils.defineLazyGetter(this, "gSubDialog", function () {
       styleSheets: [
         "chrome://browser/skin/preferences/dialog.css",
         "chrome://browser/skin/preferences/preferences.css",
+        "chrome://browser/skin/preferences/zen-preferences.css",
       ],
       resizeCallback: async ({ title, frame }) => {
         // Search within main document and highlight matched keyword.
@@ -196,6 +197,10 @@ function init_all() {
   register_module("paneSearch", gSearchPane);
   register_module("panePrivacy", gPrivacyPane);
   register_module("paneContainers", gContainersPane);
+  register_module("paneZenLooks", gZenLooksAndFeel);
+  register_module("paneZenTabManagement", gZenWorkspacesSettings);
+  register_module("paneZenCKS", gZenCKSSettings);
+  register_module("paneZenMarketplace", gZenMarketplaceManager);
 
   if (Services.prefs.getBoolPref("browser.translations.newSettingsUI.enable")) {
     register_module("paneTranslations", gTranslationsPane);
