diff --git a/browser/components/preferences/preferences.xhtml b/browser/components/preferences/preferences.xhtml
index 3affb19d1a243da4a5782c9a4a5588d165b4119f..ccb56cc323ff5c3e94013de782af54911fc0c360 100644
--- a/browser/components/preferences/preferences.xhtml
+++ b/browser/components/preferences/preferences.xhtml
@@ -43,6 +43,8 @@
   />
   <link rel="stylesheet" href="chrome://browser/skin/preferences/privacy.css" />
 
+#include zen-preferences-links.xhtml
+
   <link rel="localization" href="branding/brand.ftl"/>
   <link rel="localization" href="browser/browser.ftl"/>
     <!-- Used by fontbuilder.js -->
@@ -97,6 +99,12 @@
   <hbox flex="1">
 
     <vbox class="navigation">
+      <search-textbox
+              id="searchInput"
+              data-l10n-id="search-input-box2"
+              data-l10n-attrs="placeholder, style"
+              hidden="true"/>
+
       <!-- category list -->
       <richlistbox id="categories" data-l10n-id="category-list" data-l10n-attrs="aria-label">
         <richlistitem id="category-general"
@@ -110,6 +118,28 @@
           <label class="category-name" flex="1" data-l10n-id="pane-general-title"></label>
         </richlistitem>
 
+        <richlistitem id="category-zen-looks"
+                      class="category"
+                      value="paneZenLooks"
+                      helpTopic="prefs-main"
+                      data-l10n-id="category-zen-looks"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-zen-looks-title"></label>
+        </richlistitem>
+
+        <richlistitem id="category-zen-workspaces"
+                      class="category"
+                      value="paneZenWorkspaces"
+                      helpTopic="prefs-main"
+                      data-l10n-id="category-zen-workspaces"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-zen-workspaces-title"></label>
+        </richlistitem>
+
         <richlistitem id="category-home"
                       class="category"
                       value="paneHome"
@@ -228,11 +258,13 @@
                 <html:a href="about:policies" target="_blank" data-l10n-id="managed-notice"/>
               </hbox>
             </hbox>
+#if 0
             <search-textbox
               id="searchInput"
               data-l10n-id="search-input-box2"
               data-l10n-attrs="placeholder, style"
               hidden="true"/>
+#endif
           </hbox>
         </hbox>
         <vbox id="mainPrefPane">
@@ -246,6 +278,8 @@
 #include sync.inc.xhtml
 #include experimental.inc.xhtml
 #include moreFromMozilla.inc.xhtml
+#include zenLooksAndFeel.inc.xhtml
+#include zenWorkspaces.inc.xhtml
         </vbox>
       </vbox>
     </vbox>
