diff --git a/browser/components/preferences/preferences.xhtml b/browser/components/preferences/preferences.xhtml
index 21d951a19df06da67a28f717b9f80f8f4ebf77d2..b1f998f2b3ed99b19666e81e61428587541b0da5 100644
--- a/browser/components/preferences/preferences.xhtml
+++ b/browser/components/preferences/preferences.xhtml
@@ -44,6 +44,8 @@
   <link rel="stylesheet" href="chrome://browser/skin/preferences/privacy.css" />
   <link rel="stylesheet" href="chrome://browser/content/preferences/widgets/setting-group.css" />
 
+#include zen-preferences-links.xhtml
+
   <link rel="localization" href="branding/brand.ftl"/>
   <link rel="localization" href="browser/browser.ftl"/>
     <!-- Used by fontbuilder.js -->
@@ -104,6 +106,11 @@
   <hbox flex="1">
 
     <vbox class="navigation">
+      <html:moz-input-search
+        id="searchInput"
+        data-l10n-id="search-input-box2"
+        data-l10n-attrs="placeholder, style"
+        />
       <!-- category list -->
       <richlistbox id="categories" data-l10n-id="category-list" data-l10n-attrs="aria-label">
         <richlistitem id="category-general"
@@ -116,6 +123,50 @@
           <label class="category-name" flex="1" data-l10n-id="pane-general-title"></label>
         </richlistitem>
 
+        <richlistitem id="category-zen-looks"
+                      class="category"
+                      value="paneZenLooks"
+                      helpTopic="prefs-main"
+                      data-l10n-id="category-zen-looks"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-zen-looks-title"></label>
+        </richlistitem>
+
+        <richlistitem id="category-zen-tabs-management"
+                      class="category"
+                      value="paneZenTabManagement"
+                      helpTopic="prefs-main"
+                      data-l10n-id="category-zen-workspaces"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-zen-tabs-title"></label>
+        </richlistitem>
+
+        <richlistitem id="category-zen-CKS"
+                      class="category"
+                      value="paneZenCKS"
+                      helpTopic="prefs-main"
+                      data-l10n-id="category-zen-CKS"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-zen-CKS-title"></label>
+        </richlistitem>
+
+        <richlistitem id="category-zen-marketplace"
+                      class="category"
+                      value="paneZenMarketplace"
+                      helpTopic="prefs-main"
+                      data-l10n-id="category-zen-marketplace"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-zen-marketplace-title"></label>
+        </richlistitem>
+
         <richlistitem id="category-home"
                       class="category"
                       value="paneHome"
@@ -227,11 +278,6 @@
                 <html:a href="about:policies" target="_blank" data-l10n-id="managed-notice"/>
               </hbox>
             </hbox>
-            <html:moz-input-search
-              id="searchInput"
-              data-l10n-id="search-input-box2"
-              data-l10n-attrs="placeholder, style"
-              />
           </hbox>
         </hbox>
         <vbox id="mainPrefPane">
@@ -245,6 +291,10 @@
 #include sync.inc.xhtml
 #include experimental.inc.xhtml
 #include moreFromMozilla.inc.xhtml
+#include zenLooksAndFeel.inc.xhtml
+#include zenKeyboardShortcuts.inc.xhtml
+#include zenTabsManagement.inc.xhtml
+#include zenMarketplace.inc.xhtml
         </vbox>
       </vbox>
     </vbox>
