diff --git a/browser/components/preferences/preferences.xhtml b/browser/components/preferences/preferences.xhtml
index eee227822a772bec128d6adbe4f6aca90ed810f6..dbe59dc683c157fda716033da8a402f7d4435d19 100644
--- a/browser/components/preferences/preferences.xhtml
+++ b/browser/components/preferences/preferences.xhtml
@@ -12,7 +12,7 @@
 
 <head>
   <!-- @CSP: We should remove 'unsafe-inline' from style-src, see Bug 1579160 -->
-  <meta http-equiv="Content-Security-Policy" content="default-src chrome:; img-src chrome: moz-icon: https: blob: data:; style-src chrome: data: 'unsafe-inline'; object-src 'none'" />
+  <meta http-equiv="Content-Security-Policy" content="default-src chrome:; img-src chrome: moz-icon: https: blob: data:; style-src chrome: data:; object-src 'none'" />
 
   <title data-l10n-id="settings-page-title"></title>
 
@@ -43,6 +43,8 @@
   />
   <link rel="stylesheet" href="chrome://browser/skin/preferences/privacy.css" />
 
+#include zen-preferences-links.xhtml
+
   <link rel="localization" href="branding/brand.ftl"/>
   <link rel="localization" href="browser/browser.ftl"/>
     <!-- Used by fontbuilder.js -->
@@ -99,6 +101,12 @@
   <hbox flex="1">
 
     <vbox class="navigation">
+      <search-textbox
+              id="searchInput"
+              data-l10n-id="search-input-box2"
+              data-l10n-attrs="placeholder, style"
+              hidden="true"/>
+
       <!-- category list -->
       <richlistbox id="categories" data-l10n-id="category-list" data-l10n-attrs="aria-label">
         <richlistitem id="category-general"
@@ -112,6 +120,17 @@
           <label class="category-name" flex="1" data-l10n-id="pane-general-title"></label>
         </richlistitem>
 
+        <richlistitem id="category-zen-looks"
+                      class="category"
+                      value="paneZenLooks"
+                      helpTopic="prefs-main"
+                      data-l10n-id="category-zen-looks"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-zen-looks-title"></label>
+        </richlistitem>
+
         <richlistitem id="category-home"
                       class="category"
                       value="paneHome"
@@ -220,6 +239,7 @@
 
     <vbox class="main-content" flex="1" align="start">
       <vbox class="pane-container">
+#if 0
         <hbox class="sticky-container">
           <hbox class="sticky-inner-container" pack="end" align="start">
             <hbox id="policies-container" class="info-box-container smaller-font-size" flex="1" hidden="true">
@@ -237,6 +257,7 @@
               hidden="true"/>
           </hbox>
         </hbox>
+#endif
         <vbox id="mainPrefPane">
 #include searchResults.inc.xhtml
 #include main.inc.xhtml
@@ -248,6 +269,7 @@
 #include sync.inc.xhtml
 #include experimental.inc.xhtml
 #include moreFromMozilla.inc.xhtml
+#include zenLooksAndFeel.inc.xhtml
         </vbox>
       </vbox>
     </vbox>
