diff --git a/browser/components/preferences/preferences.js b/browser/components/preferences/preferences.js
index 57add34d876fb885275f1147209c6fbeee367a7c..5f4616d5f7d3d077326246e843775f58c293ee48 100644
--- a/browser/components/preferences/preferences.js
+++ b/browser/components/preferences/preferences.js
@@ -132,6 +132,7 @@ ChromeUtils.defineLazyGetter(this, "gSubDialog", function () {
       styleSheets: [
         "chrome://browser/skin/preferences/dialog.css",
         "chrome://browser/skin/preferences/preferences.css",
+        "chrome://browser/skin/preferences/zen-preferences.css",
       ],
       resizeCallback: async ({ title, frame }) => {
         // Search within main document and highlight matched keyword.
@@ -437,6 +438,8 @@ const CONFIG_PANES = Object.freeze({
   tabsBrowsing: {
     l10nId: "tabs-browsing-section",
     groupIds: [
+      "defaultBrowserHome",
+      "startupHome",
       "browserLayout",
       "tabs",
       "pageNavigation",
@@ -477,7 +480,7 @@ function register_module(categoryName, categoryObject) {
       }
       this._initted = true;
       let template = document.getElementById("template-" + categoryName);
-      if (template && !srdSectionPrefs.all) {
+      if (template && (!srdSectionPrefs.all || categoryName.startsWith("paneZen"))) {
         // Replace the template element with the nodes inside of it.
         template.replaceWith(template.content);
 
@@ -522,6 +525,10 @@ function init_all() {
   register_module("paneSearch", gSearchPane);
   register_module("panePrivacy", gPrivacyPane);
   register_module("paneContainers", gContainersPane);
+  register_module("paneZenLooks", gZenLooksAndFeel);
+  register_module("paneZenTabManagement", gZenWorkspacesSettings);
+  register_module("paneZenCKS", gZenCKSSettings);
+  register_module("paneZenMarketplace", gZenMarketplaceManager);
 
   // Restore the cached Firefox Labs nav button visibility so it shows
   // immediately when recipes are expected to be available, before
@@ -653,7 +660,7 @@ async function gotoPref(
   const kDefaultCategoryInternalName = redesignEnabled
     ? "paneSync"
     : "paneGeneral";
-  const kDefaultCategory = redesignEnabled ? "sync" : "general";
+  const kDefaultCategory = redesignEnabled ? "tabsBrowsing" : "general";
   let hash = document.location.hash;
   let category = aCategory || hash.substring(1) || kDefaultCategoryInternalName;
 
