diff --git a/browser/components/preferences/preferences.xhtml b/browser/components/preferences/preferences.xhtml
index 53bdc9c912a6cd0c787dbcf41f2a82b7169d5aa7..e47c4b36b6e469b0cc8b95bde0508dc6cb0b49ae 100644
--- a/browser/components/preferences/preferences.xhtml
+++ b/browser/components/preferences/preferences.xhtml
@@ -43,6 +43,8 @@
   />
   <link rel="stylesheet" href="chrome://browser/skin/preferences/privacy.css" />
 
+#include zen-preferences-links.xhtml
+
   <link rel="localization" href="branding/brand.ftl"/>
   <link rel="localization" href="browser/browser.ftl"/>
     <!-- Used by fontbuilder.js -->
@@ -97,6 +99,12 @@
   <hbox flex="1">
 
     <vbox class="navigation">
+      <search-textbox
+              id="searchInput"
+              data-l10n-id="search-input-box2"
+              data-l10n-attrs="placeholder, style"
+              />
+
       <!-- category list -->
       <richlistbox id="categories" data-l10n-id="category-list" data-l10n-attrs="aria-label">
         <richlistitem id="category-general"
@@ -110,6 +118,50 @@
           <label class="category-name" flex="1" data-l10n-id="pane-general-title"></label>
         </richlistitem>
 
+        <richlistitem id="category-zen-looks"
+                      class="category"
+                      value="paneZenLooks"
+                      helpTopic="prefs-main"
+                      data-l10n-id="category-zen-looks"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-zen-looks-title"></label>
+        </richlistitem>
+
+        <richlistitem id="category-zen-tabs-management"
+                      class="category"
+                      value="paneZenTabManagement"
+                      helpTopic="prefs-main"
+                      data-l10n-id="category-zen-workspaces"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-zen-tabs-title"></label>
+        </richlistitem>
+
+        <richlistitem id="category-zen-CKS"
+                      class="category"
+                      value="paneZenCKS"
+                      helpTopic="prefs-main"
+                      data-l10n-id="category-zen-CKS"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-zen-CKS-title"></label>
+        </richlistitem>
+
+        <richlistitem id="category-zen-marketplace"
+                      class="category"
+                      value="paneZenMarketplace"
+                      helpTopic="prefs-main"
+                      data-l10n-id="category-zen-marketplace"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-zen-marketplace-title"></label>
+        </richlistitem>
+
         <richlistitem id="category-home"
                       class="category"
                       value="paneHome"
@@ -228,11 +280,13 @@
                 <html:a href="about:policies" target="_blank" data-l10n-id="managed-notice"/>
               </hbox>
             </hbox>
+#if 0
             <search-textbox
               id="searchInput"
               data-l10n-id="search-input-box2"
               data-l10n-attrs="placeholder, style"
               />
+#endif
           </hbox>
         </hbox>
         <vbox id="mainPrefPane">
@@ -246,6 +300,10 @@
 #include sync.inc.xhtml
 #include experimental.inc.xhtml
 #include moreFromMozilla.inc.xhtml
+#include zenLooksAndFeel.inc.xhtml
+#include zenKeyboardShortcuts.inc.xhtml
+#include zenTabsManagement.inc.xhtml
+#include zenMarketplace.inc.xhtml
         </vbox>
       </vbox>
     </vbox>
