diff --git a/browser/components/preferences/preferences.xhtml b/browser/components/preferences/preferences.xhtml
index 9760a4a35b0b3bd21edec07a70c10bccc23e4a09..9f22146c259ea5b45005be660bfcb9ea2c1297ee 100644
--- a/browser/components/preferences/preferences.xhtml
+++ b/browser/components/preferences/preferences.xhtml
@@ -42,6 +42,8 @@
   <link rel="stylesheet" href="chrome://browser/content/preferences/widgets/setting-pane.css" />
   <link rel="stylesheet" href="chrome://browser/content/preferences/widgets/setting-control.css" />
 
+#include zen-preferences-links.xhtml
+
   <link rel="localization" href="branding/brand.ftl"/>
   <link rel="localization" href="browser/browser.ftl"/>
     <!-- Used by fontbuilder.js -->
@@ -121,6 +123,26 @@
         iconsrc="chrome://browser/skin/preferences/category-general.svg"
         data-l10n-id="pane-general-title">
       </html:moz-page-nav-button>
+      <html:moz-page-nav-button id="category-zen-looks"
+        view="paneZenLooks"
+        iconsrc="chrome://browser/skin/customize.svg"
+        data-l10n-id="pane-zen-looks-title">
+      </html:moz-page-nav-button>
+      <html:moz-page-nav-button id="category-zen-tabs-management"
+        view="paneZenTabManagement"
+        iconsrc="chrome://browser/skin/window.svg"
+        data-l10n-id="pane-zen-tabs-title">
+      </html:moz-page-nav-button>
+      <html:moz-page-nav-button id="category-zen-CKS"
+        view="paneZenCKS"
+        iconsrc="chrome://browser/skin/quickactions.svg"
+        data-l10n-id="pane-zen-CKS-title">
+      </html:moz-page-nav-button>
+      <html:moz-page-nav-button id="category-zen-marketplace"
+        view="paneZenMarketplace"
+        iconsrc="chrome://mozapps/skin/extensions/category-themes.svg"
+        data-l10n-id="pane-zen-marketplace-title">
+      </html:moz-page-nav-button>
       <html:moz-page-nav-button id="category-sync"
         hidden="true"
         view="paneSync"
@@ -253,6 +275,10 @@
 #include containers.inc.xhtml
 #include sync.inc.xhtml
 #include moreFromMozilla.inc.xhtml
+#include zenLooksAndFeel.inc.xhtml
+#include zenKeyboardShortcuts.inc.xhtml
+#include zenTabsManagement.inc.xhtml
+#include zenMarketplace.inc.xhtml
         </vbox>
       </vbox>
     </vbox>
