diff --git a/browser/components/preferences/sync.inc.xhtml b/browser/components/preferences/sync.inc.xhtml
index c379e1a5f82692406a92d9fcd3bca2769dfac5b2..af037dd3d995813d966524ac44a3795d0c227071 100644
--- a/browser/components/preferences/sync.inc.xhtml
+++ b/browser/components/preferences/sync.inc.xhtml
@@ -224,6 +224,10 @@
                     <image class="sync-engine-image sync-engine-prefs" alt=""/>
                     <label data-l10n-id="sync-currently-syncing-settings"/>
                   </html:div>
+                  <html:div engine_preference="services.sync.engine.workspaces">
+                    <image class="sync-engine-image sync-engine-workspaces" alt=""/>
+                    <label data-l10n-id="sync-currently-syncing-workspaces"/>
+                  </html:div>
                 </html:div>
                 <hbox>
                   <button id="syncChangeOptions"
