diff --git a/browser/components/preferences/sync.inc.xhtml b/browser/components/preferences/sync.inc.xhtml
index 86053cbf35ef1b004185ee54e9d7b9f2e2b8d550..a8fe2db034ffe99e01fd0f92d629151df7f708a8 100644
--- a/browser/components/preferences/sync.inc.xhtml
+++ b/browser/components/preferences/sync.inc.xhtml
@@ -223,6 +223,10 @@
                     <image class="sync-engine-image sync-engine-prefs" alt=""/>
                     <label data-l10n-id="sync-currently-syncing-settings"/>
                   </html:div>
+                  <html:div engine_preference="services.sync.engine.workspaces">
+                    <image class="sync-engine-image sync-engine-workspaces" alt=""/>
+                    <label data-l10n-id="sync-currently-syncing-workspaces"/>
+                  </html:div>
                 </html:div>
                 <hbox>
                   <button id="syncChangeOptions"
