diff --git a/browser/components/tabbrowser/content/browser-ctrlTab.js b/browser/components/tabbrowser/content/browser-ctrlTab.js
index 958728cde19dd7cc24692baa89e11fcab816ab82..89063cca2c26b92e0c8b23960e9cfb43ea59b557 100644
--- a/browser/components/tabbrowser/content/browser-ctrlTab.js
+++ b/browser/components/tabbrowser/content/browser-ctrlTab.js
@@ -249,7 +249,8 @@ var ctrlTab = {
   },
 
   get tabList() {
-    return this._recentlyUsedTabs;
+    const canShowPending = Services.prefs.getBoolPref("zen.ctrlTab.show-pending-tabs", false);
+    return this._recentlyUsedTabs.filter(tab => !tab.hasAttribute("pending") || canShowPending);
   },
 
   init: function ctrlTab_init() {
@@ -452,7 +453,20 @@ var ctrlTab = {
 
   remove: function ctrlTab_remove(aPreview) {
     if (aPreview._tab) {
-      gBrowser.removeTab(aPreview._tab);
+      const tab = aPreview._tab;
+      
+      if (tab.pinned) {
+        gZenPinnedTabManager.onCloseTabShortcut(
+          new KeyboardEvent('keydown', { key: 'w', ctrlKey: true }),
+          tab
+        );
+        this.detachTab(tab);
+        if (this.isOpen) {
+          this.removeClosingTabFromUI(tab);
+        }
+      } else {
+        gBrowser.removeTab(tab);
+      }
     }
   },
 
@@ -460,7 +474,7 @@ var ctrlTab = {
     // If the tab is hidden, don't add it to the list unless it's selected
     // (Normally hidden tabs would be unhidden when selected, but that doesn't
     // happen for Firefox View).
-    if (aTab.closing || (aTab.hidden && !aTab.selected)) {
+    if (aTab.closing || (aTab.hidden && !aTab.selected) || aTab.hasAttribute("zen-empty-tab")) {
       return;
     }
 
@@ -484,7 +498,7 @@ var ctrlTab = {
   },
 
   open: function ctrlTab_open() {
-    if (this.isOpen) {
+    if (this.isOpen || !this.tabCount) {
       return;
     }
 
@@ -763,7 +777,7 @@ var ctrlTab = {
   _initRecentlyUsedTabs() {
     this._recentlyUsedTabs = Array.prototype.filter.call(
       gBrowser.tabs,
-      tab => !tab.closing && !tab.hidden
+      tab => !tab.closing && !tab.hidden && !tab.hasAttribute("zen-empty-tab")
     );
     this._sortRecentlyUsedTabs();
   },
