diff --git a/browser/components/tabbrowser/content/drag-and-drop.js b/browser/components/tabbrowser/content/drag-and-drop.js
index 70b37f93c4e8ff4ad9a2452681e468d23914c1b5..dbf2d9c83dafbe0e0b4e6caf3835edc8edf9bcf4 100644
--- a/browser/components/tabbrowser/content/drag-and-drop.js
+++ b/browser/components/tabbrowser/content/drag-and-drop.js
@@ -35,6 +35,9 @@
    * @returns {MozTabbrowserTab|vbox}
    */
   const elementToMove = element => {
+    if (element.group?.hasAttribute("split-view-group")) {
+      return element.group;
+    }
     if (isTab(element) || isSplitViewWrapper(element)) {
       return element;
     }
@@ -128,6 +131,9 @@
       }
 
       let draggedTab = event.dataTransfer.mozGetDataAt(TAB_DROP_TYPE, 0);
+      if (draggedTab && dropEffect === "move") {
+        gZenViewSplitter.onBrowserDragEndToSplit(event);
+      }
       if (
         (dropEffect == "move" || dropEffect == "copy") &&
         document == draggedTab.ownerDocument &&
@@ -148,6 +154,7 @@
           !draggedTab.group.collapsed
         ) {
           draggedTab.group.collapsed = true;
+          gZenFolders.animateGroupMove(draggedTab.group);
         }
 
         if (dropEffect == "move") {
@@ -155,10 +162,6 @@
 
           // Pinned tabs in expanded vertical mode are on a grid format and require
           // different logic to drag and drop.
-          if (this._tabbrowserTabs.isContainerVerticalPinnedGrid(draggedTab)) {
-            this._animateExpandedPinnedTabMove(event);
-            return;
-          }
           this._animateTabMove(event);
           return;
         }
@@ -297,6 +300,10 @@
 
       this._tabDropIndicator.hidden = true;
       event.stopPropagation();
+      if (draggedTab && dropEffect == "move") {
+        this.handle_drop_transition?.(draggedTab._dragData.dropElement, draggedTab, movingTabs, draggedTab._dragData.dropBefore);
+        [draggedTab, movingTabs] = gZenPinnedTabManager.moveToAnotherTabContainerIfNecessary(event, draggedTab, movingTabs, this._getDropIndex(event));
+      }
       if (draggedTab && dropEffect == "copy") {
         let duplicatedDraggedTab;
         let duplicatedTabs = [];
@@ -324,12 +331,16 @@
         let translateOffsetY = oldTranslateY % tabHeight;
         let newTranslateX = oldTranslateX - translateOffsetX;
         let newTranslateY = oldTranslateY - translateOffsetY;
-        let isPinned = draggedTab.pinned;
-        let numPinned = gBrowser.pinnedTabCount;
+        let isPinned = draggedTab?.group ? draggedTab.group.pinned : draggedTab.pinned;
+        let numPinned = gBrowser._numVisiblePinTabsWithoutCollapsed;
+        let essential = draggedTab.hasAttribute("zen-essential");
         let tabs = this._tabbrowserTabs.dragAndDropElements.slice(
-          isPinned ? 0 : numPinned,
-          isPinned ? numPinned : undefined
+          isPinned ? (essential ? 0 : gBrowser._numZenEssentials) : numPinned,
+          isPinned ? (essential ? gBrowser._numZenEssentials : numPinned) : undefined
         );
+        if (tabs.length == 0) {
+          tabs = [...movingTabs];
+        }
 
         if (this._tabbrowserTabs.isContainerVerticalPinnedGrid(draggedTab)) {
           // Update both translate axis for pinned vertical expanded tabs
@@ -404,11 +415,13 @@
           this._dragToPinPromoCard,
         ];
         let shouldPin =
+          false &&  
           movingTabs.some(t => isTab(t)) &&
           !draggedTab.pinned &&
           (overPinnedDropIndicator ||
             dragToPinTargets.some(el => el.contains(event.target)));
         let shouldUnpin =
+          false &&
           isTab(draggedTab) &&
           draggedTab.pinned &&
           this._tabbrowserTabs.arrowScrollbox.contains(event.target);
@@ -484,6 +497,7 @@
           }
         }
 
+        shouldTranslate = false;
         if (shouldTranslate) {
           this.#dropAnimationEndTime = Date.now() + DROP_ANIMATION_GRACE_MS;
           let translationPromises = [];
@@ -496,7 +510,7 @@
                 item.removeAttribute("tabdrop-samewindow");
                 resolve();
               };
-              if (gReduceMotion) {
+              if (true || gReduceMotion) {
                 postTransitionCleanup();
               } else {
                 let onTransitionEnd = transitionendEvent => {
@@ -580,7 +594,7 @@
           if (tab.selected) {
             selectedTab = tab;
             indexForSelectedTab = newIndex;
-          } else if (isSplitViewWrapper(tab)) {
+          } else if (false) {
             const droppedIntoPinnedArea = dropIndex < gBrowser.pinnedTabCount;
             const newSplitView = gBrowser.adoptSplitView(tab, {
               elementIndex: droppedIntoPinnedArea
@@ -605,7 +619,7 @@
             }
           }
         }
-        if (selectedTab) {
+        if (false) {
           const newTab = gBrowser.adoptTab(selectedTab, {
             elementIndex: indexForSelectedTab,
             selectTab: selectedTab == draggedTab,
@@ -639,10 +653,6 @@
               this._tabbrowserTabs.dragAndDropElements[
                 gBrowser.pinnedTabCount + unpinnedSplitViews.length - 1
               ];
-            gBrowser.addRangeToMultiSelectedTabs(
-              firstUnpinnedSplitView.tabs.at(0),
-              lastUnpinnedSplitView.tabs.at(-1)
-            );
           }
         }
       } else {
@@ -703,6 +713,7 @@
 
           let nextItem = this._tabbrowserTabs.dragAndDropElements[newIndex];
           let tabGroup = isTab(nextItem) && nextItem.group;
+          if (gZenViewSplitter.handleTabDrop(event, urls, replace, inBackground)) return;
           gBrowser.loadTabs(urls, {
             inBackground,
             replace,
@@ -744,6 +755,22 @@
         this._expandGroupOnDrop(draggedTab);
       }
       this._resetTabsAfterDrop(draggedTab);
+      if (
+        !dt.mozUserCancelled &&
+        dt.dropEffect == "none" &&
+        !this._tabbrowserTabs._isCustomizing
+      ) {
+        const moved = gZenViewSplitter.moveTabToSplitView(event, draggedTab);
+        if (moved) {
+          delete draggedTab._dragData;
+          return;
+        }
+      } else if (dt.mozUserCancelled) {
+        gZenViewSplitter.onBrowserDragEndToSplit(event, true);
+        if (gZenViewSplitter._lastOpenedTab) {
+          gZenViewSplitter._lastOpenedTab._visuallySelected = false;
+        }
+      }
 
       if (
         dt.mozUserCancelled ||
@@ -1074,11 +1101,10 @@
         return this.#getHorizontalScrollboxDragTarget(event, ignoreSides);
       }
       while (target) {
-        if (
-          isTab(target) ||
-          isTabGroupLabel(target) ||
-          isSplitViewWrapper(target)
-        ) {
+        if (isTab(target) || isTabGroupLabel(target) || target?.classList?.contains("tab-group-label-container")) {
+          if (target.classList?.contains("tab-group-label-container")) {
+            target = target.querySelector(".tab-group-label");
+          }
           break;
         }
         target = target.parentNode;
@@ -1108,6 +1134,9 @@
           return null;
         }
       }
+      if (target?.group?.hasAttribute("split-view-group")) {
+        target = target.group.labelElement;
+      }
       return target;
     }
 
@@ -1170,14 +1199,20 @@
       let releaseReservedSpace = () =>
         this.#releaseSpaceInScrolledContent(periphery);
       if (group.collapsed) {
-        // The group's tabs animate back to their full size, so hold on to the
-        // space reserved for them until they get there.
-        group.addEventListener(
-          "TabGroupAnimationComplete",
-          releaseReservedSpace,
-          { once: true }
-        );
-        group.collapsed = false;
+        const expand = !group.hasAttribute("has-active");
+        if (expand) {
+          // The group's tabs animate back to their full size, so hold on to the
+          // space reserved for them until they get there.
+          group.addEventListener(
+            "TabGroupAnimationComplete",
+            releaseReservedSpace,
+            { once: true }
+          );
+        } else {
+          releaseReservedSpace();
+        }
+        group.collapsed = !expand;
+        gZenFolders.animateGroupMove(group, true);
       } else {
         releaseReservedSpace();
       }
@@ -1362,7 +1397,6 @@
         // using updateDragImage. On Linux, we can use a panel.
         if (platform == "win" || platform == "macosx") {
           captureListener = function () {
-            dt.updateDragImage(canvas, dragImageOffset, dragImageOffset);
           };
         } else {
           // Create a panel to use it in setDragImage
@@ -1400,7 +1434,6 @@
         );
         dragImageOffset = dragImageOffset * scale;
       }
-      dt.setDragImage(toDrag, dragImageOffset, dragImageOffset);
 
       // _dragData.offsetX/Y give the coordinates that the mouse should be
       // positioned relative to the corner of the new window created upon
@@ -1419,7 +1452,7 @@
       let dropEffect = this.getDropEffectForTabDrag(event);
       let isMovingInTabStrip = !fromTabList && dropEffect == "move";
       let collapseTabGroupDuringDrag =
-        isMovingInTabStrip && isTabGroupLabel(tab) && !tab.group.collapsed;
+        isMovingInTabStrip && isTabGroupLabel(tab) && (!tab.group.collapsed || tab.group.hasAttribute("has-active"));
 
       tab._dragData = {
         offsetX: this._tabbrowserTabs.verticalMode
@@ -1429,7 +1462,7 @@
           ? event.screenY - window.screenY - tabOffset
           : event.screenY - window.screenY,
         scrollPos:
-          this._tabbrowserTabs.verticalMode && tab.pinned
+          this._tabbrowserTabs.verticalMode && tab.pinned && false
             ? this._tabbrowserTabs.pinnedTabsContainer.scrollPosition
             : this._tabbrowserTabs.arrowScrollbox.scrollPosition,
         screenX: event.screenX,
@@ -1522,6 +1555,7 @@
       if (tabStripItemElement.hasAttribute("dragtarget")) {
         return;
       }
+      return;
       let isPinned = tab.pinned;
       let dragAndDropElements = this._tabbrowserTabs.dragAndDropElements;
       let isGrid = this._tabbrowserTabs.isContainerVerticalPinnedGrid(tab);
@@ -1895,23 +1929,6 @@
 
       // Slide the relevant tabs to their new position.
       // non-moving tabs adjust for RTL
-      for (let item of this._tabbrowserTabs.dragAndDropElements) {
-        if (
-          !tab._dragData.movingTabsSet.has(item) &&
-          (item._moveTogetherSelectedTabsData?.translateX ||
-            item._moveTogetherSelectedTabsData?.translateY) &&
-          ((item.pinned && tab.pinned) || (!item.pinned && !tab.pinned))
-        ) {
-          let element = elementToMove(item);
-          if (isGrid) {
-            element.style.transform = `translate(${(this._rtlMode ? -1 : 1) * item._moveTogetherSelectedTabsData.translateX}px, ${item._moveTogetherSelectedTabsData.translateY}px)`;
-          } else if (this._tabbrowserTabs.verticalMode) {
-            element.style.transform = `translateY(${item._moveTogetherSelectedTabsData.translateY}px)`;
-          } else {
-            element.style.transform = `translateX(${(this._rtlMode ? -1 : 1) * item._moveTogetherSelectedTabsData.translateX}px)`;
-          }
-        }
-      }
       // moving tabs don't adjust for RTL
       for (let item of selectedElements) {
         if (
@@ -1960,7 +1977,6 @@
       for (let item of this._tabbrowserTabs.dragAndDropElements) {
         delete item._moveTogetherSelectedTabsData;
         item = elementToMove(item);
-        item.style.transform = "";
         item.removeAttribute("multiselected-move-together");
       }
     }
@@ -2812,7 +2828,6 @@
         tab.style.top = "";
         tab.style.maxWidth = "";
         tab.style.pointerEvents = "";
-        tab.removeAttribute("dragtarget");
         tab.removeAttribute("small-stack");
         tab.removeAttribute("big-stack");
       }
@@ -2821,11 +2836,9 @@
       )) {
         label.style.width = "";
         label.style.maxWidth = "";
-        label.style.height = "";
         label.style.left = "";
         label.style.top = "";
         label.style.pointerEvents = "";
-        label.removeAttribute("dragtarget");
       }
       for (let label of draggedTabContainer.getElementsByClassName(
         "tab-group-label"
