diff --git a/browser/components/tabbrowser/content/drag-and-drop.js b/browser/components/tabbrowser/content/drag-and-drop.js
index 57800333445ec7850742145527e04ae8d504b0bb..a13875436a72f89178455a09c8665b9a1ef240b5 100644
--- a/browser/components/tabbrowser/content/drag-and-drop.js
+++ b/browser/components/tabbrowser/content/drag-and-drop.js
@@ -35,6 +35,9 @@
    * @returns {MozTabbrowserTab|vbox}
    */
   const elementToMove = element => {
+    if (element.group?.hasAttribute("split-view-group")) {
+      return element.group;
+    }
     if (isTab(element) || isSplitViewWrapper(element)) {
       return element;
     }
@@ -115,6 +118,9 @@
       }
 
       let draggedTab = event.dataTransfer.mozGetDataAt(TAB_DROP_TYPE, 0);
+      if (draggedTab && dropEffect === "move") {
+        gZenViewSplitter.onBrowserDragEndToSplit(event);
+      }
       if (
         (dropEffect == "move" || dropEffect == "copy") &&
         document == draggedTab.ownerDocument &&
@@ -133,10 +139,6 @@
 
           // Pinned tabs in expanded vertical mode are on a grid format and require
           // different logic to drag and drop.
-          if (this._isContainerVerticalPinnedGrid(draggedTab)) {
-            this._animateExpandedPinnedTabMove(event);
-            return;
-          }
           this._animateTabMove(event);
           return;
         }
@@ -269,6 +271,10 @@
 
       this._tabDropIndicator.hidden = true;
       event.stopPropagation();
+      if (draggedTab && dropEffect == "move") {
+        this.handle_drop_transition?.(draggedTab._dragData.dropElement, draggedTab, movingTabs, draggedTab._dragData.dropBefore);
+        gZenPinnedTabManager.moveToAnotherTabContainerIfNecessary(event, draggedTab, movingTabs, this._getDropIndex(event));
+      }
       if (draggedTab && dropEffect == "copy") {
         let duplicatedDraggedTab;
         let duplicatedTabs = [];
@@ -294,8 +300,9 @@
         let translateOffsetY = oldTranslateY % tabHeight;
         let newTranslateX = oldTranslateX - translateOffsetX;
         let newTranslateY = oldTranslateY - translateOffsetY;
-        let isPinned = draggedTab.pinned;
-        let numPinned = gBrowser.pinnedTabCount;
+        let isPinned = draggedTab?.group ? draggedTab.group.pinned : draggedTab.pinned;
+        let numPinned = gBrowser._numVisiblePinTabsWithoutCollapsed;
+        let essential = draggedTab.hasAttribute("zen-essential");
 
         if (this._isContainerVerticalPinnedGrid(draggedTab)) {
           // Update both translate axis for pinned vertical expanded tabs
@@ -311,9 +318,12 @@
           }
         } else {
           let tabs = this._tabbrowserTabs.dragAndDropElements.slice(
-            isPinned ? 0 : numPinned,
-            isPinned ? numPinned : undefined
+            isPinned ? (essential ? 0 : gBrowser._numZenEssentials) : numPinned,
+            isPinned ? (essential ? gBrowser._numZenEssentials : numPinned) : undefined 
           );
+          if (tabs.length == 0) {
+            tabs = [...movingTabs];
+          }
           let size = this._tabbrowserTabs.verticalMode ? "height" : "width";
           let screenAxis = this._tabbrowserTabs.verticalMode
             ? "screenY"
@@ -365,11 +375,13 @@
           this._dragToPinPromoCard,
         ];
         let shouldPin =
+          false &&  
           isTab(draggedTab) &&
           !draggedTab.pinned &&
           (overPinnedDropIndicator ||
             dragToPinTargets.some(el => el.contains(event.target)));
         let shouldUnpin =
+          false &&
           isTab(draggedTab) &&
           draggedTab.pinned &&
           this._tabbrowserTabs.arrowScrollbox.contains(event.target);
@@ -432,6 +444,7 @@
           }
         }
 
+        shouldTranslate = false;
         if (shouldTranslate) {
           let translationPromises = [];
           for (let item of movingTabs) {
@@ -443,7 +456,7 @@
                 item.removeAttribute("tabdrop-samewindow");
                 resolve();
               };
-              if (gReduceMotion) {
+              if (true || gReduceMotion) {
                 postTransitionCleanup();
               } else {
                 let onTransitionEnd = transitionendEvent => {
@@ -513,7 +526,7 @@
           if (tab.selected) {
             selectedTab = tab;
             indexForSelectedTab = newIndex;
-          } else {
+          } else if (false) {
             const newTab = gBrowser.adoptTab(tab, {
               elementIndex: newIndex,
               selectTab: tab == draggedTab,
@@ -523,7 +536,7 @@
             }
           }
         }
-        if (selectedTab) {
+        if (false) {
           const newTab = gBrowser.adoptTab(selectedTab, {
             elementIndex: indexForSelectedTab,
             selectTab: selectedTab == draggedTab,
@@ -534,10 +547,6 @@
         }
 
         // Restore tab selection
-        gBrowser.addRangeToMultiSelectedTabs(
-          this._tabbrowserTabs.dragAndDropElements[dropIndex],
-          this._tabbrowserTabs.dragAndDropElements[newIndex - 1]
-        );
       } else {
         // Pass true to disallow dropping javascript: or data: urls
         let links;
@@ -584,6 +593,7 @@
 
           let nextItem = this._tabbrowserTabs.dragAndDropElements[newIndex];
           let tabGroup = isTab(nextItem) && nextItem.group;
+          if (gZenViewSplitter.handleTabDrop(event, urls, replace, inBackground)) return;
           gBrowser.loadTabs(urls, {
             inBackground,
             replace,
@@ -621,7 +631,16 @@
         this._expandGroupOnDrop(draggedTab);
       }
       this._resetTabsAfterDrop(draggedTab.ownerDocument);
-
+      if (!dt.mozUserCancelled && dt.dropEffect == "none" && !this._isCustomizing) {
+        const moved = gZenViewSplitter.moveTabToSplitView(event, draggedTab);
+        if (moved) {
+          delete draggedTab._dragData;
+          return;
+        }
+      } else if (dt.mozUserCancelled) {
+        gZenViewSplitter.onBrowserDragEndToSplit(event, true);
+        if (gZenViewSplitter._lastOpenedTab) gZenViewSplitter._lastOpenedTab._visuallySelected = false;
+      }
       if (
         dt.mozUserCancelled ||
         dt.dropEffect != "none" ||
@@ -825,7 +844,10 @@
     _getDragTarget(event, { ignoreSides = false } = {}) {
       let { target } = event;
       while (target) {
-        if (isTab(target) || isTabGroupLabel(target)) {
+        if (isTab(target) || isTabGroupLabel(target) || target?.classList?.contains("tab-group-label-container")) {
+          if (target.classList?.contains("tab-group-label-container")) {
+            target = target.querySelector(".tab-group-label");
+          }
           break;
         }
         target = target.parentNode;
@@ -842,14 +864,17 @@
           return null;
         }
       }
+      if (target?.group?.hasAttribute("split-view-group")) {
+        target = target.group.labelElement;
+      }
       return target;
     }
 
     _isContainerVerticalPinnedGrid(tab) {
       return (
         this._tabbrowserTabs.verticalMode &&
-        tab.pinned &&
-        this._tabbrowserTabs.hasAttribute("expanded") &&
+        tab.hasAttribute("zen-essential") &&
+        (this._tabbrowserTabs.hasAttribute("expanded") || document.documentElement.hasAttribute("zen-sidebar-expanded")) &&
         !this._tabbrowserTabs.expandOnHover
       );
     }
@@ -880,7 +905,8 @@
         isTabGroupLabel(draggedTab) &&
         draggedTab._dragData?.expandGroupOnDrop
       ) {
-        draggedTab.group.collapsed = false;
+        draggedTab.group.collapsed = draggedTab.group.hasAttribute("has-active");
+        gZenFolders.animateGroupMove(draggedTab.group, true);
       }
     }
 
@@ -1058,7 +1084,6 @@
         // using updateDragImage. On Linux, we can use a panel.
         if (platform == "win" || platform == "macosx") {
           captureListener = function () {
-            dt.updateDragImage(canvas, dragImageOffset, dragImageOffset);
           };
         } else {
           // Create a panel to use it in setDragImage
@@ -1096,7 +1121,6 @@
         );
         dragImageOffset = dragImageOffset * scale;
       }
-      dt.setDragImage(toDrag, dragImageOffset, dragImageOffset);
 
       // _dragData.offsetX/Y give the coordinates that the mouse should be
       // positioned relative to the corner of the new window created upon
@@ -1115,7 +1139,7 @@
       let dropEffect = this.getDropEffectForTabDrag(event);
       let isMovingInTabStrip = !fromTabList && dropEffect == "move";
       let collapseTabGroupDuringDrag =
-        isMovingInTabStrip && isTabGroupLabel(tab) && !tab.group.collapsed;
+        isMovingInTabStrip && isTabGroupLabel(tab) && (!tab.group.collapsed || tab.group.hasAttribute("has-active"));
 
       tab._dragData = {
         offsetX: this._tabbrowserTabs.verticalMode
@@ -1125,7 +1149,7 @@
           ? event.screenY - window.screenY - tabOffset
           : event.screenY - window.screenY,
         scrollPos:
-          this._tabbrowserTabs.verticalMode && tab.pinned
+          this._tabbrowserTabs.verticalMode && tab.pinned && false
             ? this._tabbrowserTabs.pinnedTabsContainer.scrollPosition
             : this._tabbrowserTabs.arrowScrollbox.scrollPosition,
         screenX: event.screenX,
@@ -1152,6 +1176,7 @@
 
           if (collapseTabGroupDuringDrag) {
             tab.group.collapsed = true;
+            gZenFolders.animateGroupMove(tab.group);
           }
         }
       }
@@ -1176,6 +1201,7 @@
       if (tabStripItemElement.hasAttribute("dragtarget")) {
         return;
       }
+      return;
       let isPinned = tab.pinned;
       let numPinned = gBrowser.pinnedTabCount;
       let dragAndDropElements = this._tabbrowserTabs.dragAndDropElements;
@@ -1601,7 +1627,6 @@
 
       for (let item of this._tabbrowserTabs.dragAndDropElements) {
         item = elementToMove(item);
-        item.style.transform = "";
         item.removeAttribute("multiselected-move-together");
         delete item._moveTogetherSelectedTabsData;
       }
@@ -2429,7 +2454,6 @@
       for (let item of this._tabbrowserTabs.dragAndDropElements) {
         this._resetGroupTarget(item);
         item = elementToMove(item);
-        item.style.transform = "";
       }
       this._tabbrowserTabs.removeAttribute("movingtab-group");
       this._tabbrowserTabs.removeAttribute("movingtab-ungroup");
@@ -2460,17 +2484,14 @@
         tab.style.left = "";
         tab.style.top = "";
         tab.style.maxWidth = "";
-        tab.removeAttribute("dragtarget");
       }
       for (let label of draggedTabDocument.getElementsByClassName(
         "tab-group-label-container"
       )) {
         label.style.width = "";
-        label.style.height = "";
         label.style.left = "";
         label.style.top = "";
         label.style.maxWidth = "";
-        label.removeAttribute("dragtarget");
       }
       let periphery = draggedTabDocument.getElementById(
         "tabbrowser-arrowscrollbox-periphery"
