diff --git a/browser/components/tabbrowser/content/tabbrowser.js b/browser/components/tabbrowser/content/tabbrowser.js
index 2674dc2bebf436529a46d45c52cb56e86b82c03f..c70801bed55e0182eb56e9fd7bea5be9986c7120 100644
--- a/browser/components/tabbrowser/content/tabbrowser.js
+++ b/browser/components/tabbrowser/content/tabbrowser.js
@@ -3208,6 +3208,11 @@
         ) {
           tabWasReused = true;
           tab = this.selectedTab;
+
+          if (tabData.zenWorkspace) {
+            tab.setAttribute("zen-workspace-id", tabData.zenWorkspace);
+          }
+
           if (!tabData.pinned) {
             this.unpinTab(tab);
           } else {
@@ -7878,6 +7883,11 @@ var TabContextMenu = {
       this.contextTab.linkedBrowser,
       document.getElementById("context_sendTabToDevice")
     );
+
+    document
+      .getElementById("context_zenSplitTabs")
+      .setAttribute("data-l10n-args", tabCountInfo);
+    document.getElementById("context_zenSplitTabs").disabled = !gZenViewSplitter.contextCanSplitTabs();
   },
 
   handleEvent(aEvent) {
