diff --git a/browser/components/tabbrowser/content/tabgroup.js b/browser/components/tabbrowser/content/tabgroup.js
index 2ed2fa02b16013e40a32391b3dba0a4301d65262..f485a7e2e09d81d0699e38e19fbd6d184d6c00ea 100644
--- a/browser/components/tabbrowser/content/tabgroup.js
+++ b/browser/components/tabbrowser/content/tabgroup.js
@@ -270,7 +270,7 @@
      */
     addTabs(tabs, metricsContext) {
       for (let tab of tabs) {
-        if (tab.pinned) {
+        if (tab.pinned !== this.pinned) {
           tab.ownerGlobal.gBrowser.unpinTab(tab);
         }
         let tabToMove =
