diff --git a/browser/components/tabbrowser/content/tabgroup.js b/browser/components/tabbrowser/content/tabgroup.js
index 6dc774ea335b0c5dba7dcf76cdb23728faae1343..b0b9ef236c2e8517db4bcf3270596456bbefe11d 100644
--- a/browser/components/tabbrowser/content/tabgroup.js
+++ b/browser/components/tabbrowser/content/tabgroup.js
@@ -301,7 +301,7 @@
      */
     addTabs(tabs, metricsContext) {
       for (let tab of tabs) {
-        if (tab.pinned) {
+        if (tab.pinned !== this.pinned) {
           tab.ownerGlobal.gBrowser.unpinTab(tab);
         }
         let tabToMove =
