diff --git a/testing/mochitest/browser-test.js b/testing/mochitest/browser-test.js
index 11217c210dd8dc5f7c4bdc6043df9aeb361529c6..18c3c4d658170d0646483cd289fe8ad1f21ffae0 100644
--- a/testing/mochitest/browser-test.js
+++ b/testing/mochitest/browser-test.js
@@ -440,11 +440,11 @@ Tester.prototype = {
       this.currentTest &&
       window.gBrowser &&
       AppConstants.MOZ_APP_NAME != "thunderbird" &&
-      gBrowser.tabs.length > 1
+      gBrowser.tabs.length > 2
     ) {
       let lastURI = "";
       let lastURIcount = 0;
-      while (gBrowser.tabs.length > 1) {
+      while (gBrowser.tabs.length > 2) {
         let lastTab = gBrowser.tabs[gBrowser.tabs.length - 1];
         if (!lastTab.closing) {
           // Report the stale tab as an error only when they're not closing.
@@ -479,12 +479,12 @@ Tester.prototype = {
 
     // Replace the last tab with a fresh one
     if (window.gBrowser && AppConstants.MOZ_APP_NAME != "thunderbird") {
-      gBrowser.addTab("about:blank", {
-        skipAnimation: true,
-        triggeringPrincipal:
-          Services.scriptSecurityManager.getSystemPrincipal(),
-      });
-      gBrowser.removeTab(gBrowser.selectedTab, { skipPermitUnload: true });
+      // gBrowser.addTab("about:blank", {
+      //   skipAnimation: true,
+      //   triggeringPrincipal:
+      //     Services.scriptSecurityManager.getSystemPrincipal(),
+      // });
+      // gBrowser.removeTab(gBrowser.selectedTab, { skipPermitUnload: true });
       gBrowser.stop();
     }
 
