diff --git a/testing/mochitest/browser-test.js b/testing/mochitest/browser-test.js
index 2377812af77563d6e80178e3da47a537aa0576e2..e9e5c4a2f3b2ebb503605595f552f35eea4f9814 100644
--- a/testing/mochitest/browser-test.js
+++ b/testing/mochitest/browser-test.js
@@ -497,10 +497,10 @@ Tester.prototype = {
     // But not the Thunderbird window, it doesn't have these things!
     if (AppConstants.MOZ_APP_NAME != "thunderbird") {
       // Remove stale tabs
-      if (this.currentTest && window.gBrowser && gBrowser.tabs.length > 1) {
+      if (this.currentTest && window.gBrowser && gBrowser.tabs.length > 2) {
         let lastURI = "";
         let lastURIcount = 0;
-        while (gBrowser.tabs.length > 1) {
+        while (gBrowser.tabs.length > 2) {
           let lastTab = gBrowser.tabs[gBrowser.tabs.length - 1];
           if (!lastTab.closing) {
             // Report the stale tab as an error only when they're not closing.
@@ -1248,12 +1248,6 @@ Tester.prototype = {
     // test_end is logged, otherwise the new windows created by addTab will be
     // tracked by ShutdownLeaks as belonging to the test and cause false leak reports.
     if (window.gBrowser) {
-      gBrowser.addTab("about:blank", {
-        skipAnimation: true,
-        triggeringPrincipal:
-          Services.scriptSecurityManager.getSystemPrincipal(),
-      });
-      gBrowser.removeTab(gBrowser.selectedTab, { skipPermitUnload: true });
       gBrowser.stop();
     }
 
