diff --git a/testing/mochitest/browser-test.js b/testing/mochitest/browser-test.js
index 91bd995c2910fae5a3f67a960d15d90de3c997df..a7157331184e099ee468a719d6ec0f2d766dbed1 100644
--- a/testing/mochitest/browser-test.js
+++ b/testing/mochitest/browser-test.js
@@ -442,10 +442,10 @@ Tester.prototype = {
     // But not the Thunderbird window, it doesn't have these things!
     if (AppConstants.MOZ_APP_NAME != "thunderbird") {
       // Remove stale tabs
-      if (this.currentTest && window.gBrowser && gBrowser.tabs.length > 1) {
+      if (this.currentTest && window.gBrowser && gBrowser.tabs.length > 2) {
         let lastURI = "";
         let lastURIcount = 0;
-        while (gBrowser.tabs.length > 1) {
+        while (gBrowser.tabs.length > 2) {
           let lastTab = gBrowser.tabs[gBrowser.tabs.length - 1];
           if (!lastTab.closing) {
             // Report the stale tab as an error only when they're not closing.
@@ -480,12 +480,6 @@ Tester.prototype = {
 
       // Replace the last tab with a fresh one
       if (window.gBrowser) {
-        gBrowser.addTab("about:blank", {
-          skipAnimation: true,
-          triggeringPrincipal:
-            Services.scriptSecurityManager.getSystemPrincipal(),
-        });
-        gBrowser.removeTab(gBrowser.selectedTab, { skipPermitUnload: true });
         gBrowser.stop();
       }
 
