diff --git a/docshell/base/BrowsingContextWebProgress.cpp b/docshell/base/BrowsingContextWebProgress.cpp
index cc5bf6de7bf4875acf0c572549dc30ff06ac7f6f..4e0f3648f551fe9507a654d19f0a99e455418aa9 100644
--- a/docshell/base/BrowsingContextWebProgress.cpp
+++ b/docshell/base/BrowsingContextWebProgress.cpp
@@ -54,7 +54,7 @@ NS_IMETHODIMP BrowsingContextWebProgress::AddProgressListener(
 
   if (mListenerInfoList.Contains(listener)) {
     // The listener is already registered!
-    return NS_ERROR_FAILURE;
+    return NS_OK;
   }
 
   mListenerInfoList.AppendElement(ListenerInfo(listener, aNotifyMask));
@@ -68,7 +68,7 @@ NS_IMETHODIMP BrowsingContextWebProgress::RemoveProgressListener(
     return NS_ERROR_INVALID_ARG;
   }
 
-  return mListenerInfoList.RemoveElement(listener) ? NS_OK : NS_ERROR_FAILURE;
+  return mListenerInfoList.RemoveElement(listener) ? NS_OK : NS_OK;
 }
 
 NS_IMETHODIMP BrowsingContextWebProgress::GetBrowsingContextXPCOM(
