diff --git a/docshell/base/BrowsingContext.h b/docshell/base/BrowsingContext.h
index dc64d5c13628cb7393705ebf96110705b045471f..497c2e646e2bde4d7732fa28efe7496264d58fa0 100644
--- a/docshell/base/BrowsingContext.h
+++ b/docshell/base/BrowsingContext.h
@@ -263,6 +263,8 @@ struct EmbedderColorSchemes {
   FIELD(HistoryEntryCount, uint32_t)                                          \
   FIELD(HasRestoreData, bool)                                                 \
   FIELD(SessionStoreEpoch, uint32_t)                                          \
+  FIELD(ZenBoostsData, nscolor)                                               \
+  FIELD(IsZenBoostsInverted, bool)                                            \
   /* Whether we can execute scripts in this BrowsingContext. Has no effect    \
    * unless scripts are also allowed in the parent WindowContext. */          \
   FIELD(AllowJavascript, bool)                                                \
@@ -674,6 +676,8 @@ class BrowsingContext : public nsILoadContext, public nsWrapperCache {
 
   bool FullscreenAllowed() const;
 
+  auto ZenBoostsData() const { return GetZenBoostsData(); }
+  auto IsZenBoostsInverted() const { return GetIsZenBoostsInverted(); }
   float FullZoom() const { return GetFullZoom(); }
   float TextZoom() const { return GetTextZoom(); }
 
@@ -1274,6 +1278,8 @@ class BrowsingContext : public nsILoadContext, public nsWrapperCache {
   }
 
   void DidSet(FieldIndex<IDX_SessionStoreEpoch>, uint32_t aOldValue);
+  void DidSet(FieldIndex<IDX_ZenBoostsData>, nscolor aOldValue);
+  void DidSet(FieldIndex<IDX_IsZenBoostsInverted>, bool aOldValue);
 
   using CanSetResult = syncedcontext::CanSetResult;
 
