diff --git a/docshell/base/BrowsingContext.h b/docshell/base/BrowsingContext.h
index 6f7f58bba3be6f08b3b27e81f30864c61d683351..2516126365ee9e1c07e245a873efe5f48767aff7 100644
--- a/docshell/base/BrowsingContext.h
+++ b/docshell/base/BrowsingContext.h
@@ -265,6 +265,9 @@ struct EmbedderColorSchemes {
   FIELD(HistoryEntryCount, uint32_t)                                          \
   FIELD(HasRestoreData, bool)                                                 \
   FIELD(SessionStoreEpoch, uint32_t)                                          \
+  FIELD(ZenBoostsData, nscolor)                                               \
+  FIELD(ZenBoostsComplementaryRotation, float)                                \
+  FIELD(IsZenBoostsInverted, bool)                                            \
   /* Whether we can execute scripts in this BrowsingContext. Has no effect    \
    * unless scripts are also allowed in the parent WindowContext. */          \
   FIELD(AllowJavascript, bool)                                                \
@@ -671,6 +674,11 @@ class BrowsingContext : public nsILoadContext, public nsWrapperCache {
 
   bool FullscreenAllowed() const;
 
+  auto ZenBoostsData() const { return GetZenBoostsData(); }
+  auto ZenBoostsComplementaryRotation() const {
+    return GetZenBoostsComplementaryRotation();
+  }
+  auto IsZenBoostsInverted() const { return GetIsZenBoostsInverted(); }
   float FullZoom() const { return GetFullZoom(); }
   float TextZoom() const { return GetTextZoom(); }
 
@@ -1290,6 +1298,19 @@ class BrowsingContext : public nsILoadContext, public nsWrapperCache {
   }
 
   void DidSet(FieldIndex<IDX_SessionStoreEpoch>, uint32_t aOldValue);
+  void DidSet(FieldIndex<IDX_ZenBoostsData>, nscolor aOldValue);
+  void DidSet(FieldIndex<IDX_ZenBoostsComplementaryRotation>, float aOldValue);
+  void DidSet(FieldIndex<IDX_IsZenBoostsInverted>, bool aOldValue);
+  bool CanSet(FieldIndex<IDX_ZenBoostsData>, const nscolor&, ContentParent*) {
+    return true;
+  }
+  bool CanSet(FieldIndex<IDX_ZenBoostsComplementaryRotation>, const float&,
+              ContentParent*) {
+    return true;
+  }
+  bool CanSet(FieldIndex<IDX_IsZenBoostsInverted>, const bool&, ContentParent*) {
+    return true;
+  }
 
   // Ensure that opener is in the same BrowsingContextGroup.
   bool CanSet(FieldIndex<IDX_OpenerId>, const uint64_t& aValue,
