diff --git a/docshell/base/BrowsingContext.h b/docshell/base/BrowsingContext.h
index 2d3a14d59922045197e474f5d097d7e16ce6ad8c..2e86f53edcd1288d3ca2e0292409aec8ee5926da 100644
--- a/docshell/base/BrowsingContext.h
+++ b/docshell/base/BrowsingContext.h
@@ -258,6 +258,7 @@ struct EmbedderColorSchemes {
   FIELD(IsInBFCache, bool)                                                    \
   FIELD(HasRestoreData, bool)                                                 \
   FIELD(SessionStoreEpoch, uint32_t)                                          \
+  FIELD(ZenBoostsData, nscolor)                                               \
   /* Whether we can execute scripts in this BrowsingContext. Has no effect    \
    * unless scripts are also allowed in the parent WindowContext. */          \
   FIELD(AllowJavascript, bool)                                                \
@@ -652,6 +653,7 @@ class BrowsingContext : public nsILoadContext, public nsWrapperCache {
 
   bool FullscreenAllowed() const;
 
+  auto ZenBoostsData() const { return GetZenBoostsData(); }
   float FullZoom() const { return GetFullZoom(); }
   float TextZoom() const { return GetTextZoom(); }
 
@@ -1175,6 +1177,7 @@ class BrowsingContext : public nsILoadContext, public nsWrapperCache {
   }
 
   void DidSet(FieldIndex<IDX_SessionStoreEpoch>, uint32_t aOldValue);
+  void DidSet(FieldIndex<IDX_ZenBoostsData>, nscolor aOldValue);
 
   using CanSetResult = syncedcontext::CanSetResult;
 
