diff --git a/docshell/base/nsAboutRedirector.cpp b/docshell/base/nsAboutRedirector.cpp
index df5658c7231e73c52ee00e35eeccc47166528391..8bfdceb0fc00c618326e932990c9503af13d9e69 100644
--- a/docshell/base/nsAboutRedirector.cpp
+++ b/docshell/base/nsAboutRedirector.cpp
@@ -108,7 +108,7 @@ static const RedirEntry kRedirMap[] = {
     {"crashes", "chrome://global/content/crashes.html",
      nsIAboutModule::IS_SECURE_CHROME_UI},
 #endif
-    {"credits", "https://www.mozilla.org/credits/",
+    {"credits", "https://zen-browser.app/about/",
      nsIAboutModule::URI_SAFE_FOR_UNTRUSTED_CONTENT |
          nsIAboutModule::URI_MUST_LOAD_IN_CHILD},
     {"fingerprintingprotection",
