diff --git a/browser/components/preferences/dialogs/syncChooseWhatToSync.xhtml b/browser/components/preferences/dialogs/syncChooseWhatToSync.xhtml
index 83bd09c0c3f8fd5febf1721e6b462ca90e54327b..5d0ca4265e7b57178f1fe45068449e0355ba80e0 100644
--- a/browser/components/preferences/dialogs/syncChooseWhatToSync.xhtml
+++ b/browser/components/preferences/dialogs/syncChooseWhatToSync.xhtml
@@ -26,6 +26,10 @@
         rel="localization"
         href="browser/preferences/preferences.ftl"
       />
+      <html:link
+        rel="localization"
+        href="browser/preferences/zen-preferences.ftl"
+      />
     </linkset>
     <script src="chrome://global/content/preferencesBindings.js" />
     <script src="chrome://browser/content/preferences/dialogs/syncChooseWhatToSync.js" />
@@ -82,6 +86,12 @@
           preference="services.sync.engine.prefs"
         />
       </html:div>
+      <html:div class="sync-engine-workspaces">
+        <checkbox
+                data-l10n-id="sync-engine-workspaces"
+                preference="services.sync.engine.workspaces"
+        />
+      </html:div>
     </html:div>
   </dialog>
 </window>
