diff --git a/browser/components/preferences/dialogs/syncChooseWhatToSync.xhtml b/browser/components/preferences/dialogs/syncChooseWhatToSync.xhtml
index 6e74bbce9096179d38db422879349013f3954fa9..859f8b9640ef49411164f8e5160d4d9b8db11e54 100644
--- a/browser/components/preferences/dialogs/syncChooseWhatToSync.xhtml
+++ b/browser/components/preferences/dialogs/syncChooseWhatToSync.xhtml
@@ -27,6 +27,10 @@
         rel="localization"
         href="browser/preferences/preferences.ftl"
       />
+      <html:link
+        rel="localization"
+        href="browser/preferences/zen-preferences.ftl"
+      />
     </linkset>
     <script src="chrome://global/content/preferencesBindings.js" />
     <script src="chrome://browser/content/preferences/dialogs/syncChooseWhatToSync.js" />
@@ -79,6 +83,12 @@
           preference="services.sync.engine.prefs"
         />
       </html:div>
+      <html:div class="sync-engine-workspaces">
+        <checkbox
+                data-l10n-id="sync-engine-workspaces"
+                preference="services.sync.engine.workspaces"
+        />
+      </html:div>
     </html:div>
   </dialog>
 </window>
