diff --git a/toolkit/components/asyncshutdown/AsyncShutdown.sys.mjs b/toolkit/components/asyncshutdown/AsyncShutdown.sys.mjs
index 2aaef80411b2cef9563c49f23d36b08222b06b03..7bd15ebb0d9d09da57a287b47beb1f0e2e17d229 100644
--- a/toolkit/components/asyncshutdown/AsyncShutdown.sys.mjs
+++ b/toolkit/components/asyncshutdown/AsyncShutdown.sys.mjs
@@ -492,6 +492,23 @@ function getPhase(topic) {
       }
       return undefined;
     },
+
+    /**
+     * Reset the phase after a call to _trigger().
+     * For testing purposes only.
+     */
+    get _reset() {
+      let accepted = Services.prefs.getBoolPref(
+        "toolkit.asyncshutdown.testing",
+        false
+      );
+      if (accepted) {
+        return () => {
+          spinner = new Spinner(topic);
+        };
+      }
+      return undefined;
+    },
   });
   gPhases.set(topic, phase);
   return phase;
