diff --git a/gfx/layers/AnimationInfo.cpp b/gfx/layers/AnimationInfo.cpp
index aba548880b3250818e7f7c0f14123a1ea3a2e32a..e10b49ba484cd324bbbd31e27e4b38d61f8929b3 100644
--- a/gfx/layers/AnimationInfo.cpp
+++ b/gfx/layers/AnimationInfo.cpp
@@ -18,6 +18,7 @@
 #include "mozilla/layers/AnimationStorageData.h"
 #include "mozilla/layers/CompositorThread.h"
 #include "mozilla/layers/WebRenderLayerManager.h"
+#include "mozilla/nsZenBoostsBackend.h"
 #include "nsIContent.h"
 #include "nsLayoutUtils.h"
 #include "nsRefreshDriver.h"
@@ -367,7 +368,11 @@ static void SetAnimatable(NonCustomCSSPropertyId aProperty,
     case eCSSProperty_background_color: {
       // We don't support color animation on the compositor yet so that we can
       // resolve currentColor at this moment.
-      aAnimatable = aAnimationValue.GetColor(aFrame->StyleText()->mColor);
+      nscolor resolved = aAnimationValue.GetColor(aFrame->StyleText()->mColor);
+      aAnimatable = aAnimationValue.IsCurrentColor()
+                        ? resolved
+                        : zen::nsZenBoostsBackend::ResolveStyleColor(resolved,
+                                                                     aFrame);
       break;
     }
     case eCSSProperty_opacity:
