diff --git a/browser/components/BrowserContentHandler.sys.mjs b/browser/components/BrowserContentHandler.sys.mjs
index f4a0e1e566c363d5eb1d77e0c021df250d96b3f3..390e6377078504b84e47d44672a9352ecdfe1ae3 100644
--- a/browser/components/BrowserContentHandler.sys.mjs
+++ b/browser/components/BrowserContentHandler.sys.mjs
@@ -672,6 +672,28 @@ nsBrowserContentHandler.prototype = {
       }
     }
 
+    try {
+      let blankWindowParam = cmdLine.handleFlagWithParam("blank-window", false);
+      if (blankWindowParam !== null) {
+        let { uri, principal } = resolveURIInternal(cmdLine, blankWindowParam);
+        if (shouldLoadURI(uri)) {
+          let win = openBrowserWindow(cmdLine, principal, uri.spec);
+          win._zenStartupSyncFlag = 'unsynced';
+          cmdLine.preventDefault = true;
+        }
+      }
+    } catch (e) {
+      if (e.result != Cr.NS_ERROR_INVALID_ARG) {
+        throw e;
+      }
+      if (cmdLine.handleFlag("blank-window", false)) {
+        let win = openBrowserWindow(cmdLine, null, null);
+        win._zenStartupSyncFlag = 'unsynced';
+        cmdLine.preventDefault = true;
+      }
+    }
+
+
     var searchParam = cmdLine.handleFlagWithParam("search", false);
     if (searchParam) {
       doSearch(searchParam, cmdLine);
@@ -742,6 +764,7 @@ nsBrowserContentHandler.prototype = {
       "  --new-tab <url>    Open <url> in a new tab.\n" +
       "  --data <data-URL>  Open a data: URL.\n" +
       "  --private-window [<url>] Open <url> in a new private window.\n";
+    info += "  --blank-window [<url>]  Open the new blank window.\n";
     if (AppConstants.platform == "win") {
       info += "  --preferences      Open Options dialog.\n";
     } else {
