diff --git a/browser/components/BrowserGlue.sys.mjs b/browser/components/BrowserGlue.sys.mjs
index f4ea0c87a35a19c09f811576201a7adb865ed74c..30f81ad02cd90f478945845b4d8a47da5f3601f0 100644
--- a/browser/components/BrowserGlue.sys.mjs
+++ b/browser/components/BrowserGlue.sys.mjs
@@ -27,6 +27,7 @@ ChromeUtils.defineESModuleGetters(lazy, {
   BrowserUsageTelemetry: "resource:///modules/BrowserUsageTelemetry.sys.mjs",
   BrowserWindowTracker: "resource:///modules/BrowserWindowTracker.sys.mjs",
   BuiltInThemes: "resource:///modules/BuiltInThemes.sys.mjs",
+  ZenSidebarManager: "resource:///modules/ZenSidebarManager.sys.mjs",
   ContextualIdentityService:
     "resource://gre/modules/ContextualIdentityService.sys.mjs",
   Corroborate: "resource://gre/modules/Corroborate.sys.mjs",
@@ -1446,6 +1447,8 @@ BrowserGlue.prototype = {
 
     AboutHomeStartupCache.init();
 
+    lazy.ZenSidebarManager.init();
+
     Services.obs.notifyObservers(null, "browser-ui-startup-complete");
   },
 
@@ -4436,6 +4439,7 @@ BrowserGlue.prototype = {
   },
 
   async _maybeShowDefaultBrowserPrompt() {
+    this._ZenMaybeShowWelcomeScreen();
     // Highest priority is about:welcome window modal experiment
     // Second highest priority is the upgrade dialog, which can include a "primary
     // browser" request and is limited in various ways, e.g., major upgrades.
@@ -4894,6 +4898,16 @@ BrowserGlue.prototype = {
     "nsIObserver",
     "nsISupportsWeakReference",
   ]),
+
+  _ZenMaybeShowWelcomeScreen() {
+    const welcomeEnabled = Services.prefs.getBoolPref("zen.welcomeScreen.enabled", true)
+    const welcomeSeen = Services.prefs.getBoolPref("zen.welcomeScreen.seen", false)
+    if (welcomeEnabled && !welcomeSeen) {
+      lazy.BrowserWindowTracker.getTopWindow().gDialogBox.open(
+        "chrome://browser/content/zen-welcome/welcome.html"
+      );
+    }
+  },
 };
 
 var ContentBlockingCategoriesPrefs = {
