diff --git a/browser/components/moz.build b/browser/components/moz.build
index 0f91b90fb0ce4eac232d71ad4f9ff82ab7316635..f1596aa7d0ab7e058d3482c0dac8ab10157f8bc8 100644
--- a/browser/components/moz.build
+++ b/browser/components/moz.build
@@ -72,6 +72,9 @@ DIRS += [
     "urlbar",
 ]
 
+# Zen Components:
+DIRS += ["zen-welcome"]
+
 DIRS += ["build"]
 
 
@@ -118,3 +121,7 @@ if CONFIG["MOZ_UPDATER"]:
 MARIONETTE_MANIFESTS += ["tests/marionette/manifest.toml"]
 
 XPCSHELL_TESTS_MANIFESTS += ["tests/unit/xpcshell.toml"]
+
+EXTRA_JS_MODULES += [
+    "ZenSidebarManager.sys.mjs",
+]
