diff --git a/browser/components/asrouter/modules/FeatureCallout.sys.mjs b/browser/components/asrouter/modules/FeatureCallout.sys.mjs
index bc989826e648e2528b4142940e1ae4d192940235..4cd18e67fbb52ea33ae6a9b0a47cbf2359d1b832 100644
--- a/browser/components/asrouter/modules/FeatureCallout.sys.mjs
+++ b/browser/components/asrouter/modules/FeatureCallout.sys.mjs
@@ -809,6 +809,7 @@ export class FeatureCallout {
     ) {
       return false;
     }
+    if (el?.id === "zen-site-data-icon-button") return true;
 
     const style = this.win.getComputedStyle(el);
     return style?.visibility === "visible" && style?.display !== "none";
@@ -1107,7 +1108,6 @@ export class FeatureCallout {
             type="arrow"
             consumeoutsideclicks="never"
             norolluponanchor="true"
-            nonnative=""
             position="${panel_position.panel_position_string}"
             ${hide_arrow ? "" : 'show-arrow=""'}
             ${autohide ? "" : 'noautohide="true"'}
