diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 8c4b0e28e53f3f0e0e492dd8f351535a5604af73..968c8792f4a2358da3f89cabfa179585bfbf0874 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -3038,3 +3038,5 @@ pref("startup.homepage_override_nimbus_maxVersion", "");
 
 // Pref to enable the content relevancy feature.
 pref("toolkit.contentRelevancy.enabled", false);
+
+#include zen-browser.js
