diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 1ef5a05907f4675e4e7191dde12c33dc6da5e3c6..69279a1a1278b7034df24b9763c5722a6c623bbc 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -3037,3 +3037,5 @@ pref("browser.profiles.enabled", false);
 
 pref("startup.homepage_override_url_nimbus", "");
 pref("startup.homepage_override_nimbus_maxVersion", "");
+
+#include zen-browser.js
