diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 27c2d13fbd8599a131de6b70bba581d435198793..25aa7546d57e206b76838bc23b609f602af43a83 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -3061,3 +3061,5 @@ pref("startup.homepage_override_nimbus_maxVersion", "");
 pref("toolkit.contentRelevancy.enabled", false);
 // Pref to enable the ingestion through the Rust component.
 pref("toolkit.contentRelevancy.ingestEnabled", false);
+
+#include zen-browser.js
