diff --git a/xpfe/appshell/AppWindow.cpp b/xpfe/appshell/AppWindow.cpp
index 22dbf9028167d5555a91c7087040a1909a27e2a2..356b9722a1a4815f5fdde0ff13bf8f89d7f26f87 100644
--- a/xpfe/appshell/AppWindow.cpp
+++ b/xpfe/appshell/AppWindow.cpp
@@ -1866,7 +1866,7 @@ nsresult AppWindow::MaybeSaveEarlyWindowPersistentValues(
     }
   }
 
-  settings.verticalTabs = Preferences::GetBool("sidebar.verticalTabs", false);
+  settings.verticalTabs = Preferences::GetBool("zen.tabs.vertical", false);
 
   Unused << PersistPreXULSkeletonUIValues(settings);
 #endif
