diff --git a/xpfe/appshell/AppWindow.cpp b/xpfe/appshell/AppWindow.cpp
index 0b7788ec179d1c1021d2e5692c652b367646d33b..f5d48ca47a899b82f9d9cb7823eeb096f542c0ea 100644
--- a/xpfe/appshell/AppWindow.cpp
+++ b/xpfe/appshell/AppWindow.cpp
@@ -1873,7 +1873,7 @@ nsresult AppWindow::MaybeSaveEarlyWindowPersistentValues(
     }
   }
 
-  settings.verticalTabs = Preferences::GetBool("sidebar.verticalTabs", false);
+  settings.verticalTabs = Preferences::GetBool("zen.tabs.vertical", false);
 
   Unused << PersistPreXULSkeletonUIValues(settings);
 #endif
