diff --git a/xpfe/appshell/AppWindow.cpp b/xpfe/appshell/AppWindow.cpp
index 3efb67402877b3af707e67a9e431415c20d77a58..5dd20250aacde2d082f09deae0ad846fed90a8d6 100644
--- a/xpfe/appshell/AppWindow.cpp
+++ b/xpfe/appshell/AppWindow.cpp
@@ -1867,7 +1867,7 @@ nsresult AppWindow::MaybeSaveEarlyWindowPersistentValues(
     }
   }
 
-  settings.verticalTabs = Preferences::GetBool("sidebar.verticalTabs", false);
+  settings.verticalTabs = Preferences::GetBool("zen.tabs.vertical", false);
 
   Unused << PersistPreXULSkeletonUIValues(settings);
 #endif
