diff --git a/browser/actors/WebRTCParent.sys.mjs b/browser/actors/WebRTCParent.sys.mjs
index f327e1684d0966a0dcfdcdbf8cc70259b27a4504..c8f722bda4cf297f5d0a6aa22dafbe27c4218123 100644
--- a/browser/actors/WebRTCParent.sys.mjs
+++ b/browser/actors/WebRTCParent.sys.mjs
@@ -152,6 +152,7 @@ export class WebRTCParent extends JSWindowActorParent {
 
     let tabbrowser = browser.ownerGlobal.gBrowser;
     if (tabbrowser) {
+      browser.ownerGlobal.gZenMediaController.updateMediaSharing(state);
       tabbrowser.updateBrowserSharing(browser, {
         webRTC: state,
       });
