diff --git a/build/moz.build b/build/moz.build
index f7a912ec35dd089ea9a7e712765e954854f55cb3..a84534efbc7662f81573a4a80bc045e0a6d2ed3e 100644
--- a/build/moz.build
+++ b/build/moz.build
@@ -89,7 +89,7 @@ if CONFIG["MOZ_APP_BASENAME"]:
         if CONFIG[var]:
             appini_defines[var] = True
 
-    appini_defines["MOZ_APPUPDATE_HOST"] = "aus5.mozilla.org"
+    appini_defines["MOZ_APPUPDATE_HOST"] = "updates.zen-browser.app"
     if CONFIG["MOZ_APPUPDATE_HOST"]:
         appini_defines["MOZ_APPUPDATE_HOST"] = CONFIG["MOZ_APPUPDATE_HOST"]
 
