diff --git a/toolkit/profile/nsIToolkitProfile.idl b/toolkit/profile/nsIToolkitProfile.idl
index b194cd956382039f24844ded1f2b0edb2fbca2ee..e8a4bce343c7b7a402de3d7d387481a981640fa8 100644
--- a/toolkit/profile/nsIToolkitProfile.idl
+++ b/toolkit/profile/nsIToolkitProfile.idl
@@ -96,4 +96,9 @@ interface nsIToolkitProfile : nsISupports
      * @throws NS_ERROR_FILE_ACCESS_DENIED if the profile was already locked.
      */
     nsIProfileLock lock(out nsIProfileUnlocker aUnlocker);
+
+    /**
+     * Zen Avatar Path
+     */
+    attribute AUTF8String zenAvatarPath;
 };
