diff --git a/toolkit/profile/nsToolkitProfileService.cpp b/toolkit/profile/nsToolkitProfileService.cpp
index c9f742dc4588a21db5011b3386999a5832defd33..4a4a6fc28822200c3e254d682d303b38d10d9c67 100644
--- a/toolkit/profile/nsToolkitProfileService.cpp
+++ b/toolkit/profile/nsToolkitProfileService.cpp
@@ -83,6 +83,8 @@ using namespace mozilla;
 #define INSTALL_PREFIX_LENGTH 7
 #define STORE_ID_PREF "toolkit.profiles.storeID"
 
+#include "zenToolkitProfileServiceDefaultOverride.h"
+
 struct KeyValue {
   KeyValue(const char* aKey, const char* aValue) : key(aKey), value(aValue) {}
 
@@ -1369,7 +1371,7 @@ nsresult nsToolkitProfileService::CreateDefaultProfile(
   if (mUseDevEditionProfile) {
     name.AssignLiteral(DEV_EDITION_NAME);
   } else if (mUseDedicatedProfile) {
-    name.AppendPrintf("default-%s", mUpdateChannel.get());
+    name.AppendPrintf("Default (%s)", mUpdateChannel.get());
   } else {
     name.AssignLiteral(DEFAULT_NAME);
   }
