diff --git a/layout/style/StyleColor.cpp b/layout/style/StyleColor.cpp
index f922f7fdacc7e1a14beb8da77b48aef3f6ae0b2b..47d19242baaf2a1f25ee07c6e210c839cc7b5089 100644
--- a/layout/style/StyleColor.cpp
+++ b/layout/style/StyleColor.cpp
@@ -8,6 +8,7 @@
 #include "mozilla/StyleColorInlines.h"
 #include "nsIFrame.h"
 #include "nsStyleStruct.h"
+#include "mozilla/nsZenBoostsBackend.h"
 #include "nsStyleUtil.h"
 
 namespace mozilla {
@@ -98,23 +99,24 @@ StyleAbsoluteColor StyleColor::ResolveColor(
 
 template <>
 nscolor StyleColor::CalcColor(nscolor aColor) const {
-  return ResolveColor(StyleAbsoluteColor::FromColor(aColor)).ToColor();
+  return ResolveColor(StyleAbsoluteColor::FromColor(aColor)).ToColor(nullptr);
 }
 
 template <>
 nscolor StyleColor::CalcColor(
     const StyleAbsoluteColor& aForegroundColor) const {
-  return ResolveColor(aForegroundColor).ToColor();
+  return ResolveColor(aForegroundColor).ToColor(nullptr);
 }
 
 template <>
-nscolor StyleColor::CalcColor(const ComputedStyle& aStyle) const {
-  return ResolveColor(aStyle.StyleText()->mColor).ToColor();
+nscolor StyleColor::CalcColor(const ComputedStyle& aStyle,
+                              const nsIFrame* aFrame) const {
+  return ResolveColor(aStyle.StyleText()->mColor).ToColor(aFrame);
 }
 
 template <>
 nscolor StyleColor::CalcColor(const nsIFrame* aFrame) const {
-  return ResolveColor(aFrame->StyleText()->mColor).ToColor();
+  return ResolveColor(aFrame->StyleText()->mColor).ToColor(aFrame);
 }
 
 StyleAbsoluteColor StyleAbsoluteColor::ToColorSpace(
@@ -122,7 +124,7 @@ StyleAbsoluteColor StyleAbsoluteColor::ToColorSpace(
   return Servo_ConvertColorSpace(this, aColorSpace);
 }
 
-nscolor StyleAbsoluteColor::ToColor() const {
+nscolor StyleAbsoluteColor::ToColor(const nsIFrame* aFrame) const {
   constexpr StyleColorSpace DEST_COLOR_SPACE = StyleColorSpace::Srgb;
 
   constexpr float MIN = 0.0f;
@@ -164,11 +166,13 @@ nscolor StyleAbsoluteColor::ToColor() const {
     }
   }
 
-  return NS_RGBA(
-      nsStyleUtil::FloatToColorComponent(translatedColor.components._0),
-      nsStyleUtil::FloatToColorComponent(translatedColor.components._1),
-      nsStyleUtil::FloatToColorComponent(translatedColor.components._2),
-      nsStyleUtil::FloatToColorComponent(translatedColor.alpha));
+  return zen::nsZenBoostsBackend::ResolveStyleColor(
+      NS_RGBA(
+          nsStyleUtil::FloatToColorComponent(translatedColor.components._0),
+          nsStyleUtil::FloatToColorComponent(translatedColor.components._1),
+          nsStyleUtil::FloatToColorComponent(translatedColor.components._2),
+          nsStyleUtil::FloatToColorComponent(translatedColor.alpha)),
+      aFrame);
 }
 
 }  // namespace mozilla
