diff --git a/layout/style/StyleColor.cpp b/layout/style/StyleColor.cpp
index 95c7ae6abea5032bef0466e8d59d212374d7a4d0..8dbfbb846b786d51af288989163aacfae12e787c 100644
--- a/layout/style/StyleColor.cpp
+++ b/layout/style/StyleColor.cpp
@@ -8,6 +8,7 @@
 #include "mozilla/dom/BindingDeclarations.h"
 #include "nsIFrame.h"
 #include "nsStyleStruct.h"
+#include "mozilla/nsZenBoostsBackend.h"
 
 namespace mozilla {
 
@@ -21,6 +22,8 @@ bool StyleColor::MaybeTransparent() const {
 template <>
 StyleAbsoluteColor StyleColor::ResolveColor(
     const StyleAbsoluteColor& aForegroundColor) const {
+  auto ResolveColorInner = [this,
+                            &aForegroundColor]() -> StyleAbsoluteColor {
   if (IsAbsolute()) {
     return AsAbsolute();
   }
@@ -30,6 +33,8 @@ StyleAbsoluteColor StyleColor::ResolveColor(
   }
 
   return Servo_ResolveColor(this, &aForegroundColor);
+  };
+  return zen::nsZenBoostsBackend::ResolveStyleColor(ResolveColorInner());
 }
 
 template <>
@@ -68,10 +73,11 @@ nscolor StyleAbsoluteColor::ToColor() const {
   auto green = std::clamp(srgb.components._1, 0.0f, 1.0f);
   auto blue = std::clamp(srgb.components._2, 0.0f, 1.0f);
 
-  return NS_RGBA(nsStyleUtil::FloatToColorComponent(red),
+  return zen::nsZenBoostsBackend::ResolveStyleColor(
+         NS_RGBA(nsStyleUtil::FloatToColorComponent(red),
                  nsStyleUtil::FloatToColorComponent(green),
                  nsStyleUtil::FloatToColorComponent(blue),
-                 nsStyleUtil::FloatToColorComponent(srgb.alpha));
+                 nsStyleUtil::FloatToColorComponent(srgb.alpha)));
 }
 
 }  // namespace mozilla
