diff --git a/browser/components/places/content/bookmarkProperties.xhtml b/browser/components/places/content/bookmarkProperties.xhtml
index 4aad4e4fb4139aa3d81e00eefa82e26b697df973..a2bfac11322f3679af9b51ebc7478323aa8e4adb 100644
--- a/browser/components/places/content/bookmarkProperties.xhtml
+++ b/browser/components/places/content/bookmarkProperties.xhtml
@@ -38,6 +38,8 @@
     />
 
     <html:link rel="localization" href="browser/editBookmarkOverlay.ftl"/>
+    <html:link rel="localization" href="browser/zen-workspaces.ftl"/>
+    <html:link rel="stylesheet" href="chrome://browser/content/zen-styles/zen-popup.css" />
   </linkset>
 
   <stringbundleset id="stringbundleset">
@@ -45,6 +47,8 @@
                   src="chrome://browser/locale/places/bookmarkProperties.properties"/>
   </stringbundleset>
 
+  <script src="chrome://browser/content/zen-components/ZenWorkspacesStorage.mjs" />
+  <script src="chrome://browser/content/zenThemeModifier.js"></script>
   <script src="chrome://browser/content/places/editBookmark.js"/>
   <script src="chrome://browser/content/places/bookmarkProperties.js"/>
   <script src="chrome://global/content/globalOverlay.js"/>
