diff --git a/toolkit/components/prompts/content/commonDialog.xhtml b/toolkit/components/prompts/content/commonDialog.xhtml
index def3b93956ea9b9bfab5791b06040769dbda4645..ab9c61a5d816d17fb3f09c88514aca5ec0d2f193 100644
--- a/toolkit/components/prompts/content/commonDialog.xhtml
+++ b/toolkit/components/prompts/content/commonDialog.xhtml
@@ -28,6 +28,7 @@
 
       <html:link rel="localization" href="branding/brand.ftl" />
       <html:link rel="localization" href="toolkit/global/commonDialog.ftl" />
+      <script src="chrome://browser/content/zenThemeModifier.js"></script>
     </linkset>
     <script src="chrome://global/content/adjustableTitle.js" />
     <script src="chrome://global/content/commonDialog.js" />
