diff --git a/browser/modules/BrowserDOMWindow.sys.mjs b/browser/modules/BrowserDOMWindow.sys.mjs
index e951af5edb966718ee20422c8e46d868c177cd1c..cd41305a4da8e675417eca7df5cd4a1ecee90c3d 100644
--- a/browser/modules/BrowserDOMWindow.sys.mjs
+++ b/browser/modules/BrowserDOMWindow.sys.mjs
@@ -437,7 +437,7 @@ export class BrowserDOMWindow {
     // Passing a null-URI to only create the content window,
     // and pass true for aSkipLoad to prevent loading of
     // about:blank
-    return this.#getContentWindowOrOpenURIInFrame(
+    let res = this.#getContentWindowOrOpenURIInFrame(
       null,
       aParams,
       aWhere,
@@ -445,6 +445,10 @@ export class BrowserDOMWindow {
       aName,
       true
     );
+    if (typeof this.win?.gZenGlanceManager !== "undefined" && this.win.toolbar.visible) {
+      this.win.gZenGlanceManager.onTabOpen(res, aURI);
+    }
+    return res;
   }
 
   /**
