diff --git a/browser/modules/BrowserDOMWindow.sys.mjs b/browser/modules/BrowserDOMWindow.sys.mjs
index 29c5b1510696cfdd993388957115709b1a149d7c..6616edbc5f992ffdf3df65028cd26c17ef13f315 100644
--- a/browser/modules/BrowserDOMWindow.sys.mjs
+++ b/browser/modules/BrowserDOMWindow.sys.mjs
@@ -436,7 +436,7 @@ export class BrowserDOMWindow {
     // Passing a null-URI to only create the content window,
     // and pass true for aSkipLoad to prevent loading of
     // about:blank
-    return this.#getContentWindowOrOpenURIInFrame(
+    let res = this.#getContentWindowOrOpenURIInFrame(
       null,
       aParams,
       aWhere,
@@ -444,6 +444,10 @@ export class BrowserDOMWindow {
       aName,
       true
     );
+    if (typeof this.win?.gZenGlanceManager !== "undefined" && this.win.toolbar.visible) {
+      this.win.gZenGlanceManager.onTabOpen(res, aURI);
+    }
+    return res;
   }
 
   /**
