diff --git a/browser/modules/BrowserWindowTracker.sys.mjs b/browser/modules/BrowserWindowTracker.sys.mjs
index 0f2f66e4bbc27785d53e262001f2ad7f9a1b3216..ed8b600fd2093428e01489d7dc4aa1f5365b645e 100644
--- a/browser/modules/BrowserWindowTracker.sys.mjs
+++ b/browser/modules/BrowserWindowTracker.sys.mjs
@@ -331,6 +331,7 @@ export const BrowserWindowTracker = {
       args = null,
       remote = undefined,
       fission = undefined,
+    zenSyncedWindow = true,
     } = options;
 
     args = lazy.AIWindow.handleAIWindowOptions(options);
@@ -390,6 +391,12 @@ export const BrowserWindowTracker = {
       windowFeatures,
       args
     );
+    win._zenStartupSyncFlag = Services.prefs.getBoolPref("zen.window-sync.prefer-unsynced-windows")
+      ? (zenSyncedWindow ? 'unsynced' : 'synced')
+      : (zenSyncedWindow ? 'synced' : 'unsynced');
+    if (win._zenStartupSyncFlag === 'unsynced' && openerWindow) {
+      win._zenStartupUnsyncedUserContextId = openerWindow.gZenWorkspaces.getCurrentSpaceContainerId();
+    }
     this.registerOpeningWindow(win, isPrivate);
 
     win.addEventListener(
