diff --git a/browser/modules/ExtensionsUI.sys.mjs b/browser/modules/ExtensionsUI.sys.mjs
index 9734d95742035dbe6f608ad3cdaaefc6ed5b83bc..9a7bbb272025e84968d3173faca259f558746c5e 100644
--- a/browser/modules/ExtensionsUI.sys.mjs
+++ b/browser/modules/ExtensionsUI.sys.mjs
@@ -466,7 +466,7 @@ export var ExtensionsUI = {
         eventCallback,
         removeOnDismissal: true,
         popupOptions: {
-          position: "bottomright topright",
+          position: window.gZenUIManager.panelUIPosition,
         },
         // Pass additional options used internally by the
         // addon-webext-permissions-notification custom element
