diff --git a/browser/modules/URILoadingHelper.sys.mjs b/browser/modules/URILoadingHelper.sys.mjs
index bd2e54e914b171df9b8bcc7dcbc2388e9641f6c6..8b3cb6542bea3fb89e47adc93cb13f94447450e2 100644
--- a/browser/modules/URILoadingHelper.sys.mjs
+++ b/browser/modules/URILoadingHelper.sys.mjs
@@ -228,6 +228,7 @@ function openInWindow(url, params, sourceWindow) {
     features,
     sa
   );
+  win._zenStartupSyncFlag = Services.prefs.getBoolPref('zen.window-sync.open-link-in-new-unsynced-window') ? 'unsynced' : 'synced';
 }
 
 function openInCurrentTab(targetBrowser, url, uriObj, params) {
@@ -545,7 +546,7 @@ export const URILoadingHelper = {
       // page. If a load request bounces off for the currently selected tab,
       // we'll open a new tab instead.
       let tab = w.gBrowser.getTabForBrowser(targetBrowser);
-      if (tab == w.FirefoxViewHandler.tab) {
+      if (tab == w.FirefoxViewHandler.tab || tab.hasAttribute("zen-empty-tab")) {
         where = "tab";
         targetBrowser = null;
       } else if (
@@ -978,7 +979,7 @@ export const URILoadingHelper = {
         ignoreQueryString || replaceQueryString,
         ignoreFragmentWhenComparing
       );
-      let browsers = aWindow.gBrowser.browsers;
+      let browsers = aWindow.gZenWorkspaces.allUsedBrowsers;
       for (let i = 0; i < browsers.length; i++) {
         let browser = browsers[i];
         let browserCompare = cleanURL(
@@ -1034,7 +1035,7 @@ export const URILoadingHelper = {
               }
               aSplitView.ownerGlobal.focus();
             } else {
-              aWindow.gBrowser.tabContainer.selectedIndex = i;
+              aWindow.gZenWorkspaces.switchIfNeeded(browser);
             }
           }
 
