diff --git a/browser/modules/URILoadingHelper.sys.mjs b/browser/modules/URILoadingHelper.sys.mjs
index 9f066fe89aceb435eeead29ed9c5477e2007dd0b..2a0a398069d6dc6ce61ac9b23ab8e94e168d9f74 100644
--- a/browser/modules/URILoadingHelper.sys.mjs
+++ b/browser/modules/URILoadingHelper.sys.mjs
@@ -234,6 +234,7 @@ function openInWindow(url, params, sourceWindow) {
     features,
     sa
   );
+  win._zenStartupSyncFlag = Services.prefs.getBoolPref('zen.window-sync.open-link-in-new-unsynced-window') ? 'unsynced' : 'synced';
 }
 
 function openInCurrentTab(targetBrowser, url, uriObj, params) {
@@ -548,7 +549,7 @@ export const URILoadingHelper = {
       // page. If a load request bounces off for the currently selected tab,
       // we'll open a new tab instead.
       let tab = w.gBrowser.getTabForBrowser(targetBrowser);
-      if (tab == w.FirefoxViewHandler.tab) {
+      if (tab == w.FirefoxViewHandler.tab || tab.hasAttribute("zen-empty-tab")) {
         where = "tab";
         targetBrowser = null;
       } else if (
@@ -1006,7 +1007,7 @@ export const URILoadingHelper = {
         ignoreQueryString || replaceQueryString,
         ignoreFragmentWhenComparing
       );
-      let browsers = aWindow.gBrowser.browsers;
+      let browsers = aWindow.gZenWorkspaces.allUsedBrowsers;
       for (let i = 0; i < browsers.length; i++) {
         let browser = browsers[i];
         let browserCompare = cleanURL(
@@ -1062,7 +1063,19 @@ export const URILoadingHelper = {
               }
               aSplitView.documentGlobal.focus();
             } else {
-              aWindow.gBrowser.tabContainer.selectedIndex = i;
+              if (
+                aWindow == window &&
+                window.gURLBar?._zenPickPending &&
+                window.gURLBar._zenHandleUrlbarClose
+              ) {
+                window.gURLBar._zenHandleUrlbarClose(
+                  true,
+                  true,
+                  aWindow.gBrowser.getTabForBrowser(browser)
+                );
+              }
+
+              aWindow.gZenWorkspaces.switchIfNeeded(browser);
             }
           }
 
