diff --git a/browser/modules/URILoadingHelper.sys.mjs b/browser/modules/URILoadingHelper.sys.mjs
index a3a84f96e54e5b74ad15f3825b380c30c117695f..ca702841c1df404a6f270ac08f962c8a225549d8 100644
--- a/browser/modules/URILoadingHelper.sys.mjs
+++ b/browser/modules/URILoadingHelper.sys.mjs
@@ -542,7 +542,7 @@ export const URILoadingHelper = {
       // page. If a load request bounces off for the currently selected tab,
       // we'll open a new tab instead.
       let tab = w.gBrowser.getTabForBrowser(targetBrowser);
-      if (tab == w.FirefoxViewHandler.tab) {
+      if (tab == w.FirefoxViewHandler.tab || tab.hasAttribute("zen-empty-tab")) {
         where = "tab";
         targetBrowser = null;
       } else if (
@@ -972,7 +972,7 @@ export const URILoadingHelper = {
         ignoreQueryString || replaceQueryString,
         ignoreFragmentWhenComparing
       );
-      let browsers = aWindow.gBrowser.browsers;
+      let browsers = aWindow.gZenWorkspaces.allUsedBrowsers;
       for (let i = 0; i < browsers.length; i++) {
         let browser = browsers[i];
         let browserCompare = cleanURL(
@@ -1018,7 +1018,7 @@ export const URILoadingHelper = {
           }
 
           if (!doAdopt) {
-            aWindow.gBrowser.tabContainer.selectedIndex = i;
+            aWindow.gZenWorkspaces.switchIfNeeded(browser);
           }
 
           return true;
