diff --git a/toolkit/mozapps/extensions/internal/AddonSettings.sys.mjs b/toolkit/mozapps/extensions/internal/AddonSettings.sys.mjs
index 09bb0adc97170edb414d8da91c61ec2232aa64c6..8d6fbe2b2322e4d1ccd9de3b338edfbfb4e6448f 100644
--- a/toolkit/mozapps/extensions/internal/AddonSettings.sys.mjs
+++ b/toolkit/mozapps/extensions/internal/AddonSettings.sys.mjs
@@ -119,7 +119,7 @@ if (
 if (AppConstants.MOZ_DEV_EDITION) {
   makeConstant("DEFAULT_THEME_ID", "firefox-compact-dark@mozilla.org");
 } else {
-  makeConstant("DEFAULT_THEME_ID", "default-theme@mozilla.org");
+  makeConstant("DEFAULT_THEME_ID", "firefox-compact-dark@mozilla.org");
 }
 
 // SCOPES_SIDELOAD is a bitflag for what scopes we will load new extensions from when we scan the directories.
