diff --git a/toolkit/mozapps/extensions/internal/XPIProvider.sys.mjs b/toolkit/mozapps/extensions/internal/XPIProvider.sys.mjs
index 12d4fa1172f4f19df63521fac2bc6f065ae455df..981e3c2e22e59447343cd4b35411041656b2b7ed 100644
--- a/toolkit/mozapps/extensions/internal/XPIProvider.sys.mjs
+++ b/toolkit/mozapps/extensions/internal/XPIProvider.sys.mjs
@@ -2546,7 +2546,7 @@ export var XPIProvider = {
         );
       }
       this.maybeInstallBuiltinAddon(
-        "default-theme@mozilla.org",
+        "firefox-compact-dark@mozilla.org",
         "1.3",
         "resource://default-theme/"
       );
