diff --git a/tools/lint/eslint/__init__.py b/tools/lint/eslint/__init__.py
index c0799d3f2317c070a3c99cc91cf6e940de2c5a60..4835906279b1b48010feffb35f6c82bcc9fd1504 100644
--- a/tools/lint/eslint/__init__.py
+++ b/tools/lint/eslint/__init__.py
@@ -133,7 +133,7 @@ def lint(paths, config, binary=None, fix=None, rules=[], setup=None, **lintargs)
         [
             binary,
             os.path.join(
-                module_path, "node_modules", "prettier", "bin", "prettier.cjs"
+                module_path, "..", "node_modules", "@zen-browser", "prettier", "bin", "prettier.cjs"
             ),
             "--list-different",
             "--no-error-on-unmatched-pattern",
