diff --git a/layout/generic/nsTextFrame.cpp b/layout/generic/nsTextFrame.cpp
index 6c3615f90cdb1e5a42e15a0618cc4649cd0d0a15..17eb9d787e83093b0fea2ce43f05d7a447bd5cc9 100644
--- a/layout/generic/nsTextFrame.cpp
+++ b/layout/generic/nsTextFrame.cpp
@@ -6376,7 +6376,7 @@ void nsTextFrame::DrawSelectionDecorations(
           computedStyleFromPseudo->StyleTextReset()->mTextDecorationStyle;
       params.color =
           computedStyleFromPseudo->StyleTextReset()
-              ->mTextDecorationColor.CalcColor(*computedStyleFromPseudo);
+              ->mTextDecorationColor.CalcColor(*computedStyleFromPseudo, this);
       params.decoration =
           computedStyleFromPseudo->StyleTextReset()->mTextDecorationLine;
       params.descentLimit = -1.f;
