diff --git a/layout/generic/nsTextFrame.cpp b/layout/generic/nsTextFrame.cpp
index e7f7943a19d3ac71872d0a8439b6ed90e8fd423d..eb3e84fabc67d95b6d194b5aaf3aaef7c38480c0 100644
--- a/layout/generic/nsTextFrame.cpp
+++ b/layout/generic/nsTextFrame.cpp
@@ -6381,7 +6381,7 @@ void nsTextFrame::DrawSelectionDecorations(
           computedStyleFromPseudo->StyleTextReset()->mTextDecorationStyle;
       params.color =
           computedStyleFromPseudo->StyleTextReset()
-              ->mTextDecorationColor.CalcColor(*computedStyleFromPseudo);
+              ->mTextDecorationColor.CalcColor(*computedStyleFromPseudo, this);
       params.decoration =
           computedStyleFromPseudo->StyleTextReset()->mTextDecorationLine;
       params.descentLimit = -1.f;
