diff --git a/layout/generic/ViewportFrame.cpp b/layout/generic/ViewportFrame.cpp
index ea018fff9f549cd8e07b20f4c8073391a6fdf40d..acb9428cf2276ed04064484600024d9e2c7e2bb7 100644
--- a/layout/generic/ViewportFrame.cpp
+++ b/layout/generic/ViewportFrame.cpp
@@ -24,6 +24,7 @@
 #include "nsLayoutUtils.h"
 #include "nsPlaceholderFrame.h"
 #include "nsSubDocumentFrame.h"
+#include "mozilla/nsZenBoostsBackend.h"
 
 using namespace mozilla;
 
@@ -299,15 +300,20 @@ ViewportFrame::BuildDisplayListForViewTransitionsAndNACTopLayer(
     }
   }
 
+  bool isAnonContent = false;
   if (dom::Element* container = doc->GetCustomContentContainer()) {
     if (nsIFrame* frame = container->GetPrimaryFrame()) {
       MOZ_ASSERT(frame->StyleDisplay()->mTopLayer != StyleTopLayer::None,
                  "ua.css should ensure this");
       MOZ_ASSERT(frame->HasAnyStateBits(NS_FRAME_OUT_OF_FLOW));
+      isAnonContent = frame->ContentIsRootOfNativeAnonymousSubtree();
       BuildDisplayListForTopLayerFrame(aBuilder, frame, &topLayerList);
     }
   }
 
+  if (auto zenBackend = zen::nsZenBoostsBackend::GetInstance()) {
+    zenBackend->mCurrentFrameIsAnonymousContent = isAnonContent;
+  }
   return MaybeWrapTopLayerList(
       aBuilder, uint16_t(TopLayerIndex::ViewTransitionsAndAnonymousContent),
       topLayerList);
