diff --git a/layout/generic/ViewportFrame.cpp b/layout/generic/ViewportFrame.cpp
index 9d1c25c3834da50fd7dfa3a9583144f5817ee231..08b69a63ea3a8067861984fa2fcd25435c921c0d 100644
--- a/layout/generic/ViewportFrame.cpp
+++ b/layout/generic/ViewportFrame.cpp
@@ -23,6 +23,7 @@
 #include "nsLayoutUtils.h"
 #include "nsPlaceholderFrame.h"
 #include "nsSubDocumentFrame.h"
+#include "mozilla/nsZenBoostsBackend.h"
 
 using namespace mozilla;
 
@@ -291,15 +292,20 @@ ViewportFrame::BuildDisplayListForViewTransitionsAndNACTopLayer(
     }
   }
 
+  bool isAnonContent = false;
   if (dom::Element* container = doc->GetCustomContentContainer()) {
     if (nsIFrame* frame = container->GetPrimaryFrame()) {
       MOZ_ASSERT(frame->StyleDisplay()->mTopLayer != StyleTopLayer::None,
                  "ua.css should ensure this");
       MOZ_ASSERT(frame->HasAnyStateBits(NS_FRAME_OUT_OF_FLOW));
+      isAnonContent = frame->ContentIsRootOfNativeAnonymousSubtree();
       BuildDisplayListForTopLayerFrame(aBuilder, frame, &topLayerList);
     }
   }
 
+  if (auto zenBackend = zen::nsZenBoostsBackend::GetInstance()) {
+    zenBackend->mCurrentFrameIsAnonymousContent = isAnonContent;
+  }
   return MaybeWrapTopLayerList(
       aBuilder, uint16_t(TopLayerIndex::ViewTransitionsAndAnonymousContent),
       topLayerList);
