diff --git a/gfx/webrender_bindings/DCLayerTree.cpp b/gfx/webrender_bindings/DCLayerTree.cpp
--- a/gfx/webrender_bindings/DCLayerTree.cpp
+++ b/gfx/webrender_bindings/DCLayerTree.cpp
@@ -2182,18 +2182,18 @@
 
     updatePos = {0, 0};
   }
 
   mFirstDraw = false;
+  LayoutDeviceIntRect rect = widget::WinUtils::ToIntRect(updateRect);
+  MOZ_ASSERT(!rect.IsEmpty());
 
   hr = mCompositionSurface->BeginDraw(&updateRect, __uuidof(ID3D11Texture2D),
                                       (void**)getter_AddRefs(backBuffer),
                                       &offset);
 
   if (FAILED(hr)) {
-    LayoutDeviceIntRect rect = widget::WinUtils::ToIntRect(updateRect);
-
     gfxCriticalNote << "DCLayerCompositionSurface::Bind failed: "
                     << gfx::hexa(hr) << " " << rect;
     RenderThread::Get()->HandleWebRenderError(WebRenderError::BEGIN_DRAW);
     return;
   }
diff --git a/gfx/wr/webrender/src/renderer/composite.rs b/gfx/wr/webrender/src/renderer/composite.rs
--- a/gfx/wr/webrender/src/renderer/composite.rs
+++ b/gfx/wr/webrender/src/renderer/composite.rs
@@ -1120,11 +1120,13 @@
             // Only use supplied clear color for first content layer we encounter
             let clear_color = content_clear_color.take().unwrap_or(ColorF::TRANSPARENT);
 
             if let Some(ref mut _compositor) = self.compositor_config.layer_compositor() {
                 if let Some(PartialPresentMode::Single { dirty_rect }) = partial_present_mode {
-                    if dirty_rect.is_empty() {
+                    let device_rect = DeviceRect::from_size(device_size.to_f32());
+                    let clipped_dirty_rect = dirty_rect.intersection_unchecked(&device_rect);
+                    if clipped_dirty_rect.is_empty() {
                         continue;
                     }
                 }
             }
 

