diff --git a/layout/painting/nsDisplayList.cpp b/layout/painting/nsDisplayList.cpp
index 3f9aea06257f162ded7d7cf8b32fc6597383cf27..78e705366d74d1ce5aec3ab9e41d1e7dfbc19153 100644
--- a/layout/painting/nsDisplayList.cpp
+++ b/layout/painting/nsDisplayList.cpp
@@ -82,6 +82,7 @@
 #include "mozilla/layers/WebRenderLayerManager.h"
 #include "mozilla/layers/WebRenderMessages.h"
 #include "mozilla/layers/WebRenderScrollData.h"
+#include "mozilla/nsZenBoostsBackend.h"
 #include "nsCSSProps.h"
 #include "nsCSSRendering.h"
 #include "nsCSSRenderingGradients.h"
@@ -1158,6 +1159,11 @@ void nsDisplayListBuilder::EnterPresShell(const nsIFrame* aReferenceFrame,
   state->mFirstFrameMarkedForDisplay = mFramesMarkedForDisplay.Length();
   state->mFirstFrameWithOOFData = mFramesWithOOFData.Length();
 
+  // Ignore on anonymous content documents.
+  if (auto zenBackend = zen::nsZenBoostsBackend::GetInstance()) {
+    zenBackend->onPresShellEntered(state->mPresShell->GetPresContext());
+  }
+
   ScrollContainerFrame* sf = state->mPresShell->GetRootScrollContainerFrame();
   if (sf && IsInSubdocument()) {
     // We are forcing a rebuild of nsDisplayCanvasBackgroundColor to make sure
@@ -1343,6 +1349,13 @@ void nsDisplayListBuilder::LeavePresShell(const nsIFrame* aReferenceFrame,
   ResetMarkedFramesForDisplayList(aReferenceFrame);
   mPresShellStates.RemoveLastElement();
 
+  if (auto zenBackend = zen::nsZenBoostsBackend::GetInstance()) {
+    zenBackend->onPresShellLeave(
+      mPresShellStates.IsEmpty()
+          ? nullptr
+          : CurrentPresShellState()->mPresShell->GetPresContext());
+  }
+
   if (!mPresShellStates.IsEmpty()) {
     nsPresContext* pc = CurrentPresContext();
     nsIDocShell* docShell = pc->GetDocShell();
