diff --git a/browser/components/sessionstore/SessionStartup.sys.mjs b/browser/components/sessionstore/SessionStartup.sys.mjs
index be23213ae9ec7e59358a17276c6c3764d38d9996..b6a4110ee50e5977125feb7bd32e84e5e527adc6 100644
--- a/browser/components/sessionstore/SessionStartup.sys.mjs
+++ b/browser/components/sessionstore/SessionStartup.sys.mjs
@@ -40,6 +40,7 @@ ChromeUtils.defineESModuleGetters(lazy, {
   StartupPerformance:
     "resource:///modules/sessionstore/StartupPerformance.sys.mjs",
   sessionStoreLogger: "resource:///modules/sessionstore/SessionLogger.sys.mjs",
+  ZenSessionStore: "resource:///modules/zen/ZenSessionManager.sys.mjs",
 });
 
 const STATE_RUNNING_STR = "running";
@@ -179,6 +180,8 @@ export var SessionStartup = {
       this._initialState = parsed;
     }
 
+    lazy.ZenSessionStore.onFileRead(this._initialState);
+
     if (this._initialState == null) {
       // No valid session found.
       this._sessionType = this.NO_SESSION;
@@ -335,12 +338,7 @@ export var SessionStartup = {
   isAutomaticRestoreEnabled() {
     if (this._resumeSessionEnabled === null) {
       this._resumeSessionEnabled =
-        !lazy.PrivateBrowsingUtils.permanentPrivateBrowsing &&
-        (Services.prefs.getBoolPref(
-          "browser.sessionstore.resume_session_once"
-        ) ||
-          Services.prefs.getIntPref("browser.startup.page") ==
-            BROWSER_STARTUP_RESUME_SESSION);
+        !lazy.PrivateBrowsingUtils.permanentPrivateBrowsing;
     }
 
     return this._resumeSessionEnabled;
