diff --git a/browser/components/sessionstore/SessionSaver.sys.mjs b/browser/components/sessionstore/SessionSaver.sys.mjs
index 1f26bc5fdb83f55bdf0b9b7d01fafe1cd37f77f4..5916f741d0fa04c392dcdd10f843e595fe5e058c 100644
--- a/browser/components/sessionstore/SessionSaver.sys.mjs
+++ b/browser/components/sessionstore/SessionSaver.sys.mjs
@@ -20,6 +20,7 @@ const lazy = XPCOMUtils.declareLazy({
     "moz-src:///browser/components/sessionstore/SessionStore.sys.mjs",
   sessionStoreLogger:
     "moz-src:///browser/components/sessionstore/SessionLogger.sys.mjs",
+  ZenSessionStore: "resource:///modules/zen/ZenSessionManager.sys.mjs",
 });
 
 /*
@@ -250,7 +251,7 @@ var SessionSaverInternal = {
     // Cancel any pending timeouts.
     this.cancel();
 
-    if (lazy.PrivateBrowsingUtils.permanentPrivateBrowsing) {
+    if (false) {
       // Don't save (or even collect) anything in permanent private
       // browsing mode
 
@@ -305,6 +306,7 @@ var SessionSaverInternal = {
     this._maybeClearCookiesAndStorage(state);
 
     Glean.sessionRestore.collectData.stopAndAccumulate(timerId);
+    lazy.ZenSessionStore.saveState(state);
     return this._writeState(state);
   },
 
