diff --git a/browser/components/sessionstore/SessionStore.sys.mjs b/browser/components/sessionstore/SessionStore.sys.mjs
index f814772114948f87cbb3c3a7231c95ea1f68d776..ae35daa518ca25f4fc95ef983519c390bf088a68 100644
--- a/browser/components/sessionstore/SessionStore.sys.mjs
+++ b/browser/components/sessionstore/SessionStore.sys.mjs
@@ -3171,7 +3171,7 @@ var SessionStoreInternal = {
     if (!isPrivateWindow && tabState.isPrivate) {
       return;
     }
-    if (aTab == aWindow.FirefoxViewHandler.tab) {
+    if (aTab == aWindow.FirefoxViewHandler.tab || (aTab.hasAttribute("zen-empty-tab") && !aTab.hasAttribute("allow-split"))) {
       return;
     }
 
@@ -3876,6 +3876,7 @@ var SessionStoreInternal = {
         Math.min(tabState.index, tabState.entries.length)
       );
       tabState.pinned = false;
+      tabState.zenEssential = undefined;
 
       if (inBackground === false) {
         aWindow.gBrowser.selectedTab = newTab;
@@ -5355,14 +5356,14 @@ var SessionStoreInternal = {
     }
 
     let tabbrowser = aWindow.gBrowser;
-    let tabs = tabbrowser.tabs;
+    let tabs = aWindow.ZenWorkspaces.allStoredTabs;
     /** @type {WindowStateData} */
     let winData = this._windows[aWindow.__SSi];
     let tabsData = (winData.tabs = []);
 
     // update the internal state data for this window
     for (let tab of tabs) {
-      if (tab == aWindow.FirefoxViewHandler.tab) {
+      if (tab == aWindow.FirefoxViewHandler.tab || (tab.hasAttribute("zen-empty-tab") && !tab.hasAttribute("allow-split"))) {
         continue;
       }
       let tabData = lazy.TabState.collect(tab, TAB_CUSTOM_VALUES.get(tab));
@@ -5381,7 +5382,7 @@ var SessionStoreInternal = {
     // We don't store the Firefox View tab in Session Store, so if it was the last selected "tab" when
     // a window is closed, point to the first item in the tab strip instead (it will never be the Firefox View tab,
     // since it's only inserted into the tab strip after it's selected).
-    if (aWindow.FirefoxViewHandler.tab?.selected) {
+    if (aWindow.FirefoxViewHandler.tab?.selected || (tabbrowser.selectedTab.hasAttribute("zen-empty-tab") && !tabbrowser.selectedTab.hasAttribute("allow-split"))) {
       selectedIndex = 1;
       winData.title = tabbrowser.tabs[0].label;
     }
@@ -6086,8 +6087,23 @@ var SessionStoreInternal = {
 
     // Most of tabData has been restored, now continue with restoring
     // attributes that may trigger external events.
+    if (tabData.zenEssential) {
+      tab.setAttribute("zen-essential", "true");
+    }
+    if (tabData.zenIsEmpty) {
+      tab.setAttribute("zen-empty-tab", "true");
+      if (tabData.zenAllowSplit) {
+        tab.setAttribute("allow-split", "true");
+      }
+    }
+    if (tabData.zenHasStaticLabel) {
+      tab.setAttribute("zen-has-static-label", "true");
+    }
+    if (tabData.zenPinnedId) {
+      tab.setAttribute("zen-pin-id", tabData.zenPinnedId);
+    }
+    if (tabData.zenDefaultUserContextId) {
+      tab.setAttribute("zenDefaultUserContextId", true);
+    }
 
-    if (tabData.pinned) {
+    if (tabData.pinned || tabData.zenEssential) {
       tabbrowser.pinTab(tab);
     } else {
       tabbrowser.unpinTab(tab);
