diff --git a/browser/components/sessionstore/SessionFile.sys.mjs b/browser/components/sessionstore/SessionFile.sys.mjs
index 3c5cfef8c5c4c4439b2f4eca57a9423f9ae3f0e0..4daf32d50c7ad11fea7f073be5c32397c4c467c7 100644
--- a/browser/components/sessionstore/SessionFile.sys.mjs
+++ b/browser/components/sessionstore/SessionFile.sys.mjs
@@ -25,6 +25,7 @@ const lazy = XPCOMUtils.declareLazy({
     "moz-src:///browser/components/sessionstore/SessionStore.sys.mjs",
   SessionWriter:
     "moz-src:///browser/components/sessionstore/SessionWriter.sys.mjs",
+  ZenSessionStore: "resource:///modules/zen/ZenSessionManager.sys.mjs",
 });
 
 const PREF_UPGRADE_BACKUP = "browser.sessionstore.upgradeBackup.latestBuildID";
@@ -387,6 +388,7 @@ var SessionFileInternal = {
     }
     this._readOrigin = result.origin;
 
+    await lazy.ZenSessionStore.readFile();
     return { ...result, noFilesFound };
   },
 
