diff --git a/browser/components/sessionstore/TabGroupState.sys.mjs b/browser/components/sessionstore/TabGroupState.sys.mjs
index 1289549cf33ce6efb620c877bd1ba1c24475d2b0..ff128b9dc8cdcb418b8cc2a416026b58f8df4a1f 100644
--- a/browser/components/sessionstore/TabGroupState.sys.mjs
+++ b/browser/components/sessionstore/TabGroupState.sys.mjs
@@ -65,6 +65,11 @@ class _TabGroupState {
    */
   collect(tabGroup) {
     return {
+      pinned: tabGroup.pinned,
+      essential: tabGroup.essential,
+      splitView: tabGroup.hasAttribute("split-view-group"),
+      parentGroupId: tabGroup.group?.id,
+      isZenFolder: tabGroup.isZenFolder,
       id: tabGroup.id,
       name: tabGroup.label,
       color: tabGroup.color,
