diff --git a/browser/components/sessionstore/TabState.sys.mjs b/browser/components/sessionstore/TabState.sys.mjs
index bb7bd8c7ca88f776188fce64e0c85beb5ceb27a5..bdec77ba26172fe1b41369b1d45d5a6f9d52ae25 100644
--- a/browser/components/sessionstore/TabState.sys.mjs
+++ b/browser/components/sessionstore/TabState.sys.mjs
@@ -10,6 +10,7 @@ const lazy = XPCOMUtils.declareLazy({
     "moz-src:///browser/components/sessionstore/TabAttributes.sys.mjs",
   TabStateCache:
     "moz-src:///browser/components/sessionstore/TabStateCache.sys.mjs",
+  UrlbarShared: "chrome://browser/content/urlbar/UrlbarShared.mjs",
   sessionStoreLogger:
     "moz-src:///browser/components/sessionstore/SessionLogger.sys.mjs",
 });
@@ -106,10 +107,28 @@ class _TabState {
       }
     }
 
+    tabData.zenWorkspace = tab.getAttribute("zen-workspace-id");
+    tabData.zenSyncId = tab.getAttribute("id");
+    tabData.zenEssential = tab.getAttribute("zen-essential") === "true";
+    tabData.pinned = tabData.pinned || tabData.zenEssential;
+    tabData.zenDefaultUserContextId = tab.getAttribute("zenDefaultUserContextId");
+    tabData.zenPinnedIcon = tab.getAttribute("zen-pinned-icon");
+    tabData.zenIsEmpty = tab.hasAttribute("zen-empty-tab");
+    tabData.zenStaticLabel = tab.zenStaticLabel;
+    tabData.zenHasStaticIcon = !!tab.zenStaticIcon;
+    tabData.zenGlanceId = tab.getAttribute("glance-id");
+    tabData.zenIsGlance = tab.hasAttribute("zen-glance-tab");
+    tabData._zenPinnedInitialState = tab._zenPinnedInitialState;
+    tabData._zenIsActiveTab = tab._zenContentsVisible;
+    tabData.zenLiveFolderItemId = tab.getAttribute("zen-live-folder-item-id");
+
     tabData.searchMode = tab.documentGlobal.gURLBar.getSearchMode(
       browser,
       true
     );
+    if (tabData.searchMode?.source === lazy.UrlbarShared.RESULT_SOURCE.ZEN_ACTIONS) {
+      delete tabData.searchMode;
+    }
 
     tabData.userContextId = tab.userContextId || 0;
 
@@ -135,6 +154,10 @@ class _TabState {
       tabData.image = tabbrowser.getIcon(tab);
     }
 
+    if (tab.zenStaticIcon) {
+      tabData.image = tab.zenStaticIcon;
+    }
+
     // If there is a userTypedValue set, then either the user has typed something
     // in the URL bar, or a new tab was opened with a URI to load.
     // If so, we also track whether we were still in the process of loading something.
