diff --git a/browser/components/sessionstore/TabState.sys.mjs b/browser/components/sessionstore/TabState.sys.mjs
index 4ba4dda363b602cb6f4445ef056f2f9abb1b0e1e..6b4b407e05e26faca69d9d7ac6f31510620898fe 100644
--- a/browser/components/sessionstore/TabState.sys.mjs
+++ b/browser/components/sessionstore/TabState.sys.mjs
@@ -99,10 +99,28 @@ class _TabState {
       }
     }
 
+    tabData.zenWorkspace = tab.getAttribute("zen-workspace-id");
+    tabData.zenSyncId = tab.getAttribute("id");
+    tabData.zenEssential = tab.getAttribute("zen-essential") === "true";
+    tabData.pinned = tabData.pinned || tabData.zenEssential;
+    tabData.zenDefaultUserContextId = tab.getAttribute("zenDefaultUserContextId");
+    tabData.zenPinnedIcon = tab.getAttribute("zen-pinned-icon");
+    tabData.zenIsEmpty = tab.hasAttribute("zen-empty-tab");
+    tabData.zenStaticLabel = tab.zenStaticLabel;
+    tabData.zenHasStaticIcon = !!tab.zenStaticIcon;
+    tabData.zenGlanceId = tab.getAttribute("glance-id");
+    tabData.zenIsGlance = tab.hasAttribute("zen-glance-tab");
+    tabData._zenPinnedInitialState = tab._zenPinnedInitialState;
+    tabData._zenIsActiveTab = tab._zenContentsVisible;
+    tabData.zenLiveFolderItemId = tab.getAttribute("zen-live-folder-item-id");
+
     tabData.searchMode = tab.documentGlobal.gURLBar.getSearchMode(
       browser,
       true
     );
+    if (tabData.searchMode?.source === tab.documentGlobal.UrlbarUtils.RESULT_SOURCE.ZEN_ACTIONS) {
+      delete tabData.searchMode;
+    }
 
     tabData.userContextId = tab.userContextId || 0;
 
