diff --git a/browser/components/sessionstore/TabState.sys.mjs b/browser/components/sessionstore/TabState.sys.mjs
index 8f7ed557e6aa61e7e16ed4a8d785ad5fe651b3d8..aff63696d198055886960072a6130318e099ae42 100644
--- a/browser/components/sessionstore/TabState.sys.mjs
+++ b/browser/components/sessionstore/TabState.sys.mjs
@@ -80,10 +80,20 @@ class _TabState {
       tabData.muteReason = tab.muteReason;
     }
 
-    if (tab.group) {
+    if (tab.group && !tab.group.hasAttribute("split-view-group")) {
       tabData.groupId = tab.group.id;
     }
 
+    tabData.zenWorkspace = tab.getAttribute("zen-workspace-id");
+    tabData.zenPinnedId = tab.getAttribute("zen-pin-id");
+    tabData.zenEssential = tab.getAttribute("zen-essential");
+    tabData.pinned = tabData.pinned || tabData.zenEssential;
+    tabData.zenDefaultUserContextId = tab.getAttribute("zenDefaultUserContextId");
+    tabData.zenPinnedEntry = tab.getAttribute("zen-pinned-entry");
+    tabData.zenPinnedIcon = tab.getAttribute("zen-pinned-icon");
+    tabData.zenIsEmpty = tab.hasAttribute("zen-empty-tab");
+    tabData.zenHasStaticLabel = tab.hasAttribute("zen-has-static-label");
+    tabData.zenAllowSplit = tab.hasAttribute("allow-split");
+
     tabData.searchMode = tab.ownerGlobal.gURLBar.getSearchMode(browser, true);
 
     tabData.userContextId = tab.userContextId || 0;
