diff --git a/layout/style/test/ListCSSProperties.cpp b/layout/style/test/ListCSSProperties.cpp
index 673c1677703ef2de3fafbe86f0ffab720738f4cb..267556f66de3180fc6ae91d5b54b8648f1b41cc4 100644
--- a/layout/style/test/ListCSSProperties.cpp
+++ b/layout/style/test/ListCSSProperties.cpp
@@ -110,6 +110,7 @@ const char* gInaccessibleProperties[] = {
     "-moz-window-opacity",                // chrome-only internal properties
     "-moz-window-transform",              // chrome-only internal properties
     "-moz-window-shadow",                 // chrome-only internal properties
+    "-zen-window-transform-origin",       // chrome-only internal properties
 };
 
 inline int is_inaccessible(const char* aPropName) {
