diff --git a/browser/base/content/aboutRestartRequired.xhtml b/browser/base/content/aboutRestartRequired.xhtml
index c1f42b4ddd3b154a8643fc5265c556904805bdff..9722e7de76b02077addfeda90a9acd8b921a9cd6 100644
--- a/browser/base/content/aboutRestartRequired.xhtml
+++ b/browser/base/content/aboutRestartRequired.xhtml
@@ -29,6 +29,7 @@
     />
     <link rel="localization" href="branding/brand.ftl" />
     <link rel="localization" href="browser/aboutRestartRequired.ftl" />
+    <script src="chrome://browser/content/zenThemeModifier.js"></script>
   </head>
   <body>
     <!-- PAGE CONTAINER (for styling purposes only) -->
