diff --git a/browser/base/content/browser-places.js b/browser/base/content/browser-places.js
index e3feaebc551bfe7127b9701a7f76996022516af0..460b39f2a9f1f36d4b821c693951214098b644f7 100644
--- a/browser/base/content/browser-places.js
+++ b/browser/base/content/browser-places.js
@@ -261,6 +261,8 @@ var StarUI = {
       });
     }
 
+    this._setIconAndPreviewImage();
+
     let onPanelReady = fn => {
       let target = this.panel;
       if (target.parentNode) {
@@ -312,6 +314,21 @@ var StarUI = {
     }
   },
 
+  _setIconAndPreviewImage() {
+    let faviconImage = this._element("zenEditBookmarkPanelFavicon");
+    faviconImage.removeAttribute("iconloadingprincipal");
+    faviconImage.removeAttribute("src");
+
+    let tab = gBrowser.selectedTab;
+    if (tab.hasAttribute("image") && !tab.hasAttribute("busy")) {
+      faviconImage.setAttribute(
+        "iconloadingprincipal",
+        tab.getAttribute("iconloadingprincipal")
+      );
+      faviconImage.setAttribute("src", tab.getAttribute("image"));
+    }
+  },
+
   removeBookmarkButtonCommand: function SU_removeBookmarkButtonCommand() {
     this._removeBookmarksOnPopupHidden = true;
     this.panel.hidePopup();
@@ -1815,6 +1832,7 @@ var BookmarkingUI = {
       document.getElementById("context-bookmarkpage"),
       PanelMultiView.getViewNode(document, "panelMenuBookmarkThisPage"),
       document.getElementById("pageAction-panel-bookmark"),
+      document.getElementById("zen-site-data-icon-button"),
     ]) {
       if (!element) {
         // The page action panel element may not have been created yet.
