diff --git a/browser/base/content/browser.js b/browser/base/content/browser.js
index b3940467bc6b9170cbc75f0a1dfd161a896cc8bd..7f5105b8524c15f50d18028e04564fc9fe8b6deb 100644
--- a/browser/base/content/browser.js
+++ b/browser/base/content/browser.js
@@ -32,6 +32,7 @@ ChromeUtils.defineESModuleGetters(this, {
   ContextualIdentityService:
     "resource://gre/modules/ContextualIdentityService.sys.mjs",
   CustomizableUI: "resource:///modules/CustomizableUI.sys.mjs",
+  ZenCustomizableUI: "chrome://browser/content/ZenCustomizableUI.sys.mjs",
   DevToolsSocketStatus:
     "resource://devtools/shared/security/DevToolsSocketStatus.sys.mjs",
   DownloadUtils: "resource://gre/modules/DownloadUtils.sys.mjs",
@@ -3444,6 +3445,11 @@ var XULBrowserWindow = {
     AboutReaderParent.updateReaderButton(gBrowser.selectedBrowser);
     TranslationsParent.onLocationChange(gBrowser.selectedBrowser);
 
+    gZenViewSplitter.onLocationChange(gBrowser.selectedBrowser);
+    ZenWorkspaces.onLocationChange(gBrowser.selectedBrowser);
+    gZenTabUnloader.onLocationChange(gBrowser.selectedBrowser);
+    gZenPinnedTabManager.onLocationChange(gBrowser.selectedBrowser);
+
     PictureInPicture.updateUrlbarToggle(gBrowser.selectedBrowser);
 
     if (!gMultiProcessBrowser) {
@@ -4439,7 +4445,7 @@ nsBrowserAccess.prototype = {
     // Passing a null-URI to only create the content window,
     // and pass true for aSkipLoad to prevent loading of
     // about:blank
-    return this.getContentWindowOrOpenURIInFrame(
+    let res = this.getContentWindowOrOpenURIInFrame(
       null,
       aParams,
       aWhere,
@@ -4447,6 +4453,10 @@ nsBrowserAccess.prototype = {
       aName,
       true
     );
+    if (typeof window.gZenGlanceManager !== "undefined" && window.toolbar.visible) {
+      window.gZenGlanceManager.onTabOpen(res, aURI);
+    }
+    return res;
   },
 
   openURIInFrame: function browser_openURIInFrame(
@@ -5850,7 +5860,7 @@ function warnAboutClosingWindow() {
 
   if (!isPBWindow && !toolbar.visible) {
     return gBrowser.warnAboutClosingTabs(
-      gBrowser.openTabs.length,
+      gBrowser.visibleTabs.length - gBrowser.pinnedTabCount,
       gBrowser.closingTabsEnum.ALL
     );
   }
@@ -5890,7 +5900,7 @@ function warnAboutClosingWindow() {
     return (
       isPBWindow ||
       gBrowser.warnAboutClosingTabs(
-        gBrowser.openTabs.length,
+        gBrowser.visibleTabs.length - gBrowser.pinnedTabCount,
         gBrowser.closingTabsEnum.ALL
       )
     );
@@ -6375,7 +6385,7 @@ function switchToTabHavingURI(
       ignoreQueryString || replaceQueryString,
       ignoreFragmentWhenComparing
     );
-    let browsers = aWindow.gBrowser.browsers;
+    let browsers = aWindow.ZenWorkspaces.allUsedBrowsers;
     for (let i = 0; i < browsers.length; i++) {
       let browser = browsers[i];
       let browserCompare = cleanURL(
@@ -6416,7 +6426,7 @@ function switchToTabHavingURI(
         }
 
         if (!doAdopt) {
-          aWindow.gBrowser.tabContainer.selectedIndex = i;
+          aWindow.ZenWorkspaces.switchIfNeeded(browser, i);
         }
 
         return true;
