diff --git a/browser/base/content/browser-init.js b/browser/base/content/browser-init.js
index b64d2f52ed1eb3d0b8aa86700bc082d3fc95152b..dd8636148d0646f6949d6279d70e20c09d8c7c17 100644
--- a/browser/base/content/browser-init.js
+++ b/browser/base/content/browser-init.js
@@ -252,6 +252,7 @@ var gBrowserInit = {
   },
 
   onLoad() {
+    Services.scriptloader.loadSubScript("chrome://browser/content/zenThemeModifier.js", window);
     gBrowser.addEventListener("DOMUpdateBlockedPopups", e =>
       PopupAndRedirectBlockerObserver.handleEvent(e)
     );
@@ -446,6 +447,7 @@ var gBrowserInit = {
 
     this._handleURIToLoad();
 
+    gZenWorkspaces.selectStartPage();
     Services.obs.addObserver(gIdentityHandler, "perm-changed");
     Services.obs.addObserver(gRemoteControl, "devtools-socket");
     Services.obs.addObserver(gRemoteControl, "marionette-listening");
