diff --git a/browser/base/content/browser-init.js b/browser/base/content/browser-init.js
index bee5309c04775adff8652bfe6c54b2d466e821ac..e9741628dcc9eda3500f69441bae293548179e71 100644
--- a/browser/base/content/browser-init.js
+++ b/browser/base/content/browser-init.js
@@ -143,6 +143,7 @@ var gBrowserInit = {
     gNavToolbox.palette = document.getElementById(
       "BrowserToolbarPalette"
     ).content;
+    ZenCustomizableUI.init(window);
     for (let area of CustomizableUI.areas) {
       let type = CustomizableUI.getAreaType(area);
       if (type == CustomizableUI.TYPE_TOOLBAR) {
@@ -239,6 +240,10 @@ var gBrowserInit = {
     gPrivateBrowsingUI.init();
     BrowserSearch.init();
     BrowserPageActions.init();
+
+    Services.scriptloader.loadSubScript("chrome://browser/content/ZenStartup.mjs", window);
+    Services.scriptloader.loadSubScript("chrome://browser/content/zenThemeModifier.js", window);
+
     if (gToolbarKeyNavEnabled) {
       ToolbarKeyboardNavigator.init();
     }
