diff --git a/browser/base/content/browser-sidebar.js b/browser/base/content/browser-sidebar.js
index ea6457a5c42f7bd2e002bd9234b8eb9bd23bbcc0..9a7b75481fc0968c295bae9ca46ac5da4aec96dc 100644
--- a/browser/base/content/browser-sidebar.js
+++ b/browser/base/content/browser-sidebar.js
@@ -305,6 +305,7 @@ var SidebarUI = {
       this._box.setAttribute("positionend", true);
     } else {
       this._box.removeAttribute("positionend");
+      this._box.style.order = 0;
     }
 
     this.hideSwitcherPanel();
