diff --git a/browser/base/content/browser-init.js b/browser/base/content/browser-init.js
index 0aa4baea8e5e6d742fc6cd3333068e7eb64b5bbc..057bb2386420edd31154979338e71d00a51a7266 100644
--- a/browser/base/content/browser-init.js
+++ b/browser/base/content/browser-init.js
@@ -307,6 +307,7 @@ var gBrowserInit = {
   },
 
   onLoad() {
+    Services.scriptloader.loadSubScript("chrome://browser/content/zenThemeModifier.js", window);
     gBrowser.addEventListener("DOMUpdateBlockedPopups", e =>
       PopupAndRedirectBlockerObserver.handleEvent(e)
     );
@@ -501,6 +502,7 @@ var gBrowserInit = {
 
     this._handleURIToLoad();
 
+    gZenWorkspaces.selectStartPage();
     Services.obs.addObserver(gIdentityHandler, "perm-changed");
     Services.obs.addObserver(gRemoteControl, "devtools-socket");
     Services.obs.addObserver(gRemoteControl, "marionette-listening");
