diff --git a/browser/base/content/browser-gestureSupport.js b/browser/base/content/browser-gestureSupport.js
index a28d54bf72c0e6495b9586f220d1859aac794936..411d7255a68c48643617d77cc279a0a831fdf5c9 100644
--- a/browser/base/content/browser-gestureSupport.js
+++ b/browser/base/content/browser-gestureSupport.js
@@ -247,6 +247,10 @@ var gGestureSupport = {
         : aEvent.DIRECTION_LEFT;
     }
 
+    if (!gHistorySwipeAnimation._isSupported()) {
+      return;
+    }
+
     return canGoBack || canGoForward;
   },
 
@@ -832,7 +836,7 @@ var gHistorySwipeAnimation = {
    * @return true if there is a previous page in history, false otherwise.
    */
   canGoBack: function HSA_canGoBack() {
-    return gBrowser.webNavigation.canGoBack;
+    return gBrowser.webNavigation.canGoBack || gZenCommonActions.shouldCloseTabOnBack();
   },
 
   /**
