diff --git a/browser/base/content/browser-gestureSupport.js b/browser/base/content/browser-gestureSupport.js
index f94b39d26a1a9626b3e1f518e2395bc394d1262d..15e198f4bb440cff0bbfb8ccd97c2ce180d5724f 100644
--- a/browser/base/content/browser-gestureSupport.js
+++ b/browser/base/content/browser-gestureSupport.js
@@ -247,6 +247,10 @@ var gGestureSupport = {
         : aEvent.DIRECTION_LEFT;
     }
 
+    if (!gHistorySwipeAnimation._isSupported()) {
+      return;
+    }
+
     return canGoBack || canGoForward;
   },
 
@@ -848,7 +852,7 @@ var gHistorySwipeAnimation = {
    * @return true if there is a previous page in history, false otherwise.
    */
   canGoBack: function HSA_canGoBack() {
-    return gBrowser.webNavigation.canGoBack;
+    return gBrowser.webNavigation.canGoBack || gZenCommonActions.shouldCloseTabOnBack();
   },
 
   /**
