diff --git a/browser/base/content/browser.xhtml b/browser/base/content/browser.xhtml
index 3063370347db9e2812ad30205e585c849e44c91e..be74d9e624b8220cb9bd009bae31fb513c4ec73f 100644
--- a/browser/base/content/browser.xhtml
+++ b/browser/base/content/browser.xhtml
@@ -139,6 +139,8 @@
   window.addEventListener("DOMContentLoaded",
     gBrowserInit.onDOMContentLoaded.bind(gBrowserInit), { once: true });
 </script>
+#include zen-locales.inc.xhtml
+#include zen-assets.inc.xhtml
 </head>
 <html:body xmlns="http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul">
 # All sets except for popupsets (commands, keys, and stringbundles)
@@ -160,9 +162,12 @@
     </vbox>
   </html:template>
 
-#include navigator-toolbox.inc.xhtml
-
-#include browser-box.inc.xhtml
+  <hbox id="zen-main-app-wrapper" flex="1">
+    #include navigator-toolbox.inc.xhtml
+    <html:span id="zen-sidebar-box-container">
+    </html:span>
+    #include browser-box.inc.xhtml    
+  </hbox>
 
   <html:template id="customizationPanel">
     <box id="customization-container" flex="1" hidden="true"><![CDATA[
@@ -176,5 +181,6 @@
 
   <!-- Put it at the very end to make sure it's not covered by anything. -->
   <html:div id="fullscr-toggler" hidden="hidden"/>
+#include zen-watermark.inc.xhtml
 </html:body>
 </html>
