diff --git a/browser/base/content/browser-addons.js b/browser/base/content/browser-addons.js
index f33a300eb0eed74e286da24919ad3e8505f632c3..b45c1e5f3b93643ece3b8214e33d40b093689671 100644
--- a/browser/base/content/browser-addons.js
+++ b/browser/base/content/browser-addons.js
@@ -1069,7 +1069,7 @@ var gXPInstallObserver = {
       persistent: true,
       hideClose: true,
       popupOptions: {
-        position: "bottomright topright",
+        position: gZenUIManager.panelUIPosition(),
       },
     };

@@ -1279,7 +1279,7 @@ var gXPInstallObserver = {
       hideClose: true,
       timeout: Date.now() + 30000,
       popupOptions: {
-        position: "bottomright topright",
+        position: gZenUIManager.panelUIPosition(),
       },
     };

@@ -2211,7 +2211,7 @@ var gUnifiedExtensions = {
   // If the new ID is not added in NOTIFICATION_IDS, consider handling the case
   // in the "PopupNotificationsBeforeAnchor" handler elsewhere in this file.
   getPopupAnchorID(aBrowser, aWindow) {
-    const anchorID = "unified-extensions-button";
+    const anchorID = "zen-site-data-icon-button";
     const attr = anchorID + "popupnotificationanchor";

     if (!aBrowser[attr]) {
@@ -2222,7 +2222,7 @@ var gUnifiedExtensions = {
         anchorID
         // Anchor on the toolbar icon to position the popup right below the
         // button.
-      ).firstElementChild;
+      );
     }

     return anchorID;
@@ -2536,7 +2536,7 @@ var gUnifiedExtensions = {
     requestAnimationFrame(() => this.updateAttention());
   },

-  onToolbarVisibilityChange(toolbarId, isVisible) {
+  onToolbarVisibilityChange(toolbarId, isVisible, panel = this.panel) {
     // A list of extension widget IDs (possibly empty).
     let widgetIDs;

@@ -2550,7 +2550,7 @@ var gUnifiedExtensions = {
     }

     // The list of overflowed extensions in the extensions panel.
-    const overflowedExtensionsList = this.panel.querySelector(
+    const overflowedExtensionsList = panel.querySelector(
       "#overflowed-extensions-list"
     );

@@ -2681,7 +2681,7 @@ var gUnifiedExtensions = {

   // `aEvent` and `reason` are optional. If `reason` is specified, it should be
   // a valid argument to gUnifiedExtensions.recordButtonTelemetry().
-  async togglePanel(aEvent, reason) {
+  async togglePanel(aEvent, reason, panel = this._panel, view, button) {
     if (!CustomizationHandler.isCustomizing()) {
       if (aEvent) {
         if (
@@ -2718,32 +2718,30 @@ var gUnifiedExtensions = {
       this.blocklistAttentionInfo =
         await AddonManager.getBlocklistAttentionInfo();

-      let panel = this.panel;
-
       if (!this._listView) {
         this._listView = PanelMultiView.getViewNode(
           document,
-          "unified-extensions-view"
+          view,
         );
         this._listView.addEventListener("ViewShowing", this);
         this._listView.addEventListener("ViewHiding", this);
       }

-      if (this._button.open) {
+      if (button.open) {
         PanelMultiView.hidePopup(panel);
-        this._button.open = false;
+        button.open = false;
       } else {
         // Overflow extensions placed in collapsed toolbars, if any.
         for (const toolbarId of CustomizableUI.getCollapsedToolbarIds(window)) {
           // We pass `false` because all these toolbars are collapsed.
-          this.onToolbarVisibilityChange(toolbarId, /* isVisible */ false);
+          this.onToolbarVisibilityChange(toolbarId, /* isVisible */ false, panel);
         }

         panel.hidden = false;
         this.recordButtonTelemetry(reason || "extensions_panel_showing");
         this.ensureButtonShownBeforeAttachingPanel(panel);
-        PanelMultiView.openPopup(panel, this._button, {
-          position: "bottomright topright",
+        PanelMultiView.openPopup(panel, button, {
+          position: gZenUIManager.panelUIPosition(panel, button),
           triggerEvent: aEvent,
         });
       }
@@ -2830,7 +2828,7 @@ var gUnifiedExtensions = {
       if (
         notInPanel ||
         document.querySelector("#unified-extensions-area > :first-child")
-          ?.id === widgetId
+          ?.id === widgetIdz
       ) {
         moveUp.hidden = true;
       }
@@ -2930,18 +2928,20 @@ var gUnifiedExtensions = {
       this._maybeMoveWidgetNodeBack(widgetId);
     }

-    this.pinToToolbar(widgetId, shouldPinToToolbar);
+    await this.pinToToolbar(widgetId, shouldPinToToolbar);
   },

-  pinToToolbar(widgetId, shouldPinToToolbar) {
+  async pinToToolbar(widgetId, shouldPinToToolbar) {
     let newArea = shouldPinToToolbar
       ? CustomizableUI.AREA_NAVBAR
       : CustomizableUI.AREA_ADDONS;
     let newPosition = shouldPinToToolbar ? undefined : 0;
+    await gZenVerticalTabsManager._preCustomize();

     CustomizableUI.addWidgetToArea(widgetId, newArea, newPosition);
     // addWidgetToArea() will trigger onWidgetAdded or onWidgetMoved as needed,
     // and our handlers will call updateAttention() as needed.
+    await gZenVerticalTabsManager._postCustomize();
   },

   async moveWidget(menu, direction) {
