diff --git a/browser/base/content/browser.js b/browser/base/content/browser.js
index b18f371a47fec53882c3dfb68e34d1a474eaf446..2c183d7e7fc65e00403f5e40e7859b915a3ea700 100644
--- a/browser/base/content/browser.js
+++ b/browser/base/content/browser.js
@@ -630,6 +630,15 @@ XPCOMUtils.defineLazyPreferenceGetter(
   false
 );
 
+const ZEN_WELCOME_PATH = "zen-welcome";
+const ZEN_WELCOME_ELEMENT_ATTR = "zen-dialog-welcome-element";
+XPCOMUtils.defineLazyServiceGetter(
+  this,
+  "ProfileService",
+  "@mozilla.org/toolkit/profile-service;1",
+  "nsIToolkitProfileService"
+);
+
 customElements.setElementCreationCallback("screenshots-buttons", () => {
   Services.scriptloader.loadSubScript(
     "chrome://browser/content/screenshots/screenshots-buttons.js",
@@ -3679,6 +3688,9 @@ var XULBrowserWindow = {
     AboutReaderParent.updateReaderButton(gBrowser.selectedBrowser);
     TranslationsParent.onLocationChange(gBrowser.selectedBrowser);
 
+    gZenViewSplitter.onLocationChange(gBrowser.selectedBrowser);
+    ZenWorkspaces.onLocationChange(gBrowser.selectedBrowser);
+
     PictureInPicture.updateUrlbarToggle(gBrowser.selectedBrowser);
 
     if (!gMultiProcessBrowser) {
@@ -5001,7 +5013,7 @@ function setToolbarVisibility(
       );
     }
 
-    const overlapAttr = "BookmarksToolbarOverlapsBrowser";
+    const overlapAttr = "BookmarksToolbarOverlapsBrowser__ignore"; // Original string was "BookmarksToolbarOverlapsBrowser" but it's not used and it only bugs the UI.
     switch (isVisible) {
       case true:
       case "always":
@@ -7832,6 +7844,12 @@ var gDialogBox = {
     parentElement.showModal();
     this._didOpenHTMLDialog = true;
 
+    if (uri.includes(ZEN_WELCOME_PATH)) {
+      parentElement.setAttribute(ZEN_WELCOME_ELEMENT_ATTR, true);
+    } else if (parentElement.hasAttribute(ZEN_WELCOME_ELEMENT_ATTR)) {
+      parentElement.removeAttribute(ZEN_WELCOME_ELEMENT_ATTR);
+    }
+
     // Disable menus and shortcuts.
     this._updateMenuAndCommandState(false /* to disable */);
 
