diff --git a/browser/base/content/browser.xhtml b/browser/base/content/browser.xhtml
index aec0983a6704f09300da9a3a9ab76e627695b708..4228c96f59d6054c19c95686cafa979594e7da1f 100644
--- a/browser/base/content/browser.xhtml
+++ b/browser/base/content/browser.xhtml
@@ -140,6 +140,8 @@
   window.addEventListener("DOMContentLoaded",
     gBrowserInit.onDOMContentLoaded.bind(gBrowserInit), { once: true });
 </script>
+#include zen-locales.inc.xhtml
+#include zen-scripts.inc.xhtml
 </head>
 <html:body xmlns="http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul">
 # All sets except for popupsets (commands, keys, and stringbundles)
@@ -161,9 +163,12 @@
     </vbox>
   </html:template>
 
-#include navigator-toolbox.inc.xhtml
-
-#include browser-box.inc.xhtml
+  <hbox id="zen-main-app-wrapper" flex="1">
+    #include navigator-toolbox.inc.xhtml
+    <html:span id="zen-sidebar-box-container">
+    </html:span>
+    #include browser-box.inc.xhtml    
+  </hbox>
 
   <html:template id="customizationPanel">
     <box id="customization-container" flex="1" hidden="true"><![CDATA[
