diff --git a/browser/base/content/browser-box.inc.xhtml b/browser/base/content/browser-box.inc.xhtml
index d445abe7e743b2ec5d46d42f8f149c000e3c7997..c45459f1ae13190c5a508def867159d52cc6c1e1 100644
--- a/browser/base/content/browser-box.inc.xhtml
+++ b/browser/base/content/browser-box.inc.xhtml
@@ -2,22 +2,11 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-<hbox flex="1" id="browser">
-  <vbox id="sidebar-box" hidden="true" class="chromeclass-extrachrome">
-    <box id="sidebar-header" align="center">
-      <toolbarbutton id="sidebar-switcher-target" class="tabbable" aria-expanded="false">
-        <image id="sidebar-icon" consumeanchor="sidebar-switcher-target"/>
-        <label id="sidebar-title" crop="end" control="sidebar"/>
-        <image id="sidebar-switcher-arrow"/>
-      </toolbarbutton>
-      <image id="sidebar-throbber"/>
-      <spacer id="sidebar-spacer"/>
-      <toolbarbutton id="sidebar-close" class="close-icon tabbable" data-l10n-id="sidebar-close-button" oncommand="SidebarUI.hide();"/>
-    </box>
-    <browser id="sidebar" autoscroll="false" disablehistory="true" disablefullscreen="true" tooltip="aHTMLTooltip"/>
-  </vbox>
-  <splitter id="sidebar-splitter" class="chromeclass-extrachrome sidebar-splitter" resizebefore="sibling" resizeafter="none" hidden="true"/>
+<vbox id="browser">
   <vbox id="appcontent" flex="1">
+
+#include zen-browser-box-ui.inc.xhtml
+
     <!-- gNotificationBox will be added here lazily. -->
     <tabbox id="tabbrowser-tabbox"
             flex="1" tabcontainer="tabbrowser-tabs">
@@ -25,4 +14,4 @@
                   flex="1" selectedIndex="0"/>
     </tabbox>
   </vbox>
-</hbox>
+</vbox>
