diff --git a/browser/base/content/browser-box.inc.xhtml b/browser/base/content/browser-box.inc.xhtml
index 16559b2c0e3aef391b43e09281aee24007feb0bd..2579cea92a55fabb73fdd130c3706af08c3fd849 100644
--- a/browser/base/content/browser-box.inc.xhtml
+++ b/browser/base/content/browser-box.inc.xhtml
@@ -21,11 +21,15 @@
   </vbox>
   <splitter id="sidebar-splitter" class="chromeclass-extrachrome sidebar-splitter" resizebefore="sibling" resizeafter="none" hidden="true"/>
   <vbox id="appcontent" flex="1">
+    <html:div id="zen-appcontent-navbar-container"></html:div>
     <!-- gNotificationBox will be added here lazily. -->
+    <hbox id="zen-tabbox-wrapper" flex="1">
     <tabbox id="tabbrowser-tabbox"
             flex="1" tabcontainer="tabbrowser-tabs">
+#include zen-sidebar-panel.inc.xhtml
       <tabpanels id="tabbrowser-tabpanels"
                   flex="1" selectedIndex="0"/>
     </tabbox>
+  </hbox>
   </vbox>
 </hbox>
