diff --git a/browser/base/content/browser-sets.js b/browser/base/content/browser-sets.js
index 50845ebd925955bf9646fc499b9828eccbf266ef..5482a24576ebcd4382409cae01a4c6bcbd88a96c 100644
--- a/browser/base/content/browser-sets.js
+++ b/browser/base/content/browser-sets.js
@@ -270,7 +270,7 @@ document.addEventListener(
         }
       });
 
-    document.getElementById("mainKeyset").addEventListener("command", event => {
+    document.getElementById("zenKeyset").addEventListener("command", event => {
       const SIDEBAR_REVAMP_PREF = "sidebar.revamp";
       const SIDEBAR_REVAMP_ENABLED = Services.prefs.getBoolPref(
         SIDEBAR_REVAMP_PREF,
@@ -308,6 +308,8 @@ document.addEventListener(
               panel: SidebarController.currentID,
               opened: SidebarController._state.launcherExpanded,
             });
+          } else {
+            SidebarController.toggle();
           }
           break;
         case "key_gotoHistory":
