diff --git a/browser/base/content/browser-siteIdentity.js b/browser/base/content/browser-siteIdentity.js
index 65354e0ad3097e10de72f36e257c94472524baf7..866cf5672f30f81a1d8e061c9af0c1af5fbb5fb2 100644
--- a/browser/base/content/browser-siteIdentity.js
+++ b/browser/base/content/browser-siteIdentity.js
@@ -453,11 +453,6 @@ var gIdentityHandler = {
     // avoid a pretty ugly transition. Also hide it even
     // if the update resulted in no site data, to keep the
     // illusion that clicking the button had an effect.
-    let hidden = new Promise(c => {
-      this._identityPopup.addEventListener("popuphidden", c, { once: true });
-    });
-    PanelMultiView.hidePopup(this._identityPopup);
-    await hidden;
 
     let baseDomain = SiteDataManager.getBaseDomainFromHost(this._uri.host);
     if (SiteDataManager.promptSiteDataRemoval(window, [baseDomain])) {
@@ -834,7 +829,7 @@ var gIdentityHandler = {
       // This is a secure internal Firefox page.
       this._identityBox.className = "chromeUI";
       let brandBundle = document.getElementById("bundle_brand");
-      icon_label = brandBundle.getString("brandShorterName");
+      icon_label = brandBundle.getString("brandShortName");
     } else if (this._pageExtensionPolicy) {
       // This is a WebExtension page.
       this._identityBox.className = "extensionPage";
@@ -1163,6 +1158,12 @@ var gIdentityHandler = {
       }
     }
 
+    if (connection == "chrome" || connection == "extension" || connection == "file") {
+      document.getElementById("identity-popup-mainView-panel-header").setAttribute("hidden", "true");
+    } else {
+      document.getElementById("identity-popup-mainView-panel-header").removeAttribute("hidden");
+    }
+
     // Push the appropriate strings out to the UI.
     document.l10n.setAttributes(
       this._identityPopupMainViewHeaderLabel,
