diff --git a/browser/base/content/browser-pageActions.js b/browser/base/content/browser-pageActions.js
index 00da33bc11189db17b6a2e656acb3a778531197c..bdca62e9ed5513e1e1922f1f9498755f5459e0c2 100644
--- a/browser/base/content/browser-pageActions.js
+++ b/browser/base/content/browser-pageActions.js
@@ -449,7 +449,7 @@ var BrowserPageActions = {
       document.getElementById(
         action && this.urlbarButtonNodeIDForActionID(action.id)
       ),
-      document.getElementById(this.mainButtonNode.id),
+      document.getElementById("zen-site-data-icon-button"),
       document.getElementById("identity-icon"),
     ];
     for (let node of potentialAnchorNodes) {
