diff --git a/browser/base/content/browser.js b/browser/base/content/browser.js
index 4960f4baed32a9e9f523262591eb93550ca9c4b0..9ee825057b1b3b5a08e1d2b2c1036f1be2088da6 100644
--- a/browser/base/content/browser.js
+++ b/browser/base/content/browser.js
@@ -34,6 +34,7 @@ ChromeUtils.defineESModuleGetters(this, {
     "moz-src:///toolkit/components/contextualidentity/ContextualIdentityService.sys.mjs",
   CustomizableUI:
     "moz-src:///browser/components/customizableui/CustomizableUI.sys.mjs",
+  ZenCustomizableUI: "resource:///modules/ZenCustomizableUI.sys.mjs",
   DevToolsSocketStatus:
     "resource://devtools/shared/security/DevToolsSocketStatus.sys.mjs",
   DownloadUtils: "resource://gre/modules/DownloadUtils.sys.mjs",
@@ -831,7 +832,12 @@ function UpdateBackForwardCommands(aWebNavigation) {
 
   var backDisabled = backCommand.hasAttribute("disabled");
   var forwardDisabled = forwardCommand.hasAttribute("disabled");
-  if (backDisabled == aWebNavigation.canGoBack) {
+  var canGoBack = aWebNavigation.canGoBack;
+  if (!canGoBack) {
+    canGoBack = gZenCommonActions.shouldCloseTabOnBack();
+  }
+  
+  if (backDisabled == canGoBack) {
     if (backDisabled) {
       backCommand.removeAttribute("disabled");
     } else {
@@ -3973,7 +3979,7 @@ function warnAboutClosingWindow() {
 
   if (!isPBWindow && !toolbar.visible) {
     return gBrowser.warnAboutClosingTabs(
-      gBrowser.openTabs.length,
+      gBrowser.visibleTabs.filter(t => t.getAttribute("pending")!=="true").length,
       gBrowser.closingTabsEnum.ALL
     );
   }
@@ -4013,7 +4019,7 @@ function warnAboutClosingWindow() {
     return (
       isPBWindow ||
       gBrowser.warnAboutClosingTabs(
-        gBrowser.openTabs.length,
+        gBrowser.visibleTabs.filter(t => t.getAttribute("pending")!=="true").length,
         gBrowser.closingTabsEnum.ALL
       )
     );
@@ -4038,7 +4044,7 @@ function warnAboutClosingWindow() {
     AppConstants.platform != "macosx" ||
     isPBWindow ||
     gBrowser.warnAboutClosingTabs(
-      gBrowser.openTabs.length,
+      gBrowser.visibleTabs.filter(t => t.getAttribute("pending")!=="true").length,
       gBrowser.closingTabsEnum.ALL
     )
   );
@@ -4998,6 +5004,16 @@ var ConfirmationHint = {
     }
 
     document.l10n.setAttributes(this._message, messageId, l10nArgs);
+    if (window.gZenUIManager?.showToast) {
+      return window.gZenUIManager.showToast(messageId, {
+        hideCheckmark,
+        descriptionId,
+        l10nArgs,
+        showDescription,
+        position,
+        event,
+      });
+    }
     if (descriptionId) {
       document.l10n.setAttributes(this._description, descriptionId);
       this._description.hidden = false;
