diff --git a/browser/base/content/browser-pageActions.js b/browser/base/content/browser-pageActions.js
index 00da33bc11189db17b6a2e656acb3a778531197c..9571155baccad9a886cbe9c7bf0bd76a135331c4 100644
--- a/browser/base/content/browser-pageActions.js
+++ b/browser/base/content/browser-pageActions.js
@@ -451,6 +451,7 @@ var BrowserPageActions = {
       ),
       document.getElementById(this.mainButtonNode.id),
       document.getElementById("identity-icon"),
+      document.getElementById("zen-site-data-icon-button"),
     ];
     for (let node of potentialAnchorNodes) {
       if (node && !node.hidden) {
