diff --git a/browser/base/content/browser.js b/browser/base/content/browser.js
index e2e0526a0ddd617291f1f6c17bcfb807954b481f..7a4c3f95e9590d1f83f470f59052174d5f43f805 100644
--- a/browser/base/content/browser.js
+++ b/browser/base/content/browser.js
@@ -33,6 +33,7 @@ ChromeUtils.defineESModuleGetters(this, {
     "resource://gre/modules/ContextualIdentityService.sys.mjs",
   CustomizableUI:
     "moz-src:///browser/components/customizableui/CustomizableUI.sys.mjs",
+  ZenCustomizableUI: "resource:///modules/ZenCustomizableUI.sys.mjs",
   DevToolsSocketStatus:
     "resource://devtools/shared/security/DevToolsSocketStatus.sys.mjs",
   DownloadUtils: "resource://gre/modules/DownloadUtils.sys.mjs",
@@ -814,7 +815,12 @@ function UpdateBackForwardCommands(aWebNavigation) {
 
   var backDisabled = backCommand.hasAttribute("disabled");
   var forwardDisabled = forwardCommand.hasAttribute("disabled");
-  if (backDisabled == aWebNavigation.canGoBack) {
+  var canGoBack = aWebNavigation.canGoBack;
+  if (!canGoBack) {
+    canGoBack = gZenCommonActions.shouldCloseTabOnBack();
+  }
+  
+  if (backDisabled == canGoBack) {
     if (backDisabled) {
       backCommand.removeAttribute("disabled");
     } else {
@@ -2301,6 +2307,8 @@ var XULBrowserWindow = {
     AboutReaderParent.updateReaderButton(gBrowser.selectedBrowser);
     TranslationsParent.onLocationChange(gBrowser.selectedBrowser);
 
+    gZenPinnedTabManager.onLocationChange(gBrowser.selectedBrowser);
+
     PictureInPicture.updateUrlbarToggle(gBrowser.selectedBrowser);
 
     if (!gMultiProcessBrowser) {
@@ -3821,7 +3829,7 @@ function warnAboutClosingWindow() {
 
   if (!isPBWindow && !toolbar.visible) {
     return gBrowser.warnAboutClosingTabs(
-      gBrowser.openTabs.length,
+      gBrowser.visibleTabs.filter(t => t.getAttribute("pending")!=="true").length,
       gBrowser.closingTabsEnum.ALL
     );
   }
@@ -3861,7 +3869,7 @@ function warnAboutClosingWindow() {
     return (
       isPBWindow ||
       gBrowser.warnAboutClosingTabs(
-        gBrowser.openTabs.length,
+        gBrowser.visibleTabs.filter(t => t.getAttribute("pending")!=="true").length,
         gBrowser.closingTabsEnum.ALL
       )
     );
@@ -3886,7 +3894,7 @@ function warnAboutClosingWindow() {
     AppConstants.platform != "macosx" ||
     isPBWindow ||
     gBrowser.warnAboutClosingTabs(
-      gBrowser.openTabs.length,
+      gBrowser.visibleTabs.filter(t => t.getAttribute("pending")!=="true").length,
       gBrowser.closingTabsEnum.ALL
     )
   );
@@ -4074,8 +4082,10 @@ function safeModeRestart() {
  *  "window"      new window
  *
  * delta is the offset to the history entry that you want to load.
+ * 
+ * onRestore is called after the new tab is restored properly.
  */
-function duplicateTabIn(aTab, where, delta) {
+function duplicateTabIn(aTab, where, delta, onRestore = () => {}) {
   switch (where) {
     case "window": {
       let otherWin = OpenBrowserWindow({
@@ -4089,7 +4099,7 @@ function duplicateTabIn(aTab, where, delta) {
           Services.obs.removeObserver(delayedStartupFinished, topic);
           let otherGBrowser = otherWin.gBrowser;
           let otherTab = otherGBrowser.selectedTab;
-          SessionStore.duplicateTab(otherWin, aTab, delta);
+          SessionStore.duplicateTab(otherWin, aTab, delta, true, {}, onRestore);
           otherGBrowser.removeTab(otherTab, { animate: false });
         }
       };
@@ -4101,13 +4111,13 @@ function duplicateTabIn(aTab, where, delta) {
       break;
     }
     case "tabshifted":
-      SessionStore.duplicateTab(window, aTab, delta);
+      SessionStore.duplicateTab(window, aTab, delta, true, {}, onRestore);
       // A background tab has been opened, nothing else to do here.
       break;
     case "tab":
       SessionStore.duplicateTab(window, aTab, delta, true, {
         inBackground: false,
-      });
+      }, onRestore);
       break;
   }
   if (aTab.group) {
@@ -4809,6 +4819,9 @@ var ConfirmationHint = {
     MozXULElement.insertFTLIfNeeded("toolkit/branding/brandings.ftl");
     MozXULElement.insertFTLIfNeeded("browser/confirmationHints.ftl");
     document.l10n.setAttributes(this._message, messageId, options.l10nArgs);
+    if (window.gZenUIManager?.showToast) {
+      return window.gZenUIManager.showToast(messageId, options);
+    }
     if (options.descriptionId) {
       document.l10n.setAttributes(this._description, options.descriptionId);
       this._description.hidden = false;
