diff --git a/browser/base/content/browser-init.js b/browser/base/content/browser-init.js
index 63100defacf66c6b3232b9e0a783a5fd14e3a46a..22b79b021ff0baab4dac602447a124308208c828 100644
--- a/browser/base/content/browser-init.js
+++ b/browser/base/content/browser-init.js
@@ -162,13 +162,15 @@ var gBrowserInit = {
         elem.setAttribute("skipintoolbarset", "true");
       }
     }
+    ZenCustomizableUI.init(window);
     for (let area of CustomizableUI.areas) {
       let type = CustomizableUI.getAreaType(area);
-      if (type == CustomizableUI.TYPE_TOOLBAR) {
+      if (type == CustomizableUI.TYPE_TOOLBAR && !area.startsWith("zen-")) {
         let node = document.getElementById(area);
         CustomizableUI.registerToolbarNode(node);
       }
     }
+    ZenCustomizableUI.registerToolbarNodes(window);
     if (isVerticalTabs) {
       // Show the vertical tabs toolbar
       setToolbarVisibility(
@@ -287,6 +289,10 @@ var gBrowserInit = {
     gPrivateBrowsingUI.init();
     BrowserSearch.init();
     BrowserPageActions.init();
+
+Services.scriptloader.loadSubScript("chrome://browser/content/ZenStartup.mjs", window);
+Services.scriptloader.loadSubScript("chrome://browser/content/zenThemeModifier.js", window);
+
     if (gToolbarKeyNavEnabled) {
       ToolbarKeyboardNavigator.init();
     }
