diff --git a/browser/base/content/browser-siteProtections.js b/browser/base/content/browser-siteProtections.js
index 5364aa74cd938141e85e3a3333e8d1fa58ea3ae4..5cd1748a883b19caa2977c1047d51b9551686bf4 100644
--- a/browser/base/content/browser-siteProtections.js
+++ b/browser/base/content/browser-siteProtections.js
@@ -1904,9 +1904,14 @@ var gProtectionsHandler = {
       // the information contained there would mostly be broken and/or
       // irrelevant anyway.
       this._trackingProtectionIconContainer.hidden = true;
+      // ZEN: We make a clear distinction between native UI and websites for security reasons.
+      //   we can remove the separator for websites that dont need it, like "about:*" pages, where
+      //   they are safe to use.
+      this._nativeSeparatorForWebsitesContainer.hidden = true;
       return;
     }
     this._trackingProtectionIconContainer.hidden = false;
+    this._nativeSeparatorForWebsitesContainer.hidden = false;
 
     // Check whether the user has added an exception for this site.
     this.hasException = ContentBlockingAllowList.includes(
@@ -2882,4 +2887,12 @@ var gProtectionsHandler = {
 
     return messageEl;
   },
+
+  get _nativeSeparatorForWebsitesContainer() {
+    delete this._nativeSeparatorForWebsitesContainer;
+    return (this._nativeSeparatorForWebsitesContainer = document.getElementById(
+      "zen-website-and-native-separator"
+    ));
+  },
+
 };
