diff --git a/browser/base/content/browser-addons.js b/browser/base/content/browser-addons.js
index 73593191936cc345ee8e2c28cb251dc13f4c2fd4..e6c459c1ebc60a1f3930a55e212570f696bf07a0 100644
--- a/browser/base/content/browser-addons.js
+++ b/browser/base/content/browser-addons.js
@@ -735,7 +735,7 @@ var gXPInstallObserver = {
       persistent: true,
       hideClose: true,
       popupOptions: {
-        position: "bottomright topright",
+        position: gZenUIManager.panelUIPosition,
       },
     };
 
@@ -942,7 +942,7 @@ var gXPInstallObserver = {
       hideClose: true,
       timeout: Date.now() + 30000,
       popupOptions: {
-        position: "bottomright topright",
+        position: gZenUIManager.panelUIPosition,
       },
     };
 
@@ -2125,7 +2125,7 @@ var gUnifiedExtensions = {
 
         panel.hidden = false;
         PanelMultiView.openPopup(panel, this._button, {
-          position: "bottomright topright",
+          position: gZenUIManager.panelUIPosition,
           triggerEvent: aEvent,
         });
       }
@@ -2294,18 +2294,20 @@ var gUnifiedExtensions = {
       this._maybeMoveWidgetNodeBack(widgetId);
     }
 
-    this.pinToToolbar(widgetId, shouldPinToToolbar);
+    await this.pinToToolbar(widgetId, shouldPinToToolbar);
   },
 
-  pinToToolbar(widgetId, shouldPinToToolbar) {
+  async pinToToolbar(widgetId, shouldPinToToolbar) {
     let newArea = shouldPinToToolbar
       ? CustomizableUI.AREA_NAVBAR
       : CustomizableUI.AREA_ADDONS;
     let newPosition = shouldPinToToolbar ? undefined : 0;
+    await gZenVerticalTabsManager._preCustomize();
 
     CustomizableUI.addWidgetToArea(widgetId, newArea, newPosition);
 
     this.updateAttention();
+    await gZenVerticalTabsManager._postCustomize();
   },
 
   async moveWidget(menu, direction) {
