diff --git a/browser/base/content/browser-places.js b/browser/base/content/browser-places.js
index 3bd7b8f6025a7df4ce5cfb66916125bce9ef3991..c408ef829b7540aec72b2307e5d4bfa34109d6cd 100644
--- a/browser/base/content/browser-places.js
+++ b/browser/base/content/browser-places.js
@@ -249,6 +249,8 @@ var StarUI = {
       });
     }
 
+    this._setIconAndPreviewImage();
+
     let onPanelReady = fn => {
       let target = this.panel;
       if (target.parentNode) {
@@ -300,6 +302,21 @@ var StarUI = {
     }
   },
 
+  _setIconAndPreviewImage() {
+    let faviconImage = this._element("zenEditBookmarkPanelFavicon");
+    faviconImage.removeAttribute("iconloadingprincipal");
+    faviconImage.removeAttribute("src");
+
+    let tab = gBrowser.selectedTab;
+    if (tab.hasAttribute("image") && !tab.hasAttribute("busy")) {
+      faviconImage.setAttribute(
+        "iconloadingprincipal",
+        tab.getAttribute("iconloadingprincipal")
+      );
+      faviconImage.setAttribute("src", tab.getAttribute("image"));
+    }
+  },
+
   removeBookmarkButtonCommand: function SU_removeBookmarkButtonCommand() {
     this._removeBookmarksOnPopupHidden = true;
     this.panel.hidePopup();
@@ -1797,6 +1814,7 @@ var BookmarkingUI = {
       document.getElementById("context-bookmarkpage"),
       PanelMultiView.getViewNode(document, "panelMenuBookmarkThisPage"),
       document.getElementById("pageAction-panel-bookmark"),
+      document.getElementById("zen-site-data-icon-button"),
     ]) {
       if (!element) {
         // The page action panel element may not have been created yet.
