diff --git a/browser/base/content/browser.js b/browser/base/content/browser.js
index 6ea9b0dbcd3bfbddd74eff63df06a8637619b990..250fe064fcb5e267a27c5b645392bb1392fee3fd 100644
--- a/browser/base/content/browser.js
+++ b/browser/base/content/browser.js
@@ -34,6 +34,7 @@ ChromeUtils.defineESModuleGetters(this, {
     "moz-src:///toolkit/components/contextualidentity/ContextualIdentityService.sys.mjs",
   CustomizableUI:
     "moz-src:///browser/components/customizableui/CustomizableUI.sys.mjs",
+  ZenCustomizableUI: "resource:///modules/ZenCustomizableUI.sys.mjs",
   DevToolsSocketStatus:
     "resource://devtools/shared/security/DevToolsSocketStatus.sys.mjs",
   DownloadUtils: "resource://gre/modules/DownloadUtils.sys.mjs",
@@ -852,7 +853,12 @@ function UpdateBackForwardCommands(aWebNavigation) {
 
   var backDisabled = backCommand.hasAttribute("disabled");
   var forwardDisabled = forwardCommand.hasAttribute("disabled");
-  if (backDisabled == aWebNavigation.canGoBack) {
+  var canGoBack = aWebNavigation.canGoBack;
+  if (!canGoBack) {
+    canGoBack = gZenCommonActions.shouldCloseTabOnBack();
+  }
+  
+  if (backDisabled == canGoBack) {
     if (backDisabled) {
       backCommand.removeAttribute("disabled");
     } else {
@@ -3985,7 +3991,7 @@ function warnAboutClosingWindow() {
 
   if (!isPBWindow && !toolbar.visible) {
     return gBrowser.warnAboutClosingTabs(
-      gBrowser.openTabs.length,
+      gBrowser.visibleTabs.filter(t => t.getAttribute("pending")!=="true").length,
       gBrowser.closingTabsEnum.ALL
     );
   }
@@ -4025,7 +4031,7 @@ function warnAboutClosingWindow() {
     return (
       isPBWindow ||
       gBrowser.warnAboutClosingTabs(
-        gBrowser.openTabs.length,
+        gBrowser.visibleTabs.filter(t => t.getAttribute("pending")!=="true").length,
         gBrowser.closingTabsEnum.ALL
       )
     );
@@ -4050,7 +4056,7 @@ function warnAboutClosingWindow() {
     AppConstants.platform != "macosx" ||
     isPBWindow ||
     gBrowser.warnAboutClosingTabs(
-      gBrowser.openTabs.length,
+      gBrowser.visibleTabs.filter(t => t.getAttribute("pending")!=="true").length,
       gBrowser.closingTabsEnum.ALL
     )
   );
@@ -4763,9 +4769,13 @@ var gDialogBox = {
     // Bring the window to the front in case we're minimized or occluded:
     window.focus();
 
-    // Prevent urlbar views from showing on top of the modal.
-    for (let urlbar of document.querySelectorAll(".urlbar")) {
-      urlbar.view?.close();
+    try {
+      // Prevent urlbars from showing on top of modal
+      for (let urlbar of document.querySelectorAll(".urlbar")) {
+        urlbar.incrementPopoverBlockerCount();
+      }
+    } catch (ex) {
+      console.error(ex);
     }
 
     try {
@@ -4791,6 +4801,10 @@ var gDialogBox = {
       this._updateMenuAndCommandState(true /* to enable */);
       this._dialog = null;
       UpdatePopupNotificationsVisibility();
+      // Restore the urlbar popovers if needed
+      for (let urlbar of document.querySelectorAll(".urlbar")) {
+        urlbar.decrementPopoverBlockerCount();
+      }
     }
     if (this._queued.length) {
       setTimeout(() => this._openNextDialog(), 0);
@@ -4976,6 +4990,16 @@ var ConfirmationHint = {
     }
 
     document.l10n.setAttributes(this._message, messageId, l10nArgs);
+    if (window.gZenUIManager?.showToast) {
+      return window.gZenUIManager.showToast(messageId, {
+        hideCheckmark,
+        descriptionId,
+        l10nArgs,
+        showDescription,
+        position,
+        event,
+      });
+    }
     if (descriptionId) {
       document.l10n.setAttributes(this._description, descriptionId);
       this._description.hidden = false;
