diff --git a/browser/base/content/main-popupset.inc.xhtml b/browser/base/content/main-popupset.inc.xhtml
index e5f3424eaeeec0ba552537f167dd99e912216d94..4bdfcdb23fe9c44ad3d4de273c64f4cc31cb4034 100644
--- a/browser/base/content/main-popupset.inc.xhtml
+++ b/browser/base/content/main-popupset.inc.xhtml
@@ -181,6 +181,10 @@
            hidden="true"
            tabspecific="true"
            aria-labelledby="editBookmarkPanelTitle">
+      <html:div id="zenEditBookmarkPanelFaviconContainer">
+        <html:img id="zenEditBookmarkPanelFavicon"/>
+      </html:div>
+      <vbox>
       <box class="panel-header">
         <html:h1>
           <html:span id="editBookmarkPanelTitle"/>
@@ -206,6 +210,7 @@
                   class="footer-button"/>
         </html:moz-button-group>
       </vbox>
+    </vbox>
     </panel>
   </html:template>
 
@@ -535,6 +540,8 @@
 
 #include popup-notifications.inc.xhtml
 
+#include zen-popupset.inc.xhtml
+
 #include ../../components/customizableui/content/panelUI.inc.xhtml
 #include ../../components/controlcenter/content/identityPanel.inc.xhtml
 #include ../../components/controlcenter/content/permissionPanel.inc.xhtml
