diff --git a/browser/base/content/main-popupset.inc.xhtml b/browser/base/content/main-popupset.inc.xhtml
index 1fb595272a184f9a40f56f87d86232e3324f7750..8dd0f0ff856be524a5fa27fb8c6180c1fe058134 100644
--- a/browser/base/content/main-popupset.inc.xhtml
+++ b/browser/base/content/main-popupset.inc.xhtml
@@ -144,6 +144,10 @@
            hidden="true"
            tabspecific="true"
            aria-labelledby="editBookmarkPanelTitle">
+      <html:div id="zenEditBookmarkPanelFaviconContainer">
+        <html:img id="zenEditBookmarkPanelFavicon"/>
+      </html:div>
+      <vbox>
       <box class="panel-header">
         <html:h1>
           <html:span id="editBookmarkPanelTitle"/>
@@ -169,6 +173,7 @@
                   class="footer-button"/>
         </html:moz-button-group>
       </vbox>
+      </vbox>
     </panel>
   </html:template>
 
@@ -454,6 +459,8 @@
 
 #include popup-notifications.inc
 
+#include zen-popupset.inc.xhtml
+
 #include ../../components/customizableui/content/panelUI.inc.xhtml
 #include ../../components/controlcenter/content/identityPanel.inc.xhtml
 #include ../../components/controlcenter/content/permissionPanel.inc.xhtml
