diff --git a/browser/base/content/main-popupset.inc.xhtml b/browser/base/content/main-popupset.inc.xhtml
index ef8245938ea669227c255d85422a26b99cb2290b..48c061796072e976a45f7bd0ca1c8b3913728fc7 100644
--- a/browser/base/content/main-popupset.inc.xhtml
+++ b/browser/base/content/main-popupset.inc.xhtml
@@ -80,6 +80,12 @@
     <menuitem id="context_closeDuplicateTabs"
               data-lazy-l10n-id="tab-context-close-duplicate-tabs"
               oncommand="gBrowser.removeDuplicateTabs(TabContextMenu.contextTab);"/>
+    <menuseparator/>
+    <menuitem id="context_zenSplitTabs"
+              data-lazy-l10n-id="tab-zen-split-tabs"
+              data-l10n-args='{"tabCount": 1}'
+              oncommand="gZenViewSplitter.contextSplitTabs();"/>
+    <menuseparator/>
     <menu id="context_closeTabOptions"
           data-lazy-l10n-id="tab-context-close-multiple-tabs">
       <menupopup id="closeTabOptions">
@@ -169,6 +175,10 @@
            hidden="true"
            tabspecific="true"
            aria-labelledby="editBookmarkPanelTitle">
+      <html:div id="zenEditBookmarkPanelFaviconContainer">
+        <html:img id="zenEditBookmarkPanelFavicon"/>
+      </html:div>
+      <vbox>
       <box class="panel-header">
         <html:h1>
           <html:span id="editBookmarkPanelTitle"/>
@@ -191,12 +201,14 @@
                   class="footer-button"
                   data-l10n-id="bookmark-panel-save-button"
                   default="true"
+                  autofocus="true"
                   oncommand="StarUI.panel.hidePopup();"/>
           <button id="editBookmarkPanelRemoveButton"
                   class="footer-button"
                   oncommand="StarUI.removeBookmarkButtonCommand();"/>
         </html:moz-button-group>
       </vbox>
+      </vbox>
     </panel>
   </html:template>
 
@@ -514,6 +526,8 @@
 
 #include popup-notifications.inc
 
+#include zen-popupset.inc.xhtml
+
 #include ../../components/customizableui/content/panelUI.inc.xhtml
 #include ../../components/controlcenter/content/identityPanel.inc.xhtml
 #include ../../components/controlcenter/content/permissionPanel.inc.xhtml
