diff --git a/browser/base/content/main-popupset.inc.xhtml b/browser/base/content/main-popupset.inc.xhtml
index bff8d98b27819b149a0c51c1195e1ee3668f2e31..622cbe099be0f1da8f48132204b73e577483ceae 100644
--- a/browser/base/content/main-popupset.inc.xhtml
+++ b/browser/base/content/main-popupset.inc.xhtml
@@ -166,6 +166,10 @@
            hidden="true"
            tabspecific="true"
            aria-labelledby="editBookmarkPanelTitle">
+      <html:div id="zenEditBookmarkPanelFaviconContainer">
+        <html:img id="zenEditBookmarkPanelFavicon"/>
+      </html:div>
+      <vbox>
       <box class="panel-header">
         <html:h1>
           <html:span id="editBookmarkPanelTitle"/>
@@ -188,12 +192,14 @@
                   class="footer-button"
                   data-l10n-id="bookmark-panel-save-button"
                   default="true"
+                  autofocus="true"
                   oncommand="StarUI.panel.hidePopup();"/>
           <button id="editBookmarkPanelRemoveButton"
                   class="footer-button"
                   oncommand="StarUI.removeBookmarkButtonCommand();"/>
         </html:moz-button-group>
       </vbox>
+      </vbox>
     </panel>
   </html:template>
 
@@ -491,6 +497,8 @@
 
 #include popup-notifications.inc
 
+#include zen-popupset.inc.xhtml
+
 #include ../../components/customizableui/content/panelUI.inc.xhtml
 #include ../../components/controlcenter/content/identityPanel.inc.xhtml
 #include ../../components/controlcenter/content/permissionPanel.inc.xhtml
