diff --git a/browser/base/content/navigator-toolbox.inc.xhtml b/browser/base/content/navigator-toolbox.inc.xhtml
index 390a1300a1213af2b411ba09c67e3c25750ee6c3..253d7abfcc1ade419c8f70289719cdc53898cab4 100644
--- a/browser/base/content/navigator-toolbox.inc.xhtml
+++ b/browser/base/content/navigator-toolbox.inc.xhtml
@@ -2,7 +2,7 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-<toolbox id="navigator-toolbox">
+<toolbox id="navigator-toolbox" persist="width style">
 
   <vbox id="titlebar">
     <!-- Menu -->
@@ -32,10 +32,11 @@
 
       <hbox class="titlebar-spacer" type="pre-tabs"/>
 
-      <hbox flex="1" align="end" class="toolbar-items">
+      <hbox flex="1" align="start" class="toolbar-items">
         <toolbartabstop/>
         <hbox id="TabsToolbar-customization-target" flex="1">
           <toolbarbutton id="firefox-view-button"
+                          hidden="true"
                          class="toolbarbutton-1 chromeclass-toolbar-additional"
                          data-l10n-id="toolbar-button-firefox-view-2"
                          role="button"
@@ -50,13 +51,14 @@
                 aria-multiselectable="true"
                 setfocus="false"
                 tooltip="tabbrowser-tab-tooltip"
+                orient="vertical"
                 stopwatchid="FX_TAB_CLICK_MS">
             <hbox class="tab-drop-indicator" hidden="true"/>
 # If the name (tabbrowser-arrowscrollbox) or structure of this changes
 # significantly, there is an optimization in
 # DisplayPortUtils::MaybeCreateDisplayPortInFirstScrollFrameEncountered based
 # the current structure that we may want to revisit.
-            <arrowscrollbox id="tabbrowser-arrowscrollbox" orient="horizontal" flex="1" style="min-width: 1px;" clicktoscroll="true" scrolledtostart="true" scrolledtoend="true">
+            <arrowscrollbox id="tabbrowser-arrowscrollbox" orient="vertical" style="min-width: 1px;" clicktoscroll="true" scrolledtostart="true" scrolledtoend="true">
               <tab is="tabbrowser-tab" class="tabbrowser-tab" selected="true" visuallyselected="" fadein=""/>
               <hbox id="tabbrowser-arrowscrollbox-periphery">
                 <toolbartabstop/>
@@ -103,9 +105,10 @@
       <toolbarbutton id="content-analysis-indicator"
                      oncommand="ContentAnalysis.showPanel(this, PanelUI);"
                      class="toolbarbutton-1 content-analysis-indicator-icon"/>
-
+                     #if 0
 #include titlebar-items.inc.xhtml
-
+#endif
+#include zen-sidebar-icons.inc.xhtml
     </toolbar>
 
   </vbox>
@@ -409,6 +412,14 @@
                 <image id="star-button"
                        class="urlbar-icon"/>
               </hbox>
+              <hbox id="zen-split-views-box"
+                  hidden="true"
+                  role="button"
+                  class="urlbar-page-action"
+                  onclick="gZenViewSplitter.openSplitViewPanel(event);">
+                <image id="zen-split-views-button"
+                      class="urlbar-icon"/>
+              </hbox>
             </hbox>
           </hbox>
         </hbox>
@@ -450,6 +461,7 @@
 
       <toolbarbutton id="fxa-toolbar-menu-button" class="toolbarbutton-1 chromeclass-toolbar-additional subviewbutton-nav"
                      badged="true"
+                     hidden="true"
                      delegatesanchor="true"
                      onmousedown="gSync.toggleAccountPanel(this, event)"
                      onkeypress="gSync.toggleAccountPanel(this, event)"
@@ -501,6 +513,8 @@
                      consumeanchor="PanelUI-button"
                      data-l10n-id="appmenu-menu-button-closed2"/>
     </toolbaritem>
+
+    #include titlebar-items.inc.xhtml
   </toolbar>
 
   <toolbar id="PersonalToolbar"
