diff --git a/browser/base/content/navigator-toolbox.js b/browser/base/content/navigator-toolbox.js
index 413bad2a62058a1c434d6a44e927e44eb397289d..472eab5d3bca2bc665920707a71105167cbe75ec 100644
--- a/browser/base/content/navigator-toolbox.js
+++ b/browser/base/content/navigator-toolbox.js
@@ -8,7 +8,7 @@
 document.addEventListener(
   "DOMContentLoaded",
   () => {
-    const navigatorToolbox = document.getElementById("navigator-toolbox");
+    const navigatorToolbox = document.getElementById("browser");
     const widgetOverflow = document.getElementById("widget-overflow");
 
     function onPopupShowing(event) {
@@ -110,7 +110,7 @@ document.addEventListener(
         #pageActionButton,
         #downloads-button,
         #fxa-toolbar-menu-button,
-        #unified-extensions-button,
+        #zen-site-data-icon-button,
         #library-button
         `);
       if (!element) {
@@ -138,7 +138,7 @@ document.addEventListener(
           gSync.toggleAccountPanel(element, event);
           break;
 
-        case "unified-extensions-button":
+        case "zen-site-data-icon-button":
           gUnifiedExtensions.togglePanel(event);
           break;
 
@@ -187,6 +187,7 @@ document.addEventListener(
         #reload-button ,
         #urlbar-go-button,
         #reader-mode-button,
+        #zen-tabs-wrapper,
         #picture-in-picture-button,
         #urlbar-zoom-button,
         #star-button-box,
@@ -208,6 +209,7 @@ document.addEventListener(
         case "vertical-tabs-newtab-button":
         case "tabs-newtab-button":
         case "new-tab-button":
+        case "zen-tabs-wrapper":
           gBrowser.handleNewTabMiddleClick(element, event);
           break;
 
@@ -317,7 +319,7 @@ document.addEventListener(
         #pageActionButton,
         #downloads-button,
         #fxa-toolbar-menu-button,
-        #unified-extensions-button,
+        #zen-site-data-icon-button,
         #library-button
       `);
       if (!element) {
@@ -396,7 +398,7 @@ document.addEventListener(
           gSync.toggleAccountPanel(element, event);
           break;
 
-        case "unified-extensions-button":
+        case "zen-site-data-icon-button":
           gUnifiedExtensions.togglePanel(event);
           break;
 
