diff --git a/browser/base/content/navigator-toolbox.js b/browser/base/content/navigator-toolbox.js
index 15469e9d9b91c1eaef2578c9e43b6999edac3392..553402b41bc15f7cd99bf87c54416dc66d7c03e7 100644
--- a/browser/base/content/navigator-toolbox.js
+++ b/browser/base/content/navigator-toolbox.js
@@ -6,7 +6,7 @@
 document.addEventListener(
   "DOMContentLoaded",
   () => {
-    const navigatorToolbox = document.getElementById("navigator-toolbox");
+    const navigatorToolbox = document.getElementById("browser");
     const widgetOverflow = document.getElementById("widget-overflow");
 
     function onPopupShowing(event) {
@@ -184,6 +184,7 @@ document.addEventListener(
         #forward-button,
         #reload-button ,
         #reader-mode-button,
+        #zen-tabs-wrapper,
         #picture-in-picture-button,
         #urlbar-zoom-button,
         #star-button-box,
@@ -206,6 +207,7 @@ document.addEventListener(
         case "vertical-tabs-newtab-button":
         case "tabs-newtab-button":
         case "new-tab-button":
+        case "zen-tabs-wrapper":
           gBrowser.handleNewTabMiddleClick(element, event);
           break;
 
@@ -318,6 +320,7 @@ document.addEventListener(
         #downloads-button,
         #fxa-toolbar-menu-button,
         #unified-extensions-button,
+        #zen-site-data-icon-button,
         #library-button,
         #split-view-button
       `);
@@ -401,6 +404,16 @@ document.addEventListener(
           gUnifiedExtensions.togglePanel(event);
           break;
 
+        case "zen-site-data-icon-button":
+          gUnifiedExtensions.togglePanel(
+            event,
+            null,
+            window.gZenSiteDataPanel.unifiedPanel,
+            window.gZenSiteDataPanel.unifiedPanelView,
+            window.gZenSiteDataPanel.anchor,
+          );
+          break;
+
         case "library-button":
           PanelUI.showSubView("appMenu-libraryView", element, event);
           break;
