diff --git a/browser/base/content/navigator-toolbox.inc.xhtml b/browser/base/content/navigator-toolbox.inc.xhtml
index dce7bacdb3678a95dc97085390a7e6102090ffb2..5e3c40b384e4643b6865c4fab3d0bcea4c179d0c 100644
--- a/browser/base/content/navigator-toolbox.inc.xhtml
+++ b/browser/base/content/navigator-toolbox.inc.xhtml
@@ -30,8 +30,6 @@
              context="toolbar-context-menu"
              flex="1">
 
-      <hbox class="titlebar-spacer" type="pre-tabs"/>
-
       <hbox flex="1" align="end" class="toolbar-items">
         <toolbartabstop/>
         <hbox id="TabsToolbar-customization-target" flex="1">
@@ -107,6 +105,7 @@
 
   </vbox>
 
+#if 0
   <toolbar id="nav-bar"
            class="browser-toolbar"
            data-l10n-id="navbar-accessible"
@@ -575,7 +574,7 @@
   <html:template id="tab-notification-deck-template">
     <html:named-deck id="tab-notification-deck"></html:named-deck>
   </html:template>
-
+#endif
 
   <html:template id="BrowserToolbarPalette">
     <toolbarbutton id="import-button"
