diff --git a/browser/base/content/navigator-toolbox.js b/browser/base/content/navigator-toolbox.js
index 7b776b15d52367a008ce6bf53dcfcbbe007b7453..da23f716c753f5a43f17bb5ed7a3d335891168c2 100644
--- a/browser/base/content/navigator-toolbox.js
+++ b/browser/base/content/navigator-toolbox.js
@@ -6,7 +6,7 @@
 document.addEventListener(
   "DOMContentLoaded",
   () => {
-    const navigatorToolbox = document.getElementById("navigator-toolbox");
+    const navigatorToolbox = document.getElementById("browser");
     const widgetOverflow = document.getElementById("widget-overflow");
 
     function onPopupShowing(event) {
@@ -185,6 +185,7 @@ document.addEventListener(
         #reload-button ,
         #urlbar-go-button,
         #reader-mode-button,
+        #zen-tabs-wrapper,
         #picture-in-picture-button,
         #urlbar-zoom-button,
         #star-button-box,
@@ -206,6 +207,7 @@ document.addEventListener(
         case "vertical-tabs-newtab-button":
         case "tabs-newtab-button":
         case "new-tab-button":
+        case "zen-tabs-wrapper":
           gBrowser.handleNewTabMiddleClick(element, event);
           break;
 
@@ -316,6 +318,7 @@ document.addEventListener(
         #downloads-button,
         #fxa-toolbar-menu-button,
         #unified-extensions-button,
+        #zen-site-data-icon-button,
         #library-button
       `);
       if (!element) {
@@ -398,6 +401,16 @@ document.addEventListener(
           gUnifiedExtensions.togglePanel(event);
           break;
 
+        case "zen-site-data-icon-button":
+          gUnifiedExtensions.togglePanel(
+            event,
+            null,
+            window.gZenSiteDataPanel.unifiedPanel,
+            window.gZenSiteDataPanel.unifiedPanelView,
+            window.gZenSiteDataPanel.anchor,
+          );
+          break;
+
         case "library-button":
           PanelUI.showSubView("appMenu-libraryView", element, event);
           break;
