diff --git a/browser/base/content/nsContextMenu.sys.mjs b/browser/base/content/nsContextMenu.sys.mjs
index b77c12da637ba31aca6e8100187d0ac14db33298..4fbec0dda932bcaa77ce714826ac8ab20b9e8e63 100644
--- a/browser/base/content/nsContextMenu.sys.mjs
+++ b/browser/base/content/nsContextMenu.sys.mjs
@@ -376,6 +376,9 @@ export class nsContextMenu {
     this.initPDFItems();
     this.initTextFragmentItems();
 
+    this.showItem("context-zenSplitLink", this.onLink && !this.onMailtoLink && !this.onTelLink);
+    this.showItem("context-zenOpenLinkInGlance", this.onLink && !this.onMailtoLink && !this.onTelLink); 
+
     this.showHideSeparators(aXulMenu);
     if (!aXulMenu.showHideSeparators) {
       // Set the showHideSeparators function on the menu itself so that
