diff --git a/browser/base/content/nsContextMenu.sys.mjs b/browser/base/content/nsContextMenu.sys.mjs
index bc71ba720ef3603e8b90d295fb16d8415ba114c4..99677f70e22258f61cc9bda31e4d8745d7ca9395 100644
--- a/browser/base/content/nsContextMenu.sys.mjs
+++ b/browser/base/content/nsContextMenu.sys.mjs
@@ -1095,6 +1095,8 @@ export class nsContextMenu {
         !this.isSecureAboutPage()
     );
 
+    this.showItem("context-zenSplitLink", this.onLink && !this.onMailtoLink && !this.onTelLink);
+
     let canNotStrip =
       lazy.STRIP_ON_SHARE_CAN_DISABLE && !this.#canStripParams();
 
