diff --git a/browser/base/content/nsContextMenu.js b/browser/base/content/nsContextMenu.js
index b5c8dd44daaab18397cf01d0ef9f5288f5113925..8840b8171d5992da0a2f6216d8da9e1ec6dc1e93 100644
--- a/browser/base/content/nsContextMenu.js
+++ b/browser/base/content/nsContextMenu.js
@@ -1047,6 +1047,13 @@ class nsContextMenu {
         !this.isSecureAboutPage()
     );
 
+    this.showItem(
+      "context-zenAddToWebPanel",
+      this.onLink && !this.onMailtoLink && !this.onTelLink
+    );
+
+    this.showItem("context-zenSplitLink", this.onLink && !this.onMailtoLink && !this.onTelLink);
+
     let copyLinkSeparator = document.getElementById("context-sep-copylink");
     // Show "Copy Link", "Copy" and "Copy Clean Link" with no divider, and "copy link" and "Send link to Device" with no divider between.
     // Other cases will show a divider.
