diff --git a/browser/locales/en-US/browser/preferences/preferences.ftl b/browser/locales/en-US/browser/preferences/preferences.ftl
index d197d876dac5e9024c944d33454525123775ea02..59fa0a7921f337dc01eac3364c1d01c71afc4f7a 100644
--- a/browser/locales/en-US/browser/preferences/preferences.ftl
+++ b/browser/locales/en-US/browser/preferences/preferences.ftl
@@ -925,6 +925,7 @@ sync-currently-syncing-addresses = Addresses
 sync-currently-syncing-payment-methods = Payment methods
 sync-currently-syncing-addons = Add-ons
 sync-currently-syncing-settings = Settings
+sync-currently-syncing-workspaces = Workspaces
 
 sync-change-options =
     .label = Change…
@@ -980,6 +981,11 @@ sync-engine-settings =
     .tooltiptext = General, Privacy, and Security settings you’ve changed
     .accesskey = s
 
+sync-engine-workspaces =
+    .label = Workspaces
+    .tooltiptext = Workspace names, icons and associated containers
+    .accesskey = w
+
 ## The device name controls.
 
 sync-device-name-header = Device Name
