diff --git a/gfx/wr/webrender/src/picture.rs b/gfx/wr/webrender/src/picture.rs
index f22bcadd0624d9a22f5c3fa318d7795e102b890f..68d1eb55abde2a9b5779b1d07c5f111f46f177a8 100644
--- a/gfx/wr/webrender/src/picture.rs
+++ b/gfx/wr/webrender/src/picture.rs
@@ -7449,7 +7449,12 @@ fn get_relative_scale_offset(
         CoordinateSpaceMapping::Local => ScaleOffset::identity(),
         CoordinateSpaceMapping::ScaleOffset(scale_offset) => scale_offset,
         CoordinateSpaceMapping::Transform(m) => {
-            ScaleOffset::from_transform(&m).expect("bug: pictures caches don't support complex transforms")
+            // Temporary fix when opening extensions (#34)
+            // TODO: Look more into this
+            ScaleOffset {
+                scale: Vector2D::new(m.m11, m.m22),
+                offset: Vector2D::new(m.m41, m.m42),
+            }
         }
     };
 
