diff --git a/gfx/wr/webrender/src/picture.rs b/gfx/wr/webrender/src/picture.rs
index 79e062a40ed097fa6614abc6b5201f868c447733..91c10d17f0dca55eafec522ddc94b91bf4cd2cfb 100644
--- a/gfx/wr/webrender/src/picture.rs
+++ b/gfx/wr/webrender/src/picture.rs
@@ -8121,7 +8121,12 @@ fn get_relative_scale_offset(
         CoordinateSpaceMapping::Local => ScaleOffset::identity(),
         CoordinateSpaceMapping::ScaleOffset(scale_offset) => scale_offset,
         CoordinateSpaceMapping::Transform(m) => {
-            ScaleOffset::from_transform(&m).expect("bug: pictures caches don't support complex transforms")
+            // Temporary fix when opening extensions (#34)
+            // TODO: Look more into this
+            ScaleOffset {
+                scale: Vector2D::new(m.m11, m.m22),
+                offset: Vector2D::new(m.m41, m.m42),
+            }
         }
     };
 
