diff --git a/browser/themes/BuiltInThemeConfig.sys.mjs b/browser/themes/BuiltInThemeConfig.sys.mjs
index 28b254d5757caf35c4ba755ef4cfed44bab5af00..660f517e0e3342970c6e30837fcf4d0adf360777 100644
--- a/browser/themes/BuiltInThemeConfig.sys.mjs
+++ b/browser/themes/BuiltInThemeConfig.sys.mjs
@@ -33,11 +33,4 @@ export const BuiltInThemeConfig = new Map([
       path: "resource://builtin-themes/dark/",
     },
   ],
-  [
-    "firefox-alpenglow@mozilla.org",
-    {
-      version: "1.5",
-      path: "resource://builtin-themes/alpenglow/",
-    },
-  ],
 ]);
