diff --git a/browser/themes/BuiltInThemes.sys.mjs b/browser/themes/BuiltInThemes.sys.mjs
index c2d5dd7a18895ae8b4afbf386f122e7899c48cda..377d904bbf5adc92bf7cb0aa4c4c641794d828f3 100644
--- a/browser/themes/BuiltInThemes.sys.mjs
+++ b/browser/themes/BuiltInThemes.sys.mjs
@@ -74,7 +74,7 @@ class _BuiltInThemes {
   maybeInstallActiveBuiltInTheme() {
     const activeThemeID = Services.prefs.getStringPref(
       kActiveThemePref,
-      "default-theme@mozilla.org"
+      "firefox-compact-dark@mozilla.org"
     );
     let activeBuiltInTheme = this.builtInThemeMap.get(activeThemeID);
 
@@ -82,7 +82,7 @@ class _BuiltInThemes {
       lazy.AddonManager.maybeInstallBuiltinAddon(
         activeThemeID,
         activeBuiltInTheme.version,
-        `resource://builtin-themes/${activeBuiltInTheme.path}`
+        `${activeBuiltInTheme.path}`
       );
     }
   }
@@ -166,7 +166,7 @@ class _BuiltInThemes {
   async _uninstallExpiredThemes() {
     const activeThemeID = Services.prefs.getStringPref(
       kActiveThemePref,
-      "default-theme@mozilla.org"
+      "firefox-compact-dark@mozilla.org"
     );
     const now = new Date();
     const expiredThemes = Array.from(this.builtInThemeMap.entries()).filter(
