diff --git a/browser/themes/BuiltInThemeConfig.sys.mjs b/browser/themes/BuiltInThemeConfig.sys.mjs
index a5361f1f0ec57f575f8d76e118ff762dc7547501..db5c5eb2059b4ceef3911fd807bf985408859d1a 100644
--- a/browser/themes/BuiltInThemeConfig.sys.mjs
+++ b/browser/themes/BuiltInThemeConfig.sys.mjs
@@ -33,11 +33,4 @@ export const BuiltInThemeConfig = new Map([
       path: "resource://builtin-themes/dark/",
     },
   ],
-  [
-    "firefox-alpenglow@mozilla.org",
-    {
-      version: "1.5",
-      path: "resource://builtin-themes/alpenglow/",
-    },
-  ],
 ]);
