diff --git a/toolkit/mozapps/extensions/AddonManager.sys.mjs b/toolkit/mozapps/extensions/AddonManager.sys.mjs
index 2f855e6e48f420e3782d3be31cfdfa6c40db479b..de55b4c7ff7517169ac56693b4cb3c46b3c20736 100644
--- a/toolkit/mozapps/extensions/AddonManager.sys.mjs
+++ b/toolkit/mozapps/extensions/AddonManager.sys.mjs
@@ -1214,12 +1214,12 @@ var AddonManagerInternal = {
         ITEM_VERSION: aAddon.version,
         ITEM_STATUS: addonStatus,
         APP_ID: Services.appinfo.ID,
-        APP_VERSION: aAppVersion ? aAppVersion : Services.appinfo.version,
+        APP_VERSION: AppConstants.ZEN_FIREFOX_VERSION,
         REQ_VERSION: UPDATE_REQUEST_VERSION,
         APP_OS: Services.appinfo.OS,
         APP_ABI: Services.appinfo.XPCOMABI,
         APP_LOCALE: getLocale(),
-        CURRENT_APP_VERSION: Services.appinfo.version,
+        CURRENT_APP_VERSION: AppConstants.ZEN_FIREFOX_VERSION,
       })
     );
 
