diff --git a/browser/components/urlbar/UrlbarController.sys.mjs b/browser/components/urlbar/UrlbarController.sys.mjs
index 7751871815ac3aa4641a7752c21f625e03e34c82..418dbecd2739fb8a95ce44efb5d671ac97934f21 100644
--- a/browser/components/urlbar/UrlbarController.sys.mjs
+++ b/browser/components/urlbar/UrlbarController.sys.mjs
@@ -302,7 +302,6 @@ export class UrlbarController {
     const isMac = AppConstants.platform == "macosx";
     // Handle readline/emacs-style navigation bindings on Mac.
     if (
-      isMac &&
       this.view.isOpen &&
       event.ctrlKey &&
       (event.key == "n" || event.key == "p")
@@ -451,6 +450,8 @@ export class UrlbarController {
             });
           }
           event.preventDefault();
+        } else if (!this.input.value && !(event.ctrlKey || event.altKey || event.shiftKey)) {
+          this.browserWindow.gZenUIManager.enableCommandsMode(event);
         }
         break;
       }
