diff --git a/browser/components/urlbar/UrlbarUtils.sys.mjs b/browser/components/urlbar/UrlbarUtils.sys.mjs
index 079c9f18550b54b2a1cc030174849896fd23234e..605bb3433b61443037659d126ba17b033e5a4394 100644
--- a/browser/components/urlbar/UrlbarUtils.sys.mjs
+++ b/browser/components/urlbar/UrlbarUtils.sys.mjs
@@ -77,6 +77,7 @@ export var UrlbarUtils = {
     RESTRICT_SEARCH_KEYWORD: "restrictSearchKeyword",
     SUGGESTED_INDEX: "suggestedIndex",
     TAIL_SUGGESTION: "tailSuggestion",
+    ZEN_ACTION: "zenAction",
   }),
 
   // Defines provider types.
@@ -136,6 +137,7 @@ export var UrlbarUtils = {
     OTHER_NETWORK: 6,
     ADDON: 7,
     ACTIONS: 8,
+    ZEN_ACTIONS: 9,
   }),
 
   // Per-result exposure telemetry.
@@ -547,6 +549,8 @@ export var UrlbarUtils = {
           return this.RESULT_GROUP.HEURISTIC_FALLBACK;
         case "UrlbarProviderHistoryUrlHeuristic":
           return this.RESULT_GROUP.HEURISTIC_HISTORY_URL;
+        case "ZenUrlbarProviderGlobalActions":
+          return this.RESULT_GROUP.ZEN_ACTION;
         case "UrlbarProviderOmnibox":
           return this.RESULT_GROUP.HEURISTIC_OMNIBOX;
         case "UrlbarProviderRestrictKeywordsAutofill":
