diff --git a/browser/components/urlbar/UrlbarMuxerUnifiedComplete.sys.mjs b/browser/components/urlbar/UrlbarMuxerUnifiedComplete.sys.mjs
index dcf1af43d62..17dbaaa66aa 100644
--- a/browser/components/urlbar/UrlbarMuxerUnifiedComplete.sys.mjs
+++ b/browser/components/urlbar/UrlbarMuxerUnifiedComplete.sys.mjs
@@ -117,6 +117,21 @@ class MuxerUnifiedComplete extends UrlbarMuxer {
       // When you add state, update _copyState() as necessary.
     };
 
+    const allFromPlaces = unsortedResults.every(
+      result => result.providerName === "Places"
+    );
+  
+    // If all results are from "Places", sort by lastOpened
+    if (allFromPlaces) {
+      unsortedResults.sort((a, b) => {
+        const lastOpenedA = a.payload?.lastOpened || 0;
+        const lastOpenedB = b.payload?.lastOpened|| 0;
+        return lastOpenedB - lastOpenedA; // Descending order
+      });
+      unsortedResults[0].heuristic = true
+      unsortedResults[0].providerName = "HeuristicFallback"
+    }  
+
     // Do the first pass over all results to build some state.
     for (let result of unsortedResults) {
       // Add each result to the appropriate `resultsByGroup` map.
@@ -794,6 +807,7 @@ class MuxerUnifiedComplete extends UrlbarMuxer {
     }
 
     if (result.providerName == lazy.UrlbarProviderTabToSearch.name) {
+      return false;
       // Discard the result if a tab-to-search result was added already.
       if (!state.canAddTabToSearch) {
         return false;
