diff --git a/browser/components/urlbar/UrlbarProvidersManager.sys.mjs b/browser/components/urlbar/UrlbarProvidersManager.sys.mjs
index d9a0566c5ad2a9ae375a23769f856aecc6efd86c..3e4951f4efdf62acb06b38ad2ae4bd507ca5e71f 100644
--- a/browser/components/urlbar/UrlbarProvidersManager.sys.mjs
+++ b/browser/components/urlbar/UrlbarProvidersManager.sys.mjs
@@ -912,6 +912,7 @@ export class Query {
     if (
       result.heuristic &&
       this.context.searchMode &&
+      !(this.context.searchMode.source === lazy.UrlbarUtils.RESULT_SOURCE.ZEN_ACTIONS && result.payload?.zenAction) &&
       (!this.context.trimmedSearchString ||
         (!this.context.searchMode.engineName && !result.autofill))
     ) {
@@ -1042,6 +1043,7 @@ function updateSourcesIfEmpty(context) {
             lazy.UrlbarTokenizer.TYPE.RESTRICT_TITLE,
             lazy.UrlbarTokenizer.TYPE.RESTRICT_URL,
             lazy.UrlbarTokenizer.TYPE.RESTRICT_ACTION,
+            lazy.UrlbarTokenizer.TYPE.RESTRICT_WORKSPACE,
           ].includes(t.type)
         );
 
@@ -1099,6 +1101,14 @@ function updateSourcesIfEmpty(context) {
           acceptedSources.push(source);
         }
         break;
+      case lazy.UrlbarUtils.RESULT_SOURCE.WORKSPACES:
+        if (
+          restrictTokenType === lazy.UrlbarTokenizer.TYPE.RESTRICT_WORKSPACE ||
+          !restrictTokenType
+        ) {
+          acceptedSources.push(source);
+        }
+        break;
       case lazy.UrlbarUtils.RESULT_SOURCE.OTHER_LOCAL:
       case lazy.UrlbarUtils.RESULT_SOURCE.ADDON:
       default:
