diff --git a/browser/components/urlbar/UrlbarProvidersManager.sys.mjs b/browser/components/urlbar/UrlbarProvidersManager.sys.mjs
index f6f5e4d74296afc22e5ae651bbefeb9b0b963511..2b01feaea5558cba50c5a59944a751694a3788e2 100644
--- a/browser/components/urlbar/UrlbarProvidersManager.sys.mjs
+++ b/browser/components/urlbar/UrlbarProvidersManager.sys.mjs
@@ -931,6 +931,7 @@ export class Query {
     if (
       result.heuristic &&
       this.context.searchMode &&
+      !(this.context.searchMode.source === lazy.UrlbarShared.RESULT_SOURCE.ZEN_ACTIONS && result.payload?.zenAction) &&
       (!this.context.trimmedSearchString ||
         (!this.context.searchMode.engineName && !result.autofill))
     ) {
@@ -1082,6 +1083,7 @@ function updateSourcesIfEmpty(context) {
             lazy.UrlbarShared.TOKEN_TYPE.RESTRICT_TITLE,
             lazy.UrlbarShared.TOKEN_TYPE.RESTRICT_URL,
             lazy.UrlbarShared.TOKEN_TYPE.RESTRICT_ACTION,
+            lazy.UrlbarShared.TOKEN_TYPE.RESTRICT_WORKSPACE,
           ].includes(t.type)
         );
 
@@ -1141,6 +1143,22 @@ function updateSourcesIfEmpty(context) {
           acceptedSources.push(source);
         }
         break;
+      case lazy.UrlbarShared.RESULT_SOURCE.ZEN_PINNED_TABS:
+        if (
+          restrictTokenType === lazy.UrlbarShared.TOKEN_TYPE.RESTRICT_OPENPAGE ||
+          !restrictTokenType
+        ) {
+          acceptedSources.push(source);
+        }
+        break;
+      case lazy.UrlbarShared.RESULT_SOURCE.WORKSPACES:
+        if (
+          restrictTokenType === lazy.UrlbarShared.TOKEN_TYPE.RESTRICT_WORKSPACE ||
+          !restrictTokenType
+        ) {
+          acceptedSources.push(source);
+        }
+        break;
       case lazy.UrlbarShared.RESULT_SOURCE.OTHER_LOCAL:
       case lazy.UrlbarShared.RESULT_SOURCE.ADDON:
       default:
