diff --git a/browser/components/urlbar/UrlbarResult.sys.mjs b/browser/components/urlbar/UrlbarResult.sys.mjs
index e6eccc010ed3b4595c0aabb1e7d528b20ce1be9b..b71466ebfad0edecb8d7b78ffbe20a95cd69df98 100644
--- a/browser/components/urlbar/UrlbarResult.sys.mjs
+++ b/browser/components/urlbar/UrlbarResult.sys.mjs
@@ -178,6 +178,10 @@ export class UrlbarResult {
     return this.#heuristic;
   }
 
+  set heuristic(value) {
+    this.#heuristic = value;
+  }
+
   get hideRowLabel() {
     return this.#hideRowLabel;
   }
