diff --git a/browser/components/urlbar/UrlbarMuxerUnifiedComplete.sys.mjs b/browser/components/urlbar/UrlbarMuxerUnifiedComplete.sys.mjs
index dcf1af43d62979d3226d7f704c51a2f0bb935cc0..a8271ac083c54947756e6ef35a3b4eac407b03f9 100644
--- a/browser/components/urlbar/UrlbarMuxerUnifiedComplete.sys.mjs
+++ b/browser/components/urlbar/UrlbarMuxerUnifiedComplete.sys.mjs
@@ -117,6 +117,19 @@ class MuxerUnifiedComplete extends UrlbarMuxer {
       // When you add state, update _copyState() as necessary.
     };
 
+    let window = Services.wm.getMostRecentWindow("navigator:browser");
+    if (window && window.gZenTabSearch && window.gURLBar && window.gURLBar.searchMode && window.gURLBar.searchMode.source === 4) {
+      unsortedResults.sort((a, b) => {
+        const lastOpenedA = a.payload?.lastOpened || 0;
+        const lastOpenedB = b.payload?.lastOpened|| 0;
+        return lastOpenedB - lastOpenedA; // Descending order
+      });
+      if (window.gZenTabSearch._autoSelectResult === false || window.gZenTabSearch._counter > 0) {
+        unsortedResults[0].heuristic = true
+        unsortedResults[0].providerName = "HeuristicFallback"
+      }
+    }  
+
     // Do the first pass over all results to build some state.
     for (let result of unsortedResults) {
       // Add each result to the appropriate `resultsByGroup` map.
@@ -794,6 +805,7 @@ class MuxerUnifiedComplete extends UrlbarMuxer {
     }
 
     if (result.providerName == lazy.UrlbarProviderTabToSearch.name) {
+      return false;
       // Discard the result if a tab-to-search result was added already.
       if (!state.canAddTabToSearch) {
         return false;
