diff --git a/browser/components/urlbar/UrlbarUtils.sys.mjs b/browser/components/urlbar/UrlbarUtils.sys.mjs
index 178cd5e84fda74a28b2de59b671b3399cc28fac4..b03c6e06de4ee9a95dcb4d33d7ac9f0b053be4e1 100644
--- a/browser/components/urlbar/UrlbarUtils.sys.mjs
+++ b/browser/components/urlbar/UrlbarUtils.sys.mjs
@@ -214,6 +214,11 @@ export var UrlbarUtils = {
           return UrlbarShared.RESULT_GROUP.HEURISTIC_FALLBACK;
         case "UrlbarProviderHistoryUrlHeuristic":
           return UrlbarShared.RESULT_GROUP.HEURISTIC_HISTORY_URL;
+        case "ZenUrlbarProviderGlobalActions":
+          if (result.source == UrlbarShared.RESULT_SOURCE.WORKSPACES) {
+            return UrlbarShared.RESULT_GROUP.ZEN_WORKSPACE;
+          }
+          return UrlbarShared.RESULT_GROUP.ZEN_ACTION;
         case "UrlbarProviderOmnibox":
           return UrlbarShared.RESULT_GROUP.HEURISTIC_OMNIBOX;
         case "UrlbarProviderRestrictKeywordsAutofill":
