diff --git a/browser/components/urlbar/UrlbarMuxerStandard.sys.mjs b/browser/components/urlbar/UrlbarMuxerStandard.sys.mjs
index 65bd41b2ea14c26d490632d4a5348d6cf33ee9dd..80fbd685156949b6c7e93ed9d0ef56350dd1a0fe 100644
--- a/browser/components/urlbar/UrlbarMuxerStandard.sys.mjs
+++ b/browser/components/urlbar/UrlbarMuxerStandard.sys.mjs
@@ -925,11 +925,16 @@ class MuxerUnifiedComplete extends UrlbarMuxer {
       result.providerName == "UrlbarProviderHeuristicFallback" &&
       state.context.heuristicResult?.providerName !=
         "UrlbarProviderHeuristicFallback"
+      && !(
+        result.type == lazy.UrlbarShared.RESULT_TYPE.SEARCH &&
+        state.context.heuristicResult?.type == lazy.UrlbarShared.RESULT_TYPE.URL
+      )
     ) {
       return false;
     }
 
     if (result.providerName == "UrlbarProviderTabToSearch") {
+      return false;
       // Discard the result if a tab-to-search result was added already.
       if (!state.canAddTabToSearch) {
         return false;
@@ -1593,7 +1598,9 @@ class MuxerUnifiedComplete extends UrlbarMuxer {
       usedLimits.maxResultCount++;
     }
 
+    if (!(result.heuristic && result.source == lazy.UrlbarShared.RESULT_SOURCE.ZEN_ACTIONS)) {
     state.usedResultSpan += span;
+    }
     this._updateStatePostAdd(result, state);
 
     return true;
