diff --git a/browser/components/urlbar/UrlbarProvidersManager.sys.mjs b/browser/components/urlbar/UrlbarProvidersManager.sys.mjs
index 78d0e875978b568b79646489c28b125a44ea79fa..da8092b561c3dd8864e57f5a52a1a643db29ace1 100644
--- a/browser/components/urlbar/UrlbarProvidersManager.sys.mjs
+++ b/browser/components/urlbar/UrlbarProvidersManager.sys.mjs
@@ -1044,6 +1044,7 @@ function updateSourcesIfEmpty(context) {
             lazy.UrlbarTokenizer.TYPE.RESTRICT_TITLE,
             lazy.UrlbarTokenizer.TYPE.RESTRICT_URL,
             lazy.UrlbarTokenizer.TYPE.RESTRICT_ACTION,
+            lazy.UrlbarTokenizer.TYPE.RESTRICT_WORKSPACE,
           ].includes(t.type)
         );

@@ -1101,6 +1102,14 @@ function updateSourcesIfEmpty(context) {
           acceptedSources.push(source);
         }
         break;
+      case lazy.UrlbarUtils.RESULT_SOURCE.WORKSPACES:
+        if (
+          restrictTokenType === lazy.UrlbarTokenizer.TYPE.RESTRICT_WORKSPACE ||
+          !restrictTokenType
+        ) {
+          acceptedSources.push(source);
+        }
+        break;
       case lazy.UrlbarUtils.RESULT_SOURCE.OTHER_LOCAL:
       case lazy.UrlbarUtils.RESULT_SOURCE.ADDON:
       default:
