diff --git a/browser/components/urlbar/UrlbarUtils.sys.mjs b/browser/components/urlbar/UrlbarUtils.sys.mjs
index a4ed35032c0c60f230d95f451b3f615232784cad..4c0b034982c56a649afbeb8837556a2c77154b4e 100644
--- a/browser/components/urlbar/UrlbarUtils.sys.mjs
+++ b/browser/components/urlbar/UrlbarUtils.sys.mjs
@@ -81,6 +81,7 @@ export var UrlbarUtils = {
     RESTRICT_SEARCH_KEYWORD: "restrictSearchKeyword",
     SUGGESTED_INDEX: "suggestedIndex",
     TAIL_SUGGESTION: "tailSuggestion",
+    ZEN_ACTION: "zenAction",
   }),
 
   // Defines provider types.
@@ -140,6 +141,7 @@ export var UrlbarUtils = {
     OTHER_NETWORK: 6,
     ADDON: 7,
     ACTIONS: 8,
+    ZEN_ACTIONS: 9,
   }),
 
   // Per-result exposure telemetry.
@@ -552,6 +554,8 @@ export var UrlbarUtils = {
           return this.RESULT_GROUP.HEURISTIC_FALLBACK;
         case "UrlbarProviderHistoryUrlHeuristic":
           return this.RESULT_GROUP.HEURISTIC_HISTORY_URL;
+        case "ZenUrlbarProviderGlobalActions":
+          return this.RESULT_GROUP.ZEN_ACTION;
         case "UrlbarProviderOmnibox":
           return this.RESULT_GROUP.HEURISTIC_OMNIBOX;
         case "UrlbarProviderRestrictKeywordsAutofill":
@@ -3276,7 +3280,7 @@ export class L10nCache {
     }
 
     let messages = await l10n.formatMessages([{ id, args }]);
-    if (!messages?.length) {
+    if (!messages?.length || !messages[0]) {
       console.error(
         "l10n.formatMessages returned an unexpected value for ID: ",
         id
