diff --git a/browser/components/urlbar/UrlbarProvidersManager.sys.mjs b/browser/components/urlbar/UrlbarProvidersManager.sys.mjs
index 67f51c7996c24549f39594c9a518bc5316350ce7..d66969f2411bb687c15448ad828b64ddf43c3389 100644
--- a/browser/components/urlbar/UrlbarProvidersManager.sys.mjs
+++ b/browser/components/urlbar/UrlbarProvidersManager.sys.mjs
@@ -914,6 +914,7 @@ export class Query {
     if (
       result.heuristic &&
       this.context.searchMode &&
+      !(this.context.searchMode.source === lazy.UrlbarShared.RESULT_SOURCE.ZEN_ACTIONS && result.payload?.zenAction) &&
       (!this.context.trimmedSearchString ||
         (!this.context.searchMode.engineName && !result.autofill))
     ) {
@@ -1060,6 +1061,7 @@ function updateSourcesIfEmpty(context) {
             lazy.UrlbarShared.TOKEN_TYPE.RESTRICT_TITLE,
             lazy.UrlbarShared.TOKEN_TYPE.RESTRICT_URL,
             lazy.UrlbarShared.TOKEN_TYPE.RESTRICT_ACTION,
+            lazy.UrlbarShared.TOKEN_TYPE.RESTRICT_WORKSPACE,
           ].includes(t.type)
         );
 
@@ -1119,6 +1121,14 @@ function updateSourcesIfEmpty(context) {
           acceptedSources.push(source);
         }
         break;
+      case lazy.UrlbarShared.RESULT_SOURCE.WORKSPACES:
+        if (
+          restrictTokenType === lazy.UrlbarShared.TOKEN_TYPE.RESTRICT_WORKSPACE ||
+          !restrictTokenType
+        ) {
+          acceptedSources.push(source);
+        }
+        break;
       case lazy.UrlbarShared.RESULT_SOURCE.OTHER_LOCAL:
       case lazy.UrlbarShared.RESULT_SOURCE.ADDON:
       default:
