diff --git a/browser/components/urlbar/UrlbarUtils.sys.mjs b/browser/components/urlbar/UrlbarUtils.sys.mjs
index 2f03bdf632c6ca15c728df2801162da5159682da..6834daede8cb11854f137602100183d85a930686 100644
--- a/browser/components/urlbar/UrlbarUtils.sys.mjs
+++ b/browser/components/urlbar/UrlbarUtils.sys.mjs
@@ -109,6 +109,8 @@ export var UrlbarUtils = {
     RESTRICT_SEARCH_KEYWORD: "restrictSearchKeyword",
     SUGGESTED_INDEX: "suggestedIndex",
     TAIL_SUGGESTION: "tailSuggestion",
+    ZEN_ACTION: "zenAction",
+    ZEN_WORKSPACE: "zenWorkspace",
   }),
 
   // Defines provider types.
@@ -170,6 +172,8 @@ export var UrlbarUtils = {
     OTHER_NETWORK: 6,
     ADDON: 7,
     ACTIONS: 8,
+    ZEN_ACTIONS: 9,
+    WORKSPACES: 10,
   }),
 
   // Per-result exposure telemetry.
@@ -319,6 +323,14 @@ export var UrlbarUtils = {
         telemetryLabel: "actions",
         uiLabel: "urlbar-searchmode-actions2",
       },
+      {
+        source: this.RESULT_SOURCE.WORKSPACES,
+        restrict: lazy.UrlbarTokenizer.RESTRICT.WORKSPACE,
+        icon: "chrome://browser/skin/zen-icons/selectable/layers.svg",
+        pref: "shortcuts.workspaces",
+        telemetryLabel: "workspaces",
+        uiLabel: "urlbar-search-mode-workspaces",
+      },
     ]);
   },
 
@@ -611,6 +623,12 @@ export var UrlbarUtils = {
           return this.RESULT_GROUP.HEURISTIC_FALLBACK;
         case "UrlbarProviderHistoryUrlHeuristic":
           return this.RESULT_GROUP.HEURISTIC_HISTORY_URL;
+        case "ZenUrlbarProviderGlobalActions":
+          if (result.source == this.RESULT_SOURCE.WORKSPACES) {
+            return this.RESULT_GROUP.ZEN_WORKSPACE;
+          } else {
+            return this.RESULT_GROUP.ZEN_ACTION;
+          }
         case "UrlbarProviderOmnibox":
           return this.RESULT_GROUP.HEURISTIC_OMNIBOX;
         case "UrlbarProviderRestrictKeywordsAutofill":
