diff --git a/browser/components/urlbar/UrlbarController.sys.mjs b/browser/components/urlbar/UrlbarController.sys.mjs
index dfe0b55880bb31831531bae331de3feffcbd1029..87b5a6264cd188a21b2fdfe25fd6078a8c01e965 100644
--- a/browser/components/urlbar/UrlbarController.sys.mjs
+++ b/browser/components/urlbar/UrlbarController.sys.mjs
@@ -305,7 +305,6 @@ export class UrlbarController {
     const isMac = AppConstants.platform == "macosx";
     // Handle readline/emacs-style navigation bindings on Mac.
     if (
-      isMac &&
       this.view.isOpen &&
       event.ctrlKey &&
       (event.key == "n" || event.key == "p")
@@ -330,7 +329,7 @@ export class UrlbarController {
       }
 
       let handled = false;
-      if (lazy.UrlbarPrefs.get("scotchBonnet.enableOverride")) {
+      if (lazy.UrlbarPrefs.get("scotchBonnet.enableOverride") || this.input.sapName == "searchbar") {
         handled = this.input.searchModeSwitcher.handleKeyDown(event);
       } else if (this.view.isOpen && this._lastQueryContextWrapper) {
         let { queryContext } = this._lastQueryContextWrapper;
@@ -462,6 +461,27 @@ export class UrlbarController {
           }
         }
 
+        if (
+          this.input.sapName == "searchbar" &&
+          this.view.isOpen &&
+          !event.ctrlKey &&
+          !event.altKey
+        ) {
+          if (
+            (event.shiftKey &&
+              this.view.selectedElement ==
+                this.view.getFirstSelectableElement()) ||
+            (!event.shiftKey &&
+              this.view.selectedElement ==
+                this.view.getLastSelectableElement())
+          ) {
+            event.preventDefault();
+            this.view.selectedRowIndex = -1;
+            this.focusOnUnifiedSearchButton();
+            break;
+          }
+        }
+
         // It's always possible to tab through results when the urlbar was
         // focused with the mouse or has a search string, or when the view
         // already has a selection.
@@ -494,6 +514,8 @@ export class UrlbarController {
             });
           }
           event.preventDefault();
+        } else if (!this.input.value && !(event.ctrlKey || event.altKey || event.shiftKey)) {
+          this.browserWindow.gZenUIManager.enableCommandsMode(event);
         }
         break;
       }
