diff --git a/browser/components/urlbar/UrlbarInput.sys.mjs b/browser/components/urlbar/UrlbarInput.sys.mjs
index 6de4c2ac873b1424abee118a143ac377f26beb80..423e3aab4fd4fff9960caf8b8375d7bfcb90c733 100644
--- a/browser/components/urlbar/UrlbarInput.sys.mjs
+++ b/browser/components/urlbar/UrlbarInput.sys.mjs
@@ -67,6 +67,13 @@ XPCOMUtils.defineLazyPreferenceGetter(
   false
 );
 
+XPCOMUtils.defineLazyPreferenceGetter(
+  lazy,
+  "ZEN_URLBAR_BEHAVIOR",
+  "zen.urlbar.behavior",
+  'default'
+);
+
 const DEFAULT_FORM_HISTORY_NAME = "searchbar-history";
 const SEARCH_BUTTON_CLASS = "urlbar-search-button";
 
@@ -348,7 +355,11 @@ export class UrlbarInput {
     // See _on_select().  HTMLInputElement.select() dispatches a "select"
     // event but does not set the primary selection.
     this._suppressPrimaryAdjustment = true;
+    this.document.getElementById("navigator-toolbox").setAttribute("suppress-primary-adjustment", !this.document.getElementById("navigator-toolbox").hasAttribute("zen-has-hover"));
     this.inputField.select();
+    this.document.ownerGlobal.setTimeout(() => {
+      this.document.getElementById("navigator-toolbox").removeAttribute("suppress-primary-adjustment");
+    }, 100);
     this._suppressPrimaryAdjustment = false;
   }
 
@@ -424,6 +435,10 @@ export class UrlbarInput {
     hideSearchTerms = false,
     isSameDocument = false
   ) {
+    if (this.hasAttribute("zen-newtab")) {
+      return;
+    }
+
     // We only need to update the searchModeUI on tab switch conditionally
     // as we only persist searchMode with ScotchBonnet enabled.
     if (
@@ -697,8 +712,11 @@ export class UrlbarInput {
         return;
       }
     }
-
+    this.document.getElementById("navigator-toolbox").setAttribute("suppress-primary-adjustment", !this.document.getElementById("navigator-toolbox").hasAttribute("zen-has-hover"));
     this.handleNavigation({ event });
+    this.document.ownerGlobal.setTimeout(() => {
+      this.document.getElementById("navigator-toolbox").removeAttribute("suppress-primary-adjustment");
+    }, 200);
   }
 
   /**
@@ -1092,7 +1110,11 @@ export class UrlbarInput {
     }
 
     if (!this.#providesSearchMode(result)) {
-      this.view.close({ elementPicked: true });
+      if (this._zenHandleUrlbarClose) {
+        this._zenHandleUrlbarClose(true);
+      } else {
+        this.view.close({ elementPicked: true });
+      }
     }
 
     this.controller.recordSelectedResult(event, result);
@@ -2124,6 +2146,10 @@ export class UrlbarInput {
     await this.#updateLayoutBreakoutDimensions();
   }
 
+  async zenUpdateLayoutBreakout() {
+    await this.#updateLayoutBreakout();
+  }
+
   startLayoutExtend() {
     if (!this.#allowBreakout || this.hasAttribute("breakout-extend")) {
       // Do not expand if the Urlbar does not support being expanded or it is
@@ -2146,6 +2172,11 @@ export class UrlbarInput {
 
     this.setAttribute("breakout-extend", "true");
 
+    if (lazy.ZEN_URLBAR_BEHAVIOR == 'float' || (lazy.ZEN_URLBAR_BEHAVIOR == 'floating-on-type' && !this.focusedViaMousedown)) {
+      this.setAttribute("zen-floating-urlbar", "true");
+    } else {
+      this.removeAttribute("zen-floating-urlbar");
+    }
     // Enable the animation only after the first extend call to ensure it
     // doesn't run when opening a new window.
     if (!this.hasAttribute("breakout-extend-animate")) {
@@ -2165,6 +2196,19 @@ export class UrlbarInput {
       return;
     }
 
+    if (this._zenHandleUrlbarClose) {
+      this._zenHandleUrlbarClose();
+    }
+    
+    // Arc like URLbar: Blur the input on exit
+    this.document.getElementById("navigator-toolbox").setAttribute("supress-primary-adjustment", !this.document.getElementById("navigator-toolbox").hasAttribute("zen-has-hover"));
+    this.window.gBrowser.selectedBrowser.focus();
+    this.window.setTimeout(() => {
+      this.document.getElementById("navigator-toolbox").removeAttribute("supress-primary-adjustment");
+    }, 100);
+    
+    this.removeAttribute("zen-floating-urlbar");
+
     this.removeAttribute("breakout-extend");
     this.#updateTextboxPosition();
   }
@@ -2997,7 +3041,7 @@ export class UrlbarInput {
    */
   _trimValue(val) {
     let trimmedValue = lazy.UrlbarPrefs.get("trimURLs")
-      ? lazy.BrowserUIUtils.trimURL(val)
+      ? this._zenTrimURL(val)
       : val;
     // Only trim value if the directionality doesn't change to RTL and we're not
     // showing a strikeout https protocol.
@@ -3367,6 +3411,10 @@ export class UrlbarInput {
       }
       reuseEmpty = true;
     }
+    if (this.hasAttribute("zen-newtab")) {
+      where = "tab";
+      reuseEmpty = true;
+    }
     if (
       where == "tab" &&
       reuseEmpty &&
@@ -3374,6 +3422,9 @@ export class UrlbarInput {
     ) {
       where = "current";
     }
+    if (this.window.gBrowser.selectedTab.hasAttribute("zen-empty-tab")) {
+      return "tab"; // Always open in a new tab if the current tab is "our empty tab".
+    }
     return where;
   }
 
@@ -4048,6 +4099,11 @@ export class UrlbarInput {
   }
 
   _on_click(event) {
+    if (lazy.ZEN_URLBAR_BEHAVIOR === 'float' && event.target == this.inputField) {
+      event.zenOriginalTarget = this.textbox;
+      this._on_mousedown(event);
+    }
+
     if (
       event.target == this.inputField ||
       event.target == this._inputContainer ||
@@ -4119,7 +4175,7 @@ export class UrlbarInput {
       }
     }
 
-    if (this.focusedViaMousedown) {
+    if (this.focusedViaMousedown || this.hasAttribute("zen-newtab")) {
       this.view.autoOpen({ event });
     } else {
       if (this._untrimOnFocusAfterKeydown) {
@@ -4159,9 +4215,12 @@ export class UrlbarInput {
   }
 
   _on_mousedown(event) {
-    switch (event.currentTarget) {
+    switch (event.zenOriginalTarget || event.currentTarget) {
       case this.textbox: {
         this._mousedownOnUrlbarDescendant = true;
+        if (event.type != "click" && lazy.ZEN_URLBAR_BEHAVIOR === 'float') {
+          return true;
+        }
 
         if (
           event.target != this.inputField &&
@@ -4171,8 +4230,8 @@ export class UrlbarInput {
           break;
         }
 
-        this.focusedViaMousedown = !this.focused;
-        this._preventClickSelectsAll = this.focused;
+        this.focusedViaMousedown = !(lazy.ZEN_URLBAR_BEHAVIOR === 'default' ? this.focused : this.hasAttribute("breakout-extend"));
+        this._preventClickSelectsAll = lazy.ZEN_URLBAR_BEHAVIOR === 'default' ? this.focused : this.hasAttribute("breakout-extend");
 
         // Keep the focus status, since the attribute may be changed
         // upon calling this.focus().
