diff --git a/browser/components/urlbar/UrlbarUtils.sys.mjs b/browser/components/urlbar/UrlbarUtils.sys.mjs
index 0bc15c02f56dd8f46a21fed02b4e21a741f27f41..3b2c1218532dcbd79d5c970ab9fd075d4e58d1ff 100644
--- a/browser/components/urlbar/UrlbarUtils.sys.mjs
+++ b/browser/components/urlbar/UrlbarUtils.sys.mjs
@@ -75,6 +75,7 @@ export var UrlbarUtils = {
     RESTRICT_SEARCH_KEYWORD: "restrictSearchKeyword",
     SUGGESTED_INDEX: "suggestedIndex",
     TAIL_SUGGESTION: "tailSuggestion",
+    ZEN_ACTION: "zenAction",
   }),
 
   // Defines provider types.
@@ -553,6 +554,8 @@ export var UrlbarUtils = {
           return this.RESULT_GROUP.HEURISTIC_SEARCH_TIP;
         case "HistoryUrlHeuristic":
           return this.RESULT_GROUP.HEURISTIC_HISTORY_URL;
+        case "ZenUrlbarProviderGlobalActions":
+          return this.RESULT_GROUP.ZEN_ACTION;
         default:
           if (result.providerName.startsWith("TestProvider")) {
             return this.RESULT_GROUP.HEURISTIC_TEST;
