diff --git a/browser/components/urlbar/UrlbarMuxerStandard.sys.mjs b/browser/components/urlbar/UrlbarMuxerStandard.sys.mjs
index 000a00f3e56c2f1d3bf4eec8c4924228de1cb418..f91f8cf38f0ba9c403bc284a7cdaaee17435b258 100644
--- a/browser/components/urlbar/UrlbarMuxerStandard.sys.mjs
+++ b/browser/components/urlbar/UrlbarMuxerStandard.sys.mjs
@@ -839,11 +839,16 @@ class MuxerUnifiedComplete extends UrlbarMuxer {
       result.providerName == "UrlbarProviderHeuristicFallback" &&
       state.context.heuristicResult?.providerName !=
         "UrlbarProviderHeuristicFallback"
+      && !(
+        result.type == UrlbarUtils.RESULT_TYPE.SEARCH &&
+        state.context.heuristicResult?.type == UrlbarUtils.RESULT_TYPE.URL
+      )
     ) {
       return false;
     }
 
     if (result.providerName == "UrlbarProviderTabToSearch") {
+      return false;
       // Discard the result if a tab-to-search result was added already.
       if (!state.canAddTabToSearch) {
         return false;
@@ -1490,7 +1495,9 @@ class MuxerUnifiedComplete extends UrlbarMuxer {
       usedLimits.maxResultCount++;
     }
 
+    if (!(result.heuristic && result.source == UrlbarUtils.RESULT_SOURCE.ZEN_ACTIONS)) {
     state.usedResultSpan += span;
+    }
     this._updateStatePostAdd(result, state);
 
     return true;
