diff --git a/browser/components/urlbar/UrlbarProviderHeuristicFallback.sys.mjs b/browser/components/urlbar/UrlbarProviderHeuristicFallback.sys.mjs
index e32b02941a6019703212cede4660a24b88845120..5cb363ed20d49c1ae8abedecc3b08540406b7a1b 100644
--- a/browser/components/urlbar/UrlbarProviderHeuristicFallback.sys.mjs
+++ b/browser/components/urlbar/UrlbarProviderHeuristicFallback.sys.mjs
@@ -76,22 +76,26 @@ export class UrlbarProviderHeuristicFallback extends UrlbarProvider {
         // Since we can't tell if this is a real URL and whether the user wants
         // to visit or search for it, we provide an alternative searchengine
         // match if the string looks like an alphanumeric origin or an e-mail.
-        let str = queryContext.searchString;
-        if (!URL.canParse(str)) {
-          if (
-            queryContext.keywordEnabled &&
-            (lazy.UrlUtils.looksLikeOrigin(str, {
-              noIp: true,
-              noPort: true,
-            }) ||
-              lazy.UrlUtils.REGEXP_COMMON_EMAIL.test(str))
-          ) {
+        let trimmedSearchString = queryContext.trimmedSearchString;
+        let [searchCandidate] = lazy.UrlbarShared.stripPrefixAndTrim(
+          trimmedSearchString,
+          {
+            trimSlash: true,
+           }
+        );
+        if (
+          queryContext.keywordEnabled &&
+          (lazy.UrlUtils.looksLikeUrl(searchCandidate) ||
+            lazy.UrlUtils.looksLikeOrigin(searchCandidate, {
+              allowPartialNumericalTLDs: true,
+            }) !== lazy.UrlUtils.LOOKS_LIKE_ORIGIN.NONE ||
+            lazy.UrlUtils.REGEXP_COMMON_EMAIL.test(trimmedSearchString))
+        ) {
             let searchResult = await this._engineSearchResult({ queryContext });
             if (instance != this.queryInstance) {
               return;
             }
             addCallback(this, searchResult);
-          }
         }
         return;
       }
