diff --git a/browser/components/urlbar/UrlbarPrefs.sys.mjs b/browser/components/urlbar/UrlbarPrefs.sys.mjs
index e48f0295d803ba9eac5ab9d01ebb3b3dacbc9d4b..e0f19b7e892bf86e733a4ea4aadcd91b7cb4c5ae 100644
--- a/browser/components/urlbar/UrlbarPrefs.sys.mjs
+++ b/browser/components/urlbar/UrlbarPrefs.sys.mjs
@@ -161,6 +161,9 @@ const PREF_URLBAR_DEFAULTS = /** @type {PreferenceDefinition[]} */ ([
   // Feature gate pref for flight status suggestions in the urlbar.
   ["flightStatus.featureGate", false],
 
+  // Whether to close the urlbar when blurring the window while the urlbar is focused.
+  ["closeOnWindowBlur", true],
+
   // The minimum prefix length of a flight status keyword the user must type to
   // trigger the suggestion. 0 means the min length should be taken from Nimbus
   // or remote settings.
@@ -486,6 +489,7 @@ const PREF_URLBAR_DEFAULTS = /** @type {PreferenceDefinition[]} */ ([
   ["shortcuts.tabs", true],
   ["shortcuts.history", true],
   ["shortcuts.actions", true],
+  ["shortcuts.workspaces", true],
 
   // Boolean to determine if the providers defined in `exposureResults`
   // should be displayed in search results. This can be set by a
@@ -854,6 +858,9 @@ function makeDefaultResultGroups({
    */
   let rootGroup = {
     children: [
+      { children: [{ group: lazy.UrlbarShared.RESULT_GROUP.ZEN_PINNED_TAB }] },
+      { children: [{ group: lazy.UrlbarShared.RESULT_GROUP.ZEN_ACTION }] },
+      { children: [{ group: lazy.UrlbarShared.RESULT_GROUP.ZEN_WORKSPACE }] },
       // heuristic
       {
         maxResultCount: 1,
