diff --git a/browser/components/urlbar/UrlbarProviderPlaces.sys.mjs b/browser/components/urlbar/UrlbarProviderPlaces.sys.mjs
index 4db61038e5e476bad3a61dbdb707e5222c1f08f8..9eca13d9cfac3b762917aaaa942267effb743cf7 100644
--- a/browser/components/urlbar/UrlbarProviderPlaces.sys.mjs
+++ b/browser/components/urlbar/UrlbarProviderPlaces.sys.mjs
@@ -45,11 +45,13 @@ function defaultQuery(conditions = "") {
   let query = `
      SELECT h.url, h.title, ${SQL_BOOKMARK_TAGS_FRAGMENT}, h.id, t.open_count,
             ${lazy.PAGES_FRECENCY_FIELD} AS frecency, t.userContextId,
-            h.last_visit_date, NULLIF(t.groupId, '') groupId
+            h.last_visit_date, NULLIF(t.groupId, '') groupId, zp.url AS pinned_url, zp.title AS pinned_title
      FROM moz_places h
      LEFT JOIN moz_openpages_temp t
             ON t.url = h.url
             AND (t.userContextId = :userContextId OR (t.userContextId <> -1 AND :userContextId IS NULL))
+     LEFT JOIN zen_pins zp
+                 ON zp.url = h.url
      WHERE (
         (:switchTabsEnabled AND t.open_count > 0) OR
         ${lazy.PAGES_FRECENCY_FIELD} <> 0
@@ -63,7 +65,7 @@ function defaultQuery(conditions = "") {
                               :matchBehavior, :searchBehavior, NULL)
          ELSE
            AUTOCOMPLETE_MATCH(:searchString, h.url,
-                              h.title, '',
+                              IFNULL(zp.title, h.title), '',
                               h.visit_count, h.typed,
                               0, t.open_count,
                               :matchBehavior, :searchBehavior, NULL)
@@ -1176,11 +1178,13 @@ class Search {
       ? lazy.PlacesUtils.toDate(lastVisitPRTime).getTime()
       : undefined;
     let tabGroup = row.getResultByName("groupId");
+    let pinnedTitle = row.getResultByIndex(12);
+    let pinnedUrl = row.getResultByIndex("pinned_url");
 
     let match = {
       placeId,
-      value: url,
-      comment: bookmarkTitle || historyTitle,
+      value: pinnedUrl || url,
+      comment: pinnedTitle || bookmarkTitle || historyTitle,
       icon: UrlbarUtils.getIconForUrl(url),
       frecency: frecency || FRECENCY_DEFAULT,
       userContextId,
