diff --git a/browser/components/tabbrowser/TabsList.sys.mjs b/browser/components/tabbrowser/TabsList.sys.mjs
index b394a980869d5e5df01b6afdd81e0f41c090e6ec..236c99db0dfa5ebcba50d55c1e4eb6ab7b8110b2 100644
--- a/browser/components/tabbrowser/TabsList.sys.mjs
+++ b/browser/components/tabbrowser/TabsList.sys.mjs
@@ -87,7 +87,7 @@ class TabsListBase {
     /** @type {function(MozTabbrowserTab):boolean} */
     this.filterFn = onlyHiddenTabs
       ? tab => filterFn(tab) && tab.hidden
-      : filterFn;
+      : tab =>  !tab.hasAttribute("zen-empty-tab") && filterFn(tab);
     /** @type {Element} */
     this.containerNode = containerNode;
     /** @type {Element|null} */
