diff --git a/browser/components/tabbrowser/TabsList.sys.mjs b/browser/components/tabbrowser/TabsList.sys.mjs
index 58f91dd9f80aa34ff8d8095cbaf5ae7a6c8dc1e1..c4c9ad5d93053f5fc726a0b10fa8d56c6bbc4003 100644
--- a/browser/components/tabbrowser/TabsList.sys.mjs
+++ b/browser/components/tabbrowser/TabsList.sys.mjs
@@ -87,7 +87,7 @@ class TabsListBase {
     /** @type {function(MozTabbrowserTab):boolean} */
     this.filterFn = onlyHiddenTabs
       ? tab => filterFn(tab) && tab.hidden
-      : filterFn;
+      : tab =>  !tab.hasAttribute("zen-empty-tab") && filterFn(tab);
     /** @type {Element} */
     this.containerNode = containerNode;
     /** @type {Element|null} */
