diff --git a/browser/components/tabbrowser/TabUnloader.sys.mjs b/browser/components/tabbrowser/TabUnloader.sys.mjs
index a6d6c5405f98c839b7c0f2007e053e874bb7ce07..9d8c6066ce917e15fbf9e19284a3db50aa5a3815 100644
--- a/browser/components/tabbrowser/TabUnloader.sys.mjs
+++ b/browser/components/tabbrowser/TabUnloader.sys.mjs
@@ -51,7 +51,7 @@ let CRITERIA_WEIGHT = 1;
  */
 let DefaultTabUnloaderMethods = {
   isNonDiscardable(tab, weight) {
-    if (tab.undiscardable || tab.selected) {
+    if (tab.undiscardable || tab.selected || tab.zenModeActive) {
       return weight;
     }
 
@@ -105,7 +105,7 @@ let DefaultTabUnloaderMethods = {
 
   *iterateTabs() {
     for (let win of Services.wm.getEnumerator("navigator:browser")) {
-      for (let tab of win.gBrowser.tabs) {
+      for (let tab of win.gZenWorkspaces.allStoredTabs) {
         yield { tab, gBrowser: win.gBrowser };
       }
     }
