diff --git a/browser/app/splash.rc b/browser/app/splash.rc
index 04c1d3e1db7d067d579512999a08de85a0aa212c..4e04eb5008ca1701c731c3109ba184ccf465b51b 100644
--- a/browser/app/splash.rc
+++ b/browser/app/splash.rc
@@ -5,6 +5,9 @@
 #include <windows.h>
 #include "nsNativeAppSupportWin.h"
 
+#define MANIFEST_PATH "firefox.exe.manifest" 
+CREATEPROCESS_MANIFEST_RESOURCE_ID  RT_MANIFEST             MANIFEST_PATH
+
 IDI_APPICON ICON FIREFOX_ICO
 IDI_DOCUMENT ICON DOCUMENT_ICO
 IDI_APPLICATION ICON FIREFOX_ICO
