diff --git a/dom/base/Document.cpp b/dom/base/Document.cpp
index 7bbbaf6fb936eda9cf87f164f93f4a96a50cf733..b1a50d8b4932f60f882b7f60c7d65ce3a9fde99d 100644
--- a/dom/base/Document.cpp
+++ b/dom/base/Document.cpp
@@ -461,6 +461,7 @@
 #include "prtime.h"
 #include "prtypes.h"
 #include "xpcpublic.h"
+#include "mozilla/ZenStyleSheetCache.h"
 
 // clang-format off
 #include "mozilla/Encoding.h"
@@ -3446,6 +3447,10 @@ void Document::FillStyleSetUserAndUASheets() {
   for (StyleSheet* sheet : *sheetService->UserStyleSheets()) {
     styleSet.AppendStyleSheet(*sheet);
   }
+  if (auto sheet = zen::ZenStyleSheetCache::Singleton()->GetModsSheet(); sheet && IsInChromeDocShell()) {
+    // The mods sheet is only used in the chrome docshell.
+    styleSet.AppendStyleSheet(*sheet);
+  }
 
   StyleSheet* sheet = IsInChromeDocShell() ? cache->GetUserChromeSheet()
                                            : cache->GetUserContentSheet();
