diff --git a/dom/base/Document.cpp b/dom/base/Document.cpp
index 0c2ef36f9765b6904659ac0ed4329468ca89225f..d06552e8d4ac102dd50a1afea10938ca32aeb788 100644
--- a/dom/base/Document.cpp
+++ b/dom/base/Document.cpp
@@ -466,6 +466,7 @@
 #include "prtime.h"
 #include "prtypes.h"
 #include "xpcpublic.h"
+#include "mozilla/ZenStyleSheetCache.h"
 
 // clang-format off
 #include "mozilla/Encoding.h"
@@ -3428,6 +3429,10 @@ void Document::FillStyleSetUserAndUASheets() {
   for (StyleSheet* sheet : *sheetService->UserStyleSheets()) {
     styleSet.AppendStyleSheet(*sheet);
   }
+  if (auto sheet = zen::ZenStyleSheetCache::Singleton()->GetModsSheet(); sheet && IsInChromeDocShell()) {
+    // The mods sheet is only used in the chrome docshell.
+    styleSet.AppendStyleSheet(*sheet);
+  }
 
   StyleSheet* sheet = IsInChromeDocShell() ? cache->GetUserChromeSheet()
                                            : cache->GetUserContentSheet();
