diff --git a/dom/base/Document.cpp b/dom/base/Document.cpp
index 40da41ae4ac7d8d61c664e25bec8a74091d8b3e0..59773e80c6220115d0c51e268bb1eff0fe208010 100644
--- a/dom/base/Document.cpp
+++ b/dom/base/Document.cpp
@@ -10,6 +10,7 @@
 
 #include "mozilla/dom/Document.h"
 #include "mozilla/dom/DocumentInlines.h"
+#include "mozilla/ZenStyleSheetCache.h"
 
 #include <inttypes.h>
 #include <stdlib.h>
@@ -3306,6 +3307,11 @@ void Document::FillStyleSetUserAndUASheets() {
     styleSet.AppendStyleSheet(*sheet);
   }
 
+  if (auto sheet = zen::ZenStyleSheetCache::Singleton()->GetModsSheet(); sheet && IsInChromeDocShell()) {
+    // The mods sheet is only used in the chrome docshell.
+    styleSet.AppendStyleSheet(*sheet);
+  }
+
   StyleSheet* sheet = IsInChromeDocShell() ? cache->GetUserChromeSheet()
                                            : cache->GetUserContentSheet();
   if (sheet) {
