diff --git a/dom/base/Document.cpp b/dom/base/Document.cpp
index d9917d78b93d3200cbe15a1ee316cf01c0ad88de..4d7e4f9d57f0d121c717758ead06e554e01aea91 100644
--- a/dom/base/Document.cpp
+++ b/dom/base/Document.cpp
@@ -462,6 +462,7 @@
 #include "prtime.h"
 #include "prtypes.h"
 #include "xpcpublic.h"
+#include "mozilla/ZenStyleSheetCache.h"
 
 // clang-format off
 #include "mozilla/Encoding.h"
@@ -3327,6 +3328,10 @@ void Document::FillStyleSetUserAndUASheets() {
   for (StyleSheet* sheet : *sheetService->UserStyleSheets()) {
     styleSet.AppendStyleSheet(*sheet);
   }
+  if (auto sheet = zen::ZenStyleSheetCache::Singleton()->GetModsSheet(); sheet && IsInChromeDocShell()) {
+    // The mods sheet is only used in the chrome docshell.
+    styleSet.AppendStyleSheet(*sheet);
+  }
 
   StyleSheet* sheet = IsInChromeDocShell() ? cache->GetUserChromeSheet()
                                            : cache->GetUserContentSheet();
