diff --git a/browser/components/extensions/ExtensionPopups.sys.mjs b/browser/components/extensions/ExtensionPopups.sys.mjs
index 1cd35175507bd442aae6857e65015cfed00058e3..376377629e9732b292085ba94309e47cc53ee2ef 100644
--- a/browser/components/extensions/ExtensionPopups.sys.mjs
+++ b/browser/components/extensions/ExtensionPopups.sys.mjs
@@ -418,6 +418,8 @@ export class BasePopup {
       this.browser.style.minWidth = `${width}px`;
       this.browser.style.height = `${height}px`;
       this.browser.style.minHeight = `${height}px`;
+
+      this.lastCalculatedInViewHeight = height;
     }
 
     let event = new this.window.CustomEvent("WebExtPopupResized", { detail });
