diff --git a/toolkit/components/downloads/DownloadList.sys.mjs b/toolkit/components/downloads/DownloadList.sys.mjs
index 8e6d1987c359c9388ae73aeeb57ec6c94be033e5..0162a3f3033e97be95d1230852d38c9fe0c524a1 100644
--- a/toolkit/components/downloads/DownloadList.sys.mjs
+++ b/toolkit/components/downloads/DownloadList.sys.mjs
@@ -50,6 +50,7 @@ const FILE_EXTENSIONS = [
   "jpg",
   "jpeg",
   "json",
+  "jxl",
   "m4a",
   "mdb",
   "mid",
