diff --git a/browser/components/sessionstore/content/aboutSessionRestore.xhtml b/browser/components/sessionstore/content/aboutSessionRestore.xhtml
index 05538be5d95492903e44076dc43e195cbc76c10d..3c2ee6a9d44802657c9e0d5ef6c78e0b3920a9a5 100644
--- a/browser/components/sessionstore/content/aboutSessionRestore.xhtml
+++ b/browser/components/sessionstore/content/aboutSessionRestore.xhtml
@@ -20,6 +20,7 @@
     <link rel="localization" href="browser/aboutSessionRestore.ftl"/>
     <link rel="localization" href="branding/brand.ftl"/>
     <script src="chrome://browser/content/aboutSessionRestore.js"/>
+    <script src="chrome://browser/content/zenThemeModifier.js"></script>
   </head>
 
   <body>
