diff --git a/devtools/client/jar.mn b/devtools/client/jar.mn
index 8dabb821f0491d5811d57059a8a28848df0f96d4..2af9f8644225b6d4367f202f5330596c99a3bcbc 100644
--- a/devtools/client/jar.mn
+++ b/devtools/client/jar.mn
@@ -234,6 +234,7 @@ devtools.jar:
     content/shared/sourceeditor/codemirror/codemirror.bundle.js (shared/sourceeditor/codemirror/codemirror.bundle.js)
     content/shared/sourceeditor/codemirror/lib/codemirror.css (shared/sourceeditor/codemirror/lib/codemirror.css)
     content/shared/sourceeditor/codemirror/mozilla.css (shared/sourceeditor/codemirror/mozilla.css)
+    content/shared/sourceeditor/codemirror/material-darker.css (shared/sourceeditor/codemirror/material-darker.css)
     content/shared/sourceeditor/codemirror/cmiframe.html (shared/sourceeditor/codemirror/cmiframe.html)
 
     # Debugger
