diff --git a/build/moz.configure/lto-pgo.configure b/build/moz.configure/lto-pgo.configure
index df2b0bf0081edfdc2c7bccaefc780d2b0c6555ff..8619fc5fbf284c7a5660d5b7a77494404e448e44 100644
--- a/build/moz.configure/lto-pgo.configure
+++ b/build/moz.configure/lto-pgo.configure
@@ -256,7 +256,6 @@ option(
     help="Indicate that ld64 is free of symbol aliasing bugs",
 )
 
-imply_option("MOZ_LD64_KNOWN_GOOD", moz_automation)
 
 use_fat_lto = cxx_compiler.try_link(
     ldflags=depends(stlport_libs)(
