diff --git a/build/moz.configure/update-programs.configure b/build/moz.configure/update-programs.configure
index 4edcccf4c49749e6d80ee75c1e43cfa32a555933..9c7c69c97740e5c02766f913d40bd4f059940d3a 100644
--- a/build/moz.configure/update-programs.configure
+++ b/build/moz.configure/update-programs.configure
@@ -71,7 +71,7 @@ def mac_prod_requirements_string(identifier):
         f'identifier "{identifier}" and anchor apple generic and '
         "certificate 1[field.1.2.840.113635.100.6.2.6] and "
         "certificate leaf[field.1.2.840.113635.100.6.1.13] and "
-        'certificate leaf[subject.OU] = "43AQ936H96"'
+        'certificate leaf[subject.OU] = "9V5K9TP787"'
     )
 
 
