diff --git a/toolkit/themes/shared/desktop-jar.inc.mn b/toolkit/themes/shared/desktop-jar.inc.mn
index 5907a9773fe608520b0088c1fc40c23a7003530f..535a7fae17f295a3bc97cf807d9c9dd496ec2352 100644
--- a/toolkit/themes/shared/desktop-jar.inc.mn
+++ b/toolkit/themes/shared/desktop-jar.inc.mn
@@ -162,3 +162,5 @@
   skin/classic/global/reader/content-width-20.svg          (../../shared/reader/content-width-20.svg)
   skin/classic/global/tree/sort-asc.svg                    (../../shared/tree/sort-asc.svg)
   skin/classic/global/tree/sort-dsc.svg                    (../../shared/tree/sort-dsc.svg)
+
+#include zen-desktop.jar.inc.mn
\ No newline at end of file
