diff --git a/browser/components/tabpreview/tab-preview-panel.mjs b/browser/components/tabpreview/tab-preview-panel.mjs
index 683b2c17ec893a97680d885c622029591dda7f29..605f8682368d691f7cc18dfb1775fa2622006ebd 100644
--- a/browser/components/tabpreview/tab-preview-panel.mjs
+++ b/browser/components/tabpreview/tab-preview-panel.mjs
@@ -7,7 +7,7 @@ var { XPCOMUtils } = ChromeUtils.importESModule(
 );
 
 const POPUP_OPTIONS = {
-  position: "bottomleft topleft",
+  position: "topright topleft",
   x: 0,
   y: -2,
 };
