diff --git a/browser/components/tabpreview/tabpreview.mjs b/browser/components/tabpreview/tabpreview.mjs
index 5256ab22ff3a79d9c020dac2ce375f5e42f9c13c..8dacc6b4e04b8b1848b1b6280e8132202d1b51b6 100644
--- a/browser/components/tabpreview/tabpreview.mjs
+++ b/browser/components/tabpreview/tabpreview.mjs
@@ -44,7 +44,7 @@ export default class TabPreview extends MozLitElement {
       this,
       "_prefDisplayThumbnail",
       TAB_PREVIEW_USE_THUMBNAILS_PREF,
-      false
+      true
     );
   }
 
@@ -109,7 +109,7 @@ export default class TabPreview extends MozLitElement {
 
   showPreview() {
     this.panel.openPopup(this.tab, {
-      position: "bottomleft topleft",
+      position: "topright topleft",
       y: -2,
       isContextMenu: false,
     });
