diff --git a/browser/components/controlcenter/content/identityPanel.inc.xhtml b/browser/components/controlcenter/content/identityPanel.inc.xhtml
index 0344fcb566e8e0bedd92b0d45e42aea38de5c329..0d5296700590dd0c8a90929976b8cbc5370e7b47 100644
--- a/browser/components/controlcenter/content/identityPanel.inc.xhtml
+++ b/browser/components/controlcenter/content/identityPanel.inc.xhtml
@@ -30,7 +30,7 @@
       <toolbarseparator/>
       <vbox class="panel-subview-body">
         <toolbarbutton id="identity-popup-security-button"
-                       class="identity-popup-security-connection subviewbutton subviewbutton-iconic"
+                       class="identity-popup-security-connection subviewbutton"
                        oncommand="gIdentityHandler.showSecuritySubView();">
           <image class="identity-popup-security-connection-icon toolbarbutton-icon"></image>
           <hbox class="toolbarbutton-text" flex="1">
