diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index 44b91b7bb3e0196b7f1f923c43fb8967f569cd28..907f7059cb44b990b5eb3a90d5df523075cd6ae1 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -22,6 +22,7 @@ def check_moz_app_id(moz_app_id, build_project):
 
 project_flag(
     env="MOZ_APP_VENDOR",
+    default="Zen Team",
     nargs=1,
     help='Used for application.ini\'s "Vendor" field, which also impacts profile location and user-visible fields',
 )
@@ -35,6 +36,7 @@ project_flag(
 
 project_flag(
     "MOZ_APP_PROFILE",
+    default="zen",
     nargs=1,
     help='Used for application.ini\'s "Profile" field, which controls profile location',
 )
@@ -86,10 +88,13 @@ option(
 )
 set_config("MOZ_INCLUDE_SOURCE_INFO", True, when="MOZ_INCLUDE_SOURCE_INFO")
 
+option(env="ZEN_FIREFOX_VERSION", help="Set Zen version of Firefox", nargs=1)
+set_config("ZEN_FIREFOX_VERSION", depends_if("ZEN_FIREFOX_VERSION")(lambda v: v[0]))
+
 option(
     "--with-distribution-id",
     nargs=1,
-    default="org.mozilla",
+    default="app.zen-browser",
     help="Set distribution-specific id",
 )
 set_config("MOZ_DISTRIBUTION_ID", depends("--with-distribution-id")(lambda v: v[0]))
@@ -931,9 +936,9 @@ set_config("MOZ_SYSTEM_AV1", True, when="--with-system-av1")
 option("--disable-jxl", help="Disable jxl image support")
 
 
-@depends("--disable-jxl", milestone.is_nightly)
-def jxl(value, is_nightly):
-    if is_nightly and value:
+@depends("--disable-jxl")
+def jxl(value):
+    if value:
         return True
 
 
@@ -2064,7 +2069,7 @@ set_define("A11Y_LOG", True, when=a11y_log)
 # ==============================================================
 @depends(milestone)
 def require_signing(milestone):
-    return milestone.is_release_or_beta and not milestone.is_esr
+    return False
 
 
 option(
