diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index 77069eaacf383214e705f04733cd4f5db1fca1e8..9abbd6a9ba58759a33f9e9a0bea934a5fe022893 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -22,6 +22,7 @@ def check_moz_app_id(moz_app_id, build_project):
 
 project_flag(
     env="MOZ_APP_VENDOR",
+    default="Zen Team",
     nargs=1,
     help='Used for application.ini\'s "Vendor" field, which also impacts profile location and user-visible fields',
 )
@@ -35,6 +36,7 @@ project_flag(
 
 project_flag(
     "MOZ_APP_PROFILE",
+    default="zen",
     nargs=1,
     help='Used for application.ini\'s "Profile" field, which controls profile location',
 )
@@ -87,10 +89,13 @@ option(
 )
 set_config("MOZ_INCLUDE_SOURCE_INFO", True, when="MOZ_INCLUDE_SOURCE_INFO")
 
+option(env="ZEN_FIREFOX_VERSION", help="Set Zen version of Firefox", nargs=1)
+set_config("ZEN_FIREFOX_VERSION", depends_if("ZEN_FIREFOX_VERSION")(lambda v: v[0]))
+
 option(
     "--with-distribution-id",
     nargs=1,
-    default="org.mozilla",
+    default="app.zen-browser",
     help="Set distribution-specific id",
 )
 set_config("MOZ_DISTRIBUTION_ID", depends("--with-distribution-id")(lambda v: v[0]))
@@ -932,9 +937,9 @@ set_config("MOZ_SYSTEM_AV1", True, when="--with-system-av1")
 option("--disable-jxl", help="Disable jxl image support")
 
 
-@depends("--disable-jxl", milestone.is_nightly)
-def jxl(value, is_nightly):
-    if is_nightly and value:
+@depends("--disable-jxl")
+def jxl(value):
+    if value:
         return True
 
 
@@ -2065,7 +2070,7 @@ set_define("A11Y_LOG", True, when=a11y_log)
 # ==============================================================
 @depends(milestone)
 def require_signing(milestone):
-    return milestone.is_release_or_beta and not milestone.is_esr
+    return False
 
 
 option(
