diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index 1f85d2831f2f71ebe4c7216bd604926d71e5189e..fb0d07554a592339f5a794151d953cb9269afd75 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -81,6 +81,13 @@ option(
 )
 set_config("MOZ_INCLUDE_SOURCE_INFO", True, when="MOZ_INCLUDE_SOURCE_INFO")
 
+option(
+    env="ZEN_FIREFOX_VERSION",
+    default="130.0",
+    help="Set the version of the browser",
+)
+set_config("ZEN_FIREFOX_VERSION", "130.0", when="ZEN_FIREFOX_VERSION")
+
 option(
     "--with-distribution-id",
     nargs=1,
@@ -905,9 +912,9 @@ set_define("MOZ_AV1", av1)
 option("--disable-jxl", help="Disable jxl image support")
 
 
-@depends("--disable-jxl", milestone.is_nightly)
-def jxl(value, is_nightly):
-    if is_nightly and value:
+@depends("--disable-jxl")
+def jxl(value):
+    if value:
         return True
 
 
