diff --git a/modules/libpref/moz.build b/modules/libpref/moz.build
index 2c8b218c618f8fa73ab4b21f4155d5f063bc1bb4..e8b525665d41e10309b4bb1779547d4b47e08f90 100644
--- a/modules/libpref/moz.build
+++ b/modules/libpref/moz.build
@@ -90,6 +90,7 @@ pref_groups = [
     "view_source",
     "webgl",
     "widget",
+    "zen",
     "zoom",
 ]
 if CONFIG["OS_TARGET"] == "Android":
@@ -140,8 +141,8 @@ GeneratedFile(
     script="init/generate_static_pref_list.py",
     entry_point="emit_code",
     inputs=gen_inputs,
+    extra_deps=["init/zen-static-prefs.inc"],
 )
-
 PYTHON_UNITTEST_MANIFESTS += [
     "test/python.toml",
 ]
