diff --git a/toolkit/modules/LightweightThemeConsumer.sys.mjs b/toolkit/modules/LightweightThemeConsumer.sys.mjs
index 756cf47d4dd5e096af96a95ab63c20ac9348572b..a74c1ffd73c19ee54ed87318d97898d93b974ea7 100644
--- a/toolkit/modules/LightweightThemeConsumer.sys.mjs
+++ b/toolkit/modules/LightweightThemeConsumer.sys.mjs
@@ -34,7 +34,7 @@ const toolkitVariableMap = [
       lwtProperty: "accentcolor",
       processColor(rgbaChannels) {
         if (!rgbaChannels || rgbaChannels.a == 0) {
-          return "white";
+          return "var(--zen-colors-primary)";
         }
         // Remove the alpha channel
         const { r, g, b } = rgbaChannels;
@@ -366,7 +366,7 @@ LightweightThemeConsumer.prototype = {
     let forcedColorsThemeOverride =
       this.FORCED_COLORS_OVERRIDE_ENABLED &&
       this.forcedColorsMediaQuery?.matches;
-    if (!theme || forcedColorsThemeOverride) {
+    if (!theme || forcedColorsThemeOverride || Services.prefs.getBoolPref("zen.theme.disable-lightweight")) {
       theme = { id: DEFAULT_THEME_ID };
     }
     let builtinThemeConfig = lazy.BuiltInThemeConfig.get(theme.id);
