diff --git a/toolkit/modules/LightweightThemeConsumer.sys.mjs b/toolkit/modules/LightweightThemeConsumer.sys.mjs
index c62ad0b054296d0302aaeb60d78a30eba257647e..68f5ff2761ab4ded10cd98fb12e5668ece511e14 100644
--- a/toolkit/modules/LightweightThemeConsumer.sys.mjs
+++ b/toolkit/modules/LightweightThemeConsumer.sys.mjs
@@ -34,7 +34,7 @@ const toolkitVariableMap = [
       lwtProperty: "accentcolor",
       processColor(rgbaChannels) {
         if (!rgbaChannels || rgbaChannels.a == 0) {
-          return "white";
+          return "var(--zen-colors-primary)";
         }
         // Remove the alpha channel
         const { r, g, b } = rgbaChannels;
@@ -347,7 +347,7 @@ LightweightThemeConsumer.prototype = {
     let forcedColorsThemeOverride =
       this.FORCED_COLORS_OVERRIDE_ENABLED &&
       this.forcedColorsMediaQuery?.matches;
-    if (!theme || forcedColorsThemeOverride) {
+    if (!theme || forcedColorsThemeOverride || Services.prefs.getBoolPref("zen.theme.disable-lightweight")) {
       theme = { id: DEFAULT_THEME_ID };
     }
     let builtinThemeConfig = lazy.BuiltInThemeConfig.get(theme.id);
