diff --git a/toolkit/modules/LightweightThemeConsumer.sys.mjs b/toolkit/modules/LightweightThemeConsumer.sys.mjs
index 1d16b3f191a53b6877122086de9bba1fbcaa71f4..64df5577aac095ddd3d9a381bf118496bf279276 100644
--- a/toolkit/modules/LightweightThemeConsumer.sys.mjs
+++ b/toolkit/modules/LightweightThemeConsumer.sys.mjs
@@ -31,7 +31,7 @@ const toolkitVariableMap = [
       lwtProperty: "accentcolor",
       processColor(rgbaChannels) {
         if (!rgbaChannels || rgbaChannels.a == 0) {
-          return "white";
+          return "var(--zen-colors-primary)";
         }
         // Remove the alpha channel
         const { r, g, b } = rgbaChannels;
@@ -320,7 +320,7 @@ LightweightThemeConsumer.prototype = {
     }
 
     let theme = useDarkTheme ? themeData.darkTheme : themeData.theme;
-    if (!theme) {
+    if (!theme || Services.prefs.getBoolPref("zen.theme.disable-lightweight")) {
       theme = { id: DEFAULT_THEME_ID };
     }
     let hasTheme = theme.id != DEFAULT_THEME_ID;
