diff --git a/toolkit/modules/LightweightThemeConsumer.sys.mjs b/toolkit/modules/LightweightThemeConsumer.sys.mjs
index c12ba1cd3891dab6ccf07e0ee069d44cc11f3c8f..d7e35aa2dc93629f74e62d6437e3a064242e8974 100644
--- a/toolkit/modules/LightweightThemeConsumer.sys.mjs
+++ b/toolkit/modules/LightweightThemeConsumer.sys.mjs
@@ -32,7 +32,7 @@ const toolkitVariableMap = [
       lwtProperty: "accentcolor",
       processColor(rgbaChannels) {
         if (!rgbaChannels || rgbaChannels.a == 0) {
-          return "white";
+          return "var(--zen-colors-primary)";
         }
         // Remove the alpha channel
         const { r, g, b } = rgbaChannels;
@@ -320,7 +320,7 @@ LightweightThemeConsumer.prototype = {
     let forcedColorsThemeOverride =
       this.FORCED_COLORS_OVERRIDE_ENABLED &&
       this.forcedColorsMediaQuery?.matches;
-    if (!theme || forcedColorsThemeOverride) {
+    if (!theme || forcedColorsThemeOverride || Services.prefs.getBoolPref("zen.theme.disable-lightweight")) {
       theme = { id: DEFAULT_THEME_ID };
     }
     let builtinThemeConfig = lazy.BuiltInThemeConfig.get(theme.id);
