diff --git a/toolkit/modules/LightweightThemeConsumer.sys.mjs b/toolkit/modules/LightweightThemeConsumer.sys.mjs
index c42269ca92e21d1058046d7247e19dc5fde4c1c0..b12d38aa8999d4c2c4c7218b299cd9ed1566fc7d 100644
--- a/toolkit/modules/LightweightThemeConsumer.sys.mjs
+++ b/toolkit/modules/LightweightThemeConsumer.sys.mjs
@@ -31,7 +31,7 @@ const toolkitVariableMap = [
       lwtProperty: "accentcolor",
       processColor(rgbaChannels) {
         if (!rgbaChannels || rgbaChannels.a == 0) {
-          return "white";
+          return "var(--zen-colors-primary)";
         }
         // Remove the alpha channel
         const { r, g, b } = rgbaChannels;
@@ -320,7 +320,7 @@ LightweightThemeConsumer.prototype = {
     }
 
     let theme = useDarkTheme ? themeData.darkTheme : themeData.theme;
-    if (!theme) {
+    if (!theme || Services.prefs.getBoolPref("zen.theme.disable-lightweight")) {
       theme = { id: DEFAULT_THEME_ID };
     }
     let hasTheme = theme.id != DEFAULT_THEME_ID;
