diff --git a/toolkit/modules/PopupNotifications.sys.mjs b/toolkit/modules/PopupNotifications.sys.mjs
index 5e88f279e570edbf95823bb9dc9835bae89feb32..f50db584dff740583665239d2cc2c5d00c5010e9 100644
--- a/toolkit/modules/PopupNotifications.sys.mjs
+++ b/toolkit/modules/PopupNotifications.sys.mjs
@@ -1532,6 +1532,15 @@ PopupNotifications.prototype = {
   ) {
     for (let anchorElement of anchorElements) {
       anchorElement.setAttribute(ICON_ATTRIBUTE_SHOWING, "true");
+      let toolbox = anchorElement.closest("#navigator-toolbox");
+      if (toolbox) {
+        // Disable transitions for now, see gh-11667
+        toolbox.style.transition = "none";
+        toolbox.setAttribute("zen-compact-mode-active", "true");
+        anchorElement.documentGlobal.setTimeout(() => {
+          toolbox.style.transition = "";
+        }, 0);
+      }
     }
   },
 
