diff --git a/toolkit/modules/ShortcutUtils.sys.mjs b/toolkit/modules/ShortcutUtils.sys.mjs
index e52e88a09a281e69ecf15e7b84b8b98736e1dbdd..f47fbe24e0860195e11fe5231ab27e52aad96d0b 100644
--- a/toolkit/modules/ShortcutUtils.sys.mjs
+++ b/toolkit/modules/ShortcutUtils.sys.mjs
@@ -414,13 +414,11 @@ export var ShortcutUtils = {
         }
         break;
       case event.DOM_VK_UP: // fall through
-      case event.DOM_VK_LEFT:
         if (metaAltAccel) {
           return ShortcutUtils.PREVIOUS_TAB;
         }
         break;
       case event.DOM_VK_DOWN: // fall through
-      case event.DOM_VK_RIGHT:
         if (metaAltAccel) {
           return ShortcutUtils.NEXT_TAB;
         }
