diff --git a/toolkit/modules/UpdateUtils.sys.mjs b/toolkit/modules/UpdateUtils.sys.mjs
index 2c8a093013d40623aa16864eca18945887add7c8..592f8a77665f8e8b4c2f64fe6ae44f3f1f3935e1 100644
--- a/toolkit/modules/UpdateUtils.sys.mjs
+++ b/toolkit/modules/UpdateUtils.sys.mjs
@@ -81,7 +81,7 @@ export var UpdateUtils = {
           replacement = Services.appinfo.name;
           break;
         case "VERSION":
-          replacement = Services.appinfo.version;
+          replacement = AppConstants.ZEN_FIREFOX_VERSION;
           break;
         case "BUILD_ID":
           replacement = Services.appinfo.appBuildID;
