diff --git a/browser/themes/addons/jar.mn b/browser/themes/addons/jar.mn
index 69bb3000d6d2d249d1d77262dc2a0057d0318308..bbed04a6b6f11c582e17e08201db260b9ed496b7 100644
--- a/browser/themes/addons/jar.mn
+++ b/browser/themes/addons/jar.mn
@@ -5,8 +5,8 @@
 browser.jar:
 %  resource builtin-themes %content/builtin-themes/
 
-  content/builtin-themes/alpenglow                 (alpenglow/*.svg)
-  content/builtin-themes/alpenglow/manifest.json   (alpenglow/manifest.json)
+#  content/builtin-themes/alpenglow                 (alpenglow/*.svg)
+#  content/builtin-themes/alpenglow/manifest.json   (alpenglow/manifest.json)
 
   content/builtin-themes/dark                      (dark/*.svg)
   content/builtin-themes/dark                      (dark/*.css)
@@ -100,3 +100,5 @@ browser.jar:
   content/builtin-themes/colorways/2022innovator/balanced/manifest.json       (colorways/2022-independent-voices/innovator/balanced/manifest.json)
   content/builtin-themes/colorways/2022innovator/bold                         (colorways/2022-independent-voices/innovator/bold/*.svg)
   content/builtin-themes/colorways/2022innovator/bold/manifest.json           (colorways/2022-independent-voices/innovator/bold/manifest.json)
+
+#include zen-addons.inc.mn
\ No newline at end of file
