diff --git a/netwerk/protocol/http/nsHttpHandler.cpp b/netwerk/protocol/http/nsHttpHandler.cpp
index 0ddb378e2b2ca20edc9cf7494d80d308d8cf51eb..f311cf92a16d652e24d479038f2ccad5f19b27d4 100644
--- a/netwerk/protocol/http/nsHttpHandler.cpp
+++ b/netwerk/protocol/http/nsHttpHandler.cpp
@@ -861,12 +861,12 @@ void nsHttpHandler::BuildUserAgent() {
   mUserAgent += mProductSub;
 
   bool isFirefox = mAppName.EqualsLiteral("Firefox");
-  if (isFirefox || mCompatFirefoxEnabled) {
+  if (isFirefox || mCompatFirefoxEnabled || true) {
     // "Firefox/x.y" (compatibility) app token
     mUserAgent += ' ';
     mUserAgent += mCompatFirefox;
   }
-  if (!isFirefox) {
+  if (!isFirefox && false) {
     // App portion
     mUserAgent += ' ';
     mUserAgent += mAppName;
