diff --git a/browser/themes/shared/jar.inc.mn b/browser/themes/shared/jar.inc.mn
index 37b5f04cddde9a3769d31c6e3a93b43da76f44e5..e7c79a280c9899675b8da8467960386106d42285 100644
--- a/browser/themes/shared/jar.inc.mn
+++ b/browser/themes/shared/jar.inc.mn
@@ -285,3 +285,6 @@
   skin/classic/browser/weather/night-partly-cloudy-with-showers.svg (../shared/weather/night-partly-cloudy-with-showers.svg)
   skin/classic/browser/weather/night-partly-cloudy-with-thunderstorms.svg (../shared/weather/night-partly-cloudy-with-thunderstorms.svg)
   skin/classic/browser/weather/night-mostly-cloudy-with-flurries.svg (../shared/weather/night-mostly-cloudy-with-flurries.svg)
+
+  skin/classic/browser/zen-browser-shared.css                      (../shared/zen-browser-shared.css)
+  skin/classic/browser/zen-sidebar.css                             (../shared/zen-sidebar.css)
