diff --git a/browser/components/extensions/parent/ext-browser.js b/browser/components/extensions/parent/ext-browser.js
index 91adc6bc3649af47745818e6f9d00cbbdffc35a5..e221b077dbfbe20c8696732f003bd5809d761a02 100644
--- a/browser/components/extensions/parent/ext-browser.js
+++ b/browser/components/extensions/parent/ext-browser.js
@@ -362,6 +362,7 @@ class TabTracker extends TabTrackerBase {
   }
 
   getId(nativeTab) {
+    if (nativeTab.hasAttribute("zen-empty-tab")) return -1;
     let id = this._tabs.get(nativeTab);
     if (id) {
       return id;
@@ -395,6 +396,7 @@ class TabTracker extends TabTrackerBase {
     if (nativeTab.documentGlobal.closed) {
       throw new Error("Cannot attach ID to a tab in a closed window.");
     }
+    if (nativeTab.hasAttribute("zen-empty-tab")) return;
 
     this._tabs.set(nativeTab, id);
     this._tabIds.set(id, nativeTab);
@@ -1262,6 +1264,10 @@ class TabManager extends TabManagerBase {
   }
 
   canAccessTab(nativeTab) {
+    if (nativeTab.hasAttribute("zen-empty-tab")) {
+      return false
+    }
+
     // Check private browsing access at browser window level.
     if (!this.extension.canAccessWindow(nativeTab.documentGlobal)) {
       return false;
