diff --git a/browser/components/extensions/parent/ext-tabs.js b/browser/components/extensions/parent/ext-tabs.js
index 7e32fc3f3c325b7317533acf5142bb912556fdbe..04bbe6bece881a05d2eadf81411bd4df2b745017 100644
--- a/browser/components/extensions/parent/ext-tabs.js
+++ b/browser/components/extensions/parent/ext-tabs.js
@@ -501,6 +501,7 @@ this.tabs = class extends ExtensionAPIPersistent {
         }
 
         let tab = tabManager.getWrapper(updatedTab);
+        if (!tab) return;
 
         let changeInfo = {};
         for (let prop of needed) {
@@ -855,6 +856,7 @@ this.tabs = class extends ExtensionAPIPersistent {
               });
             }
 
+            window.gZenCompactModeManager._nextTimeWillBeActive = active;
             let nativeTab = window.gBrowser.addTab(url, options);
 
             if (active) {
