diff --git a/browser/components/extensions/parent/ext-browser.js b/browser/components/extensions/parent/ext-browser.js
index 89b0bb3b92c15b89499ffc6cf35dcee7ebb89e48..327afaea3821cdca8d7f58bfaa65c7ce3dbfa7a3 100644
--- a/browser/components/extensions/parent/ext-browser.js
+++ b/browser/components/extensions/parent/ext-browser.js
@@ -354,6 +354,7 @@ class TabTracker extends TabTrackerBase {
   }
 
   getId(nativeTab) {
+    if (nativeTab.hasAttribute("zen-empty-tab")) return -1;
     let id = this._tabs.get(nativeTab);
     if (id) {
       return id;
@@ -388,6 +389,7 @@ class TabTracker extends TabTrackerBase {
     if (nativeTab.ownerGlobal.closed) {
       throw new Error("Cannot attach ID to a tab in a closed window.");
     }
+    if (nativeTab.hasAttribute("zen-empty-tab")) return;
 
     this._tabs.set(nativeTab, id);
     if (nativeTab.linkedBrowser) {
@@ -1271,6 +1273,10 @@ class TabManager extends TabManagerBase {
   }
 
   canAccessTab(nativeTab) {
+    if (nativeTab.hasAttribute("zen-empty-tab")) {
+      return false
+    }
+
     // Check private browsing access at browser window level.
     if (!this.extension.canAccessWindow(nativeTab.ownerGlobal)) {
       return false;
