diff --git a/browser/installer/package-manifest.in b/browser/installer/package-manifest.in
index 46725ca049..f41fa58b7f 100644
--- a/browser/installer/package-manifest.in
+++ b/browser/installer/package-manifest.in
@@ -163,6 +163,7 @@
 #endif
 @BINPATH@/@MOZ_APP_NAME@
 #endif
+@BINPATH@/zen-firefox-password-import@BIN_SUFFIX@
 @RESPATH@/application.ini
 #ifdef MOZ_UPDATER
 # update-settings.ini has been removed on macOS.
@@ -395,9 +396,9 @@ bin/libfreebl_64int_3.so
 ;
 #ifdef MOZ_CRASHREPORTER
 #ifdef XP_MACOSX
-@BINPATH@/crashreporter.app/
+;@BINPATH@/crashreporter.app/
 #else
-@BINPATH@/crashreporter@BIN_SUFFIX@
+;@BINPATH@/crashreporter@BIN_SUFFIX@
 #if defined(XP_WIN)
 @BINPATH@/@DLL_PREFIX@mozwer@DLL_SUFFIX@
 #endif
