diff --git a/browser/installer/package-manifest.in b/browser/installer/package-manifest.in
index 70f268914f1078ef45e86d295f4bb2ce179a05e0..9eb138023bc9de6211a3e814b0525da612156a78 100644
--- a/browser/installer/package-manifest.in
+++ b/browser/installer/package-manifest.in
@@ -369,9 +369,9 @@ bin/libfreebl_64int_3.so
 ;
 #ifdef MOZ_CRASHREPORTER
 #ifdef XP_MACOSX
-@BINPATH@/crashreporter.app/
+;@BINPATH@/crashreporter.app/
 #else
-@BINPATH@/crashreporter@BIN_SUFFIX@
+;@BINPATH@/crashreporter@BIN_SUFFIX@
 #if defined(XP_WIN)
 @BINPATH@/@DLL_PREFIX@mozwer@DLL_SUFFIX@
 #endif
