diff --git a/browser/installer/package-manifest.in b/browser/installer/package-manifest.in
index dbfb287747ed9059cfb46838800421ac9ffe78c1..47279559a7dbea8b08bafb75e968f7b219e1d818 100644
--- a/browser/installer/package-manifest.in
+++ b/browser/installer/package-manifest.in
@@ -374,17 +374,17 @@ bin/libfreebl_64int_3.so
 ; [MaintenanceService]
 ;
 #ifdef MOZ_MAINTENANCE_SERVICE
-@BINPATH@/maintenanceservice.exe
-@BINPATH@/maintenanceservice_installer.exe
+;@BINPATH@/maintenanceservice.exe
+;@BINPATH@/maintenanceservice_installer.exe
 #endif
 
 ; [Crash Reporter]
 ;
 #ifdef MOZ_CRASHREPORTER
 #ifdef XP_MACOSX
-@BINPATH@/crashreporter.app/
+;@BINPATH@/crashreporter.app/
 #else
-@BINPATH@/crashreporter@BIN_SUFFIX@
+;@BINPATH@/crashreporter@BIN_SUFFIX@
 #if defined(XP_WIN)
 @BINPATH@/@DLL_PREFIX@mozwer@DLL_SUFFIX@
 #endif
