diff --git a/browser/components/newtab/lib/AboutPreferences.sys.mjs b/browser/components/newtab/lib/AboutPreferences.sys.mjs
index 33f7ecdaeb7074c7633dd229a4446f2a8ed4b247..f422a9bfb9694b190e84e81a430a509b4ab8c986 100644
--- a/browser/components/newtab/lib/AboutPreferences.sys.mjs
+++ b/browser/components/newtab/lib/AboutPreferences.sys.mjs
@@ -34,7 +34,7 @@ const PREFS_BEFORE_SECTIONS = () => [
       titleString: "home-prefs-shortcuts-header",
       descString: "home-prefs-shortcuts-description",
       get nestedPrefs() {
-        return Services.prefs.getBoolPref("browser.topsites.useRemoteSetting")
+        return Services.prefs.getBoolPref("browser.topsites.useRemoteSetting") && false
           ? [
               {
                 name: "showSponsoredTopSites",
