diff --git a/browser/components/customizableui/content/customizeMode.inc.xhtml b/browser/components/customizableui/content/customizeMode.inc.xhtml
index 2788cc6a8fb42b7b968eb45aafc02f8d6fcf771d..5142e65576aebaac0e31677b707a5e05f88dae76 100644
--- a/browser/components/customizableui/content/customizeMode.inc.xhtml
+++ b/browser/components/customizableui/content/customizeMode.inc.xhtml
@@ -94,7 +94,7 @@
         class="footer-button"
         oncommand="gCustomizeMode.togglePong(this.checked);"
         hidden="true"/>
-
+#include zenCustomizationMode.inc.xhtml
 <spacer id="customization-footer-spacer"/>
 #ifdef XP_MACOSX
     <button id="customization-touchbar-button"
