diff --git a/browser/components/customizableui/content/panelUI.js b/browser/components/customizableui/content/panelUI.js
index d6393080aa8aff6c0f7b2043fd408b6f6249d0c4..bf37b46c4743b7619393c730720204380635666a 100644
--- a/browser/components/customizableui/content/panelUI.js
+++ b/browser/components/customizableui/content/panelUI.js
@@ -608,10 +608,12 @@ const PanelUI = {
     if (hasKids && !this.navbar.hasAttribute("nonemptyoverflow")) {
       this.navbar.setAttribute("nonemptyoverflow", "true");
       this.overflowPanel.setAttribute("hasfixeditems", "true");
+      window.document.getElementById("zen-sidebar-top-buttons").setAttribute("nonemptyoverflow", "true");
     } else if (!hasKids && this.navbar.hasAttribute("nonemptyoverflow")) {
       PanelMultiView.hidePopup(this.overflowPanel);
       this.overflowPanel.removeAttribute("hasfixeditems");
       this.navbar.removeAttribute("nonemptyoverflow");
+      window.document.getElementById("zen-sidebar-top-buttons").removeAttribute("nonemptyoverflow");
     }
   },
 
@@ -959,7 +961,7 @@ const PanelUI = {
         el.removeAttribute("data-lazy-l10n-id");
       });
 
-    this.notificationPanel.openPopup(anchor, "bottomright topright");
+    this.notificationPanel.openPopup(anchor, gZenUIManager.panelUIPosition(this.notificationPanel, anchor));
   },
 
   _clearNotificationPanel() {
