diff --git a/browser/components/customizableui/content/panelUI.inc.xhtml b/browser/components/customizableui/content/panelUI.inc.xhtml
index c78f08bdd3fc4d01e10552ef65c2d5813dd053e8..154fb1de2ab9384937d02c14cb7c9c607b330da9 100644
--- a/browser/components/customizableui/content/panelUI.inc.xhtml
+++ b/browser/components/customizableui/content/panelUI.inc.xhtml
@@ -132,7 +132,7 @@
   <panel id="appMenu-notification-popup"
          class="popup-notification-panel panel-no-padding"
          type="arrow"
-         position="after_start"
+         position="after_end"
          flip="slide"
          orient="vertical"
          noautofocus="true"
