diff --git a/browser/components/customizableui/content/panelUI.inc.xhtml b/browser/components/customizableui/content/panelUI.inc.xhtml
index 5c248c0d49f3d9378260230c24ab60a4157dc438..1e87a1d040ed82c7edfb31f9e5853a4fb299b503 100644
--- a/browser/components/customizableui/content/panelUI.inc.xhtml
+++ b/browser/components/customizableui/content/panelUI.inc.xhtml
@@ -139,7 +139,7 @@
   <panel id="appMenu-notification-popup"
          class="popup-notification-panel panel-no-padding"
          type="arrow"
-         position="after_start"
+         position="after_end"
          flip="slide"
          orient="vertical"
          noautofocus="true"
