diff --git a/browser/components/customizableui/content/panelUI.inc.xhtml b/browser/components/customizableui/content/panelUI.inc.xhtml
index 6cb22be6526b17d952a90fd414c49a31e9afde7a..d224686db275f79b712479b32762a1b836e3cb29 100644
--- a/browser/components/customizableui/content/panelUI.inc.xhtml
+++ b/browser/components/customizableui/content/panelUI.inc.xhtml
@@ -139,7 +139,7 @@
   <panel id="appMenu-notification-popup"
          class="popup-notification-panel panel-no-padding"
          type="arrow"
-         position="after_start"
+         position="after_end"
          flip="slide"
          orient="vertical"
          noautofocus="true"
@@ -278,6 +278,7 @@
        hidden="true"
        flip="slide"
        position="bottomright topright"
+       hidepopovertail="true"
        noautofocus="true">
   <panelmultiview id="appMenu-multiView" mainViewId="appMenu-mainView"
                   viewCacheId="appMenu-viewCache">
