diff --git a/browser/components/places/jar.mn b/browser/components/places/jar.mn
index 7334914664484267ccf80eeeb2917c2001a1acaf..e0636a13798d6fbf41cbf69e51e460822b00ab1a 100644
--- a/browser/components/places/jar.mn
+++ b/browser/components/places/jar.mn
@@ -25,3 +25,5 @@ browser.jar:
     content/browser/places/interactionsViewer.html       (metadataViewer/interactionsViewer.html)
     content/browser/places/interactionsViewer.js         (metadataViewer/interactionsViewer.js)
 #endif
+    content/browser/places/zenNewWebPanel.xhtml      (content/zenNewWebPanel.xhtml)
+    content/browser/places/zenNewWebPanel.js         (content/zenNewWebPanel.js)
