diff --git a/toolkit/actors/PictureInPictureChild.sys.mjs b/toolkit/actors/PictureInPictureChild.sys.mjs
index e4dea54a29e2a1575d76091061781a504da38465..d5248eebdd018feca7bb9d5ee3284d6f253a4b35 100644
--- a/toolkit/actors/PictureInPictureChild.sys.mjs
+++ b/toolkit/actors/PictureInPictureChild.sys.mjs
@@ -291,6 +291,7 @@ export class PictureInPictureLauncherChild extends JSWindowActorChild {
       if (
         video &&
         PictureInPictureChild.videoIsPlaying(video) &&
+        !video.muted &&
         PictureInPictureChild.videoIsPiPEligible(video)
       ) {
         this.togglePictureInPicture({ video, reason: "AutoPip" }, false);
