diff --git a/toolkit/actors/PictureInPictureChild.sys.mjs b/toolkit/actors/PictureInPictureChild.sys.mjs
index 25ae1e984e3dbd610b60930492e1fa63ef889b9a..2680f2bdabf9884bc2c68b2db7b2fbedbb8af317 100644
--- a/toolkit/actors/PictureInPictureChild.sys.mjs
+++ b/toolkit/actors/PictureInPictureChild.sys.mjs
@@ -295,6 +295,7 @@ export class PictureInPictureLauncherChild extends JSWindowActorChild {
       if (
         video &&
         PictureInPictureChild.videoIsPlaying(video) &&
+        !video.muted &&
         PictureInPictureChild.videoIsPiPEligible(video)
       ) {
         this.togglePictureInPicture({ video, reason: "AutoPip" }, false);
