diff --git a/toolkit/actors/SelectParent.sys.mjs b/toolkit/actors/SelectParent.sys.mjs
index fa8ce002f0cd64395d04bcd1f93f6f8a89417a62..c7d2865e8e78015e253f0ef1a82f1ec57a4fe255 100644
--- a/toolkit/actors/SelectParent.sys.mjs
+++ b/toolkit/actors/SelectParent.sys.mjs
@@ -103,7 +103,7 @@ export var SelectParentHelper = {
     }
 
     menupopup.setAttribute("style", "");
-    menupopup.style.colorScheme = isDarkBackground ? "dark" : "light";
+    //menupopup.style.colorScheme = isDarkBackground ? "dark" : "light";
     menupopup.style.direction = selectStyle.direction;
 
     stylesheet = doc.createElementNS("http://www.w3.org/1999/xhtml", "style");
