diff --git a/toolkit/content/widgets/browser-custom-element.mjs b/toolkit/content/widgets/browser-custom-element.mjs
index 8eeea2a403ba2007a1a1e6074bb982b24f40fef5..29bae869dfd8c9df6039b3c9b8b1603092e517ea 100644
--- a/toolkit/content/widgets/browser-custom-element.mjs
+++ b/toolkit/content/widgets/browser-custom-element.mjs
@@ -482,11 +482,11 @@ class MozBrowser extends MozElements.MozElementMixin(XULFrameElement) {
     if (!this.browsingContext) {
       return;
     }
-    this.browsingContext.isActive = val;
+    this.browsingContext.isActive = val || this.zenModeActive;
     if (this.isRemoteBrowser) {
       let remoteTab = this.frameLoader?.remoteTab;
       if (remoteTab) {
-        remoteTab.renderLayers = val;
+        remoteTab.renderLayers = val || this.zenModeActive;
       }
     }
   }
