diff --git a/toolkit/content/widgets/browser-custom-element.mjs b/toolkit/content/widgets/browser-custom-element.mjs
index 22fcec0368d0356a24dd125373cb767836a6760d..7a78f5c1ddb06f3bc1b1e76f7489d57ee17a28e9 100644
--- a/toolkit/content/widgets/browser-custom-element.mjs
+++ b/toolkit/content/widgets/browser-custom-element.mjs
@@ -504,11 +504,11 @@ export class MozBrowser extends MozElements.MozElementMixin(XULFrameElement) {
     if (!this.browsingContext) {
       return;
     }
-    this.browsingContext.isActive = val;
+    this.browsingContext.isActive = val || this.zenModeActive;
     if (this.isRemoteBrowser) {
       let remoteTab = this.frameLoader?.remoteTab;
       if (remoteTab) {
-        remoteTab.renderLayers = val;
+        remoteTab.renderLayers = val || this.zenModeActive;
       }
     }
   }
