diff --git a/toolkit/content/widgets/browser-custom-element.mjs b/toolkit/content/widgets/browser-custom-element.mjs
index 44f3d5f5875fbd83f54a0a06d4881ded95e396e5..43f107b2a8231f4a0994afd724ff66a5e1e1a0e8 100644
--- a/toolkit/content/widgets/browser-custom-element.mjs
+++ b/toolkit/content/widgets/browser-custom-element.mjs
@@ -495,11 +495,11 @@ export class MozBrowser extends MozElements.MozElementMixin(XULFrameElement) {
     if (!this.browsingContext) {
       return;
     }
-    this.browsingContext.isActive = val;
+    this.browsingContext.isActive = val || this.zenModeActive;
     if (this.isRemoteBrowser) {
       let remoteTab = this.frameLoader?.remoteTab;
       if (remoteTab) {
-        remoteTab.renderLayers = val;
+        remoteTab.renderLayers = val || this.zenModeActive;
       }
     }
   }
