diff --git a/toolkit/content/widgets/findbar.js b/toolkit/content/widgets/findbar.js
index ff9bb99e4ca481feb97d060e34eae37846ba55cf..6517111456d85d975d5857a044c642bdc1294be4 100644
--- a/toolkit/content/widgets/findbar.js
+++ b/toolkit/content/widgets/findbar.js
@@ -1189,6 +1189,7 @@
      *   e.g. <command name="cmd_find" oncommand="gFindBar.onFindCommand();"/>
      */
     onFindCommand() {
+      if (gBrowser?.selectedTab?.hasAttribute("zen-empty-tab")) return;
       return this.startFind(this.FIND_NORMAL);
     }
 
