diff --git a/toolkit/content/widgets/findbar.js b/toolkit/content/widgets/findbar.js
index 589d96886b1c9a1bd0e819957978a8968402e98b..4df4be8dbbe67063f0e1920baa2db3f5b909b11b 100644
--- a/toolkit/content/widgets/findbar.js
+++ b/toolkit/content/widgets/findbar.js
@@ -1188,6 +1188,7 @@
      *   e.g. <command name="cmd_find" oncommand="gFindBar.onFindCommand();"/>
      */
     onFindCommand() {
+      if (gBrowser?.selectedTab?.hasAttribute("zen-empty-tab")) return;
       return this.startFind(this.FIND_NORMAL);
     }
 
