diff --git a/toolkit/content/widgets/findbar.js b/toolkit/content/widgets/findbar.js
index 89550d1700d4cb7e8bdef5e99ae4eb42b3ce3b16..b6d23961739b170d34f7a1bdc5f5c2e0ef005882 100644
--- a/toolkit/content/widgets/findbar.js
+++ b/toolkit/content/widgets/findbar.js
@@ -1189,6 +1189,7 @@
      *   e.g. <command name="cmd_find" oncommand="gFindBar.onFindCommand();"/>
      */
     onFindCommand() {
+      if (gBrowser?.selectedTab?.hasAttribute("zen-empty-tab")) return;
       return this.startFind(this.FIND_NORMAL);
     }
 
