diff --git a/toolkit/content/widgets/findbar.js b/toolkit/content/widgets/findbar.js
index 8ca512e0b212bb6e25ed37e7edb4f8e2587d5bb0..5abfff92b3a8df5d138082af96419668a2c8a9bb 100644
--- a/toolkit/content/widgets/findbar.js
+++ b/toolkit/content/widgets/findbar.js
@@ -1188,6 +1188,7 @@
      *   e.g. <command name="cmd_find" oncommand="gFindBar.onFindCommand();"/>
      */
     onFindCommand() {
+      if (gBrowser?.selectedTab?.hasAttribute("zen-empty-tab")) return;
       return this.startFind(this.FIND_NORMAL);
     }
 
