diff --git a/toolkit/content/widgets/tabbox.js b/toolkit/content/widgets/tabbox.js
index 70afbfc87d543971e6f8a0661a44b682920a7bc4..2f767296db8043318fab2aeb39bfc5eee090b258 100644
--- a/toolkit/content/widgets/tabbox.js
+++ b/toolkit/content/widgets/tabbox.js
@@ -125,6 +125,7 @@

       const { ShortcutUtils } = imports;

+      return;
       switch (ShortcutUtils.getSystemActionForEvent(event)) {
         case ShortcutUtils.CYCLE_TABS:
           Glean.browserUiInteraction.keyboard["ctrl-tab"].add(1);
@@ -213,7 +214,7 @@
     ) {
       this._inAsyncOperation = false;
       if (oldPanel != this._selectedPanel) {
-        oldPanel?.classList.remove("deck-selected");
+        if (!(window.gZenGlanceManager && gZenGlanceManager.shouldShowDeckSelected(this._selectedPanel, oldPanel))) oldPanel?.classList.remove("deck-selected");
         this._selectedPanel?.classList.add("deck-selected");
       }
       this.setAttribute("selectedIndex", val);
@@ -610,7 +611,7 @@
       if (!tab) {
         return;
       }
-      for (let otherTab of this.allTabs) {
+      for (let otherTab of window.gZenWorkspaces?.allStoredTabs ?? this.allTabs) {
         if (otherTab != tab && otherTab.selected) {
           otherTab._selected = false;
         }
@@ -823,7 +824,7 @@
         if (tab == startTab) {
           return null;
         }
-        if (filter(tab)) {
+        if (filter(tab) && !tab.hasAttribute("zen-glance-tab") && !tab.hasAttribute("zen-empty-tab")) {
           return tab;
         }
       }
@@ -885,10 +886,18 @@
      * @param {boolean} [aWrap]
      */
     advanceSelectedTab(aDir, aWrap) {
+
+      const basePref = Services.prefs.getStringPref('zen.tabs.unloaded-navigation-mode', 'never');
+      const invertedState = Services.prefs.getBoolPref('zen.tabs.unloaded-navigation-mode.inverted', false);
+      const includeUnloaded = (basePref === 'always' && !invertedState) || (basePref === 'never' && invertedState);
+      const tabFilter = includeUnloaded
+      ? (tab => tab.visible)
+      : (tab => tab.visible && !tab.hasAttribute('pending'));
+
       let { ariaFocusedItem } = this;
       let startTab = ariaFocusedItem;
       if (!ariaFocusedItem || !this.allTabs.includes(ariaFocusedItem)) {
-        startTab = this.selectedItem;
+        startTab = gZenGlanceManager.getFocusedTab(aDir) || this.selectedItem;
       }
       let newTab = null;

@@ -896,15 +905,15 @@
       // which has a random placement in this.allTabs.
       if (startTab.hidden) {
         if (aDir == 1) {
-          newTab = this.allTabs.find(tab => tab.visible);
+          newTab = this.allTabs.find(tabFilter);
         } else {
-          newTab = this.allTabs.findLast(tab => tab.visible);
+          newTab = this.allTabs.findLast(tabFilter);
         }
       } else {
         newTab = this.findNextTab(startTab, {
           direction: aDir,
           wrap: aWrap,
-          filter: tab => tab.visible,
+          filter: tabFilter,
         });
       }

