diff --git a/toolkit/content/widgets/tabbox.js b/toolkit/content/widgets/tabbox.js
index 344984211e70c67bcc65fbbee54495f381da2b9f..1346ab8b820ae60b13511f6e35b713060b2453d7 100644
--- a/toolkit/content/widgets/tabbox.js
+++ b/toolkit/content/widgets/tabbox.js
@@ -11,6 +11,23 @@
     "resource://gre/modules/AppConstants.sys.mjs"
   );
 
+  const { XPCOMUtils } = ChromeUtils.importESModule(
+    "resource://gre/modules/XPCOMUtils.sys.mjs"
+  );
+  const lazyZenPrefs = {};
+  XPCOMUtils.defineLazyPreferenceGetter(
+    lazyZenPrefs,
+    "cycleByAttribute",
+    "zen.tabs.ctrl-tab.ignore-essential-tabs",
+    false
+  );
+  XPCOMUtils.defineLazyPreferenceGetter(
+    lazyZenPrefs,
+    "ignorePendingTabs",
+    "zen.tabs.ctrl-tab.ignore-pending-tabs",
+    false
+  );
+
   let imports = {};
   ChromeUtils.defineESModuleGetters(imports, {
     ShortcutUtils: "resource://gre/modules/ShortcutUtils.sys.mjs",
@@ -213,7 +230,7 @@
     ) {
       this._inAsyncOperation = false;
       if (oldPanel != this._selectedPanel) {
-        oldPanel?.classList.remove("deck-selected");
+        if (!(window.gZenGlanceManager && gZenGlanceManager.shouldShowDeckSelected(this._selectedPanel, oldPanel))) oldPanel?.classList.remove("deck-selected");
         this._selectedPanel?.classList.add("deck-selected");
       }
       this.setAttribute("selectedIndex", val);
@@ -736,7 +753,7 @@
       if (!tab) {
         return;
       }
-      for (let otherTab of this.allTabs) {
+      for (let otherTab of window.gZenWorkspaces?.allStoredTabs ?? this.allTabs) {
         if (otherTab != tab && otherTab.selected) {
           otherTab._selected = false;
         }
@@ -772,6 +789,7 @@
      * @param {MozTab|null} [val]
      */
     set selectedItem(val) {
+      val = window.gZenGlanceManager?.getTabOrGlanceChild(val) || val;
       if (val && !val.selected) {
         // The selectedIndex setter ignores invalid values
         // such as -1 if |val| isn't one of our child nodes.
@@ -949,7 +967,7 @@
         if (tab == startTab) {
           return null;
         }
-        if (filter(tab)) {
+        if (filter(tab) && !tab.hasAttribute("zen-glance-tab") && !tab.hasAttribute("zen-empty-tab")) {
           return tab;
         }
       }
@@ -1011,13 +1029,30 @@
      * @param {boolean} [aWrap]
      */
     advanceSelectedTab(aDir, aWrap) {
+      if (window?.gZenGlanceManager?._animating) return;
       let { ariaFocusedItem } = this;
       let startTab = ariaFocusedItem;
       if (!ariaFocusedItem || !this.allTabs.includes(ariaFocusedItem)) {
-        startTab = this.selectedItem;
+        startTab = gZenGlanceManager.getFocusedTab(aDir) || this.selectedItem;
       }
       let newTab = null;
 
+      const tabFilter = tab => {
+        if (!tab.visible) {
+          return false
+        }
+        if (lazyZenPrefs.ignorePendingTabs && tab.hasAttribute("pending")) {
+          return false
+        }
+        if (!lazyZenPrefs.cycleByAttribute) {
+          return true
+        }
+        if (startTab.hasAttribute("zen-essential")) {
+          return tab.hasAttribute("zen-essential")
+        }
+        return !tab.hasAttribute("zen-essential")
+      }
+
       // Handle keyboard navigation for a hidden tab that can be selected, like the Firefox View tab,
       // which has a random placement in this.allTabs.
       if (startTab.hidden) {
@@ -1030,7 +1065,7 @@
         newTab = this.findNextTab(startTab, {
           direction: aDir,
           wrap: aWrap,
-          filter: tab => tab.visible,
+          filter: tabFilter,
         });
       }
 
