diff --git a/gfx/webrender_bindings/src/bindings.rs b/gfx/webrender_bindings/src/bindings.rs
index 578c2f80a9dbcb3112c127a9b6b128810c80454c..08823b7fea222e11dcd7b7cf4a6ddd4976722d70 100644
--- a/gfx/webrender_bindings/src/bindings.rs
+++ b/gfx/webrender_bindings/src/bindings.rs
@@ -2144,12 +2144,6 @@ pub extern "C" fn wr_window_new(
         }
     };
 
-    let enable_dithering = if !software && static_prefs::pref!("gfx.webrender.dithering") {
-        true
-    } else {
-        false
-    };
-
     let opts = WebRenderOptions {
         enable_aa: true,
         enable_subpixel_aa,
@@ -2204,7 +2198,6 @@ pub extern "C" fn wr_window_new(
         reject_software_rasterizer,
         low_quality_pinch_zoom,
         max_shared_surface_size,
-        enable_dithering,
         ..Default::default()
     };
 
@@ -4615,11 +4608,8 @@ pub extern "C" fn wr_shaders_new(
 
     device.begin_frame();
 
-    let mut options = WebRenderOptions::default();
-    options.enable_dithering = static_prefs::pref!("gfx.webrender.dithering");
-
     let gl_type = device.gl().get_type();
-    let mut shaders = match Shaders::new(&mut device, gl_type, &options) {
+    let mut shaders = match Shaders::new(&mut device, gl_type, &WebRenderOptions::default()) {
         Ok(shaders) => shaders,
         Err(e) => {
             warn!(" Failed to create a Shaders: {:?}", e);
