diff --git a/gfx/wr/webrender/res/border_shared.glsl b/gfx/wr/webrender/res/border_shared.glsl
--- a/gfx/wr/webrender/res/border_shared.glsl
+++ b/gfx/wr/webrender/res/border_shared.glsl
@@ -27,25 +27,33 @@
     vec4 rect;
     vec4 color0;
     vec4 color1;
     vec2 widths;
     vec2 radii;
+#ifdef WR_FEATURE_SUPERELLIPSE
     float shape;
     vec2 shape_offset;
     vec2 inset;
+#endif
 };
 
 BorderInstanceGpuData fetch_gpu_data(int index) {
     BorderInstanceGpuData data;
 
+#ifdef WR_FEATURE_SUPERELLIPSE
     vec4 texels[6] = fetch_from_gpu_buffer_6f(index);
+#else
+    vec4 texels[4] = fetch_from_gpu_buffer_4f(index);
+#endif
     data.rect = texels[0];
     data.color0 = texels[1];
     data.color1 = texels[2];
     data.widths = texels[3].xy;
     data.radii = texels[3].zw;
+#ifdef WR_FEATURE_SUPERELLIPSE
     data.shape = texels[4].x;
     data.shape_offset = texels[4].yz;
     data.inset = texels[5].xy;
+#endif
 
     return data;
 }
diff --git a/gfx/wr/webrender/res/cs_border_segment.glsl b/gfx/wr/webrender/res/cs_border_segment.glsl
--- a/gfx/wr/webrender/res/cs_border_segment.glsl
+++ b/gfx/wr/webrender/res/cs_border_segment.glsl
@@ -30,11 +30,13 @@
 flat varying highp vec4 vClipCenter_Sign;
 
 // An outer and inner elliptical radii for border
 // corner clipping.
 flat varying highp vec4 vClipRadii;
+#ifdef WR_FEATURE_SUPERELLIPSE
 flat varying highp vec4 vClipOffsets;
+#endif
 
 // Reference point for determine edge clip lines.
 flat varying highp vec4 vEdgeReference;
 
 // Stores widths/2 and widths/3 to save doing this in FS.
@@ -192,26 +194,34 @@
     vColor00 = color0[0];
     vColor01 = color0[1];
     vec4[2] color1 = get_colors_for_side(data.color1, style1);
     vColor10 = color1[0];
     vColor11 = color1[1];
+#ifdef WR_FEATURE_SUPERELLIPSE
     vClipCenter_Sign = vec4(outer + clip_sign * (data.radii + data.shape_offset), clip_sign);
+#else
+    vClipCenter_Sign = vec4(outer + clip_sign * data.radii, clip_sign);
+#endif
     vClipRadii = vec4(data.radii, max(data.radii - data.widths, 0.0));
+#ifdef WR_FEATURE_SUPERELLIPSE
     vClipOffsets = vec4(0.0);
+#endif
     vColorLine = vec4(outer, data.widths.y * -clip_sign.y, data.widths.x * clip_sign.x);
     vEdgeReference = vec4(edge_reference, edge_reference + data.widths);
     vClipParams1 = aClipParams1;
     vClipParams2 = aClipParams2;
 
+#ifdef WR_FEATURE_SUPERELLIPSE
     if (data.shape != 1.0)
     {
         vec2 reference_radii = (data.radii == vec2(0.0)) ? vec2(0.0) : data.radii + data.inset;
         vec4 contour1 = compute_contoured_superellipse(reference_radii, data.shape, data.inset);
         vec4 contour2 = compute_contoured_superellipse(reference_radii, data.shape, data.inset + data.widths);
         vClipOffsets = vec4(contour1.xy, contour2.xy);
         vClipRadii = vec4(contour1.zw, contour2.zw);
     }
+#endif
 
     // For the case of dot and dash clips, optimize the number of pixels that
     // are hit to just include the dot itself.
     if (clip_mode == CLIP_DOT) {
         float radius = aClipParams1.z;
@@ -258,19 +268,22 @@
             // also 0.67 of the radii. Use these to form a
             // SDF subtraction which will clip out the inside
             // third of the rounded edge.
             float d_radii_a;
             float d_radii_b;
+#ifdef WR_FEATURE_SUPERELLIPSE
             if (data.shape == 1.0) {
+#endif
                 d_radii_a = distance_to_ellipse(
                     clip_relative_pos,
                     clip_radii.xy - vPartialWidths.xy
                 );
                 d_radii_b = distance_to_ellipse(
                     clip_relative_pos,
                     clip_radii.xy - 2.0 * vPartialWidths.xy
                 );
+#ifdef WR_FEATURE_SUPERELLIPSE
             } else {
                 d_radii_a = distance_to_superellipse(
                     clip_relative_pos - mix(vClipOffsets.xy, vClipOffsets.zw, 1.0 / 3.0),
                     mix(clip_radii.xy, clip_radii.zw, 1.0 / 3.0),
                     data.shape
@@ -286,24 +299,28 @@
                 d_radii_a = max(d_radii_a, -min(included_region_a.x, included_region_a.y));
 
                 vec2 included_region_b = data.radii - 2.0 * vPartialWidths.xy - clip_relative_pos;
                 d_radii_b = max(d_radii_b, -min(included_region_b.x, included_region_b.y));
             }
+#endif
 
             float d = min(-d_radii_a, d_radii_b);
 
             color0 *= distance_aa(aa_range, d);
             break;
         }
         case BORDER_STYLE_GROOVE:
         case BORDER_STYLE_RIDGE: {
             float d;
+#ifdef WR_FEATURE_SUPERELLIPSE
             if (data.shape == 1.0) {
+#endif
                 d = distance_to_ellipse(
                     clip_relative_pos,
                     clip_radii.xy - vPartialWidths.zw
                 );
+#ifdef WR_FEATURE_SUPERELLIPSE
             } else {
                 d = distance_to_superellipse(
                     clip_relative_pos - mix(vClipOffsets.xy, vClipOffsets.zw, 0.5),
                     mix(clip_radii.xy, clip_radii.zw, 0.5),
                     data.shape
@@ -311,10 +328,11 @@
 
                 // Clamp to the middle of the adjacent borders
                 vec2 included_region = data.radii - vPartialWidths.zw - clip_relative_pos;
                 d = max(d, -min(included_region.x, included_region.y));
             }
+#endif
             float alpha = distance_aa(aa_range, d);
             float swizzled_factor;
             switch (segment) {
                 case SEGMENT_TOP_LEFT: swizzled_factor = 0.0; break;
                 case SEGMENT_TOP_RIGHT: swizzled_factor = mix_factor; break;
@@ -435,22 +453,26 @@
 
     if (in_clip_region) {
         float d_radii_a;
         float d_radii_b;
 
+#ifdef WR_FEATURE_SUPERELLIPSE
         if (data.shape == 1.0) {
+#endif
             d_radii_a = distance_to_ellipse(clip_relative_pos, vClipRadii.xy);
             d_radii_b = distance_to_ellipse(clip_relative_pos, vClipRadii.zw);
+#ifdef WR_FEATURE_SUPERELLIPSE
         } else {
             clip_relative_pos = abs(clip_relative_pos) - data.shape_offset;
             d_radii_a = distance_to_superellipse(clip_relative_pos - vClipOffsets.xy, vClipRadii.xy, data.shape);
             d_radii_b = distance_to_superellipse(clip_relative_pos - vClipOffsets.zw, vClipRadii.zw, data.shape);
 
             // exclude the straight border part from the subtracted region
             vec2 included_region = data.radii - data.widths - clip_relative_pos;
             d_radii_b = max(d_radii_b, -min(included_region.x, included_region.y));
         }
+#endif
 
         float d_radii = max(d_radii_a, -d_radii_b);
         d = max(d, d_radii);
         
         color0 = evaluate_color_for_style_in_corner(
diff --git a/gfx/wr/webrender/res/cs_border_solid.glsl b/gfx/wr/webrender/res/cs_border_solid.glsl
--- a/gfx/wr/webrender/res/cs_border_solid.glsl
+++ b/gfx/wr/webrender/res/cs_border_solid.glsl
@@ -31,11 +31,13 @@
 flat varying highp vec4 vClipCenter_Sign;
 
 // An outer and inner elliptical radii for border
 // corner clipping.
 flat varying highp vec4 vClipRadii;
+#ifdef WR_FEATURE_SUPERELLIPSE
 flat varying highp vec4 vClipOffsets;
+#endif
 
 // Position, scale, and radii of horizontally and vertically adjacent corner clips.
 flat varying highp vec4 vHorizontalClipCenter_Sign;
 flat varying highp vec2 vHorizontalClipRadii;
 flat varying highp vec4 vVerticalClipCenter_Sign;
@@ -74,11 +76,11 @@
 void main(void) {
     BorderInstanceGpuData data = fetch_gpu_data(aGpuDataAddress);
     vGpuDataAddress.x = aGpuDataAddress;
 
     int segment = aFlags & 0xff;
-    bool do_aa = ((aFlags >> 24) & 0xf0) != 0;
+    bool do_aa = ((aFlags >> 28) & 0x1) != 0;
 
     vec2 outer_scale = get_outer_corner_scale(segment);
     vec2 size = data.rect.zw - data.rect.xy;
     vec2 outer = outer_scale * size;
     vec2 clip_sign = 1.0 - 2.0 * outer_scale;
@@ -100,23 +102,31 @@
     vMixColors.x = mix_colors;
     vPos = size * aPosition.xy;
 
     vColor0 = data.color0;
     vColor1 = data.color1;
+#ifdef WR_FEATURE_SUPERELLIPSE
     vClipCenter_Sign = vec4(outer + clip_sign * (data.radii + data.shape_offset), clip_sign);
+#else
+    vClipCenter_Sign = vec4(outer + clip_sign * data.radii, clip_sign);
+#endif
     vClipRadii = vec4(data.radii, max(data.radii - data.widths, 0.0));
+#ifdef WR_FEATURE_SUPERELLIPSE
     vClipOffsets = vec4(0.0);
+#endif
     vColorLine = vec4(outer, data.widths.y * -clip_sign.y, data.widths.x * clip_sign.x);
 
+#ifdef WR_FEATURE_SUPERELLIPSE
     if (data.shape != 1.0)
     {
         vec2 reference_radii = (data.radii == vec2(0.0)) ? vec2(0.0) : data.radii + data.inset;
         vec4 contour1 = compute_contoured_superellipse(reference_radii, data.shape, data.inset);
         vec4 contour2 = compute_contoured_superellipse(reference_radii, data.shape, data.inset + data.widths);
         vClipOffsets = vec4(contour1.xy, contour2.xy);
         vClipRadii = vec4(contour1.zw, contour2.zw);
     }
+#endif
 
     vec2 horizontal_clip_sign = vec2(-clip_sign.x, clip_sign.y);
     vHorizontalClipCenter_Sign = vec4(aClipParams1.xy +
                                       horizontal_clip_sign * aClipParams1.zw,
                                       horizontal_clip_sign);
@@ -156,22 +166,26 @@
     float d = -1.0;
     if (in_clip_region) {
         float d_radii_a;
         float d_radii_b;
 
+#ifdef WR_FEATURE_SUPERELLIPSE
         if (data.shape == 1.0) {
+#endif
             d_radii_a = distance_to_ellipse(clip_relative_pos, vClipRadii.xy);
             d_radii_b = distance_to_ellipse(clip_relative_pos, vClipRadii.zw);
+#ifdef WR_FEATURE_SUPERELLIPSE
         } else {
             clip_relative_pos = abs(clip_relative_pos) - data.shape_offset;
             d_radii_a = distance_to_superellipse(clip_relative_pos - vClipOffsets.xy, vClipRadii.xy, data.shape);
             d_radii_b = distance_to_superellipse(clip_relative_pos - vClipOffsets.zw, vClipRadii.zw, data.shape);
 
             // exclude the straight border part from the subtracted region
             vec2 included_region = data.radii - data.widths - clip_relative_pos;
             d_radii_b = max(d_radii_b, -min(included_region.x, included_region.y));
         }
+#endif
 
         d = max(d_radii_a, -d_radii_b);
     }
 
     // And again for horizontally-adjacent corner
diff --git a/gfx/wr/webrender/src/border.rs b/gfx/wr/webrender/src/border.rs
--- a/gfx/wr/webrender/src/border.rs
+++ b/gfx/wr/webrender/src/border.rs
@@ -855,14 +855,17 @@
     h_adjacent_corner_radius: DeviceSize,
     v_adjacent_corner_outer: DevicePoint,
     v_adjacent_corner_radius: DeviceSize,
     gpu_buffer_builder: &mut GpuBufferBuilderF,
 ) {
+    let superellipse = shape != 1.0;
+
     let base_flags = (segment as i32) |
                      ((style0 as i32) << 8) |
                      ((style1 as i32) << 16) |
-                     ((do_aa as i32) << 28);
+                     ((do_aa as i32) << 28) |
+                     ((superellipse as i32) << 29);
 
     let instance_gpu_data = BorderInstanceGpuData {
         local_rect: task_rect,
         color0: color0.premultiplied(),
         color1: color1.premultiplied(),
@@ -875,11 +878,11 @@
 
     let base_instance = BorderInstance {
         task_origin: DevicePoint::zero(),
         flags: base_flags,
         clip_params: [0.0; 8],
-        gpu_data_address: instance_gpu_data.write(gpu_buffer_builder)
+        gpu_data_address: instance_gpu_data.write(superellipse, gpu_buffer_builder)
     };
 
     match segment {
         BorderSegment::TopLeft |
         BorderSegment::TopRight |
diff --git a/gfx/wr/webrender/src/gpu_types.rs b/gfx/wr/webrender/src/gpu_types.rs
--- a/gfx/wr/webrender/src/gpu_types.rs
+++ b/gfx/wr/webrender/src/gpu_types.rs
@@ -192,18 +192,21 @@
     pub shape_offset: DeviceSize,
     pub inset: DeviceSize,
 }
 
 impl BorderInstanceGpuData {
-    pub fn write(&self, gpu_buffer_builder: &mut GpuBufferBuilderF) -> GpuBufferAddress {
-        let mut writer = gpu_buffer_builder.write_blocks(6);
+    pub fn write(&self, superellipse: bool, gpu_buffer_builder: &mut GpuBufferBuilderF) -> GpuBufferAddress {
+        let block_count = if superellipse { 6 } else { 4 };
+        let mut writer = gpu_buffer_builder.write_blocks(block_count);
         writer.push_one(self.local_rect);
         writer.push_one(self.color0);
         writer.push_one(self.color1);
         writer.push_one([self.widths.width, self.widths.height, self.radius.width, self.radius.height]);
-        writer.push_one([self.shape, self.shape_offset.width, self.shape_offset.height, 0.0]);
-        writer.push_one([self.inset.width, self.inset.height, 0.0, 0.0]);
+        if superellipse {
+            writer.push_one([self.shape, self.shape_offset.width, self.shape_offset.height, 0.0]);
+            writer.push_one([self.inset.width, self.inset.height, 0.0, 0.0]);
+        }
 
         writer.finish()
     }
 }
 
diff --git a/gfx/wr/webrender/src/render_target.rs b/gfx/wr/webrender/src/render_target.rs
--- a/gfx/wr/webrender/src/render_target.rs
+++ b/gfx/wr/webrender/src/render_target.rs
@@ -33,10 +33,11 @@
 use crate::spatial_tree::SpatialNodeIndex;
 
 
 const STYLE_SOLID: i32 = ((BorderStyle::Solid as i32) << 8) | ((BorderStyle::Solid as i32) << 16);
 const STYLE_MASK: i32 = 0x00FF_FF00;
+const SUPERELLIPSE_MASK: i32 = 1 << 29;
 
 /// A tag used to identify the output format of a `RenderTarget`.
 #[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
 #[cfg_attr(feature = "capture", derive(Serialize))]
 #[cfg_attr(feature = "replay", derive(Deserialize))]
@@ -163,10 +164,12 @@
 
     pub clip_masks: ClipMaskInstanceList,
 
     pub border_segments_complex: FrameVec<BorderInstance>,
     pub border_segments_solid: FrameVec<BorderInstance>,
+    pub border_segments_complex_superellipse: FrameVec<BorderInstance>,
+    pub border_segments_solid_superellipse: FrameVec<BorderInstance>,
     pub line_decorations: FrameVec<LineDecorationJob>,
 
     // Clearing render targets has a fair amount of special cases.
     // The general rules are:
     // - Depth (for at least the used potion of the target) is always cleared if it
@@ -223,10 +226,12 @@
             prim_instances: std::array::from_fn(|_| FastHashMap::default()),
             prim_instances_with_scissor: FastHashMap::default(),
             clip_masks: ClipMaskInstanceList::new(memory),
             border_segments_complex: memory.new_vec(),
             border_segments_solid: memory.new_vec(),
+            border_segments_complex_superellipse: memory.new_vec(),
+            border_segments_solid_superellipse: memory.new_vec(),
             clears: memory.new_vec(),
             line_decorations: memory.new_vec(),
         }
     }
 
@@ -547,14 +552,23 @@
                 let instances = task_info.instances.clone();
                 for mut instance in instances {
                     // TODO(gw): It may be better to store the task origin in
                     //           the render task data instead of per instance.
                     instance.task_origin = task_origin;
+                    let superellipse = (instance.flags & SUPERELLIPSE_MASK) != 0;
                     if instance.flags & STYLE_MASK == STYLE_SOLID {
-                        self.border_segments_solid.push(instance);
+                        if superellipse {
+                            self.border_segments_solid_superellipse.push(instance);
+                        } else {
+                            self.border_segments_solid.push(instance);
+                        }
                     } else {
-                        self.border_segments_complex.push(instance);
+                        if superellipse {
+                            self.border_segments_complex_superellipse.push(instance);
+                        } else {
+                            self.border_segments_complex.push(instance);
+                        }
                     }
                 }
             }
             RenderTaskKind::Image(..) |
             RenderTaskKind::Cached(..) |
diff --git a/gfx/wr/webrender/src/renderer/mod.rs b/gfx/wr/webrender/src/renderer/mod.rs
--- a/gfx/wr/webrender/src/renderer/mod.rs
+++ b/gfx/wr/webrender/src/renderer/mod.rs
@@ -3394,11 +3394,13 @@
             draw_target,
         );
 
         // Draw any borders for this target.
         if !target.border_segments_solid.is_empty() ||
-           !target.border_segments_complex.is_empty()
+           !target.border_segments_complex.is_empty() ||
+           !target.border_segments_solid_superellipse.is_empty() ||
+           !target.border_segments_complex_superellipse.is_empty()
         {
             let _timer = self.gpu_profiler.start_timer(GPU_TAG_CACHE_BORDER);
 
             self.set_blend(true, FramebufferKind::Other);
             self.set_blend_mode_premultiplied_alpha(FramebufferKind::Other);
@@ -3437,10 +3439,46 @@
                     &BatchTextures::empty(),
                     stats,
                 );
             }
 
+            if !target.border_segments_solid_superellipse.is_empty() {
+                self.shaders.borrow_mut().cs_border_solid_superellipse().bind(
+                    &mut self.device,
+                    &projection,
+                    None,
+                    &mut self.renderer_errors,
+                    &mut self.profile,
+                    &mut self.command_log,
+                );
+
+                self.draw_instanced_batch(
+                    &target.border_segments_solid_superellipse,
+                    VertexArrayKind::Border,
+                    &BatchTextures::empty(),
+                    stats,
+                );
+            }
+
+            if !target.border_segments_complex_superellipse.is_empty() {
+                self.shaders.borrow_mut().cs_border_segment_superellipse().bind(
+                    &mut self.device,
+                    &projection,
+                    None,
+                    &mut self.renderer_errors,
+                    &mut self.profile,
+                    &mut self.command_log,
+                );
+
+                self.draw_instanced_batch(
+                    &target.border_segments_complex_superellipse,
+                    VertexArrayKind::Border,
+                    &BatchTextures::empty(),
+                    stats,
+                );
+            }
+
             self.set_blend(false, FramebufferKind::Other);
         }
 
         // Draw any line decorations for this target.
         if !target.line_decorations.is_empty() {
diff --git a/gfx/wr/webrender/src/renderer/shade.rs b/gfx/wr/webrender/src/renderer/shade.rs
--- a/gfx/wr/webrender/src/renderer/shade.rs
+++ b/gfx/wr/webrender/src/renderer/shade.rs
@@ -413,10 +413,12 @@
     // draw intermediate results to cache targets. The results
     // of these shaders are then used by the primitive shaders.
     cs_blur_rgba8: ShaderHandle,
     cs_border_segment: ShaderHandle,
     cs_border_solid: ShaderHandle,
+    cs_border_segment_superellipse: ShaderHandle,
+    cs_border_solid_superellipse: ShaderHandle,
     cs_scale: Vec<Option<ShaderHandle>>,
     cs_line_decoration: ShaderHandle,
     cs_svg_filter_node: ShaderHandle,
 
     // The are "primitive shaders". These shaders draw and blend
@@ -774,19 +776,35 @@
             "cs_border_solid",
             &[],
             &shader_list,
         )?;
 
+        let cs_border_segment_superellipse = loader.create_shader(
+            ShaderKind::Cache(VertexArrayKind::Border),
+            "cs_border_segment",
+             &[SUPERELLIPSE_FEATURE],
+            &shader_list,
+        )?;
+
+        let cs_border_solid_superellipse = loader.create_shader(
+            ShaderKind::Cache(VertexArrayKind::Border),
+            "cs_border_solid",
+            &[SUPERELLIPSE_FEATURE],
+            &shader_list,
+        )?;
+
         let composite = CompositorShaders::new(device, gl_type, &mut loader)?;
 
         Ok(Shaders {
             loader,
 
             cs_blur_rgba8,
             cs_border_segment,
-            cs_line_decoration,
             cs_border_solid,
+            cs_border_segment_superellipse,
+            cs_border_solid_superellipse,
+            cs_line_decoration,
             cs_scale,
             cs_svg_filter_node,
             ps_text_run,
             ps_text_run_dual_source,
             ps_quad_textured,
@@ -979,10 +997,12 @@
     }
 
     pub fn cs_blur_rgba8(&mut self) -> &mut LazilyCompiledShader { self.loader.get(self.cs_blur_rgba8) }
     pub fn cs_border_segment(&mut self) -> &mut LazilyCompiledShader { self.loader.get(self.cs_border_segment) }
     pub fn cs_border_solid(&mut self) -> &mut LazilyCompiledShader { self.loader.get(self.cs_border_solid) }
+    pub fn cs_border_segment_superellipse(&mut self) -> &mut LazilyCompiledShader { self.loader.get(self.cs_border_segment_superellipse) }
+    pub fn cs_border_solid_superellipse(&mut self) -> &mut LazilyCompiledShader { self.loader.get(self.cs_border_solid_superellipse) }
     pub fn cs_line_decoration(&mut self) -> &mut LazilyCompiledShader { self.loader.get(self.cs_line_decoration) }
     pub fn cs_svg_filter_node(&mut self) -> &mut LazilyCompiledShader { self.loader.get(self.cs_svg_filter_node) }
     pub fn ps_quad_textured(&mut self) -> &mut LazilyCompiledShader {
         self.loader.get(self.ps_quad_textured)
     }
diff --git a/gfx/wr/webrender_build/src/shader_features.rs b/gfx/wr/webrender_build/src/shader_features.rs
--- a/gfx/wr/webrender_build/src/shader_features.rs
+++ b/gfx/wr/webrender_build/src/shader_features.rs
@@ -69,17 +69,22 @@
 
     shaders.insert("ps_quad_mask", vec![String::new(), "FAST_PATH".to_string(), "SUPERELLIPSE".to_string()]);
 
     for name in &[
         "cs_line_decoration",
-        "cs_border_segment",
-        "cs_border_solid",
         "cs_svg_filter_node",
     ] {
         shaders.insert(name, vec![String::new()]);
     }
 
+    for name in &[
+        "cs_border_segment",
+        "cs_border_solid",
+    ] {
+        shaders.insert(name, vec![String::new(), "SUPERELLIPSE".to_string()]);
+    }
+
     let mut base_prim_features = FeatureList::new();
 
     let mut texture_types = vec!["TEXTURE_2D"];
     if flags.contains(ShaderFeatureFlags::GL) {
         texture_types.push("TEXTURE_RECT");

