diff --git a/devtools/client/responsive/test/browser/browser.toml b/devtools/client/responsive/test/browser/browser.toml
--- a/devtools/client/responsive/test/browser/browser.toml
+++ b/devtools/client/responsive/test/browser/browser.toml
@@ -275,7 +275,10 @@
 skip-if = [
   "os == 'win' && os_version == '11.26100' && arch == 'x86_64' && asan", # Bug 2016837
 ]
 
 ["browser_window_sizing.js"]
+skip-if = [
+  "os == 'win' && opt && standalone", # Bug 2063169
+]
 
 ["browser_zoom.js"]

