diff --git a/browser/components/preferences/config/account-sync.mjs b/browser/components/preferences/config/account-sync.mjs
index b01fd4bb163ad53139fcbd8e79e12ef994d1fa13..dd60be4fe268618d6596cd5cfac3342975bd241a 100644
--- a/browser/components/preferences/config/account-sync.mjs
+++ b/browser/components/preferences/config/account-sync.mjs
@@ -46,6 +46,7 @@ Preferences.addAll([
   { id: "services.sync.engine.creditcards", type: "bool" },
   { id: "services.sync.engine.addons", type: "bool" },
   { id: "services.sync.engine.prefs", type: "bool" },
+  { id: "services.sync.engine.spaces", type: "bool" },
 ]);
 
 /**
@@ -557,6 +558,7 @@ const SYNC_ENGINE_SETTINGS = [
     type: "payments",
   },
   { id: "syncAddons", pref: "services.sync.engine.addons", type: "addons" },
+  { id: "syncSpaces", pref: "services.sync.engine.spaces", type: "workspaces" },
   { id: "syncSettings", pref: "services.sync.engine.prefs", type: "settings" },
 ];
 
@@ -564,6 +566,12 @@ SYNC_ENGINE_SETTINGS.forEach(({ id, pref }) => {
   Preferences.addSetting({ id, pref });
 });
 
+Preferences.addSetting({ id: "zenSyncWorkspacesGroup" });
+Preferences.addSetting({
+  id: "zenSyncWorkspaces",
+  pref: "services.sync.engine.spaces",
+});
+
 Preferences.addSetting({
   id: "syncEngines",
   deps: SYNC_ENGINE_SETTINGS.map(({ id }) => id),
@@ -995,6 +1003,20 @@ SettingGroupManager.registerGroups({
             id: "syncEnginesList",
             control: "sync-engines-list",
           },
+          {
+            id: "zenSyncWorkspacesGroup",
+            control: "moz-box-item",
+            items: [
+              {
+                id: "zenSyncWorkspaces",
+                l10nId: "zen-settings-workspaces-sync",
+                control: "moz-toggle",
+                controlAttrs: {
+                  inputlayout: "inline-end",
+                },
+              },
+            ],
+          },
           {
             id: "syncChangeOptions",
             control: "moz-box-button",
