diff --git a/browser/components/preferences/config/account-sync.mjs b/browser/components/preferences/config/account-sync.mjs
index b01fd4bb163ad53139fcbd8e79e12ef994d1fa13..7e3c2f6852b77192b4505bb3d5a7b58586da6548 100644
--- a/browser/components/preferences/config/account-sync.mjs
+++ b/browser/components/preferences/config/account-sync.mjs
@@ -46,6 +46,8 @@ Preferences.addAll([
   { id: "services.sync.engine.creditcards", type: "bool" },
   { id: "services.sync.engine.addons", type: "bool" },
   { id: "services.sync.engine.prefs", type: "bool" },
+  { id: "services.sync.engine.spaces", type: "bool" },
+  { id: "zen.spaces-sync.normal-tabs", type: "bool" },
 ]);
 
 /**
@@ -557,6 +559,7 @@ const SYNC_ENGINE_SETTINGS = [
     type: "payments",
   },
   { id: "syncAddons", pref: "services.sync.engine.addons", type: "addons" },
+  { id: "syncSpaces", pref: "services.sync.engine.spaces", type: "workspaces" },
   { id: "syncSettings", pref: "services.sync.engine.prefs", type: "settings" },
 ];
 
@@ -564,6 +567,18 @@ SYNC_ENGINE_SETTINGS.forEach(({ id, pref }) => {
   Preferences.addSetting({ id, pref });
 });
 
+Preferences.addSetting({ id: "zenSyncWorkspacesGroup" });
+Preferences.addSetting({
+  id: "zenSyncWorkspaces",
+  pref: "services.sync.engine.spaces",
+});
+Preferences.addSetting({
+  id: "zenSyncNormalTabs",
+  pref: "zen.spaces-sync.normal-tabs",
+  deps: ["zenSyncWorkspaces"],
+  visible: deps => !!deps.zenSyncWorkspaces.value,
+});
+
 Preferences.addSetting({
   id: "syncEngines",
   deps: SYNC_ENGINE_SETTINGS.map(({ id }) => id),
@@ -995,6 +1010,27 @@ SettingGroupManager.registerGroups({
             id: "syncEnginesList",
             control: "sync-engines-list",
           },
+          {
+            id: "zenSyncWorkspacesGroup",
+            control: "moz-box-item",
+            items: [
+              {
+                id: "zenSyncWorkspaces",
+                l10nId: "zen-settings-workspaces-sync",
+                control: "moz-toggle",
+                controlAttrs: {
+                  inputlayout: "inline-end",
+                },
+                items: [
+                  {
+                    id: "zenSyncNormalTabs",
+                    l10nId: "zen-settings-normal-tabs-sync",
+                    control: "moz-checkbox",
+                  },
+                ],
+              },
+            ],
+          },
           {
             id: "syncChangeOptions",
             control: "moz-box-button",
