diff --git a/browser/components/preferences/config/account-sync.mjs b/browser/components/preferences/config/account-sync.mjs
index b503987a08e2ce64bfc01c4e3a21e5e19ef834f0..b1c03a0b219fd3eddd93c1deaeb18ab79c85f168 100644
--- a/browser/components/preferences/config/account-sync.mjs
+++ b/browser/components/preferences/config/account-sync.mjs
@@ -42,6 +42,7 @@ Preferences.addAll([
   { id: "services.sync.engine.creditcards", type: "bool" },
   { id: "services.sync.engine.addons", type: "bool" },
   { id: "services.sync.engine.prefs", type: "bool" },
+  { id: "services.sync.engine.spaces", type: "bool" },
 ]);
 
 /**
@@ -546,6 +547,7 @@ const SYNC_ENGINE_SETTINGS = [
     type: "payments",
   },
   { id: "syncAddons", pref: "services.sync.engine.addons", type: "addons" },
+  { id: "syncSpaces", pref: "services.sync.engine.spaces", type: "workspaces" },
   { id: "syncSettings", pref: "services.sync.engine.prefs", type: "settings" },
 ];
 
