diff --git a/widget/gtk/nsDragService.cpp b/widget/gtk/nsDragService.cpp
index f760a6bb887b321376761431886c771f14fd03af..d52110ba52a95ef7c7bfb915e120b8dd636c9448 100644
--- a/widget/gtk/nsDragService.cpp
+++ b/widget/gtk/nsDragService.cpp
@@ -23,6 +23,7 @@
 #include "mozilla/StaticPrefs_widget.h"
 #include "mozilla/WidgetUtils.h"
 #include "mozilla/WidgetUtilsGtk.h"
+#include "mozilla/nsZenDragAndDrop.h"
 #include "nsAppShell.h"
 #include "nsArrayUtils.h"
 #include "nsCRT.h"
@@ -2304,11 +2305,15 @@ bool nsDragSession::SetAlphaPixmap(SourceSurface* aSurface,
       cairo_image_surface_get_stride(surf), SurfaceFormat::B8G8R8A8);
   if (!dt) return false;
 
+  auto drag_translucency = DRAG_IMAGE_ALPHA_LEVEL;
+  if (auto zenDragAndDrop = zen::nsZenDragAndDrop::GetZenDragAndDropInstance()) {
+    drag_translucency = zenDragAndDrop->GetDragImageOpacity();
+  }
   dt->ClearRect(Rect(0, 0, dragRect.width, dragRect.height));
   dt->DrawSurface(
       aSurface, Rect(0, 0, dragRect.width, dragRect.height),
       Rect(0, 0, dragRect.width, dragRect.height), DrawSurfaceOptions(),
-      DrawOptions(DRAG_IMAGE_ALPHA_LEVEL, CompositionOp::OP_SOURCE));
+      DrawOptions(drag_translucency, CompositionOp::OP_SOURCE));
 
   cairo_surface_mark_dirty(surf);
   cairo_surface_set_device_offset(surf, -aXOffset, -aYOffset);
