diff --git a/widget/gtk/nsWindow.cpp b/widget/gtk/nsWindow.cpp
index e8cef42106..1eff90f9ee 100644
--- a/widget/gtk/nsWindow.cpp
+++ b/widget/gtk/nsWindow.cpp
@@ -794,2 +794,11 @@ bool nsWindow::WidgetTypeSupportsNativeCompositing() {
 
+// Small wrapper for hints that changes them to xdg_popups based on pref
+static GdkWindowTypeHint GetGdkTypeHint(GdkWindowTypeHint aOriginalHint) {
+  if (Preferences::GetBool("zen.widget.wayland.force_native_menus", false)) {
+    return GDK_WINDOW_TYPE_HINT_POPUP_MENU;
+  }
+
+  return aOriginalHint;
+}
+
 static bool IsPenEvent(GdkEvent* aEvent, bool* isEraser) {
@@ -1793,3 +1802,3 @@ bool nsWindow::WaylandPopupConfigure() {
     case PopupType::Tooltip:
-      gtkTypeHint = GDK_WINDOW_TYPE_HINT_TOOLTIP;
+      gtkTypeHint = GetGdkTypeHint(GDK_WINDOW_TYPE_HINT_TOOLTIP);
       LOG("  popup type Tooltip");
@@ -1797,3 +1806,3 @@ bool nsWindow::WaylandPopupConfigure() {
     default:
-      gtkTypeHint = GDK_WINDOW_TYPE_HINT_UTILITY;
+      gtkTypeHint = GetGdkTypeHint(GDK_WINDOW_TYPE_HINT_UTILITY);
       LOG("  popup type Utility");
@@ -1806,3 +1815,3 @@ bool nsWindow::WaylandPopupConfigure() {
     LOG("  not tracked in popup hierarchy, switch to Utility");
-    gtkTypeHint = GDK_WINDOW_TYPE_HINT_UTILITY;
+    gtkTypeHint = GetGdkTypeHint(GDK_WINDOW_TYPE_HINT_UTILITY);
   }
@@ -2692,3 +2701,3 @@ void nsWindow::WaylandPopupPrepareForMove() {
                                              ? GDK_WINDOW_TYPE_HINT_POPUP_MENU
-                                             : GDK_WINDOW_TYPE_HINT_UTILITY;
+                                             : GetGdkTypeHint(GDK_WINDOW_TYPE_HINT_UTILITY);
 
@@ -2719,5 +2728,5 @@ void nsWindow::WaylandPopupMovePlain(int aX, int aY) {
   MOZ_DIAGNOSTIC_ASSERT(gtk_window_get_type_hint(GTK_WINDOW(mShell)) ==
-                            GDK_WINDOW_TYPE_HINT_UTILITY ||
+                            GetGdkTypeHint(GDK_WINDOW_TYPE_HINT_UTILITY) ||
                         gtk_window_get_type_hint(GTK_WINDOW(mShell)) ==
-                            GDK_WINDOW_TYPE_HINT_TOOLTIP);
+                            GetGdkTypeHint(GDK_WINDOW_TYPE_HINT_TOOLTIP));
 
@@ -6232,3 +6241,3 @@ nsresult nsWindow::Create(nsIWidget* aParent, const LayoutDeviceIntRect& aRect,
     LOG("  Is Media PiP window\n");
-    gtk_window_set_type_hint(GTK_WINDOW(mShell), GDK_WINDOW_TYPE_HINT_UTILITY);
+    gtk_window_set_type_hint(GTK_WINDOW(mShell), GetGdkTypeHint(GDK_WINDOW_TYPE_HINT_UTILITY));
   } else if (mIsAlert) {
@@ -6236,3 +6245,3 @@ nsresult nsWindow::Create(nsIWidget* aParent, const LayoutDeviceIntRect& aRect,
     gtk_window_set_type_hint(GTK_WINDOW(mShell),
-                             GDK_WINDOW_TYPE_HINT_NOTIFICATION);
+                             GetGdkTypeHint(GDK_WINDOW_TYPE_HINT_NOTIFICATION));
     gtk_window_set_skip_taskbar_hint(GTK_WINDOW(mShell), TRUE);
@@ -6242,3 +6251,3 @@ nsresult nsWindow::Create(nsIWidget* aParent, const LayoutDeviceIntRect& aRect,
     SetDefaultIcon();
-    gtk_window_set_type_hint(GTK_WINDOW(mShell), GDK_WINDOW_TYPE_HINT_DIALOG);
+    gtk_window_set_type_hint(GTK_WINDOW(mShell), GetGdkTypeHint(GDK_WINDOW_TYPE_HINT_DIALOG));
     LOG("  nsWindow::Create(): dialog");
@@ -6255,3 +6264,3 @@ nsresult nsWindow::Create(nsIWidget* aParent, const LayoutDeviceIntRect& aRect,
     if (mIsDragPopup) {
-      gtk_window_set_type_hint(GTK_WINDOW(mShell), GDK_WINDOW_TYPE_HINT_DND);
+      gtk_window_set_type_hint(GTK_WINDOW(mShell), GetGdkTypeHint(GDK_WINDOW_TYPE_HINT_DND));
       LOG("  nsWindow::Create() Drag popup\n");
@@ -6266,6 +6275,6 @@ nsresult nsWindow::Create(nsIWidget* aParent, const LayoutDeviceIntRect& aRect,
         case PopupType::Tooltip:
-          gtkTypeHint = GDK_WINDOW_TYPE_HINT_TOOLTIP;
+          gtkTypeHint = GetGdkTypeHint(GDK_WINDOW_TYPE_HINT_TOOLTIP);
           break;
         default:
-          gtkTypeHint = GDK_WINDOW_TYPE_HINT_UTILITY;
+          gtkTypeHint = GetGdkTypeHint(GDK_WINDOW_TYPE_HINT_UTILITY);
           break;
@@ -7210,3 +7219,3 @@ FullscreenTransitionWindow::FullscreenTransitionWindow(GtkWidget* aWidget) {
 
-  gtk_window_set_type_hint(gtkWin, GDK_WINDOW_TYPE_HINT_SPLASHSCREEN);
+  gtk_window_set_type_hint(gtkWin, GetGdkTypeHint(GDK_WINDOW_TYPE_HINT_SPLASHSCREEN));
   GtkWindowSetTransientFor(gtkWin, GTK_WINDOW(aWidget));
@@ -9897,3 +9906,3 @@ bool nsWindow::ApplyEnterLeaveMutterWorkaround() {
       gtk_window_get_type_hint(GTK_WINDOW(mWaylandPopupNext->GetGtkWidget())) ==
-          GDK_WINDOW_TYPE_HINT_UTILITY) {
+          GetGdkTypeHint(GDK_WINDOW_TYPE_HINT_UTILITY)) {
     LOG("nsWindow::ApplyEnterLeaveMutterWorkaround(): leave toplevel");
@@ -9904,3 +9913,3 @@ bool nsWindow::ApplyEnterLeaveMutterWorkaround() {
       gtk_window_get_type_hint(GTK_WINDOW(mShell)) ==
-          GDK_WINDOW_TYPE_HINT_UTILITY) {
+          GetGdkTypeHint(GDK_WINDOW_TYPE_HINT_UTILITY)) {
     LOG("nsWindow::ApplyEnterLeaveMutterWorkaround(): leave popup");
