diff --git a/browser/components/newtab/data/content/abouthomecache/page.html.template b/browser/components/newtab/data/content/abouthomecache/page.html.template
index 60898ed6b82ec298198cb9529387f4efd8561cc1..fbd93ccc7ee3619415192164a55f5fcbfd387a51 100644
--- a/browser/components/newtab/data/content/abouthomecache/page.html.template
+++ b/browser/components/newtab/data/content/abouthomecache/page.html.template
@@ -32,6 +32,7 @@
     <link rel="stylesheet" href="chrome://global/skin/design-system/tokens-brand.css">
     <link rel="stylesheet" href="chrome://browser/content/contentSearchUI.css" />
     <link rel="stylesheet" href="chrome://activity-stream/content/css/activity-stream.css" />
+    <script src="chrome://browser/content/zenThemeModifier.js"></script>
   </head>
   <!-- Cached: {{ CACHE_TIME }} -->
   <body class="activity-stream">
