diff --git a/widget/cocoa/nsCocoaUtils.mm b/widget/cocoa/nsCocoaUtils.mm
index 49cb2b2d6960cbb0b6a8a17c409cfc909072213d..2406d1b576d1bfd38add149522c64281b9a20889 100644
--- a/widget/cocoa/nsCocoaUtils.mm
+++ b/widget/cocoa/nsCocoaUtils.mm
@@ -1630,7 +1630,8 @@ bool static ShouldConsiderStartingSwipeFromEvent(NSEvent* anEvent) {
   return [anEvent type] == NSEventTypeScrollWheel &&
          eventPhase == NSEventPhaseBegan &&
          [anEvent hasPreciseScrollingDeltas] &&
-         [NSEvent isSwipeTrackingFromScrollEventsEnabled];
+         ([NSEvent isSwipeTrackingFromScrollEventsEnabled] ||
+              Preferences::GetBool("zen.widget.macos.override-system-swipe-gestures"));
 }
 
 PanGestureInput nsCocoaUtils::CreatePanGestureEvent(
