diff --git a/widget/cocoa/VibrancyManager.mm b/widget/cocoa/VibrancyManager.mm
index bc62ff28285691c605fe0680e7a98927f11d5882..63403611b9c5c5d12fe221c2bb56f884fabe85ba 100644
--- a/widget/cocoa/VibrancyManager.mm
+++ b/widget/cocoa/VibrancyManager.mm
@@ -9,6 +9,7 @@
 #import <objc/message.h>
 
 #include "mozilla/StaticPrefs_widget.h"
+#include "mozilla/StaticPrefs_zen.h"
 #include "nsCocoaWindow.h"
 
 using namespace mozilla;
@@ -28,6 +29,9 @@ static NSVisualEffectState VisualEffectStateForVibrancyType(
     case VibrancyType::Sidebar:
       break;
   }
+  if (!StaticPrefs::zen_view_grey_out_inactive_windows()) {
+    return NSVisualEffectStateActive;
+  }
   return NSVisualEffectStateFollowsWindowActiveState;
 }
 
@@ -35,7 +39,23 @@ static NSVisualEffectMaterial VisualEffectMaterialForVibrancyType(
     VibrancyType aType) {
   switch (aType) {
     case VibrancyType::Sidebar:
-      return NSVisualEffectMaterialSidebar;
+      switch (StaticPrefs::zen_widget_macos_window_material()) {
+        case 1:
+          return NSVisualEffectMaterialHUDWindow;
+        case 2:
+          return NSVisualEffectMaterialFullScreenUI;
+        case 3:
+          return NSVisualEffectMaterialPopover;
+        case 4:
+          return NSVisualEffectMaterialMenu;
+        case 5:
+          return NSVisualEffectMaterialToolTip;
+        case 6:
+          return NSVisualEffectMaterialHeaderView;
+        case 7:
+        default:
+          return NSVisualEffectMaterialUnderWindowBackground;
+      }
     case VibrancyType::Titlebar:
       return NSVisualEffectMaterialTitlebar;
   }
@@ -75,6 +95,7 @@ - (NSView*)hitTest:(NSPoint)aPoint {
 
 - (void)prefChanged {
   self.blendingMode = VisualEffectBlendingModeForVibrancyType(mType);
+  self.material = VisualEffectMaterialForVibrancyType(mType);
 }
 @end
 
@@ -85,6 +106,7 @@ static void PrefChanged(const char* aPref, void* aClosure) {
 static constexpr nsLiteralCString kObservedPrefs[] = {
     "widget.macos.sidebar-blend-mode.behind-window"_ns,
     "widget.macos.titlebar-blend-mode.behind-window"_ns,
+    "zen.widget.macos.window-material"_ns,
 };
 
 VibrancyManager::VibrancyManager(const nsCocoaWindow& aCoordinateConverter,
