diff --git a/layout/svg/SVGUtils.h b/layout/svg/SVGUtils.h
index 1f1f7ad6b78e5c411651fee30bbbfeb784257e0a..28c7332aae411c4220bd00af08b491d31d2eb090 100644
--- a/layout/svg/SVGUtils.h
+++ b/layout/svg/SVGUtils.h
@@ -428,7 +428,7 @@ class SVGUtils final {
 
   static nscolor GetFallbackOrPaintColor(
       const ComputedStyle&, StyleSVGPaint nsStyleSVG::* aFillOrStroke,
-      nscolor aDefaultContextFallbackColor);
+      nscolor aDefaultContextFallbackColor, const nsIFrame* aFrame);
 
   static void MakeFillPatternFor(nsIFrame* aFrame, gfxContext* aContext,
                                  GeneralPattern* aOutPattern,
