diff --git a/browser/components/urlbar/tests/unit/test_providerHeuristicFallback.js b/browser/components/urlbar/tests/unit/test_providerHeuristicFallback.js
index 6b8243ea39..67b63ae8f0 100644
--- a/browser/components/urlbar/tests/unit/test_providerHeuristicFallback.js
+++ b/browser/components/urlbar/tests/unit/test_providerHeuristicFallback.js
@@ -97,6 +97,9 @@ add_task(async function () {
         fallbackTitle: `${query}/`,
         heuristic: true,
       }),
+      makeSearchResult(context, {
+        engineName: SUGGESTIONS_ENGINE_NAME,
+      }),
     ],
   });
 
@@ -112,6 +115,9 @@ add_task(async function () {
         fallbackTitle: `${query}/`,
         heuristic: true,
       }),
+      makeSearchResult(context, {
+        engineName: SUGGESTIONS_ENGINE_NAME,
+      }),
     ],
   });
 
@@ -127,6 +133,9 @@ add_task(async function () {
         fallbackTitle: `${query}/`,
         heuristic: true,
       }),
+      makeSearchResult(context, {
+        engineName: SUGGESTIONS_ENGINE_NAME,
+      }),
     ],
   });
 
@@ -159,6 +168,9 @@ add_task(async function () {
         fallbackTitle: query,
         heuristic: true,
       }),
+      makeSearchResult(context, {
+        engineName: SUGGESTIONS_ENGINE_NAME,
+      }),
     ],
   });
 
@@ -174,6 +186,10 @@ add_task(async function () {
         fallbackTitle: `${query}/`,
         heuristic: true,
       }),
+      makeSearchResult(context, {
+        engineName: SUGGESTIONS_ENGINE_NAME,
+        query: `${query}\u2028`,
+      }),
     ],
   });
 
@@ -199,6 +215,10 @@ add_task(async function () {
         iconUri: "page-icon:http://mozilla.org/",
         heuristic: true,
       }),
+      makeSearchResult(context, {
+        engineName: SUGGESTIONS_ENGINE_NAME,
+        query: `${query}\u2028`,
+      }),
     ],
   });
   await PlacesUtils.history.clear();
@@ -266,6 +286,9 @@ add_task(async function () {
         iconUri: "page-icon:http://firefox/",
         heuristic: true,
       }),
+      makeSearchResult(context, {
+        engineName: SUGGESTIONS_ENGINE_NAME,
+      }),
     ],
   });
 
@@ -286,6 +309,9 @@ add_task(async function () {
         iconUri: "page-icon:http://mozilla/",
         heuristic: true,
       }),
+      makeSearchResult(context, {
+        engineName: SUGGESTIONS_ENGINE_NAME,
+      }),
     ],
   });
 
@@ -302,6 +328,9 @@ add_task(async function () {
         fallbackTitle: `${query}/`,
         heuristic: true,
       }),
+      makeSearchResult(context, {
+        engineName: SUGGESTIONS_ENGINE_NAME,
+      }),
     ],
   });
 
@@ -317,6 +346,9 @@ add_task(async function () {
         fallbackTitle: `${query}/`,
         heuristic: true,
       }),
+      makeSearchResult(context, {
+        engineName: SUGGESTIONS_ENGINE_NAME,
+      }),
     ],
   });
 
@@ -443,6 +475,9 @@ add_task(async function () {
         fallbackTitle: `${query}/`,
         heuristic: true,
       }),
+      makeSearchResult(context, {
+        engineName: SUGGESTIONS_ENGINE_NAME,
+      }),
     ],
   });
 
@@ -452,10 +487,8 @@ add_task(async function () {
   await check_results({
     context,
     matches: [
-      makeVisitResult(context, {
-        source: UrlbarUtils.RESULT_SOURCE.OTHER_LOCAL,
-        uri: `${query}/`,
-        fallbackTitle: `${query}/`,
+      makeSearchResult(context, {
+        engineName: SUGGESTIONS_ENGINE_NAME,
         heuristic: true,
       }),
     ],
@@ -473,6 +506,9 @@ add_task(async function () {
         fallbackTitle: `${query}/`,
         heuristic: true,
       }),
+      makeSearchResult(context, {
+        engineName: SUGGESTIONS_ENGINE_NAME,
+      }),
     ],
   });
 
@@ -519,6 +555,9 @@ add_task(async function () {
         fallbackTitle: query,
         heuristic: true,
       }),
+      makeSearchResult(context, {
+        engineName: SUGGESTIONS_ENGINE_NAME,
+      }),
     ],
   });
 
@@ -534,6 +573,9 @@ add_task(async function () {
         fallbackTitle: query,
         heuristic: true,
       }),
+      makeSearchResult(context, {
+        engineName: SUGGESTIONS_ENGINE_NAME,
+      }),
     ],
   });
 
@@ -721,6 +763,9 @@ add_task(async function () {
         fallbackTitle: "user:pass@not-host/",
         heuristic: true,
       }),
+      makeSearchResult(context, {
+        engineName: SUGGESTIONS_ENGINE_NAME,
+      }),
     ],
   });
 
