diff --git a/toolkit/components/pictureinpicture/content/player.xhtml b/toolkit/components/pictureinpicture/content/player.xhtml
index b38789882149c97a3263c405b783999bc60a5c71..009b7b7aa2ded0d88247c62171bfff59222a8e28 100644
--- a/toolkit/components/pictureinpicture/content/player.xhtml
+++ b/toolkit/components/pictureinpicture/content/player.xhtml
@@ -18,6 +18,7 @@
     <link rel="localization" href="browser/browserSets.ftl"/>
     <script src="chrome://global/content/pictureinpicture/player.js"></script>
     <title data-l10n-id="pictureinpicture-player-title"></title>
+    <link rel="localization" href="browser/zen-general.ftl"/>
   </head>
 
   <body>
@@ -53,13 +54,22 @@
       tabindex="10"
 #endif
       />
-      <button id="unpip"
-      class="control-item control-button tooltip-under-controls" data-l10n-id="pictureinpicture-unpip-btn" data-l10n-attrs="tooltip"
+      <button id="minimize"
+      class="control-item control-button tooltip-under-controls" data-l10n-id="pictureinpicture-minimize-btn" data-l10n-attrs="tooltip"
 #ifdef XP_MACOSX
       mac="true"
       tabindex="10"
 #else
       tabindex="9"
+#endif
+      />
+      <button id="unpip"
+      class="control-item control-button tooltip-under-controls" data-l10n-id="pictureinpicture-unpip-btn" data-l10n-attrs="tooltip"
+#ifdef XP_MACOSX
+      mac="true"
+      tabindex="11"
+#else
+      tabindex="10"
 #endif
       />
     <div id="controls-bottom-gradient" class="control-item"></div>
