diff --git a/toolkit/components/pictureinpicture/content/player.js b/toolkit/components/pictureinpicture/content/player.js
index a23e4a3ef87dfe7ca74cc20abd53e5fa1b2664dd..afc9ade19441bf9e23ee91d87ee2644e952543d9 100644
--- a/toolkit/components/pictureinpicture/content/player.js
+++ b/toolkit/components/pictureinpicture/content/player.js
@@ -734,6 +734,11 @@ let Player = {
         document.getElementById("large").click();
         break;
       }
+
+      case "minimize": {
+        this.closePipWindow({ reason: "CloseButton" });
+        break;
+      }
     }
     // If the click came from a element that is not inside the subtitles settings panel
     // then we want to hide the panel
