diff --git a/dom/chrome-webidl/BrowsingContext.webidl b/dom/chrome-webidl/BrowsingContext.webidl
index 43f28a6aad9d0dd3e3c908e8472244fc6aa32a74..60bfd83a9da2923ab3811f363320db474323e636 100644
--- a/dom/chrome-webidl/BrowsingContext.webidl
+++ b/dom/chrome-webidl/BrowsingContext.webidl
@@ -179,6 +179,9 @@ interface BrowsingContext {
 
   [SetterThrows] attribute float textZoom;
 
+  [SetterThrows] attribute long zenBoostsData;
+  [SetterThrows] attribute boolean isZenBoostsInverted;
+
   // Override the dots-per-CSS-pixel scaling factor in this BrowsingContext
   // and all of its descendants. May only be set on the top BC, and should
   // only be set from the parent process.
