diff --git a/dom/chrome-webidl/IOUtils.webidl b/dom/chrome-webidl/IOUtils.webidl
index 8b28cd2f9d746b082ca421d9817e81b51be02777..47a3dd96730f7845dcbbd06d2d76012126b863c5 100644
--- a/dom/chrome-webidl/IOUtils.webidl
+++ b/dom/chrome-webidl/IOUtils.webidl
@@ -95,19 +95,19 @@ namespace IOUtils {
   [NewObject]
   Promise<unsigned long long> writeUTF8(DOMString path, UTF8String string, optional WriteOptions options = {});
   /**
-   * Attempts to serialize |value| into a JSON string and encode it as into a
-   * UTF-8 string, then safely write the result to a file at |path|. Works
-   * exactly like |write|.
+   * Attempts to serialize |value| into a JSON string and encode it as a UTF-8
+   * string, then safely write the result to a file at |path|. Works exactly
+   * like |write|.
    *
    * @param path      An absolute file path
    * @param value     The value to be serialized.
    * @param options   Options for writing the file. The "append" mode is not supported.
    *
-   * @return Resolves with the number of bytes successfully written to the file,
-   *         otherwise rejects with a DOMException.
+   * @return Resolves with the pre-compression size of the serialized JSON in
+   *         bytes (UTF-8), otherwise rejects with a DOMException.
    */
   [NewObject]
-  Promise<unsigned long long> writeJSON(DOMString path, any value, optional WriteOptions options = {});
+  Promise<WriteJSONResult> writeJSON(DOMString path, any value, optional WriteJSONOptions options = {});
   /**
    * Moves the file from |sourcePath| to |destPath|, creating necessary parents.
    * If |destPath| is a directory, then the source file will be moved into the
@@ -566,6 +566,41 @@ dictionary WriteOptions {
    * If true, compress the data with LZ4-encoding before writing to the file.
    */
   boolean compress = false;
+  /**
+   * For |writeJSON|, a hint for the expected JSON string length in UTF-16 code
+   * units. When provided, the JSON serializer pre-allocates a buffer of this
+   * size to avoid incremental reallocations.
+   */
+  unsigned long long jsonLengthHint = 0;
+};
+
+/**
+ * Options to be passed to the |IOUtils.writeJSON| method.
+ */
+dictionary WriteJSONOptions: WriteOptions {
+  /**
+   * An optional length hint that will be used to pre-allocate the buffer that
+   * will hold the stringified JSON.
+   *
+   * This is the *length* and not the size (i.e., it is the number of UTF-16
+   * codepoints and not the number of bytes).
+   */
+  unsigned long long lengthHint = 0;
+};
+
+/**
+ * Information about a WriteJSON operation.
+ */
+dictionary WriteJSONResult {
+  /**
+   * The number of bytes written.
+   */
+  required unsigned long long size;
+
+  /**
+   * The length of the stringified JSON (in UTF-16 codepoints).
+   */
+  required unsigned long long jsonLength;
 };
 
 /**
