diff --git a/toolkit/components/pictureinpicture/PictureInPicture.sys.mjs b/toolkit/components/pictureinpicture/PictureInPicture.sys.mjs
index ad325b45b882e8444b1b4bfc9c78950cf090b411..14f7d840bb91dfef0887537ef95de733d5591265 100644
--- a/toolkit/components/pictureinpicture/PictureInPicture.sys.mjs
+++ b/toolkit/components/pictureinpicture/PictureInPicture.sys.mjs
@@ -122,6 +122,9 @@ export class PictureInPictureToggleParent extends JSWindowActorParent {
         if (browser.ownerGlobal.gBrowser.selectedBrowser == browser) {
           break;
         }
+        if (browser.audioMuted) {
+          break;
+        }
         let actor = browsingContext.currentWindowGlobal.getActor(
           "PictureInPictureLauncher"
         );
@@ -490,7 +493,7 @@ export var PictureInPicture = {
     // focus the tab's window
     tab.ownerGlobal.focus();
 
-    gBrowser.selectedTab = tab;
+    browser?.ownerGlobal?.gZenWorkspaces.switchIfNeeded(browser);
     await this.closeSinglePipWindow({ reason: "Unpip", actorRef: pipActor });
   },
 
