diff --git a/browser/components/sidebar/browser-sidebar.js b/browser/components/sidebar/browser-sidebar.js
index 6cbac7c0826856adfcdafd3507e05f3efac95a2d..966289f04246deefe3dcb233f7fcb755fccfdbc6 100644
--- a/browser/components/sidebar/browser-sidebar.js
+++ b/browser/components/sidebar/browser-sidebar.js
@@ -404,6 +404,7 @@ var SidebarController = {
     } else {
       this._box.removeAttribute("positionend");
       sidebarMain.removeAttribute("positionend");
+      this._box.style.order = 0;
     }
 
     this.hideSwitcherPanel();
