diff --git a/browser/components/sidebar/browser-sidebar.js b/browser/components/sidebar/browser-sidebar.js
index c68a32794996348d95f3b1ee363be2eec587065e..3a163c75ac10a204754648dfd1332c79895799c7 100644
--- a/browser/components/sidebar/browser-sidebar.js
+++ b/browser/components/sidebar/browser-sidebar.js
@@ -521,7 +521,7 @@ var SidebarController = {
    */
   setPosition() {
     // First reset all ordinals to match DOM ordering.
-    let browser = document.getElementById("browser");
+    let browser = document.getElementById("tabbrowser-tabbox");
     [...browser.children].forEach((node, i) => {
       node.style.order = i + 1;
     });
@@ -535,9 +535,11 @@ var SidebarController = {
       let boxOrdinal = this._box.style.order;
       this._box.style.order = appcontent.style.order;
 
-      appcontent.style.order = boxOrdinal;
+      // appcontent.style.order = boxOrdinal;
       // the launcher should be on the right of the sidebar-box
-      sidebarContainer.style.order = parseInt(this._box.style.order) + 1;
+      this._box.style.order = browser.children.length + 2;
+      this._splitter.style.order = browser.children.length + 1;
+      this._splitter.style.marginRight = 0;
       // Indicate we've switched ordering to the box
       this._box.setAttribute("positionend", true);
       sidebarMain.setAttribute("positionend", true);
@@ -546,6 +548,9 @@ var SidebarController = {
       this._box.removeAttribute("positionend");
       sidebarMain.removeAttribute("positionend");
       sidebarContainer.removeAttribute("positionend");
+      this._box.style.order = 1;
+      this._splitter.style.order = 2;
+      this._splitter.style.removeProperty("margin-right");
     }
 
     this.hideSwitcherPanel();
