diff --git a/browser/components/sidebar/browser-sidebar.js b/browser/components/sidebar/browser-sidebar.js
index d05d561b6d01b11a36c762fc39dc13e14181bce7..cbb85d7019a68f283e38c45172eda605c7b90144 100644
--- a/browser/components/sidebar/browser-sidebar.js
+++ b/browser/components/sidebar/browser-sidebar.js
@@ -471,6 +471,7 @@ var SidebarController = {
       this._box.removeAttribute("positionend");
       sidebarMain.removeAttribute("positionend");
       sidebarContainer.removeAttribute("positionend");
+      this._box.style.order = 0;
     }
 
     this.hideSwitcherPanel();
