diff --git a/browser/components/sidebar/sidebar-customize.mjs b/browser/components/sidebar/sidebar-customize.mjs
--- a/browser/components/sidebar/sidebar-customize.mjs
+++ b/browser/components/sidebar/sidebar-customize.mjs
@@ -18,27 +18,14 @@ const l10nMap = new Map([
   ["viewGenaiChatSidebar", "sidebar-menu-genai-chat-label"],
   ["viewGenaiPageAssistSidebar", "sidebar-menu-genai-page-assist-label"],
   ["viewHistorySidebar", "sidebar-menu-history-label"],
-  ["viewTabsSidebar", "sidebar-menu-synced-tabs-label"],
   ["viewBookmarksSidebar", "sidebar-menu-bookmarks-label"],
   ["viewCPMSidebar", "sidebar-menu-contextual-password-manager-label"],
 ]);
-const VISIBILITY_SETTING_PREF = "sidebar.visibility";
-const EXPAND_ON_HOVER_PREF = "sidebar.expandOnHover";
 const POSITION_SETTING_PREF = "sidebar.position_start";
-const TAB_DIRECTION_SETTING_PREF = "sidebar.verticalTabs";
 
 export class SidebarCustomize extends SidebarPage {
   constructor() {
     super();
-    XPCOMUtils.defineLazyPreferenceGetter(
-      this.#prefValues,
-      "visibility",
-      VISIBILITY_SETTING_PREF,
-      "always-show",
-      (_aPreference, _previousValue, newValue) => {
-        this.visibility = newValue;
-      }
-    );
     XPCOMUtils.defineLazyPreferenceGetter(
       this.#prefValues,
       "isPositionStart",
@@ -48,38 +35,14 @@ export class SidebarCustomize extends SidebarPage {
         this.isPositionStart = newValue;
       }
     );
-    XPCOMUtils.defineLazyPreferenceGetter(
-      this.#prefValues,
-      "verticalTabsEnabled",
-      TAB_DIRECTION_SETTING_PREF,
-      false,
-      (_aPreference, _previousValue, newValue) => {
-        this.verticalTabsEnabled = newValue;
-      }
-    );
-    XPCOMUtils.defineLazyPreferenceGetter(
-      this.#prefValues,
-      "expandOnHoverEnabled",
-      EXPAND_ON_HOVER_PREF,
-      false,
-      (_aPreference, _previousValue, newValue) => {
-        this.expandOnHoverEnabled = newValue;
-      }
-    );
-    this.visibility = this.#prefValues.visibility;
     this.isPositionStart = this.#prefValues.isPositionStart;
-    this.verticalTabsEnabled = this.#prefValues.verticalTabsEnabled;
-    this.expandOnHoverEnabled = this.#prefValues.expandOnHoverEnabled;
     this.boundObserve = (...args) => this.observe(...args);
   }
 
   #prefValues = {};
 
   static properties = {
-    visibility: { type: String },
     isPositionStart: { type: Boolean },
-    verticalTabsEnabled: { type: Boolean },
-    expandOnHoverEnabled: { type: Boolean },
   };
 
   static queries = {
@@ -87,9 +50,6 @@ export class SidebarCustomize extends SidebarPage {
     extensionInputs: { all: ".extensions > .tool" },
     extensionLink: ".extension-item a",
     positionInput: "#position",
-    visibilityInput: "#hide-sidebar",
-    verticalTabsInput: "#vertical-tabs",
-    expandOnHoverInput: "#expand-on-hover",
   };
 
   connectedCallback() {
@@ -136,11 +96,6 @@ export class SidebarCustomize extends SidebarPage {
           checked: e.target.checked,
         });
         break;
-      case "viewTabsSidebar":
-        Glean.sidebarCustomize.syncedTabsEnabled.record({
-          checked: e.target.checked,
-        });
-        break;
       case "viewHistorySidebar":
         Glean.sidebarCustomize.historyEnabled.record({
           checked: e.target.checked,
@@ -218,50 +173,6 @@ export class SidebarCustomize extends SidebarPage {
         </sidebar-panel-header>
         <div class="sidebar-panel-scrollable-content">
           <moz-fieldset class="customize-group no-end-margin" data-l10n-id="sidebar-settings">
-            <moz-checkbox
-              type="checkbox"
-              id="vertical-tabs"
-              name="verticalTabs"
-              iconsrc="chrome://browser/skin/sidebar-collapsed.svg"
-              data-l10n-id="sidebar-vertical-tabs"
-              @change=${this.#handleTabDirectionChange}
-              ?checked=${this.verticalTabsEnabled}
-            >
-            ${when(
-              this.verticalTabsEnabled,
-              () => html`
-                ${when(
-                  this.expandOnHoverEnabled,
-                  () => html`
-                    <moz-checkbox
-                      slot="nested"
-                      type="checkbox"
-                      id="expand-on-hover"
-                      name="expand-on-hover"
-                      data-l10n-id="expand-sidebar-on-hover"
-                      @change=${this.#toggleExpandOnHover}
-                      ?checked=${this.getWindow().SidebarController._state
-                        .revampVisibility === "expand-on-hover"}
-                      ?disabled=${this.visibility == "hide-sidebar"}
-                    ></moz-checkbox>
-                  `
-                )}
-                <moz-checkbox
-                  slot="nested"
-                  type="checkbox"
-                  id="hide-sidebar"
-                  name="hideSidebar"
-                  data-l10n-id="sidebar-hide-tabs-and-sidebar"
-                  @change=${this.#handleVisibilityChange}
-                  ?checked=${this.visibility == "hide-sidebar"}
-                  ?disabled=${this.getWindow().SidebarController._state
-                    .revampVisibility === "expand-on-hover"}
-                ></moz-checkbox>
-              `
-            )}
-            </moz-checkbox>
-          </moz-fieldset>
-          <moz-fieldset class="customize-group medium-top-margin no-label">
             <moz-checkbox
               type="checkbox"
               id="position"
@@ -269,7 +180,7 @@ export class SidebarCustomize extends SidebarPage {
               data-l10n-id=${document.dir == "rtl" ? "sidebar-show-on-the-left" : "sidebar-show-on-the-right"}
               @change=${this.reversePosition}
               ?checked=${!this.isPositionStart}
-          ></moz-checkbox>
+            ></moz-checkbox>
           </moz-fieldset>
           <moz-fieldset class="customize-group tools" data-l10n-id="sidebar-customize-firefox-tools-header">
             ${this.getWindow()
@@ -319,38 +230,6 @@ export class SidebarCustomize extends SidebarPage {
       </div>
     `;
   }
-
-  #handleVisibilityChange(e) {
-    e.stopPropagation();
-    this.visibility = e.target.checked ? "hide-sidebar" : "always-show";
-    Services.prefs.setStringPref(
-      VISIBILITY_SETTING_PREF,
-      e.target.checked ? "hide-sidebar" : "always-show"
-    );
-    Glean.sidebarCustomize.sidebarDisplay.record({
-      preference: e.target.checked ? "hide" : "always",
-    });
-  }
-
-  #toggleExpandOnHover(e) {
-    e.stopPropagation();
-    if (e.target.checked) {
-      Services.prefs.setStringPref("sidebar.visibility", "expand-on-hover");
-      Glean.sidebarCustomize.expandOnHoverEnabled.record({
-        checked: true,
-      });
-    } else {
-      Services.prefs.setStringPref("sidebar.visibility", "always-show");
-    }
-  }
-
-  #handleTabDirectionChange({ target: { checked } }) {
-    const verticalTabsEnabled = checked;
-    Services.prefs.setBoolPref(TAB_DIRECTION_SETTING_PREF, verticalTabsEnabled);
-    Glean.sidebarCustomize.tabsLayout.record({
-      orientation: verticalTabsEnabled ? "vertical" : "horizontal",
-    });
-  }
 }
 
 customElements.define("sidebar-customize", SidebarCustomize);
