diff --git a/browser/installer/windows/nsis/defines.nsi.in b/browser/installer/windows/nsis/defines.nsi.in
index a9bf2d67ce522054442be664bda1434c63609ea1..dc2b861668a1c7d33d5b18569ed51b6eb87b4e3c 100644
--- a/browser/installer/windows/nsis/defines.nsi.in
+++ b/browser/installer/windows/nsis/defines.nsi.in
@@ -137,6 +137,10 @@ VIAddVersionKey "CompanyName"     "${CompanyName}"
 VIAddVersionKey "LegalCopyright"  "${CompanyName}"
 VIAddVersionKey "FileVersion"     "${AppVersion}"
 VIAddVersionKey "ProductVersion"  "${AppVersion}"
+VIAddVersionKey "SupportLink" "https://github.com/Hrishikeshmind/astradesktop"
+VIAddVersionKey "URLUpdateInfo" "https://github.com/Hrishikeshmind/astradesktop/releases"
+VIAddVersionKey "URLInfoAbout" "https://github.com/Hrishikeshmind/astradesktop"
+VIAddVersionKey "Comments" "Astra Browser - Made for India"
 # Comments is not used but left below commented out for future reference
 # VIAddVersionKey "Comments"        "Comments"
 
@@ -147,7 +151,7 @@ VIAddVersionKey "ProductVersion"  "${AppVersion}"
 !define APPROXIMATE_REQUIRED_SPACE_MB "145"
 
 # Constants for parts of the telemetry submission URL
-!define TELEMETRY_BASE_URL https://incoming.telemetry.mozilla.org/submit
+!define TELEMETRY_BASE_URL
 !define TELEMETRY_NAMESPACE firefox-installer
 !define TELEMETRY_INSTALL_PING_VERSION 1
 !define TELEMETRY_INSTALL_PING_DOCTYPE install
