diff --git a/browser/components/customizableui/CustomizableUI.sys.mjs b/browser/components/customizableui/CustomizableUI.sys.mjs
index 289c3df7dcd6fa6b35681ce61c1e920b6a2651d7..a31dc07c8b756e09f43cd5f81bad8d4a3c5f7fce 100644
--- a/browser/components/customizableui/CustomizableUI.sys.mjs
+++ b/browser/components/customizableui/CustomizableUI.sys.mjs
@@ -13,6 +13,7 @@ ChromeUtils.defineESModuleGetters(lazy, {
   AddonManagerPrivate: "resource://gre/modules/AddonManager.sys.mjs",
   BrowserUsageTelemetry: "resource:///modules/BrowserUsageTelemetry.sys.mjs",
   CustomizableWidgets: "resource:///modules/CustomizableWidgets.sys.mjs",
+  ZenCustomizableUI: "chrome://browser/content/ZenCustomizableUI.sys.mjs",
   HomePage: "resource:///modules/HomePage.sys.mjs",
   PanelMultiView: "resource:///modules/PanelMultiView.sys.mjs",
   PrivateBrowsingUtils: "resource://gre/modules/PrivateBrowsingUtils.sys.mjs",
@@ -25,7 +26,7 @@ ChromeUtils.defineLazyGetter(lazy, "gWidgetsBundle", function () {
   return Services.strings.createBundle(kUrl);
 });
 
-const kDefaultThemeID = "default-theme@mozilla.org";
+const kDefaultThemeID = "firefox-compact-dark@mozilla.org";
 
 const kSpecialWidgetPfx = "customizableui-special-";
 
@@ -253,13 +254,14 @@ var CustomizableUIInternal = {
       "spring",
       "urlbar-container",
       "spring",
-      "save-to-pocket-button",
       "downloads-button",
       AppConstants.MOZ_DEV_EDITION ? "developer-button" : null,
-      "fxa-toolbar-menu-button",
+      "wrapper-sidebar-button",
       lazy.resetPBMToolbarButtonEnabled ? "reset-pbm-toolbar-button" : null,
     ].filter(name => name);
 
+    lazy.ZenCustomizableUI.startup(this);
+
     this.registerArea(
       CustomizableUI.AREA_NAVBAR,
       {
@@ -288,10 +290,10 @@ var CustomizableUIInternal = {
       {
         type: CustomizableUI.TYPE_TOOLBAR,
         defaultPlacements: [
-          "firefox-view-button",
+//          "firefox-view-button",
           "tabbrowser-tabs",
-          "new-tab-button",
-          "alltabs-button",
+//          "new-tab-button",
+//          "alltabs-button",
         ],
         defaultCollapsed: null,
       },
@@ -331,6 +333,7 @@ var CustomizableUIInternal = {
       CustomizableUI.AREA_NAVBAR,
       CustomizableUI.AREA_BOOKMARKS,
       CustomizableUI.AREA_TABSTRIP,
+      "zen-sidebar-top-buttons",
     ]);
     if (AppConstants.platform != "macosx") {
       toolbars.add(CustomizableUI.AREA_MENUBAR);
@@ -1033,6 +1036,9 @@ var CustomizableUIInternal = {
         placements = gPlacements.get(area);
       }
 
+      // remove "zen-sidebar-top-buttons" from the placements
+      placements = placements.filter(p => p != "zen-sidebar-top-buttons");
+
       // For toolbars that need it, mark as dirty.
       let defaultPlacements = areaProperties.get("defaultPlacements");
       if (
@@ -3289,6 +3295,9 @@ var CustomizableUIInternal = {
         gSeenWidgets.add(widgetId);
       }
     }
+
+    
+
     if (gSeenWidgets.size || gNewElementCount) {
       gDirty = true;
       this.saveState();
