diff --git a/browser/components/customizableui/PanelMultiView.sys.mjs b/browser/components/customizableui/PanelMultiView.sys.mjs
index a97889f08a61e68ebf5bc5d3749362a40d48bb77..5770b0e86bbe01180c174f5d512133c577c4e0ac 100644
--- a/browser/components/customizableui/PanelMultiView.sys.mjs
+++ b/browser/components/customizableui/PanelMultiView.sys.mjs
@@ -1205,7 +1205,7 @@ export var PanelMultiView = class extends AssociatedToNode {
     } else {
       let anchorScreenBottom = anchor.screenY + anchorRect.height;
       let cssAvailHeight = availHeight.value / screen.defaultCSSScaleFactor;
-      maxHeight = cssAvailTop + cssAvailHeight - anchorScreenBottom;
+      maxHeight = cssAvailTop + cssAvailHeight;
     }
 
     // To go from the maximum height of the panel to the maximum height of
