diff --git a/toolkit/locales/en-US/toolkit/about/aboutProfiles.ftl b/toolkit/locales/en-US/toolkit/about/aboutProfiles.ftl
index c11b89b3214cc17826a0be745f5d14d8c5c5cae3..417295cc42f1c61a02d9f26d1df442237b8e720c 100644
--- a/toolkit/locales/en-US/toolkit/about/aboutProfiles.ftl
+++ b/toolkit/locales/en-US/toolkit/about/aboutProfiles.ftl
@@ -20,6 +20,7 @@ profiles-flush-restart-button = Restart { -brand-short-name }
 profiles-name = Profile: { $name }
 profiles-is-default = Default Profile
 profiles-rootdir = Root Directory
+profiles-avatar = Profile Avatar
 
 # localDir is used to show the directory corresponding to
 # the main profile directory that exists for the purpose of storing data on the
