diff --git a/browser/components/aboutwelcome/actors/AboutWelcomeParent.sys.mjs b/browser/components/aboutwelcome/actors/AboutWelcomeParent.sys.mjs
index 1eb77da0d8f9db02b3afad9d3cdd438cb6f58cd7..25806b026b4da05add168f2f0e556a877d772e1a 100644
--- a/browser/components/aboutwelcome/actors/AboutWelcomeParent.sys.mjs
+++ b/browser/components/aboutwelcome/actors/AboutWelcomeParent.sys.mjs
@@ -42,9 +42,9 @@ const AWTerminate = {
 };
 const LIGHT_WEIGHT_THEMES = {
   AUTOMATIC: "default-theme@mozilla.org",
-  DARK: "firefox-compact-dark@mozilla.org",
-  LIGHT: "firefox-compact-light@mozilla.org",
-  ALPENGLOW: "firefox-alpenglow@mozilla.org",
+  DARK: "zen-compact-dark@mozilla.org",
+  LIGHT: "zen-compact-light@mozilla.org",
+  ALPENGLOW: "zen-compact-universe@mozilla.org",
 };
 
 class AboutWelcomeObserver {
