diff --git a/.stylelintrc.js b/.stylelintrc.js
index 876f76417f800ce8889d398b6ab478cc089c4125..170b6e81dc1cec343d8678649e3248b89d115e39 100644
--- a/.stylelintrc.js
+++ b/.stylelintrc.js
@@ -67,7 +67,7 @@ module.exports = {
     ],
 
     "max-nesting-depth": [
-      3,
+      6,
       {
         ignore: ["blockless-at-rules"],
       },
@@ -282,7 +282,7 @@ module.exports = {
     "stylelint-plugin-mozilla/no-background-without-text-color": true,
     "stylelint-plugin-mozilla/no-base-design-tokens": true,
     "stylelint-plugin-mozilla/no-has-selector": true,
-    "stylelint-plugin-mozilla/use-design-tokens": true,
+    "stylelint-plugin-mozilla/use-design-tokens": null,
     "stylelint-plugin-mozilla/use-paired-color-tokens": true,
   },
 
