diff --git a/Cargo.toml b/Cargo.toml
index 8c6b99bc629ec26e01ab3746f2ae46174bfaa0b1..5791c7470923c7c15986089b40357ce80365e942 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -233,6 +233,9 @@ rure = { path = "third_party/rust/rure" }
 # Patch `plist` to work with `indexmap` 2.*
 plist = { path = "third_party/rust/plist" }
 
+# Patch mime_guess to add missing mime types
+mime_guess = { path = "third_party/rust/mime_guess" }
+
 # Patch `unicode-width` 0.1.* to 0.2.
 unicode-width = { path = "build/rust/unicode-width" }
 
