diff --git a/Cargo.toml b/Cargo.toml
index c41cc4f4e68e5b30e1a290e07b2c5061db4400cc..7fa5402b32116922766ee1463b441d3fc86c1eff 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -218,6 +218,8 @@ moz_asserts = { path = "mozglue/static/rust/moz_asserts" }
 # Workaround for https://github.com/rust-lang/cargo/issues/11232
 rure = { path = "third_party/rust/rure" }
 
+mime_guess = { path = "third_party/rust/mime_guess" }
+
 # Patch `unicode-width` 0.1.* to 0.2.
 unicode-width = { path = "build/rust/unicode-width" }
 
