diff --git a/Cargo.toml b/Cargo.toml
index d402d783e2bbc30b87b2c59cf02e38b74d0484c1..b718ee7ed0519cac20d2828fee0350b44fc3f45a 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -227,6 +227,9 @@ rure = { path = "third_party/rust/rure" }
 # Patch `plist` to work with `indexmap` 2.*
 plist = { path = "third_party/rust/plist" }
 
+# Patch mime_guess to add missing mime types
+mime_guess = { path = "third_party/rust/mime_guess" }
+
 # Patch `unicode-width` 0.1.* to 0.2.
 unicode-width = { path = "build/rust/unicode-width" }
 
