diff --git a/eslint-ignores.config.mjs b/eslint-ignores.config.mjs
index cad7ef0ef8890ee4af411fdbfa59419c5931388c..0bbded34c89004727ce9893d17bbdacc1fdc2517 100644
--- a/eslint-ignores.config.mjs
+++ b/eslint-ignores.config.mjs
@@ -307,4 +307,8 @@ export default [
   "dom/base/test/jsmodules/import_circular_1.mjs",
   "dom/base/test/jsmodules/importmaps/multiple/import_circular.mjs",
   "dom/base/test/jsmodules/importmaps/multiple/import_circular_1.mjs",
+
+  "zen/common/emojis/ZenEmojisData.min.mjs",
+  "zen/tests/**",
+  "zen/vendor/**",
 ];
