diff --git a/eslint-ignores.config.mjs b/eslint-ignores.config.mjs
index bfb30a0a35af081a2496a05fd78d9efea15ee37f..04fee66309192b3e38986ae09e77ec7a4e1ad83b 100644
--- a/eslint-ignores.config.mjs
+++ b/eslint-ignores.config.mjs
@@ -314,4 +314,8 @@ export default [
   "dom/base/test/jsmodules/import_circular_1.mjs",
   "dom/base/test/jsmodules/importmaps/multiple/import_circular.mjs",
   "dom/base/test/jsmodules/importmaps/multiple/import_circular_1.mjs",
+
+  "zen/common/emojis/ZenEmojisData.min.mjs",
+  "zen/tests/**",
+  "zen/vendor/**",
 ];
