diff --git a/eslint-test-paths.config.mjs b/eslint-test-paths.config.mjs
index 0d3bd58e7fb9802223588a46fd83ae6b8e14837f..b8f91f6a6d52d58b47c7556650910f3ac0ec2842 100644
--- a/eslint-test-paths.config.mjs
+++ b/eslint-test-paths.config.mjs
@@ -219,6 +219,7 @@ const extraBrowserTestPaths = [
   "toolkit/components/windowwatcher/test/",
   "toolkit/mozapps/extensions/test/xpinstall/",
   "uriloader/exthandler/tests/mochitest/",
+  "zen/tests/",
 ];
 
 // DO NOT add more items to this list. Please see the note at the top
