diff --git a/dom/events/KeyEventHandler.cpp b/dom/events/KeyEventHandler.cpp
index 3f3ca808e8..0828b37f51 100644
--- a/dom/events/KeyEventHandler.cpp
+++ b/dom/events/KeyEventHandler.cpp
@@ -641,7 +641,7 @@ void KeyEventHandler::BuildModifiers(nsAString& aModifiers) {
       } else if (strcmp(token, "control") == 0) {
         mKeyMask |= cControl | cControlMask;
       } else if (strcmp(token, "accel") == 0) {
-        mKeyMask |= AccelKeyMask();
+        mKeyMask |= cControl | cControlMask;
       } else if (strcmp(token, "access") == 0) {
         mKeyMask |= KeyToMask(LookAndFeel::GetMenuAccessKey());
       } else if (strcmp(token, "any") == 0) {
