diff --git a/browser/components/search/SearchUIUtils.sys.mjs b/browser/components/search/SearchUIUtils.sys.mjs
index 6ef224ce377cf3ff511e435f3c7a5dc1de819c60..5b4c395c222c6317fd88499a1aa5307032ea13f8 100644
--- a/browser/components/search/SearchUIUtils.sys.mjs
+++ b/browser/components/search/SearchUIUtils.sys.mjs
@@ -426,7 +426,7 @@ export var SearchUIUtils = {
         triggeringSearchEngine: engine.name,
       },
     });
-
+    window.gZenGlanceManager?.onSearchSelectCommand(where);
     return { engine, url: submission.uri };
   },
 
