diff --git a/browser/components/search/SearchUIUtils.sys.mjs b/browser/components/search/SearchUIUtils.sys.mjs
index 22680c91da251adc21bb46debc73b3dbf6cdf08c..f717b4a29bbd29f571635d4f0717463eabdbfa60 100644
--- a/browser/components/search/SearchUIUtils.sys.mjs
+++ b/browser/components/search/SearchUIUtils.sys.mjs
@@ -445,6 +445,7 @@ export var SearchUIUtils = {
       },
     });
 
+    this.#openGlanceForSearch(currentTab, parentTab);
     lazy.BrowserSearchTelemetry.recordSearch(
       window.gBrowser.selectedBrowser,
       engine,
