diff --git a/browser/components/urlbar/content/UrlbarView.mjs b/browser/components/urlbar/content/UrlbarView.mjs
index 99ee8ec42297a56fd72d344179b021e604fefd48..931d33c4a576ec7917393797d48ddd85cde8950e 100644
--- a/browser/components/urlbar/content/UrlbarView.mjs
+++ b/browser/components/urlbar/content/UrlbarView.mjs
@@ -818,7 +818,7 @@ export class UrlbarView {
       !this.input.value ||
       this.input.getAttribute("pageproxystate") == "valid"
     ) {
-      if (!this.isOpen && ["mousedown", "command"].includes(event.type)) {
+      if (!this.isOpen && ["mousedown", "click", "command"].includes(event.type)) {
         // Try to reuse the cached top-sites context. If it's not cached, then
         // there will be a gap of time between when the input is focused and
         // when the view opens that can be perceived as flicker.
@@ -962,10 +962,6 @@ export class UrlbarView {
     }
 
     // If search mode isn't active, close the view.
-    if (!this.input.searchMode) {
-      this.close();
-      return;
-    }
 
     // Search mode is active.  If the one-offs should be shown, make sure they
     // are enabled and show the view.
@@ -1042,6 +1038,15 @@ export class UrlbarView {
         // them, resembling tab-to-search. In that case, the input value is
         // still associated with the first result.
         this.input.setResultForCurrentValue(firstResult);
+      } else if (
+        !queryContext.searchString &&
+        firstResult.providerName == "ZenUrlbarProviderRecentHistory"
+      ) {
+        // Preselect the most recent page when the urlbar opens empty.
+        this.#selectElement(this.getFirstSelectableElement(), {
+          updateInput: false,
+          setAccessibleFocus: this.controller.userSelectionBehavior == "arrow",
+        });
       }
     }
 
@@ -3290,6 +3295,8 @@ export class UrlbarView {
       if (row?.hasAttribute("row-selectable")) {
         row?.toggleAttribute("selected", true);
       }
+      // only scroll if its not already fully visible
+      element.scrollIntoView({ behavior: "smooth", block: "nearest" });
       if (element != row) {
         row?.toggleAttribute("descendant-selected", true);
       }
@@ -3956,7 +3963,7 @@ export class UrlbarView {
   }
 
   #enableOrDisableRowWrap() {
-    let wrap = getBoundsWithoutFlushing(this.input).width < 650;
+    let wrap = false;
     this.#rows.toggleAttribute("wrap", wrap);
     this.oneOffSearchButtons?.container.toggleAttribute("wrap", wrap);
   }
@@ -4450,6 +4457,12 @@ export class UrlbarView {
   }
 
   on_blur() {
+    if (
+      this.document.commandDispatcher.focusedElement == this.input.inputField &&
+      !UrlbarPrefs.get("closeOnWindowBlur")
+    ) {
+      return;
+    }
     // If the view is open without the input being focused, it will not close
     // automatically when the window loses focus. We might be in this state
     // after a Search Tip is shown on an engine homepage.
