diff --git a/browser/locales/en-US/browser/appExtensionFields.ftl b/browser/locales/en-US/browser/appExtensionFields.ftl
index 09d7a076eccd4235fac66156770d8b9ae01a7d16..f32affde0d6f77dadd421ff68bf2050bda9f0cd0 100644
--- a/browser/locales/en-US/browser/appExtensionFields.ftl
+++ b/browser/locales/en-US/browser/appExtensionFields.ftl
@@ -8,13 +8,13 @@
 extension-default-theme-name-auto=System theme — auto
 extension-default-theme-description=Follow the operating system setting for buttons, menus, and windows.
 
-extension-firefox-compact-light-name=Light
+extension-firefox-compact-light-name=Zen Light
 extension-firefox-compact-light-description=A theme with a light color scheme.
 
-extension-firefox-compact-dark-name=Dark
+extension-firefox-compact-dark-name=Zen Dark
 extension-firefox-compact-dark-description=A theme with a dark color scheme.
 
-extension-firefox-alpenglow-name=Firefox Alpenglow
+extension-firefox-alpenglow-name=Zen Universe
 extension-firefox-alpenglow-description=Use a colorful appearance for buttons, menus, and windows.
 
 ## Colorway Themes
@@ -28,3 +28,9 @@ extension-colorways-balanced-name={ $colorway-name } — Balanced
 # "Bold" is used in the sense of bravery or courage, not in the sense of
 # emphasized text.
 extension-colorways-bold-name={ $colorway-name } — Bold
+
+extension-firefox-compact-galaxy-name=Zen Galaxy
+extension-firefox-compact-galaxy-description=A theme with a dark color scheme and a nice galaxy touch!
+
+extension-firefox-compact-dream-name=Zen Dream
+extension-firefox-compact-dream-description=A theme with a light color scheme and a nice dream-y touch!
\ No newline at end of file
