diff --git a/browser/locales/en-US/browser/appExtensionFields.ftl b/browser/locales/en-US/browser/appExtensionFields.ftl
index 09d7a076eccd4235fac66156770d8b9ae01a7d16..31c7632c9dce24f2e4e5135abfca96a47e961eed 100644
--- a/browser/locales/en-US/browser/appExtensionFields.ftl
+++ b/browser/locales/en-US/browser/appExtensionFields.ftl
@@ -8,13 +8,13 @@
 extension-default-theme-name-auto=System theme — auto
 extension-default-theme-description=Follow the operating system setting for buttons, menus, and windows.
 
-extension-firefox-compact-light-name=Light
+extension-firefox-compact-light-name=Zen Light
 extension-firefox-compact-light-description=A theme with a light color scheme.
 
-extension-firefox-compact-dark-name=Dark
+extension-firefox-compact-dark-name=Zen Dark
 extension-firefox-compact-dark-description=A theme with a dark color scheme.
 
-extension-firefox-alpenglow-name=Firefox Alpenglow
+extension-firefox-alpenglow-name=Zen Universe
 extension-firefox-alpenglow-description=Use a colorful appearance for buttons, menus, and windows.
 
 ## Colorway Themes
