diff --git a/browser/locales/en-US/browser/browser.ftl b/browser/locales/en-US/browser/browser.ftl
index 02ba7bb1faa074128a995107735c31dbbd280fd1..28439a06063ec626dcfb73f7dedc0a53fea22f59 100644
--- a/browser/locales/en-US/browser/browser.ftl
+++ b/browser/locales/en-US/browser/browser.ftl
@@ -347,7 +347,7 @@ bookmark-panel-cancel =
 bookmark-panel-remove =
     .label =
         { $count ->
-            [1] Remove bookmark
+            [1] Remove
            *[other] Remove { $count } bookmarks
         }
     .accesskey = R
