diff --git a/browser/locales/en-US/browser/customizeMode.ftl b/browser/locales/en-US/browser/customizeMode.ftl
index e5f54b231738fc8fee505c8c8dcccde820995cfc..ec81ef8e451faa25878830b7d4796708e15fe395 100644
--- a/browser/locales/en-US/browser/customizeMode.ftl
+++ b/browser/locales/en-US/browser/customizeMode.ftl
@@ -36,3 +36,5 @@ customize-mode-touchbar-cmd =
     .label = Customize Touch Bar…
 customize-mode-downloads-button-autohide =
     .label = Hide button when empty
+
+customization-zen-browser-space = Content Space
\ No newline at end of file
