diff --git a/browser/locales/en-US/browser/sidebarMenu.ftl b/browser/locales/en-US/browser/sidebarMenu.ftl
index 746a2084df954f224ace4ce699623dfc618f0ada..ee84f78fe7ec3992d269d3ba103b62c846dc22b8 100644
--- a/browser/locales/en-US/browser/sidebarMenu.ftl
+++ b/browser/locales/en-US/browser/sidebarMenu.ftl
@@ -16,3 +16,6 @@ sidebar-menu-close =
 
 sidebar-close-button =
   .tooltiptext = Close sidebar
+
+sidebar-zen-sidepanel =
+  .label = Side Panels
