diff --git a/browser/locales/en-US/browser/sidebarMenu.ftl b/browser/locales/en-US/browser/sidebarMenu.ftl
index e050a2302c6699ffcabad2db4d45267912297519..d4ff03ace4d08482b4a6de62c4f00016486cad84 100644
--- a/browser/locales/en-US/browser/sidebarMenu.ftl
+++ b/browser/locales/en-US/browser/sidebarMenu.ftl
@@ -19,3 +19,8 @@ sidebar-menu-close =
 
 sidebar-close-button =
   .tooltiptext = Close sidebar
+
+sidebar-zen-sidepanel =
+  .label = Side Panels
+sidebar-zen-expand =
+  .label = Expand Sidebar
\ No newline at end of file
