diff --git a/browser/locales/en-US/browser/tabContextMenu.ftl b/browser/locales/en-US/browser/tabContextMenu.ftl
index df58df794c5e7101aa1ea684edd009532d74240d..0b4237dfde4e8e30e279f0a7022c1a36e67a04e8 100644
--- a/browser/locales/en-US/browser/tabContextMenu.ftl
+++ b/browser/locales/en-US/browser/tabContextMenu.ftl
@@ -101,6 +101,14 @@ tab-context-move-tabs =
         }
     .accesskey = v
 
+tab-zen-split-tabs =
+    .label =
+        { $tabCount ->
+            [1] Split Tab (multiple selected tabs needed)
+           *[other] Split { $tabCount } Tabs
+        }
+    .accesskey = S
+
 tab-context-send-tabs-to-device =
     .label =
         { $tabCount ->
