diff --git a/browser/locales/en-US/browser/unifiedExtensions.ftl b/browser/locales/en-US/browser/unifiedExtensions.ftl
index 8db0d9488ac334ada07bf3f537f78cdc99577fb7..acc1d60d5577ffafe982557757c868fa691a1d96 100644
--- a/browser/locales/en-US/browser/unifiedExtensions.ftl
+++ b/browser/locales/en-US/browser/unifiedExtensions.ftl
@@ -10,6 +10,8 @@ unified-extensions-header-title = Extensions
 unified-extensions-manage-extensions =
     .label = Manage extensions
 
+unified-extensions-description = Extensions are used to bring more extra functionality into { -brand-short-name }. 
+
 ## An extension in the main list
 
 # Each extension in the unified extensions panel (list) has a secondary button
