diff --git a/browser/components/newtab/content-src/components/CustomizeMenu/ContentSection/ContentSection.jsx b/browser/components/newtab/content-src/components/CustomizeMenu/ContentSection/ContentSection.jsx
index 57ed935e93198bb50555d119080552934c49e428..21e8fb35501091d448502dbd0c53ee01ce8b7bc0 100644
--- a/browser/components/newtab/content-src/components/CustomizeMenu/ContentSection/ContentSection.jsx
+++ b/browser/components/newtab/content-src/components/CustomizeMenu/ContentSection/ContentSection.jsx
@@ -153,7 +153,7 @@ export class ContentSection extends React.PureComponent {
                     data-l10n-args='{"num": 4}'
                   />
                 </select>
-                {mayHaveSponsoredTopSites && (
+                {false && (
                   <div className="check-wrapper" role="presentation">
                     <input
                       id="sponsored-shortcuts"
