diff --git a/browser/tools/mozscreenshots/mozscreenshots/extension/configurations/LightweightThemes.sys.mjs b/browser/tools/mozscreenshots/mozscreenshots/extension/configurations/LightweightThemes.sys.mjs
index 91c3349ec737aa12518b596f1e6278eb27f0f5ec..15331c3d170acb1b12c7f4a9ac6134e497b31a96 100644
--- a/browser/tools/mozscreenshots/mozscreenshots/extension/configurations/LightweightThemes.sys.mjs
+++ b/browser/tools/mozscreenshots/mozscreenshots/extension/configurations/LightweightThemes.sys.mjs
@@ -42,7 +42,7 @@ export var LightweightThemes = {
       selectors: [],
       async applyConfig() {
         let addon = await AddonManager.getAddonByID(
-          "firefox-alpenglow@mozilla.org"
+          "zen-compact-universe@mozilla.org"
         );
         await addon.enable();
       },
